--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/EventSubscriberTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/EventSubscriberTransformer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,11 +19,9 @@
 
 package net.minecraftforge.fml.common.asm.transformers;
 
-import java.lang.reflect.Modifier;
-import java.util.List;
-
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
 import net.minecraft.launchwrapper.IClassTransformer;
-
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.Opcodes;
@@ -31,8 +29,8 @@
 import org.objectweb.asm.tree.ClassNode;
 import org.objectweb.asm.tree.MethodNode;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Iterables;
+import java.lang.reflect.Modifier;
+import java.util.List;
 
 public class EventSubscriberTransformer implements IClassTransformer
 {
