--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/FieldRedirectTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/FieldRedirectTransformer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,20 +19,14 @@
 
 package net.minecraftforge.fml.common.asm.transformers;
 
-import java.util.ListIterator;
-
 import net.minecraft.launchwrapper.IClassTransformer;
-
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.Opcodes;
-import org.objectweb.asm.tree.AbstractInsnNode;
-import org.objectweb.asm.tree.ClassNode;
-import org.objectweb.asm.tree.FieldInsnNode;
-import org.objectweb.asm.tree.FieldNode;
-import org.objectweb.asm.tree.MethodInsnNode;
-import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.*;
 
+import java.util.ListIterator;
+
 public class FieldRedirectTransformer implements IClassTransformer
 {
     private final String clsName;
