--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/PatchingTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/PatchingTransformer.java
@@ -24,8 +24,7 @@
 
 public class PatchingTransformer implements IClassTransformer {
     @Override
-    public byte[] transform(String name, String transformedName, byte[] bytes)
-    {
+    public byte[] transform(String name, String transformedName, byte[] bytes) {
         return ClassPatchManager.INSTANCE.applyPatch(name, transformedName, bytes);
     }
 
