--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/PotionEffectTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/PotionEffectTransformer.java
@@ -19,10 +19,8 @@
 
 package net.minecraftforge.fml.common.asm.transformers;
 
-public class PotionEffectTransformer extends FieldRedirectTransformer
-{
-    public PotionEffectTransformer()
-    {
+public class PotionEffectTransformer extends FieldRedirectTransformer {
+    public PotionEffectTransformer() {
         super("net.minecraft.potion.PotionEffect", "Lnet/minecraft/potion/Potion;", "getPotionRaw");
     }
 }
