--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/TerminalTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/TerminalTransformer.java
@@ -63,7 +63,7 @@
         @Override
         public MethodVisitor visitMethod(int mAccess, final String mName, final String mDesc, String mSignature, String[] mExceptions)
         {
-            final boolean warn = !(clsName.equals("net/minecraft/client/Minecraft") ||
+        	final boolean warn = !(clsName.equals("net/minecraft/client/Minecraft") ||
                                    clsName.equals("net/minecraft/server/dedicated/DedicatedServer") ||
                                    clsName.equals("net/minecraft/server/dedicated/ServerHangWatchdog") ||
                                    clsName.equals("net/minecraft/server/dedicated/ServerHangWatchdog$1") ||
