--- ../src-base/minecraft/net/minecraftforge/client/model/animation/AnimationTESR.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/animation/AnimationTESR.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -35,7 +35,6 @@
 import net.minecraftforge.common.model.animation.IAnimationStateMachine;
 import net.minecraftforge.common.property.IExtendedBlockState;
 import net.minecraftforge.common.property.Properties;
-
 import org.apache.commons.lang3.tuple.Pair;
 
 /**
