--- ../src-base/minecraft/net/minecraftforge/client/model/animation/AnimationItemOverrideList.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/animation/AnimationItemOverrideList.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,8 +19,6 @@
 
 package net.minecraftforge.client.model.animation;
 
-import java.util.List;
-
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.block.model.IBakedModel;
 import net.minecraft.client.renderer.block.model.ItemOverride;
@@ -37,10 +35,10 @@
 import net.minecraftforge.common.model.animation.CapabilityAnimation;
 import net.minecraftforge.common.model.animation.IAnimationStateMachine;
 
+import javax.annotation.Nullable;
+import java.util.List;
 import java.util.function.Function;
 
-import javax.annotation.Nullable;
-
 public final class AnimationItemOverrideList extends ItemOverrideList
 {
     private final IModel model;
