--- ../src-base/minecraft/net/minecraftforge/client/model/animation/Animation.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/animation/Animation.java
@@ -23,30 +23,25 @@
 
 import java.lang.ref.WeakReference;
 
-public enum Animation
-{
+public enum Animation {
     INSTANCE;
 
-    private float clientPartialTickTime;
-
     private static long epochTime;
     private static WeakReference<World> worldRef;
+    private float clientPartialTickTime;
 
     /**
      * Get the global world time for the current tick, in seconds.
      */
-    public static float getWorldTime(World world)
-    {
+    public static float getWorldTime(World world) {
         return getWorldTime(world, 0);
     }
 
     /**
      * Get the global world time for the current tick + partial tick progress, in seconds.
      */
-    public static float getWorldTime(World world, float tickProgress)
-    {
-        if (worldRef == null || worldRef.get() != world)
-        {
+    public static float getWorldTime(World world, float tickProgress) {
+        if (worldRef == null || worldRef.get() != world) {
             epochTime = world.getTotalWorldTime();
             worldRef = new WeakReference<>(world);
         }
@@ -56,8 +51,7 @@
     /**
      * Get current partialTickTime.
      */
-    public static float getPartialTickTime()
-    {
+    public static float getPartialTickTime() {
         return INSTANCE.clientPartialTickTime;
     }
 
