--- ../src-base/minecraft/net/minecraftforge/client/model/animation/FastTESR.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/animation/FastTESR.java
@@ -19,23 +19,20 @@
 
 package net.minecraftforge.client.model.animation;
 
-import org.lwjgl.opengl.GL11;
-
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.RenderHelper;
 import net.minecraft.client.renderer.Tessellator;
-import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.texture.TextureMap;
 import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
 import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
 import net.minecraft.tileentity.TileEntity;
+import org.lwjgl.opengl.GL11;
 
-public abstract class FastTESR<T extends TileEntity> extends TileEntitySpecialRenderer<T>
-{
+public abstract class FastTESR<T extends TileEntity> extends TileEntitySpecialRenderer<T> {
     @Override
-    public final void render(T te, double x, double y, double z, float partialTicks, int destroyStage, float partial)
-    {
+    public final void render(T te, double x, double y, double z, float partialTicks, int destroyStage, float partial) {
         Tessellator tessellator = Tessellator.getInstance();
         BufferBuilder buffer = tessellator.getBuffer();
         this.bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
@@ -44,12 +41,9 @@
         GlStateManager.enableBlend();
         GlStateManager.disableCull();
 
-        if (Minecraft.isAmbientOcclusionEnabled())
-        {
+        if (Minecraft.isAmbientOcclusionEnabled()) {
             GlStateManager.shadeModel(GL11.GL_SMOOTH);
-        }
-        else
-        {
+        } else {
             GlStateManager.shadeModel(GL11.GL_FLAT);
         }
 
