--- ../src-base/minecraft/net/minecraftforge/fml/common/network/NetworkEventFiringHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/NetworkEventFiringHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,14 +19,11 @@
 
 package net.minecraftforge.fml.common.network;
 
-import net.minecraftforge.fml.common.FMLLog;
-import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
-
-import org.apache.logging.log4j.Level;
-
+import cn.pfcraft.server.Mohist;
 import io.netty.channel.ChannelHandler.Sharable;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.SimpleChannelInboundHandler;
+import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
 
 /**
  * Use this handler as the only thing in your channel, to receive network events
@@ -59,7 +56,7 @@
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception
     {
-        FMLLog.log.error("NetworkEventFiringHandler exception", cause);
+        Mohist.LOGGER.error("NetworkEventFiringHandler exception", cause);
         super.exceptionCaught(ctx, cause);
     }
 }
