--- ../src-base/minecraft/net/minecraftforge/fml/common/network/NetworkCheckHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/NetworkCheckHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,13 +19,13 @@
 
 package net.minecraftforge.fml.common.network;
 
+import net.minecraftforge.fml.relauncher.Side;
+
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-import net.minecraftforge.fml.relauncher.Side;
-
 /**
  * A method annotated with this will be called when a remote network connection is offered.
  * The method should have two parameters, of types Map<String,String> and {@link Side}. It should return a boolean
