--- ../src-base/minecraft/net/minecraft/server/DebugLoggingPrintStream.java
+++ ../src-work/minecraft/net/minecraft/server/DebugLoggingPrintStream.java
@@ -1,8 +1,9 @@
 package net.minecraft.server;
 
-import java.io.OutputStream;
 import net.minecraft.util.LoggingPrintStream;
 
+import java.io.OutputStream;
+
 public class DebugLoggingPrintStream extends LoggingPrintStream
 {
     public DebugLoggingPrintStream(String domainIn, OutputStream outStream)
