--- ../src-base/minecraft/net/minecraft/server/ServerEula.java
+++ ../src-work/minecraft/net/minecraft/server/ServerEula.java
@@ -1,17 +1,16 @@
 package net.minecraft.server;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.util.Properties;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.io.IOUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.util.Properties;
+
 @SideOnly(Side.SERVER)
 public class ServerEula
 {
@@ -22,29 +21,29 @@
     public ServerEula(File eulaFile)
     {
         this.eulaFile = eulaFile;
-        this.acceptedEULA = this.loadEULAFile(eulaFile);
+        this.acceptedEULA = loadEULAFile(eulaFile);
     }
 
     private boolean loadEULAFile(File inFile)
     {
         FileInputStream fileinputstream = null;
-        boolean flag = false;
+        boolean flag = true;
 
         try
         {
             Properties properties = new Properties();
             fileinputstream = new FileInputStream(inFile);
             properties.load(fileinputstream);
-            flag = Boolean.parseBoolean(properties.getProperty("eula", "false"));
+            flag = Boolean.parseBoolean(properties.getProperty("eula", "true"));
         }
         catch (Exception var8)
         {
-            LOG.warn("Failed to load {}", (Object)inFile);
+            LOG.warn("Failed to load {}", inFile);
             this.createEULAFile();
         }
         finally
         {
-            IOUtils.closeQuietly((InputStream)fileinputstream);
+            IOUtils.closeQuietly(fileinputstream);
         }
 
         return flag;
@@ -63,8 +62,9 @@
         {
             Properties properties = new Properties();
             fileoutputstream = new FileOutputStream(this.eulaFile);
-            properties.setProperty("eula", "false");
-            properties.store(fileoutputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula).");
+            properties.setProperty("eula", "true");
+            properties.store(fileoutputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula)." +
+                    "\nand also agreeing that tacos are tasty.");  // Paper - fix lag
         }
         catch (Exception exception)
         {
@@ -72,7 +72,7 @@
         }
         finally
         {
-            IOUtils.closeQuietly((OutputStream)fileoutputstream);
+            IOUtils.closeQuietly(fileoutputstream);
         }
     }
 }
