--- ../src-base/minecraft/net/minecraftforge/common/capabilities/Capability.java
+++ ../src-work/minecraft/net/minecraftforge/common/capabilities/Capability.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,14 +19,12 @@
 
 package net.minecraftforge.common.capabilities;
 
-import java.util.concurrent.Callable;
-
 import com.google.common.base.Throwables;
-
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.util.EnumFacing;
 
 import javax.annotation.Nullable;
+import java.util.concurrent.Callable;
 
 /**
  * This is the core holder object Capabilities.
