--- ../src-base/minecraft/net/minecraft/world/gen/structure/MapGenStructureIO.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/MapGenStructureIO.java
@@ -1,13 +1,14 @@
 package net.minecraft.world.gen.structure;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.world.World;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+
 public class MapGenStructureIO
 {
     private static final Logger LOGGER = LogManager.getLogger();
