--- ../src-base/minecraft/net/minecraft/world/gen/structure/MapGenEndCity.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/MapGenEndCity.java
@@ -1,12 +1,13 @@
 package net.minecraft.world.gen.structure;
 
-import java.util.Random;
 import net.minecraft.util.Rotation;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.ChunkPrimer;
 import net.minecraft.world.gen.ChunkGeneratorEnd;
 
+import java.util.Random;
+
 public class MapGenEndCity extends MapGenStructure
 {
     private final int citySpacing = 20;
@@ -59,7 +60,7 @@
 
     protected StructureStart getStructureStart(int chunkX, int chunkZ)
     {
-        return new MapGenEndCity.Start(this.world, this.endProvider, this.rand, chunkX, chunkZ);
+        return new Start(this.world, this.endProvider, this.rand, chunkX, chunkZ);
     }
 
     public BlockPos getNearestStructurePos(World worldIn, BlockPos pos, boolean findUnexplored)
