--- ../src-base/minecraft/net/minecraft/world/gen/structure/StructureComponentTemplate.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/StructureComponentTemplate.java
@@ -1,8 +1,5 @@
 package net.minecraft.world.gen.structure;
 
-import java.util.Map;
-import java.util.Random;
-import java.util.Map.Entry;
 import net.minecraft.init.Blocks;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.EnumFacing;
@@ -14,6 +11,10 @@
 import net.minecraft.world.gen.structure.template.Template;
 import net.minecraft.world.gen.structure.template.TemplateManager;
 
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+
 public abstract class StructureComponentTemplate extends StructureComponent
 {
     private static final PlacementSettings DEFAULT_PLACE_SETTINGS = new PlacementSettings();
