--- ../src-base/minecraft/net/minecraft/world/gen/structure/StructureBoundingBox.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/StructureBoundingBox.java
@@ -13,6 +13,8 @@
     public int maxX;
     public int maxY;
     public int maxZ;
+    public Vec3i getLowPosition() { return new Vec3i(minX, minY, minZ); } // Paper
+    public Vec3i getHighPosition() { return new Vec3i(maxX, maxY, maxZ); } // Paper
 
     public StructureBoundingBox()
     {
