--- ../src-base/minecraft/net/minecraftforge/event/terraingen/OreGenEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/terraingen/OreGenEvent.java
@@ -19,17 +19,16 @@
 
 package net.minecraftforge.event.terraingen;
 
-import java.util.Random;
-
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeDecorator;
+import net.minecraft.world.gen.feature.WorldGenerator;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.world.World;
-import net.minecraft.world.gen.feature.WorldGenerator;
 
+import java.util.Random;
+
 /**
  * OreGenEvent is fired when an event involving ore generation occurs.<br>
  * If a method utilizes this {@link Event} as its parameter, the method will
@@ -41,31 +40,26 @@
  * <br>
  * All children of this event are fired on the {@link MinecraftForge#ORE_GEN_BUS}.<br>
  **/
-public class OreGenEvent extends Event
-{
+public class OreGenEvent extends Event {
     private final World world;
     private final Random rand;
     private final BlockPos pos;
 
-    public OreGenEvent(World world, Random rand, BlockPos pos)
-    {
+    public OreGenEvent(World world, Random rand, BlockPos pos) {
         this.world = world;
         this.rand = rand;
         this.pos = pos;
     }
 
-    public World getWorld()
-    {
+    public World getWorld() {
         return world;
     }
 
-    public Random getRand()
-    {
+    public Random getRand() {
         return rand;
     }
 
-    public BlockPos getPos()
-    {
+    public BlockPos getPos() {
         return pos;
     }
 
@@ -80,10 +74,8 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#ORE_GEN_BUS}.<br>
      **/
-    public static class Pre extends OreGenEvent
-    {
-        public Pre(World world, Random rand, BlockPos pos)
-        {
+    public static class Pre extends OreGenEvent {
+        public Pre(World world, Random rand, BlockPos pos) {
             super(world, rand, pos);
         }
     }
@@ -99,10 +91,8 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#ORE_GEN_BUS}.<br>
      **/
-    public static class Post extends OreGenEvent
-    {
-        public Post(World world, Random rand, BlockPos pos)
-        {
+    public static class Post extends OreGenEvent {
+        public Post(World world, Random rand, BlockPos pos) {
             super(world, rand, pos);
         }
     }
@@ -123,28 +113,23 @@
      * This event is fired on the {@link MinecraftForge#ORE_GEN_BUS}.<br>
      **/
     @HasResult
-    public static class GenerateMinable extends OreGenEvent
-    {
-        public static enum EventType { COAL, DIAMOND, DIRT, GOLD, GRAVEL, IRON, LAPIS, REDSTONE, QUARTZ, DIORITE, GRANITE, ANDESITE, EMERALD, SILVERFISH, CUSTOM }
-
+    public static class GenerateMinable extends OreGenEvent {
         private final EventType type;
         private final WorldGenerator generator;
-
-        public GenerateMinable(World world, Random rand, WorldGenerator generator, BlockPos pos, EventType type)
-        {
+        public GenerateMinable(World world, Random rand, WorldGenerator generator, BlockPos pos, EventType type) {
             super(world, rand, pos);
             this.generator = generator;
             this.type = type;
         }
 
-        public EventType getType()
-        {
+        public EventType getType() {
             return type;
         }
 
-        public WorldGenerator getGenerator()
-        {
+        public WorldGenerator getGenerator() {
             return generator;
         }
+
+        public static enum EventType {COAL, DIAMOND, DIRT, GOLD, GRAVEL, IRON, LAPIS, REDSTONE, QUARTZ, DIORITE, GRANITE, ANDESITE, EMERALD, SILVERFISH, CUSTOM}
     }
 }
