--- ../src-base/minecraft/net/minecraftforge/event/terraingen/WorldTypeEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/terraingen/WorldTypeEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,12 @@
 
 package net.minecraftforge.event.terraingen;
 
+import net.minecraft.world.WorldType;
 import net.minecraft.world.biome.BiomeProvider;
+import net.minecraft.world.gen.layer.GenLayer;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraft.world.gen.layer.GenLayer;
-import net.minecraft.world.WorldType;
 
 /**
  * WorldTypeEvent is fired when an event involving the world occurs.<br>
