--- ../src-base/minecraft/net/minecraft/entity/ai/attributes/AttributeModifier.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/attributes/AttributeModifier.java
@@ -1,10 +1,11 @@
 package net.minecraft.entity.ai.attributes;
 
 import io.netty.util.internal.ThreadLocalRandom;
-import java.util.UUID;
 import net.minecraft.util.math.MathHelper;
 import org.apache.commons.lang3.Validate;
 
+import java.util.UUID;
+
 public class AttributeModifier
 {
     private final double amount;
