--- ../src-base/minecraft/net/minecraft/stats/RecipeBook.java
+++ ../src-work/minecraft/net/minecraft/stats/RecipeBook.java
@@ -1,12 +1,13 @@
 package net.minecraft.stats;
 
-import java.util.BitSet;
-import javax.annotation.Nullable;
 import net.minecraft.item.crafting.CraftingManager;
 import net.minecraft.item.crafting.IRecipe;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.BitSet;
+
 public class RecipeBook
 {
     protected final BitSet recipes = new BitSet();
