--- ../src-base/minecraft/net/minecraft/stats/RecipeBookServer.java
+++ ../src-work/minecraft/net/minecraft/stats/RecipeBookServer.java
@@ -1,8 +1,6 @@
 package net.minecraft.stats;
 
 import com.google.common.collect.Lists;
-import java.util.Collections;
-import java.util.List;
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.crafting.CraftingManager;
@@ -15,6 +13,9 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.Collections;
+import java.util.List;
+
 public class RecipeBookServer extends RecipeBook
 {
     private static final Logger LOGGER = LogManager.getLogger();
@@ -55,6 +56,7 @@
 
     private void sendPacket(SPacketRecipeBook.State state, EntityPlayerMP player, List<IRecipe> recipesIn)
     {
+        if (player.connection == null) return; // SPIGOT-4478 during PlayerLoginEvent
         net.minecraftforge.common.ForgeHooks.sendRecipeBook(player.connection, state, recipesIn, Collections.emptyList(), this.isGuiOpen, this.isFilteringCraftable);
     }
 
@@ -67,7 +69,11 @@
 
         for (IRecipe irecipe : this.getRecipes())
         {
-            nbttaglist.appendTag(new NBTTagString(((ResourceLocation)CraftingManager.REGISTRY.getNameForObject(irecipe)).toString()));
+            // Paper start - ignore missing recipes
+            ResourceLocation key = CraftingManager.REGISTRY.getNameForObject(irecipe);
+            if (key == null) continue;
+            nbttaglist.appendTag(new NBTTagString(key.toString()));
+            // Paper end
         }
 
         nbttagcompound.setTag("recipes", nbttaglist);
@@ -75,7 +81,11 @@
 
         for (IRecipe irecipe1 : this.getDisplayedRecipes())
         {
-            nbttaglist1.appendTag(new NBTTagString(((ResourceLocation)CraftingManager.REGISTRY.getNameForObject(irecipe1)).toString()));
+            // Paper start - ignore missing recipes
+            ResourceLocation key = CraftingManager.REGISTRY.getNameForObject(irecipe1);
+            if (key == null) continue;
+            nbttaglist.appendTag(new NBTTagString(key.toString()));
+            // Paper end
         }
 
         nbttagcompound.setTag("toBeDisplayed", nbttaglist1);
