--- ../src-base/minecraft/net/minecraftforge/common/ticket/ChunkTicketManager.java
+++ ../src-work/minecraft/net/minecraftforge/common/ticket/ChunkTicketManager.java
@@ -26,31 +26,26 @@
 import java.util.Set;
 import java.util.WeakHashMap;
 
-public class ChunkTicketManager<T> implements ITicketGetter<T>
-{
-    private final Set<SimpleTicket<T>> tickets = Collections.newSetFromMap(new WeakHashMap<>());
+public class ChunkTicketManager<T> implements ITicketGetter<T> {
     public final ChunkPos pos;
+    private final Set<SimpleTicket<T>> tickets = Collections.newSetFromMap(new WeakHashMap<>());
 
-    public ChunkTicketManager(ChunkPos pos)
-    {
+    public ChunkTicketManager(ChunkPos pos) {
         this.pos = pos;
     }
 
     @Override
-    public void add(SimpleTicket<T> ticket)
-    {
+    public void add(SimpleTicket<T> ticket) {
         this.tickets.add(ticket);
     }
 
     @Override
-    public void remove(SimpleTicket<T> ticket)
-    {
+    public void remove(SimpleTicket<T> ticket) {
         this.tickets.remove(ticket);
     }
 
     @Override
-    public Collection<SimpleTicket<T>> getTickets()
-    {
+    public Collection<SimpleTicket<T>> getTickets() {
         return tickets;
     }
 }
