--- ../src-base/minecraft/net/minecraftforge/common/ticket/MultiTicketManager.java
+++ ../src-work/minecraft/net/minecraftforge/common/ticket/MultiTicketManager.java
@@ -19,26 +19,22 @@
 
 package net.minecraftforge.common.ticket;
 
-public class MultiTicketManager<T> implements ITicketManager<T>
-{
+public class MultiTicketManager<T> implements ITicketManager<T> {
     private final ITicketGetter<T>[] ticketManagers;
 
     @SafeVarargs
-    public MultiTicketManager(ITicketGetter<T>... ticketManagers)
-    {
+    public MultiTicketManager(ITicketGetter<T>... ticketManagers) {
         this.ticketManagers = ticketManagers;
     }
 
     @Override
-    public void add(SimpleTicket<T> ticket)
-    {
+    public void add(SimpleTicket<T> ticket) {
         for (ITicketGetter<T> manager : ticketManagers)
             manager.add(ticket);
     }
 
     @Override
-    public void remove(SimpleTicket<T> ticket)
-    {
+    public void remove(SimpleTicket<T> ticket) {
         for (ITicketGetter<T> manager : ticketManagers)
             manager.remove(ticket);
     }
