--- ../src-base/minecraft/net/minecraft/world/storage/loot/properties/EntityOnFire.java
+++ ../src-work/minecraft/net/minecraft/world/storage/loot/properties/EntityOnFire.java
@@ -4,11 +4,12 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonPrimitive;
 import com.google.gson.JsonSerializationContext;
-import java.util.Random;
 import net.minecraft.entity.Entity;
 import net.minecraft.util.JsonUtils;
 import net.minecraft.util.ResourceLocation;
 
+import java.util.Random;
+
 public class EntityOnFire implements EntityProperty
 {
     private final boolean onFire;
