--- ../src-base/minecraft/net/minecraft/client/gui/toasts/RecipeToast.java
+++ ../src-work/minecraft/net/minecraft/client/gui/toasts/RecipeToast.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.gui.toasts;
 
 import com.google.common.collect.Lists;
-import java.util.List;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.RenderHelper;
 import net.minecraft.client.resources.I18n;
@@ -11,6 +10,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class RecipeToast implements IToast
 {
@@ -23,7 +24,7 @@
         this.recipesOutputs.add(p_i47489_1_);
     }
 
-    public IToast.Visibility draw(GuiToast toastGui, long delta)
+    public Visibility draw(GuiToast toastGui, long delta)
     {
         if (this.hasNewOutputs)
         {
@@ -33,7 +34,7 @@
 
         if (this.recipesOutputs.isEmpty())
         {
-            return IToast.Visibility.HIDE;
+            return Visibility.HIDE;
         }
         else
         {
@@ -44,7 +45,7 @@
             toastGui.getMinecraft().fontRenderer.drawString(I18n.format("recipe.toast.description"), 30, 18, -16777216);
             RenderHelper.enableGUIStandardItemLighting();
             toastGui.getMinecraft().getRenderItem().renderItemAndEffectIntoGUI((EntityLivingBase)null, this.recipesOutputs.get((int)(delta * (long)this.recipesOutputs.size() / 5000L % (long)this.recipesOutputs.size())), 8, 8); //Forge: fix math so that it doesn't divide by 0 when there are more than 5000 recipes
-            return delta - this.firstDrawTime >= 5000L ? IToast.Visibility.HIDE : IToast.Visibility.SHOW;
+            return delta - this.firstDrawTime >= 5000L ? Visibility.HIDE : Visibility.SHOW;
         }
     }
 
