--- ../src-base/minecraft/net/minecraft/client/shader/ShaderLoader.java
+++ ../src-work/minecraft/net/minecraft/client/shader/ShaderLoader.java
@@ -1,11 +1,6 @@
 package net.minecraft.client.shader;
 
 import com.google.common.collect.Maps;
-import java.io.BufferedInputStream;
-import java.io.Closeable;
-import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.util.Map;
 import net.minecraft.client.renderer.OpenGlHelper;
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.resources.IResourceManager;
@@ -17,15 +12,21 @@
 import org.apache.commons.lang3.StringUtils;
 import org.lwjgl.BufferUtils;
 
+import java.io.BufferedInputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
 public class ShaderLoader
 {
-    private final ShaderLoader.ShaderType shaderType;
+    private final ShaderType shaderType;
     private final String shaderFilename;
     private final int shader;
     private int shaderAttachCount;
 
-    private ShaderLoader(ShaderLoader.ShaderType type, int shaderId, String filename)
+    private ShaderLoader(ShaderType type, int shaderId, String filename)
     {
         this.shaderType = type;
         this.shader = shaderId;
@@ -54,7 +55,7 @@
         return this.shaderFilename;
     }
 
-    public static ShaderLoader loadShader(IResourceManager resourceManager, ShaderLoader.ShaderType type, String filename) throws IOException
+    public static ShaderLoader loadShader(IResourceManager resourceManager, ShaderType type, String filename) throws IOException
     {
         ShaderLoader shaderloader = (ShaderLoader)type.getLoadedShaders().get(filename);
 
