--- ../src-base/minecraft/net/minecraftforge/server/terminalconsole/MinecraftFormattingConverter.java
+++ ../src-work/minecraft/net/minecraftforge/server/terminalconsole/MinecraftFormattingConverter.java
@@ -23,20 +23,15 @@
 
 package net.minecraftforge.server.terminalconsole;
 
-import javax.annotation.Nullable;
-
 import org.apache.logging.log4j.core.LogEvent;
 import org.apache.logging.log4j.core.config.Configuration;
 import org.apache.logging.log4j.core.config.plugins.Plugin;
 import org.apache.logging.log4j.core.layout.PatternLayout;
-import org.apache.logging.log4j.core.pattern.ConverterKeys;
-import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
-import org.apache.logging.log4j.core.pattern.PatternConverter;
-import org.apache.logging.log4j.core.pattern.PatternFormatter;
-import org.apache.logging.log4j.core.pattern.PatternParser;
+import org.apache.logging.log4j.core.pattern.*;
 import org.apache.logging.log4j.util.PerformanceSensitive;
 import org.apache.logging.log4j.util.PropertiesUtil;
 
+import javax.annotation.Nullable;
 import java.util.List;
 
 /**
