--- ../src-base/minecraft/net/minecraftforge/server/terminalconsole/MinecraftFormattingConverter.java
+++ ../src-work/minecraft/net/minecraftforge/server/terminalconsole/MinecraftFormattingConverter.java
@@ -76,7 +76,7 @@
      * {@code true}, all Minecraft formatting codes will be replaced
      * or stripped from the console output.</p>
      */
-    public static final String KEEP_FORMATTING_PROPERTY = TerminalConsoleAppender.PROPERTY_PREFIX + ".keepMinecraftFormatting";
+    public static final String KEEP_FORMATTING_PROPERTY = "terminal.keepMinecraftFormatting";
 
     private static final boolean KEEP_FORMATTING = PropertiesUtil.getProperties().getBooleanProperty(KEEP_FORMATTING_PROPERTY);
 
@@ -143,7 +143,7 @@
         }
 
         String content = toAppendTo.substring(start);
-        format(content, toAppendTo, start, ansi && TerminalConsoleAppender.isAnsiSupported());
+        format(content, toAppendTo, start, ansi && true);
     }
 
     private static void format(String s, StringBuilder result, int start, boolean ansi)
