--- ../src-base/minecraft/net/minecraftforge/server/terminalconsole/package-info.java
+++ ../src-work/minecraft/net/minecraftforge/server/terminalconsole/package-info.java
@@ -2,6 +2,6 @@
 @MethodsReturnNonnullByDefault
 package net.minecraftforge.server.terminalconsole;
 
-import javax.annotation.ParametersAreNonnullByDefault;
-
 import mcp.MethodsReturnNonnullByDefault;
+
+import javax.annotation.ParametersAreNonnullByDefault;
