--- ../src-base/minecraft/net/minecraft/world/chunk/BlockStatePaletteLinear.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/BlockStatePaletteLinear.java
@@ -1,12 +1,13 @@
 package net.minecraft.world.chunk;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.network.PacketBuffer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public class BlockStatePaletteLinear implements IBlockStatePalette
 {
     private final IBlockState[] states;
