--- ../src-base/minecraft/net/minecraft/advancements/AdvancementRewards.java
+++ ../src-work/minecraft/net/minecraft/advancements/AdvancementRewards.java
@@ -1,14 +1,6 @@
 package net.minecraft.advancements;
 
-import com.google.gson.JsonArray;
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonDeserializer;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParseException;
-import com.google.gson.JsonSyntaxException;
-import java.lang.reflect.Type;
-import java.util.Arrays;
+import com.google.gson.*;
 import net.minecraft.command.CommandResultStats;
 import net.minecraft.command.FunctionObject;
 import net.minecraft.command.ICommandSender;
@@ -30,6 +22,9 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.loot.LootContext;
 
+import java.lang.reflect.Type;
+import java.util.Arrays;
+
 public class AdvancementRewards
 {
     public static final AdvancementRewards EMPTY = new AdvancementRewards(0, new ResourceLocation[0], new ResourceLocation[0], FunctionObject.CacheableFunction.EMPTY);
@@ -89,56 +84,67 @@
 
         if (functionobject != null)
         {
-            ICommandSender icommandsender = new ICommandSender()
-            {
-                public String getName()
-                {
-                    return player.getName();
-                }
-                public ITextComponent getDisplayName()
-                {
-                    return player.getDisplayName();
-                }
-                public void sendMessage(ITextComponent component)
-                {
-                }
-                public boolean canUseCommand(int permLevel, String commandName)
-                {
-                    return permLevel <= 2;
-                }
-                public BlockPos getPosition()
-                {
-                    return player.getPosition();
-                }
-                public Vec3d getPositionVector()
-                {
-                    return player.getPositionVector();
-                }
-                public World getEntityWorld()
-                {
-                    return player.world;
-                }
-                public Entity getCommandSenderEntity()
-                {
-                    return player;
-                }
-                public boolean sendCommandFeedback()
-                {
-                    return minecraftserver.worlds[0].getGameRules().getBoolean("commandBlockOutput");
-                }
-                public void setCommandStat(CommandResultStats.Type type, int amount)
-                {
-                    player.setCommandStat(type, amount);
-                }
-                public MinecraftServer getServer()
-                {
-                    return player.getServer();
-                }
-            };
+            ICommandSender icommandsender = new AdvancementCommandListener(player, minecraftserver);
+
             minecraftserver.getFunctionManager().execute(functionobject, icommandsender);
         }
+
     }
 
+    public static class AdvancementCommandListener implements ICommandSender {
+
+        private final EntityPlayerMP player;
+        private final MinecraftServer minecraftserver;
+
+        public AdvancementCommandListener(EntityPlayerMP entityplayer, MinecraftServer minecraftserver) {
+            this.player = entityplayer;
+            this.minecraftserver = minecraftserver;
+        }
+
+        public String getName() {
+            return player.getName();
+        }
+
+        public ITextComponent getDisplayName() {
+            return player.getDisplayName();
+        }
+
+        public void sendMessage(ITextComponent component) {
+        }
+
+        public boolean canUseCommand(int permLevel, String commandName) {
+            return permLevel <= 2;
+        }
+
+        public BlockPos getPosition() {
+            return player.getPosition();
+        }
+
+        public Vec3d getPositionVector() {
+            return player.getPositionVector();
+        }
+
+        public World getEntityWorld() {
+            return player.world;
+        }
+
+        public Entity getCommandSenderEntity() {
+            return player;
+        }
+
+        public boolean sendCommandFeedback() {
+            return minecraftserver.worlds[0].getGameRules().getBoolean("commandBlockOutput");
+        }
+
+        public void setCommandStat(CommandResultStats.Type type, int amount) {
+            player.setCommandStat(type, amount);
+        }
+
+        public MinecraftServer getServer() {
+            return player.getServer();
+        }
+    }
+
     public String toString()
     {
         return "AdvancementRewards{experience=" + this.experience + ", loot=" + Arrays.toString((Object[])this.loot) + ", recipes=" + Arrays.toString((Object[])this.recipes) + ", function=" + this.function + '}';
