--- ../src-base/minecraft/net/minecraft/advancements/Advancement.java
+++ ../src-work/minecraft/net/minecraft/advancements/Advancement.java
@@ -6,12 +6,6 @@
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSyntaxException;
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Map;
-import java.util.Set;
-import java.util.function.Function;
-import javax.annotation.Nullable;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.util.JsonUtils;
 import net.minecraft.util.ResourceLocation;
@@ -22,8 +16,15 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.lang3.ArrayUtils;
 
-public class Advancement
-{
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+public class Advancement {
+    public final org.bukkit.advancement.Advancement bukkit = new org.bukkit.craftbukkit.v1_12_R1.advancement.CraftAdvancement(this); // CraftBukkit
     private final Advancement parent;
     private final DisplayInfo display;
     private final AdvancementRewards rewards;
@@ -33,8 +34,7 @@
     private final Set<Advancement> children = Sets.<Advancement>newLinkedHashSet();
     private final ITextComponent displayText;
 
-    public Advancement(ResourceLocation id, @Nullable Advancement parentIn, @Nullable DisplayInfo displayIn, AdvancementRewards rewardsIn, Map<String, Criterion> criteriaIn, String[][] requirementsIn)
-    {
+    public Advancement(ResourceLocation id, @Nullable Advancement parentIn, @Nullable DisplayInfo displayIn, AdvancementRewards rewardsIn, Map<String, Criterion> criteriaIn, String[][] requirementsIn) {
         this.id = id;
         this.display = displayIn;
         this.criteria = ImmutableMap.copyOf(criteriaIn);
@@ -42,17 +42,13 @@
         this.rewards = rewardsIn;
         this.requirements = requirementsIn;
 
-        if (parentIn != null)
-        {
+        if (parentIn != null) {
             parentIn.addChild(this);
         }
 
-        if (displayIn == null)
-        {
+        if (displayIn == null) {
             this.displayText = new TextComponentString(id.toString());
-        }
-        else
-        {
+        } else {
             this.displayText = new TextComponentString("[");
             this.displayText.getStyle().setColor(displayIn.getFrame().getFormat());
             ITextComponent itextcomponent = displayIn.getTitle().createCopy();
@@ -68,263 +64,208 @@
         }
     }
 
-    public Advancement.Builder copy()
-    {
+    public Advancement.Builder copy() {
         return new Advancement.Builder(this.parent == null ? null : this.parent.getId(), this.display, this.rewards, this.criteria, this.requirements);
     }
 
     @Nullable
-    public Advancement getParent()
-    {
+    public Advancement getParent() {
         return this.parent;
     }
 
     @Nullable
-    public DisplayInfo getDisplay()
-    {
+    public DisplayInfo getDisplay() {
         return this.display;
     }
 
-    public AdvancementRewards getRewards()
-    {
+    public AdvancementRewards getRewards() {
         return this.rewards;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return "SimpleAdvancement{id=" + this.getId() + ", parent=" + (this.parent == null ? "null" : this.parent.getId()) + ", display=" + this.display + ", rewards=" + this.rewards + ", criteria=" + this.criteria + ", requirements=" + Arrays.deepToString(this.requirements) + '}';
     }
 
-    public Iterable<Advancement> getChildren()
-    {
+    public Iterable<Advancement> getChildren() {
         return this.children;
     }
 
-    public Map<String, Criterion> getCriteria()
-    {
+    public Map<String, Criterion> getCriteria() {
         return this.criteria;
     }
 
     @SideOnly(Side.CLIENT)
-    public int getRequirementCount()
-    {
+    public int getRequirementCount() {
         return this.requirements.length;
     }
 
-    public void addChild(Advancement advancementIn)
-    {
+    public void addChild(Advancement advancementIn) {
         this.children.add(advancementIn);
     }
 
-    public ResourceLocation getId()
-    {
+    public ResourceLocation getId() {
         return this.id;
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        if (this == p_equals_1_)
-        {
+    public boolean equals(Object p_equals_1_) {
+        if (this == p_equals_1_) {
             return true;
-        }
-        else if (!(p_equals_1_ instanceof Advancement))
-        {
+        } else if (!(p_equals_1_ instanceof Advancement)) {
             return false;
-        }
-        else
-        {
-            Advancement advancement = (Advancement)p_equals_1_;
+        } else {
+            Advancement advancement = (Advancement) p_equals_1_;
             return this.id.equals(advancement.id);
         }
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return this.id.hashCode();
     }
 
-    public String[][] getRequirements()
-    {
+    public String[][] getRequirements() {
         return this.requirements;
     }
 
-    public ITextComponent getDisplayText()
-    {
+    public ITextComponent getDisplayText() {
         return this.displayText;
     }
 
-    public static class Builder
-        {
-            private final ResourceLocation parentId;
-            private Advancement parent;
-            private final DisplayInfo display;
-            private final AdvancementRewards rewards;
-            private final Map<String, Criterion> criteria;
-            private final String[][] requirements;
+    public static class Builder {
+        private final ResourceLocation parentId;
+        private final DisplayInfo display;
+        private final AdvancementRewards rewards;
+        private final Map<String, Criterion> criteria;
+        private final String[][] requirements;
+        private Advancement parent;
 
-            Builder(@Nullable ResourceLocation p_i47414_1_, @Nullable DisplayInfo p_i47414_2_, AdvancementRewards p_i47414_3_, Map<String, Criterion> p_i47414_4_, String[][] p_i47414_5_)
-            {
-                this.parentId = p_i47414_1_;
-                this.display = p_i47414_2_;
-                this.rewards = p_i47414_3_;
-                this.criteria = p_i47414_4_;
-                this.requirements = p_i47414_5_;
-            }
+        Builder(@Nullable ResourceLocation p_i47414_1_, @Nullable DisplayInfo p_i47414_2_, AdvancementRewards p_i47414_3_, Map<String, Criterion> p_i47414_4_, String[][] p_i47414_5_) {
+            this.parentId = p_i47414_1_;
+            this.display = p_i47414_2_;
+            this.rewards = p_i47414_3_;
+            this.criteria = p_i47414_4_;
+            this.requirements = p_i47414_5_;
+        }
 
-            public boolean resolveParent(Function<ResourceLocation, Advancement> lookup)
-            {
-                if (this.parentId == null)
-                {
-                    return true;
-                }
-                else
-                {
-                    this.parent = lookup.apply(this.parentId);
-                    return this.parent != null;
-                }
-            }
+        public static Advancement.Builder deserialize(JsonObject json, JsonDeserializationContext context) {
+            ResourceLocation resourcelocation = json.has("parent") ? new ResourceLocation(JsonUtils.getString(json, "parent")) : null;
+            DisplayInfo displayinfo = json.has("display") ? DisplayInfo.deserialize(JsonUtils.getJsonObject(json, "display"), context) : null;
+            AdvancementRewards advancementrewards = (AdvancementRewards) JsonUtils.deserializeClass(json, "rewards", AdvancementRewards.EMPTY, context, AdvancementRewards.class);
+            Map<String, Criterion> map = Criterion.criteriaFromJson(JsonUtils.getJsonObject(json, "criteria"), context);
 
-            public Advancement build(ResourceLocation id)
-            {
-                return new Advancement(id, this.parent, this.display, this.rewards, this.criteria, this.requirements);
-            }
+            if (map.isEmpty()) {
+                throw new JsonSyntaxException("Advancement criteria cannot be empty");
+            } else {
+                JsonArray jsonarray = JsonUtils.getJsonArray(json, "requirements", new JsonArray());
+                String[][] astring = new String[jsonarray.size()][];
 
-            public void writeTo(PacketBuffer buf)
-            {
-                if (this.parentId == null)
-                {
-                    buf.writeBoolean(false);
-                }
-                else
-                {
-                    buf.writeBoolean(true);
-                    buf.writeResourceLocation(this.parentId);
-                }
+                for (int i = 0; i < jsonarray.size(); ++i) {
+                    JsonArray jsonarray1 = JsonUtils.getJsonArray(jsonarray.get(i), "requirements[" + i + "]");
+                    astring[i] = new String[jsonarray1.size()];
 
-                if (this.display == null)
-                {
-                    buf.writeBoolean(false);
+                    for (int j = 0; j < jsonarray1.size(); ++j) {
+                        astring[i][j] = JsonUtils.getString(jsonarray1.get(j), "requirements[" + i + "][" + j + "]");
+                    }
                 }
-                else
-                {
-                    buf.writeBoolean(true);
-                    this.display.write(buf);
-                }
 
-                Criterion.serializeToNetwork(this.criteria, buf);
-                buf.writeVarInt(this.requirements.length);
+                if (astring.length == 0) {
+                    astring = new String[map.size()][];
+                    int k = 0;
 
-                for (String[] astring : this.requirements)
-                {
-                    buf.writeVarInt(astring.length);
-
-                    for (String s : astring)
-                    {
-                        buf.writeString(s);
+                    for (String s2 : map.keySet()) {
+                        astring[k++] = new String[]{s2};
                     }
                 }
-            }
 
-            public String toString()
-            {
-                return "Task Advancement{parentId=" + this.parentId + ", display=" + this.display + ", rewards=" + this.rewards + ", criteria=" + this.criteria + ", requirements=" + Arrays.deepToString(this.requirements) + '}';
-            }
+                for (String[] astring1 : astring) {
+                    if (astring1.length == 0 && map.isEmpty()) {
+                        throw new JsonSyntaxException("Requirement entry cannot be empty");
+                    }
 
-            public static Advancement.Builder deserialize(JsonObject json, JsonDeserializationContext context)
-            {
-                ResourceLocation resourcelocation = json.has("parent") ? new ResourceLocation(JsonUtils.getString(json, "parent")) : null;
-                DisplayInfo displayinfo = json.has("display") ? DisplayInfo.deserialize(JsonUtils.getJsonObject(json, "display"), context) : null;
-                AdvancementRewards advancementrewards = (AdvancementRewards)JsonUtils.deserializeClass(json, "rewards", AdvancementRewards.EMPTY, context, AdvancementRewards.class);
-                Map<String, Criterion> map = Criterion.criteriaFromJson(JsonUtils.getJsonObject(json, "criteria"), context);
-
-                if (map.isEmpty())
-                {
-                    throw new JsonSyntaxException("Advancement criteria cannot be empty");
-                }
-                else
-                {
-                    JsonArray jsonarray = JsonUtils.getJsonArray(json, "requirements", new JsonArray());
-                    String[][] astring = new String[jsonarray.size()][];
-
-                    for (int i = 0; i < jsonarray.size(); ++i)
-                    {
-                        JsonArray jsonarray1 = JsonUtils.getJsonArray(jsonarray.get(i), "requirements[" + i + "]");
-                        astring[i] = new String[jsonarray1.size()];
-
-                        for (int j = 0; j < jsonarray1.size(); ++j)
-                        {
-                            astring[i][j] = JsonUtils.getString(jsonarray1.get(j), "requirements[" + i + "][" + j + "]");
+                    for (String s : astring1) {
+                        if (!map.containsKey(s)) {
+                            throw new JsonSyntaxException("Unknown required criterion '" + s + "'");
                         }
                     }
+                }
 
-                    if (astring.length == 0)
-                    {
-                        astring = new String[map.size()][];
-                        int k = 0;
+                for (String s1 : map.keySet()) {
+                    boolean flag = false;
 
-                        for (String s2 : map.keySet())
-                        {
-                            astring[k++] = new String[] {s2};
+                    for (String[] astring2 : astring) {
+                        if (ArrayUtils.contains(astring2, s1)) {
+                            flag = true;
+                            break;
                         }
                     }
 
-                    for (String[] astring1 : astring)
-                    {
-                        if (astring1.length == 0 && map.isEmpty())
-                        {
-                            throw new JsonSyntaxException("Requirement entry cannot be empty");
-                        }
-
-                        for (String s : astring1)
-                        {
-                            if (!map.containsKey(s))
-                            {
-                                throw new JsonSyntaxException("Unknown required criterion '" + s + "'");
-                            }
-                        }
+                    if (!flag) {
+                        throw new JsonSyntaxException("Criterion '" + s1 + "' isn't a requirement for completion. This isn't supported behaviour, all criteria must be required.");
                     }
+                }
 
-                    for (String s1 : map.keySet())
-                    {
-                        boolean flag = false;
+                return new Advancement.Builder(resourcelocation, displayinfo, advancementrewards, map, astring);
+            }
+        }
 
-                        for (String[] astring2 : astring)
-                        {
-                            if (ArrayUtils.contains(astring2, s1))
-                            {
-                                flag = true;
-                                break;
-                            }
-                        }
+        public static Advancement.Builder readFrom(PacketBuffer buf) throws IOException {
+            ResourceLocation resourcelocation = buf.readBoolean() ? buf.readResourceLocation() : null;
+            DisplayInfo displayinfo = buf.readBoolean() ? DisplayInfo.read(buf) : null;
+            Map<String, Criterion> map = Criterion.criteriaFromNetwork(buf);
+            String[][] astring = new String[buf.readVarInt()][];
 
-                        if (!flag)
-                        {
-                            throw new JsonSyntaxException("Criterion '" + s1 + "' isn't a requirement for completion. This isn't supported behaviour, all criteria must be required.");
-                        }
-                    }
+            for (int i = 0; i < astring.length; ++i) {
+                astring[i] = new String[buf.readVarInt()];
 
-                    return new Advancement.Builder(resourcelocation, displayinfo, advancementrewards, map, astring);
+                for (int j = 0; j < astring[i].length; ++j) {
+                    astring[i][j] = buf.readString(32767);
                 }
             }
 
-            public static Advancement.Builder readFrom(PacketBuffer buf) throws IOException
-            {
-                ResourceLocation resourcelocation = buf.readBoolean() ? buf.readResourceLocation() : null;
-                DisplayInfo displayinfo = buf.readBoolean() ? DisplayInfo.read(buf) : null;
-                Map<String, Criterion> map = Criterion.criteriaFromNetwork(buf);
-                String[][] astring = new String[buf.readVarInt()][];
+            return new Advancement.Builder(resourcelocation, displayinfo, AdvancementRewards.EMPTY, map, astring);
+        }
 
-                for (int i = 0; i < astring.length; ++i)
-                {
-                    astring[i] = new String[buf.readVarInt()];
+        public boolean resolveParent(Function<ResourceLocation, Advancement> lookup) {
+            if (this.parentId == null) {
+                return true;
+            } else {
+                this.parent = lookup.apply(this.parentId);
+                return this.parent != null;
+            }
+        }
 
-                    for (int j = 0; j < astring[i].length; ++j)
-                    {
-                        astring[i][j] = buf.readString(32767);
-                    }
-                }
+        public Advancement build(ResourceLocation id) {
+            return new Advancement(id, this.parent, this.display, this.rewards, this.criteria, this.requirements);
+        }
 
-                return new Advancement.Builder(resourcelocation, displayinfo, AdvancementRewards.EMPTY, map, astring);
+        public void writeTo(PacketBuffer buf) {
+            if (this.parentId == null) {
+                buf.writeBoolean(false);
+            } else {
+                buf.writeBoolean(true);
+                buf.writeResourceLocation(this.parentId);
             }
+
+            if (this.display == null) {
+                buf.writeBoolean(false);
+            } else {
+                buf.writeBoolean(true);
+                this.display.write(buf);
+            }
+
+            Criterion.serializeToNetwork(this.criteria, buf);
+            buf.writeVarInt(this.requirements.length);
+
+            for (String[] astring : this.requirements) {
+                buf.writeVarInt(astring.length);
+
+                for (String s : astring) {
+                    buf.writeString(s);
+                }
+            }
         }
+
+        public String toString() {
+            return "Task Advancement{parentId=" + this.parentId + ", display=" + this.display + ", rewards=" + this.rewards + ", criteria=" + this.criteria + ", requirements=" + Arrays.deepToString(this.requirements) + '}';
+        }
+    }
 }
