--- ../src-base/minecraft/net/minecraft/advancements/AdvancementList.java
+++ ../src-work/minecraft/net/minecraft/advancements/AdvancementList.java
@@ -3,25 +3,26 @@
 import com.google.common.base.Functions;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Set;
-import java.util.Map.Entry;
-import java.util.function.Function;
-import javax.annotation.Nullable;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+
 public class AdvancementList
 {
     private static final Logger LOGGER = LogManager.getLogger();
-    private final Map<ResourceLocation, Advancement> advancements = Maps.<ResourceLocation, Advancement>newHashMap();
+    public final Map<ResourceLocation, Advancement> advancements = Maps.<ResourceLocation, Advancement>newHashMap();
     private final Set<Advancement> roots = Sets.<Advancement>newLinkedHashSet();
     private final Set<Advancement> nonRoots = Sets.<Advancement>newLinkedHashSet();
-    private AdvancementList.Listener listener;
+    private Listener listener;
 
     @SideOnly(Side.CLIENT)
     private void remove(Advancement advancementIn)
@@ -84,7 +85,7 @@
 
             while (iterator.hasNext())
             {
-                Entry<ResourceLocation, Advancement.Builder> entry = (Entry)iterator.next();
+                Entry<ResourceLocation, Advancement.Builder> entry = iterator.next();
                 ResourceLocation resourcelocation = entry.getKey();
                 Advancement.Builder advancement$builder = entry.getValue();
 
@@ -133,7 +134,7 @@
             }
         }
 
-        LOGGER.info("Loaded " + this.advancements.size() + " advancements");
+       // LOGGER.info("Loaded " + this.advancements.size() + " advancements"); // CraftBukkit - moved to AdvancementDataWorld#reload
     }
 
     public void clear()
