--- ../src-base/minecraft/net/minecraft/advancements/AdvancementManager.java
+++ ../src-work/minecraft/net/minecraft/advancements/AdvancementManager.java
@@ -1,32 +1,7 @@
 package net.minecraft.advancements;
 
 import com.google.common.collect.Maps;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonDeserializer;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParseException;
-import java.io.BufferedReader;
-import java.io.Closeable;
-import java.io.File;
-import java.io.IOException;
-import java.io.Reader;
-import java.lang.reflect.Type;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.net.URL;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.FileSystem;
-import java.nio.file.FileSystems;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.Map;
-import javax.annotation.Nullable;
+import com.google.gson.*;
 import net.minecraft.item.crafting.CraftingManager;
 import net.minecraft.util.EnumTypeAdapterFactory;
 import net.minecraft.util.JsonUtils;
@@ -39,6 +14,19 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.*;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+
 public class AdvancementManager
 {
     private static final Logger LOGGER = LogManager.getLogger();
@@ -50,8 +38,8 @@
             return Advancement.Builder.deserialize(jsonobject, p_deserialize_3_);
         }
     }).registerTypeAdapter(AdvancementRewards.class, new AdvancementRewards.Deserializer()).registerTypeHierarchyAdapter(ITextComponent.class, new ITextComponent.Serializer()).registerTypeHierarchyAdapter(Style.class, new Style.Serializer()).registerTypeAdapterFactory(new EnumTypeAdapterFactory()).create();
-    private static final AdvancementList ADVANCEMENT_LIST = new AdvancementList();
-    private final File advancementsDir;
+    public static final AdvancementList ADVANCEMENT_LIST = new AdvancementList();
+    public final File advancementsDir;
     private boolean hasErrored;
 
     public AdvancementManager(@Nullable File advancementsDirIn)
@@ -173,7 +161,11 @@
                         Path path2 = path.relativize(path1);
                         String s = FilenameUtils.removeExtension(path2.toString()).replaceAll("\\\\", "/");
                         ResourceLocation resourcelocation = new ResourceLocation("minecraft", s);
-
+                        // Spigot start
+                        if (org.spigotmc.SpigotConfig.disabledAdvancements != null && (org.spigotmc.SpigotConfig.disabledAdvancements.contains("*") || org.spigotmc.SpigotConfig.disabledAdvancements.contains(resourcelocation.toString()))) {
+                            continue;
+                        }
+                        // Spigot end
                         if (!map.containsKey(resourcelocation))
                         {
                             BufferedReader bufferedreader = null;
