--- ../src-base/minecraft/net/minecraft/advancements/FrameType.java
+++ ../src-work/minecraft/net/minecraft/advancements/FrameType.java
@@ -4,8 +4,7 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public enum FrameType
-{
+public enum FrameType {
     TASK("task", 0, TextFormatting.GREEN),
     CHALLENGE("challenge", 26, TextFormatting.DARK_PURPLE),
     GOAL("goal", 52, TextFormatting.GREEN);
@@ -14,24 +13,15 @@
     private final int icon;
     private final TextFormatting format;
 
-    private FrameType(String nameIn, int iconIn, TextFormatting formatIn)
-    {
+    private FrameType(String nameIn, int iconIn, TextFormatting formatIn) {
         this.name = nameIn;
         this.icon = iconIn;
         this.format = formatIn;
     }
 
-    public String getName()
-    {
-        return this.name;
-    }
-
-    public static FrameType byName(String nameIn)
-    {
-        for (FrameType frametype : values())
-        {
-            if (frametype.name.equals(nameIn))
-            {
+    public static FrameType byName(String nameIn) {
+        for (FrameType frametype : values()) {
+            if (frametype.name.equals(nameIn)) {
                 return frametype;
             }
         }
@@ -39,14 +29,16 @@
         throw new IllegalArgumentException("Unknown frame type '" + nameIn + "'");
     }
 
+    public String getName() {
+        return this.name;
+    }
+
     @SideOnly(Side.CLIENT)
-    public int getIcon()
-    {
+    public int getIcon() {
         return this.icon;
     }
 
-    public TextFormatting getFormat()
-    {
+    public TextFormatting getFormat() {
         return this.format;
     }
 }
