--- ../src-base/minecraft/net/minecraft/realms/RealmsBufferBuilder.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsBufferBuilder.java
@@ -1,11 +1,12 @@
 package net.minecraft.realms;
 
-import java.nio.ByteBuffer;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.vertex.VertexFormat;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.ByteBuffer;
+
 @SideOnly(Side.CLIENT)
 public class RealmsBufferBuilder
 {
