--- ../src-base/minecraft/net/minecraft/realms/RealmsVertexFormatElement.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsVertexFormatElement.java
@@ -5,52 +5,42 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RealmsVertexFormatElement
-{
+public class RealmsVertexFormatElement {
     private final VertexFormatElement v;
 
-    public RealmsVertexFormatElement(VertexFormatElement vIn)
-    {
+    public RealmsVertexFormatElement(VertexFormatElement vIn) {
         this.v = vIn;
     }
 
-    public VertexFormatElement getVertexFormatElement()
-    {
+    public VertexFormatElement getVertexFormatElement() {
         return this.v;
     }
 
-    public boolean isPosition()
-    {
+    public boolean isPosition() {
         return this.v.isPositionElement();
     }
 
-    public int getIndex()
-    {
+    public int getIndex() {
         return this.v.getIndex();
     }
 
-    public int getByteSize()
-    {
+    public int getByteSize() {
         return this.v.getSize();
     }
 
-    public int getCount()
-    {
+    public int getCount() {
         return this.v.getElementCount();
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return this.v.hashCode();
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
+    public boolean equals(Object p_equals_1_) {
         return this.v.equals(p_equals_1_);
     }
 
-    public String toString()
-    {
+    public String toString() {
         return this.v.toString();
     }
 }
