--- ../src-base/minecraft/net/minecraft/realms/RealmsScreen.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsScreen.java
@@ -1,7 +1,6 @@
 package net.minecraft.realms;
 
 import com.mojang.util.UUIDTypeAdapter;
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.entity.AbstractClientPlayer;
 import net.minecraft.client.gui.Gui;
@@ -13,6 +12,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class RealmsScreen
 {
