--- ../src-base/minecraft/net/minecraft/realms/RealmsDefaultVertexFormat.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsDefaultVertexFormat.java
@@ -6,8 +6,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RealmsDefaultVertexFormat
-{
+public class RealmsDefaultVertexFormat {
     public static final RealmsVertexFormat BLOCK = new RealmsVertexFormat(new VertexFormat());
     public static final RealmsVertexFormat BLOCK_NORMALS = new RealmsVertexFormat(new VertexFormat());
     public static final RealmsVertexFormat ENTITY = new RealmsVertexFormat(new VertexFormat());
@@ -27,8 +26,7 @@
     public static final RealmsVertexFormatElement ELEMENT_NORMAL = new RealmsVertexFormatElement(new VertexFormatElement(0, VertexFormatElement.EnumType.BYTE, VertexFormatElement.EnumUsage.NORMAL, 3));
     public static final RealmsVertexFormatElement ELEMENT_PADDING = new RealmsVertexFormatElement(new VertexFormatElement(0, VertexFormatElement.EnumType.BYTE, VertexFormatElement.EnumUsage.PADDING, 1));
 
-    static
-    {
+    static {
         BLOCK.addElement(ELEMENT_POSITION);
         BLOCK.addElement(ELEMENT_COLOR);
         BLOCK.addElement(ELEMENT_UV0);
