--- ../src-base/minecraft/net/minecraft/realms/RealmsBridge.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsBridge.java
@@ -1,6 +1,5 @@
 package net.minecraft.realms;
 
-import java.lang.reflect.Constructor;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.gui.GuiScreenRealmsProxy;
@@ -9,6 +8,8 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.lang.reflect.Constructor;
+
 @SideOnly(Side.CLIENT)
 public class RealmsBridge extends RealmsScreen
 {
