--- ../src-base/minecraft/net/minecraft/realms/RealmsAnvilLevelStorageSource.java
+++ ../src-work/minecraft/net/minecraft/realms/RealmsAnvilLevelStorageSource.java
@@ -1,7 +1,6 @@
 package net.minecraft.realms;
 
 import com.google.common.collect.Lists;
-import java.util.List;
 import net.minecraft.client.AnvilConverterException;
 import net.minecraft.util.IProgressUpdate;
 import net.minecraft.world.storage.ISaveFormat;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class RealmsAnvilLevelStorageSource
 {
