--- ../src-base/minecraft/net/minecraftforge/fml/repackage/com/nothome/delta/ByteBufferSeekableSource.java
+++ ../src-work/minecraft/net/minecraftforge/fml/repackage/com/nothome/delta/ByteBufferSeekableSource.java
@@ -11,7 +11,7 @@
  * and/or sell copies of the Software, and to permit persons to whom the
  * Software is furnished to do so, subject to the following conditions:
  *
- * The above copyright notice and this permission notice shall be included in 
+ * The above copyright notice and this permission notice shall be included in
  * all copies or substantial portions of the Software.
  *
  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
@@ -33,17 +33,17 @@
  * Wraps a byte buffer as a source
  */
 public class ByteBufferSeekableSource implements SeekableSource {
-    
+
     private ByteBuffer bb;
     private ByteBuffer cur;
-    
+
     /**
      * Constructs a new ByteArraySeekableSource.
      */
     public ByteBufferSeekableSource(byte[] source) {
         this(ByteBuffer.wrap(source));
     }
-    
+
     /**
      * Constructs a new ByteArraySeekableSource.
      */
@@ -58,7 +58,7 @@
             throw new RuntimeException(e);
         }
     }
-    
+
     @Override
     public void seek(long pos) throws IOException {
         cur = bb.slice();
@@ -66,7 +66,7 @@
             throw new IOException("pos " + pos + " cannot seek " + cur.limit());
         cur.position((int) pos);
     }
-    
+
     @Override
     public int read(ByteBuffer dest) throws IOException {
         if (!cur.hasRemaining())
@@ -78,7 +78,7 @@
         }
         return c;
     }
-    
+
     @Override
     public void close() throws IOException {
         bb = null;
@@ -89,12 +89,11 @@
      * Returns a debug <code>String</code>.
      */
     @Override
-    public String toString()
-    {
+    public String toString() {
         return "BBSeekable" +
-            " bb=" + this.bb.position() + "-" + bb.limit() +
-            " cur=" + this.cur.position() + "-" + cur.limit() +
-            "";
+                " bb=" + this.bb.position() + "-" + bb.limit() +
+                " cur=" + this.cur.position() + "-" + cur.limit() +
+                "";
     }
-    
+
 }
