--- ../src-base/minecraft/net/minecraftforge/fml/repackage/com/nothome/delta/DiffWriter.java
+++ ../src-work/minecraft/net/minecraftforge/fml/repackage/com/nothome/delta/DiffWriter.java
@@ -41,11 +41,11 @@
      * Add a GDIFF data instruction.
      * Implementors should buffer the data.
      */
-	public void addData(byte b) throws IOException;
+    public void addData(byte b) throws IOException;
 
-	/**
-	 * Flushes to output, e.g. any data added.
-	 */
+    /**
+     * Flushes to output, e.g. any data added.
+     */
     public void flush() throws IOException;
 
     /**
