--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/libraries/MemoryModList.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/libraries/MemoryModList.java
@@ -21,19 +21,17 @@
 
 import java.io.IOException;
 
-public class MemoryModList extends ModList
-{
-    MemoryModList(Repository repo)
-    {
+public class MemoryModList extends ModList {
+    MemoryModList(Repository repo) {
         super(repo);
     }
 
     @Override
-    public void save() throws IOException {}
+    public void save() throws IOException {
+    }
 
     @Override
-    public String getName()
-    {
+    public String getName() {
         return "MEMORY";
     }
 }
