--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/libraries/SnapshotJson.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/libraries/SnapshotJson.java
@@ -16,9 +16,14 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
-
 package net.minecraftforge.fml.relauncher.libraries;
 
+import cn.pfcraft.server.Mohist;
+import com.google.common.io.Files;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
 import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
@@ -28,13 +33,6 @@
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
-
-import com.google.common.io.Files;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonSyntaxException;
-
-import net.minecraftforge.fml.common.FMLLog;
 /**
  * This is different from the standard maven snapshot metadata.
  * Because none of that data is exposed to us as a user of gradle/maven/whatever.
@@ -71,11 +69,11 @@
         }
         catch (JsonSyntaxException jse)
         {
-            FMLLog.log.info(FMLLog.log.getMessageFactory().newMessage("Failed to parse snapshot json file {}.", target), jse);
+            Mohist.LOGGER.info(Mohist.LOGGER.getMessageFactory().newMessage("Failed to parse snapshot json file {}.", target), jse);
         }
         catch (IOException ioe)
         {
-            FMLLog.log.info(FMLLog.log.getMessageFactory().newMessage("Failed to read snapshot json file {}.", target), ioe);
+            Mohist.LOGGER.info(Mohist.LOGGER.getMessageFactory().newMessage("Failed to read snapshot json file {}.", target), ioe);
         }
 
         return new SnapshotJson();
