--- ../src-base/minecraft/net/minecraft/client/gui/advancements/AdvancementState.java
+++ ../src-work/minecraft/net/minecraft/client/gui/advancements/AdvancementState.java
@@ -4,20 +4,17 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public enum AdvancementState
-{
+public enum AdvancementState {
     OBTAINED(0),
     UNOBTAINED(1);
 
     private final int id;
 
-    private AdvancementState(int id)
-    {
+    private AdvancementState(int id) {
         this.id = id;
     }
 
-    public int getId()
-    {
+    public int getId() {
         return this.id;
     }
 }
