--- ../src-base/minecraft/net/minecraft/client/gui/advancements/GuiAdvancementTab.java
+++ ../src-work/minecraft/net/minecraft/client/gui/advancements/GuiAdvancementTab.java
@@ -1,8 +1,6 @@
 package net.minecraft.client.gui.advancements;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.advancements.Advancement;
 import net.minecraft.advancements.DisplayInfo;
 import net.minecraft.client.Minecraft;
@@ -16,6 +14,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
 public class GuiAdvancementTab extends Gui
 {
