--- ../src-base/minecraft/net/minecraft/block/material/MaterialLogic.java
+++ ../src-work/minecraft/net/minecraft/block/material/MaterialLogic.java
@@ -1,25 +1,20 @@
 package net.minecraft.block.material;
 
-public class MaterialLogic extends Material
-{
-    public MaterialLogic(MapColor color)
-    {
+public class MaterialLogic extends Material {
+    public MaterialLogic(MapColor color) {
         super(color);
         this.setAdventureModeExempt();
     }
 
-    public boolean isSolid()
-    {
+    public boolean isSolid() {
         return false;
     }
 
-    public boolean blocksLight()
-    {
+    public boolean blocksLight() {
         return false;
     }
 
-    public boolean blocksMovement()
-    {
+    public boolean blocksMovement() {
         return false;
     }
 }
