--- ../src-base/minecraft/net/minecraft/block/material/MaterialTransparent.java
+++ ../src-work/minecraft/net/minecraft/block/material/MaterialTransparent.java
@@ -1,25 +1,20 @@
 package net.minecraft.block.material;
 
-public class MaterialTransparent extends Material
-{
-    public MaterialTransparent(MapColor color)
-    {
+public class MaterialTransparent extends Material {
+    public MaterialTransparent(MapColor color) {
         super(color);
         this.setReplaceable();
     }
 
-    public boolean isSolid()
-    {
+    public boolean isSolid() {
         return false;
     }
 
-    public boolean blocksLight()
-    {
+    public boolean blocksLight() {
         return false;
     }
 
-    public boolean blocksMovement()
-    {
+    public boolean blocksMovement() {
         return false;
     }
 }
