--- ../src-base/minecraft/net/minecraftforge/fml/common/gameevent/InputEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/gameevent/InputEvent.java
@@ -22,6 +22,9 @@
 import net.minecraftforge.fml.common.eventhandler.Event;
 
 public class InputEvent extends Event {
-    public static class MouseInputEvent extends InputEvent {}
-    public static class KeyInputEvent extends InputEvent {}
+    public static class MouseInputEvent extends InputEvent {
+    }
+
+    public static class KeyInputEvent extends InputEvent {
+    }
 }
