--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/CoreModManager.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/CoreModManager.java
@@ -223,7 +223,7 @@
             {
                 FMLLog.log.info("Detected deobfuscated environment, loading log configs for colored console logs.");
                 // use server logging configs in deobfuscated environment so developers get nicely colored console logs
-                System.setProperty("log4j.configurationFile", "log4j2_server.xml");
+                System.setProperty("log4j.configurationFile", "log4j2_mohist.xml");
                 ((LoggerContext) LogManager.getContext(false)).reconfigure();
             }
         }
