--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/FileListHelper.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/FileListHelper.java
@@ -26,24 +26,23 @@
 import java.util.Comparator;
 
 public final class FileListHelper {
-    private enum CaseInsensitiveFileComparator implements Comparator<File>
-    {
-        INSTANCE;
-        @Override
-        public int compare(File o1, File o2)
-        {
-            return o1 != null && o2 != null ? o1.getName().compareToIgnoreCase(o2.getName()) : o1 == null ? -1 : 1;
-        }
-
-    }
-    public static File[] sortFileList(File[] files)
-    {
+    public static File[] sortFileList(File[] files) {
         Arrays.sort(files, CaseInsensitiveFileComparator.INSTANCE);
         return files;
     }
-    public static File[] sortFileList(File dir, @Nullable FilenameFilter filter)
-    {
+
+    public static File[] sortFileList(File dir, @Nullable FilenameFilter filter) {
         File[] files = dir.listFiles(filter);
         return sortFileList(files);
     }
+
+    private enum CaseInsensitiveFileComparator implements Comparator<File> {
+        INSTANCE;
+
+        @Override
+        public int compare(File o1, File o2) {
+            return o1 != null && o2 != null ? o1.getName().compareToIgnoreCase(o2.getName()) : o1 == null ? -1 : 1;
+        }
+
+    }
 }
