--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/FMLSecurityManager.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/FMLSecurityManager.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -28,6 +28,7 @@
  * @author cpw
  *
  */
+
 public class FMLSecurityManager extends SecurityManager {
     @Override
     public void checkPermission(Permission perm)
@@ -43,7 +44,9 @@
                     || "net.minecraft.server.dedicated.ServerHangWatchdog$1".equals(callingClass)
                     || "net.minecraft.server.dedicated.ServerHangWatchdog".equals(callingClass)
                     || ( "net.minecraft.client.Minecraft".equals(callingClass) && "net.minecraft.client.Minecraft".equals(callingParent))
-                    || ("net.minecraft.server.dedicated.DedicatedServer".equals(callingClass) && "net.minecraft.server.MinecraftServer".equals(callingParent)))
+                    || ("net.minecraft.server.dedicated.DedicatedServer".equals(callingClass) && "net.minecraft.server.MinecraftServer".equals(callingParent))
+                    || "com.intellij.rt.execution.junit.JUnitStarter".equals(callingClass) // JUnitStarter is needed for test running
+                    || "worker.org.gradle.process.internal.worker.GradleWorkerMain".equals(callingClass)) // GradleWorkerMain is needed for test running from Gradle test task
                     )
             {
                 throw new ExitTrappedException();
