--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/IFMLLoadingPlugin.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/IFMLLoadingPlugin.java
@@ -33,8 +33,7 @@
  * @author cpw
  *
  */
-public interface IFMLLoadingPlugin
-{
+public interface IFMLLoadingPlugin {
     /**
      * Return a list of classes that implements the IClassTransformer interface
      * @return a list of classes that implements the IClassTransformer interface
@@ -93,8 +92,7 @@
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
-    @interface TransformerExclusions
-    {
+    @interface TransformerExclusions {
         String[] value() default "";
     }
 
@@ -107,8 +105,7 @@
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
-    @interface MCVersion
-    {
+    @interface MCVersion {
         String value() default "";
     }
 
@@ -119,15 +116,13 @@
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
-    @interface Name
-    {
+    @interface Name {
         String value() default "";
     }
 
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
-    @interface DependsOn
-    {
+    @interface DependsOn {
         String[] value() default {};
     }
 
@@ -138,8 +133,7 @@
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
-    @interface SortingIndex
-    {
+    @interface SortingIndex {
         int value() default 0;
     }
 
