--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/ServerLaunchWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/ServerLaunchWrapper.java
@@ -19,47 +19,53 @@
 
 package net.minecraftforge.fml.relauncher;
 
-import java.lang.reflect.Method;
+import red.mohist.Mohist;
 
+import red.mohist.down.DownloadLibraries;
+import red.mohist.down.DownloadServer;
+import red.mohist.i18n.Message;
 import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.core.LoggerContext;
 
+import java.lang.reflect.Method;
+
 public class ServerLaunchWrapper {
 
-    /**
-     * @param args
-     */
-    public static void main(String[] args)
-    {
-        new ServerLaunchWrapper().run(args);
-    }
+    public static long j;
 
-    private ServerLaunchWrapper()
+    public ServerLaunchWrapper()
     {
 
     }
 
-    private void run(String[] args)
+    public void run(String[] args)
     {
         if (System.getProperty("log4j.configurationFile") == null)
         {
             // Set this early so we don't need to reconfigure later
-            System.setProperty("log4j.configurationFile", "log4j2_server.xml");
+            System.setProperty("log4j.configurationFile", "log4j2_mohist.xml");
         }
         Class<?> launchwrapper = null;
         try
         {
             launchwrapper = Class.forName("net.minecraft.launchwrapper.Launch",true,getClass().getClassLoader());
             Class.forName("org.objectweb.asm.Type",true,getClass().getClassLoader());
+            Mohist.LOGGER = LogManager.getLogger("Mohist");
+            Mohist.LOGGER.info(Message.getString(Message.Mohist_Start));
         }
         catch (Exception e)
         {
-            System.err.printf("We appear to be missing one or more essential library files.\n" +
-            		"You will need to add them to your server before FML and Forge will run successfully.");
-            e.printStackTrace(System.err);
-            System.exit(1);
+            System.out.println(Message.getString(Message.Not_Have_Library));
+            //e.printStackTrace(System.err);
+            Thread t = new Thread(new DownloadServer(),"Server Download Thread");
+            Thread t1 = new Thread(new DownloadLibraries(),"Server Libraries Thread");
+            t.start();
+            t1.start();
         }
+        j = System.nanoTime();
 
+        //Thread t = new Thread(new Update(),"Mohist Update Thread");
+        //t.run();
+
         try
         {
             Method main = launchwrapper.getMethod("main", String[].class);
@@ -71,9 +77,7 @@
         }
         catch (Exception e)
         {
-            System.err.printf("A problem occurred running the Server launcher.");
-            e.printStackTrace(System.err);
-            System.exit(1);
+            System.out.println(Message.rb.getString(Message.Mohist_Start_Error.toString()));
         }
     }
 
