--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/ServerLaunchWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/ServerLaunchWrapper.java
@@ -19,33 +19,30 @@
 
 package net.minecraftforge.fml.relauncher;
 
-import java.lang.reflect.Method;
+import red.mohist.Mohist;
 
+import red.mohist.down.DownloadLibraries;
+import red.mohist.down.DownloadServer;
+import red.mohist.i18n.Message;
 import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.core.LoggerContext;
 
+import java.lang.reflect.Method;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.Logger;
+
 public class ServerLaunchWrapper {
 
-    /**
-     * @param args
-     */
-    public static void main(String[] args)
-    {
-        new ServerLaunchWrapper().run(args);
-    }
+    public static long j;
 
-    private ServerLaunchWrapper()
+    public ServerLaunchWrapper()
     {
 
     }
 
-    private void run(String[] args)
+    public void run(String[] args)
     {
-        if (System.getProperty("log4j.configurationFile") == null)
-        {
-            // Set this early so we don't need to reconfigure later
-            System.setProperty("log4j.configurationFile", "log4j2_server.xml");
-        }
         Class<?> launchwrapper = null;
         try
         {
@@ -54,11 +51,11 @@
         }
         catch (Exception e)
         {
-            System.err.printf("We appear to be missing one or more essential library files.\n" +
-            		"You will need to add them to your server before FML and Forge will run successfully.");
+            System.out.println(Message.getString(Message.Not_Have_Library));
             e.printStackTrace(System.err);
             System.exit(1);
         }
+        j = System.nanoTime();
 
         try
         {
@@ -71,9 +68,8 @@
         }
         catch (Exception e)
         {
-            System.err.printf("A problem occurred running the Server launcher.");
+            System.out.println(Message.rb.getString(Message.Mohist_Start_Error.toString()));
             e.printStackTrace(System.err);
-            System.exit(1);
         }
     }
 
