--- ../src-base/minecraft/net/minecraftforge/fluids/capability/FluidTankProperties.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/capability/FluidTankProperties.java
@@ -19,80 +19,68 @@
 
 package net.minecraftforge.fluids.capability;
 
-import javax.annotation.Nullable;
-
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.FluidTankInfo;
 
+import javax.annotation.Nullable;
+
 /**
  * Basic implementation of {@link IFluidTankProperties}.
  */
-public class FluidTankProperties implements IFluidTankProperties
-{
-    public static FluidTankProperties[] convert(FluidTankInfo[] fluidTankInfos)
-    {
-        FluidTankProperties[] properties = new FluidTankProperties[fluidTankInfos.length];
-        for (int i = 0; i < fluidTankInfos.length; i++)
-        {
-            FluidTankInfo info = fluidTankInfos[i];
-            properties[i] = new FluidTankProperties(info.fluid, info.capacity);
-        }
-        return properties;
-    }
-
+public class FluidTankProperties implements IFluidTankProperties {
     @Nullable
     private final FluidStack contents;
     private final int capacity;
     private final boolean canFill;
     private final boolean canDrain;
-
-    public FluidTankProperties(@Nullable FluidStack contents, int capacity)
-    {
+    public FluidTankProperties(@Nullable FluidStack contents, int capacity) {
         this(contents, capacity, true, true);
     }
 
-    public FluidTankProperties(@Nullable FluidStack contents, int capacity, boolean canFill, boolean canDrain)
-    {
+    public FluidTankProperties(@Nullable FluidStack contents, int capacity, boolean canFill, boolean canDrain) {
         this.contents = contents;
         this.capacity = capacity;
         this.canFill = canFill;
         this.canDrain = canDrain;
     }
 
+    public static FluidTankProperties[] convert(FluidTankInfo[] fluidTankInfos) {
+        FluidTankProperties[] properties = new FluidTankProperties[fluidTankInfos.length];
+        for (int i = 0; i < fluidTankInfos.length; i++) {
+            FluidTankInfo info = fluidTankInfos[i];
+            properties[i] = new FluidTankProperties(info.fluid, info.capacity);
+        }
+        return properties;
+    }
+
     @Nullable
     @Override
-    public FluidStack getContents()
-    {
+    public FluidStack getContents() {
         return contents == null ? null : contents.copy();
     }
 
     @Override
-    public int getCapacity()
-    {
+    public int getCapacity() {
         return capacity;
     }
 
     @Override
-    public boolean canFill()
-    {
+    public boolean canFill() {
         return canFill;
     }
 
     @Override
-    public boolean canDrain()
-    {
+    public boolean canDrain() {
         return canDrain;
     }
 
     @Override
-    public boolean canFillFluidType(FluidStack fluidStack)
-    {
+    public boolean canFillFluidType(FluidStack fluidStack) {
         return canFill;
     }
 
     @Override
-    public boolean canDrainFluidType(FluidStack fluidStack)
-    {
+    public boolean canDrainFluidType(FluidStack fluidStack) {
         return canDrain;
     }
 }
