--- ../src-base/minecraft/net/minecraftforge/fluids/capability/IFluidHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/capability/IFluidHandler.java
@@ -19,18 +19,18 @@
 
 package net.minecraftforge.fluids.capability;
 
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.IFluidTank;
+
 import javax.annotation.Nullable;
 
-import net.minecraftforge.fluids.*;
-
 /**
  * Implement this interface as a capability which should handle fluids, generally storing them in
  * one or more internal {@link IFluidTank} objects.
  * <p/>
  * A reference implementation is provided {@link TileFluidHandler}.
  */
-public interface IFluidHandler
-{
+public interface IFluidHandler {
     /**
      * Returns an array of objects which represent the internal tanks.
      * These objects cannot be used to manipulate the internal tanks.
