--- ../src-base/minecraft/net/minecraftforge/fluids/capability/IFluidTankProperties.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/capability/IFluidTankProperties.java
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.fluids.capability;
 
-import javax.annotation.Nullable;
-
 import net.minecraftforge.fluids.FluidStack;
 
+import javax.annotation.Nullable;
+
 /**
  * Simplified Read-only Information about the internals of an {@link IFluidHandler}.
  * This is useful for displaying information, and as hints for interacting with it.
@@ -32,8 +32,7 @@
  * for real fluid transactions you must use {@link IFluidHandler} to simulate, check, and then interact.
  * None of the information in these properties is required to successfully interact using a {@link IFluidHandler}.
  */
-public interface IFluidTankProperties
-{
+public interface IFluidTankProperties {
     /**
      * @return A copy of the fluid contents of this tank. May be null.
      * To modify the contents, use {@link IFluidHandler}.
