--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingDamageEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingDamageEvent.java
@@ -19,11 +19,11 @@
 
 package net.minecraftforge.event.entity.living;
 
+import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.DamageSource;
 import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraft.util.DamageSource;
-import net.minecraft.entity.EntityLivingBase;
 
 /**
  * LivingDamageEvent is fired just before damage is applied to entity.<br>
@@ -45,20 +45,25 @@
  * @see LivingHurtEvent
  **/
 @Cancelable
-public class LivingDamageEvent extends LivingEvent
-{
+public class LivingDamageEvent extends LivingEvent {
     private final DamageSource source;
     private float amount;
-    public LivingDamageEvent(EntityLivingBase entity, DamageSource source, float amount)
-    {
+
+    public LivingDamageEvent(EntityLivingBase entity, DamageSource source, float amount) {
         super(entity);
         this.source = source;
         this.amount = amount;
     }
 
-    public DamageSource getSource() { return source; }
+    public DamageSource getSource() {
+        return source;
+    }
 
-    public float getAmount() { return amount; }
+    public float getAmount() {
+        return amount;
+    }
 
-    public void setAmount(float amount) { this.amount = amount; }
+    public void setAmount(float amount) {
+        this.amount = amount;
+    }
 }
