--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingDeathEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingDeathEvent.java
@@ -19,13 +19,13 @@
 
 package net.minecraftforge.event.entity.living;
 
+import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.DamageSource;
 import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraft.util.DamageSource;
-import net.minecraft.entity.EntityLivingBase;
 
 /**
  * LivingDeathEvent is fired when an Entity dies. <br>
@@ -46,17 +46,15 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class LivingDeathEvent extends LivingEvent
-{
+public class LivingDeathEvent extends LivingEvent {
     private final DamageSource source;
-    public LivingDeathEvent(EntityLivingBase entity, DamageSource source)
-    {
+
+    public LivingDeathEvent(EntityLivingBase entity, DamageSource source) {
         super(entity);
         this.source = source;
     }
 
-    public DamageSource getSource()
-    {
+    public DamageSource getSource() {
         return source;
     }
 }
