--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingKnockBackEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingKnockBackEvent.java
@@ -51,17 +51,15 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class LivingKnockBackEvent extends LivingEvent
-{
-    protected Entity attacker;
-    protected float strength;
-    protected double ratioX, ratioZ;
+public class LivingKnockBackEvent extends LivingEvent {
     protected final Entity originalAttacker;
     protected final float originalStrength;
     protected final double originalRatioX, originalRatioZ;
+    protected Entity attacker;
+    protected float strength;
+    protected double ratioX, ratioZ;
 
-    public LivingKnockBackEvent(EntityLivingBase target, Entity attacker, float strength, double ratioX, double ratioZ)
-    {
+    public LivingKnockBackEvent(EntityLivingBase target, Entity attacker, float strength, double ratioX, double ratioZ) {
         super(target);
         this.attacker = this.originalAttacker = attacker;
         this.strength = this.originalStrength = strength;
@@ -69,27 +67,51 @@
         this.ratioZ = this.originalRatioZ = ratioZ;
     }
 
-    public Entity getAttacker() {return this.attacker;}
+    public Entity getAttacker() {
+        return this.attacker;
+    }
 
-    public float getStrength() {return this.strength;}
+    public void setAttacker(Entity attacker) {
+        this.attacker = attacker;
+    }
 
-    public double getRatioX() {return this.ratioX;}
+    public float getStrength() {
+        return this.strength;
+    }
 
-    public double getRatioZ() {return this.ratioZ;}
+    public void setStrength(float strength) {
+        this.strength = strength;
+    }
 
-    public Entity getOriginalAttacker() {return this.originalAttacker;}
+    public double getRatioX() {
+        return this.ratioX;
+    }
 
-    public float getOriginalStrength() {return this.originalStrength;}
+    public void setRatioX(double ratioX) {
+        this.ratioX = ratioX;
+    }
 
-    public double getOriginalRatioX() {return this.originalRatioX;}
+    public double getRatioZ() {
+        return this.ratioZ;
+    }
 
-    public double getOriginalRatioZ() {return this.originalRatioZ;}
+    public void setRatioZ(double ratioZ) {
+        this.ratioZ = ratioZ;
+    }
 
-    public void setAttacker(Entity attacker) {this.attacker = attacker;}
+    public Entity getOriginalAttacker() {
+        return this.originalAttacker;
+    }
 
-    public void setStrength(float strength) {this.strength = strength;}
+    public float getOriginalStrength() {
+        return this.originalStrength;
+    }
 
-    public void setRatioX(double ratioX) {this.ratioX = ratioX;}
+    public double getOriginalRatioX() {
+        return this.originalRatioX;
+    }
 
-    public void setRatioZ(double ratioZ) {this.ratioZ = ratioZ;}
+    public double getOriginalRatioZ() {
+        return this.originalRatioZ;
+    }
 }
