--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingAttackEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingAttackEvent.java
@@ -19,12 +19,12 @@
 
 package net.minecraftforge.event.entity.living;
 
+import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.DamageSource;
 import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraft.util.DamageSource;
-import net.minecraft.entity.EntityLivingBase;
 
 /**
  * LivingAttackEvent is fired when a living Entity is attacked. <br>
@@ -45,17 +45,21 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class LivingAttackEvent extends LivingEvent
-{
+public class LivingAttackEvent extends LivingEvent {
     private final DamageSource source;
     private final float amount;
-    public LivingAttackEvent(EntityLivingBase entity, DamageSource source, float amount)
-    {
+
+    public LivingAttackEvent(EntityLivingBase entity, DamageSource source, float amount) {
         super(entity);
         this.source = source;
         this.amount = amount;
     }
 
-    public DamageSource getSource() { return source; }
-    public float getAmount() { return amount; }
+    public DamageSource getSource() {
+        return source;
+    }
+
+    public float getAmount() {
+        return amount;
+    }
 }
