--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingEvent.java
@@ -19,14 +19,14 @@
 
 package net.minecraftforge.event.entity.living;
 
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraftforge.fml.common.eventhandler.Event;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.monster.EntityMagmaCube;
 import net.minecraft.entity.passive.EntityHorse;
 import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.entity.EntityEvent;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
 
 /**
  * LivingEvent is fired whenever an event involving Living entities occurs.<br>
@@ -35,17 +35,15 @@
  * <br>
  * All children of this event are fired on the {@link MinecraftForge#EVENT_BUS}.<br>
  **/
-public class LivingEvent extends EntityEvent
-{
+public class LivingEvent extends EntityEvent {
     private final EntityLivingBase entityLiving;
-    public LivingEvent(EntityLivingBase entity)
-    {
+
+    public LivingEvent(EntityLivingBase entity) {
         super(entity);
         entityLiving = entity;
     }
 
-    public EntityLivingBase getEntityLiving()
-    {
+    public EntityLivingBase getEntityLiving() {
         return entityLiving;
     }
 
@@ -64,11 +62,12 @@
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
      **/
     @Cancelable
-    public static class LivingUpdateEvent extends LivingEvent
-    {
-        public LivingUpdateEvent(EntityLivingBase e){ super(e); }
+    public static class LivingUpdateEvent extends LivingEvent {
+        public LivingUpdateEvent(EntityLivingBase e) {
+            super(e);
+        }
     }
-    
+
     /**
      * LivingJumpEvent is fired when an Entity jumps.<br>
      * This event is fired whenever an Entity jumps in 
@@ -83,8 +82,9 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
      **/
-    public static class LivingJumpEvent extends LivingEvent
-    {
-        public LivingJumpEvent(EntityLivingBase e){ super(e); }
+    public static class LivingJumpEvent extends LivingEvent {
+        public LivingJumpEvent(EntityLivingBase e) {
+            super(e);
+        }
     }
 }
