--- ../src-base/minecraft/net/minecraftforge/event/entity/living/LivingEquipmentChangeEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/LivingEquipmentChangeEvent.java
@@ -43,23 +43,29 @@
  * <br>
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
-public class LivingEquipmentChangeEvent extends LivingEvent
-{
+public class LivingEquipmentChangeEvent extends LivingEvent {
     private final EntityEquipmentSlot slot;
     private final ItemStack from;
     private final ItemStack to;
 
-    public LivingEquipmentChangeEvent(EntityLivingBase entity, EntityEquipmentSlot slot, @Nonnull ItemStack from, @Nonnull ItemStack to)
-    {
+    public LivingEquipmentChangeEvent(EntityLivingBase entity, EntityEquipmentSlot slot, @Nonnull ItemStack from, @Nonnull ItemStack to) {
         super(entity);
         this.slot = slot;
         this.from = from;
         this.to = to;
     }
 
-    public EntityEquipmentSlot getSlot() { return this.slot; }
+    public EntityEquipmentSlot getSlot() {
+        return this.slot;
+    }
+
     @Nonnull
-    public ItemStack getFrom() { return this.from; }
+    public ItemStack getFrom() {
+        return this.from;
+    }
+
     @Nonnull
-    public ItemStack getTo() { return this.to; }
+    public ItemStack getTo() {
+        return this.to;
+    }
 }
