--- ../src-base/minecraft/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,11 +19,12 @@
 
 package net.minecraftforge.event.entity.living;
 
-import java.util.Collection;
-import java.util.Collections;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.potion.PotionEffect;
 
+import java.util.Collection;
+import java.util.Collections;
+
 /**
  * Fires after Potion Color Calculation.
  * 
