--- ../src-base/minecraft/net/minecraft/world/storage/loot/functions/EnchantWithLevels.java
+++ ../src-work/minecraft/net/minecraft/world/storage/loot/functions/EnchantWithLevels.java
@@ -3,7 +3,6 @@
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
-import java.util.Random;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.JsonUtils;
@@ -12,6 +11,8 @@
 import net.minecraft.world.storage.loot.RandomValueRange;
 import net.minecraft.world.storage.loot.conditions.LootCondition;
 
+import java.util.Random;
+
 public class EnchantWithLevels extends LootFunction
 {
     private final RandomValueRange randomLevel;
