--- ../src-base/minecraft/net/minecraft/util/registry/IRegistry.java
+++ ../src-work/minecraft/net/minecraft/util/registry/IRegistry.java
@@ -1,10 +1,11 @@
 package net.minecraft.util.registry;
 
-import java.util.Set;
-import javax.annotation.Nullable;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Set;
+
 public interface IRegistry<K, V> extends Iterable<V>
 {
     @Nullable
