--- ../src-base/minecraft/net/minecraft/util/registry/RegistryDefaulted.java
+++ ../src-work/minecraft/net/minecraft/util/registry/RegistryDefaulted.java
@@ -3,19 +3,16 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-public class RegistryDefaulted<K, V> extends RegistrySimple<K, V>
-{
+public class RegistryDefaulted<K, V> extends RegistrySimple<K, V> {
     private final V defaultObject;
 
-    public RegistryDefaulted(V defaultObjectIn)
-    {
+    public RegistryDefaulted(V defaultObjectIn) {
         this.defaultObject = defaultObjectIn;
     }
 
     @Nonnull
-    public V getObject(@Nullable K name)
-    {
-        V v = (V)super.getObject(name);
-        return (V)(v == null ? this.defaultObject : v);
+    public V getObject(@Nullable K name) {
+        V v = (V) super.getObject(name);
+        return (V) (v == null ? this.defaultObject : v);
     }
 }
