--- ../src-base/minecraft/net/minecraft/util/text/TextComponentTranslation.java
+++ ../src-work/minecraft/net/minecraft/util/text/TextComponentTranslation.java
@@ -3,13 +3,14 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.Iterators;
 import com.google.common.collect.Lists;
+import net.minecraft.util.text.translation.I18n;
+
 import java.util.Arrays;
 import java.util.IllegalFormatException;
 import java.util.Iterator;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import net.minecraft.util.text.translation.I18n;
 
 public class TextComponentTranslation extends TextComponentBase
 {
