--- ../src-base/minecraft/net/minecraft/util/text/TextComponentUtils.java
+++ ../src-work/minecraft/net/minecraft/util/text/TextComponentUtils.java
@@ -1,6 +1,5 @@
 package net.minecraft.util.text;
 
-import java.util.List;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.EntityNotFoundException;
 import net.minecraft.command.EntitySelector;
@@ -8,6 +7,8 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 
+import java.util.List;
+
 public class TextComponentUtils
 {
     public static ITextComponent processComponent(ICommandSender commandSender, ITextComponent component, Entity entityIn) throws CommandException
