--- ../src-base/minecraft/net/minecraftforge/client/event/ClientChatEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/ClientChatEvent.java
@@ -20,11 +20,11 @@
 package net.minecraftforge.client.event;
 
 import com.google.common.base.Strings;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.ForgeEventFactory;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraftforge.event.ForgeEventFactory;
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraft.client.gui.GuiScreen;
 
 /**
  * ClientChatEvent is fired whenever the client is about to send a chat message or command to the server. <br>
@@ -42,28 +42,24 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class ClientChatEvent extends Event
-{
-    private String message;
+public class ClientChatEvent extends Event {
     private final String originalMessage;
-    public ClientChatEvent(String message)
-    {
+    private String message;
+
+    public ClientChatEvent(String message) {
         this.setMessage(message);
         this.originalMessage = Strings.nullToEmpty(message);
     }
 
-    public String getMessage()
-    {
+    public String getMessage() {
         return message;
     }
 
-    public void setMessage(String message)
-    {
+    public void setMessage(String message) {
         this.message = Strings.nullToEmpty(message);
     }
 
-    public String getOriginalMessage()
-    {
+    public String getOriginalMessage() {
         return originalMessage;
     }
 }
