--- ../src-base/minecraft/net/minecraftforge/client/event/FOVUpdateEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/FOVUpdateEvent.java
@@ -26,36 +26,30 @@
  * Author: MachineMuse (Claire Semple)
  * Created: 6:07 PM, 9/5/13
  */
-public class FOVUpdateEvent extends Event
-{
+public class FOVUpdateEvent extends Event {
     private final EntityPlayer entity;
     private final float fov;
     private float newfov;
 
-    public FOVUpdateEvent(EntityPlayer entity, float fov)
-    {
+    public FOVUpdateEvent(EntityPlayer entity, float fov) {
         this.entity = entity;
         this.fov = fov;
         this.setNewfov(fov);
     }
 
-    public EntityPlayer getEntity()
-    {
+    public EntityPlayer getEntity() {
         return entity;
     }
 
-    public float getFov()
-    {
+    public float getFov() {
         return fov;
     }
 
-    public float getNewfov()
-    {
+    public float getNewfov() {
         return newfov;
     }
 
-    public void setNewfov(float newfov)
-    {
+    public void setNewfov(float newfov) {
         this.newfov = newfov;
     }
 }
