--- ../src-base/minecraft/net/minecraftforge/client/event/GuiContainerEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/GuiContainerEvent.java
@@ -25,18 +25,15 @@
 /**
  * Event class for handling GuiContainer specific events.
  */
-public class GuiContainerEvent extends Event
-{
+public class GuiContainerEvent extends Event {
 
     private final GuiContainer guiContainer;
 
-    public GuiContainerEvent(GuiContainer guiContainer)
-    {
+    public GuiContainerEvent(GuiContainer guiContainer) {
         this.guiContainer = guiContainer;
     }
 
-    public GuiContainer getGuiContainer()
-    {
+    public GuiContainer getGuiContainer() {
         return guiContainer;
     }
 
@@ -46,8 +43,7 @@
      * This is useful for any slot / item specific overlays.
      * Things that need to be on top of All GUI elements but bellow tooltips and dragged stacks.
      */
-    public static class DrawForeground extends GuiContainerEvent
-    {
+    public static class DrawForeground extends GuiContainerEvent {
         private final int mouseX;
         private final int mouseY;
 
@@ -58,20 +54,17 @@
          * @param mouseX       The current X position of the players mouse.
          * @param mouseY       The current Y position of the players mouse.
          */
-        public DrawForeground(GuiContainer guiContainer, int mouseX, int mouseY)
-        {
+        public DrawForeground(GuiContainer guiContainer, int mouseX, int mouseY) {
             super(guiContainer);
             this.mouseX = mouseX;
             this.mouseY = mouseY;
         }
 
-        public int getMouseX()
-        {
+        public int getMouseX() {
             return mouseX;
         }
 
-        public int getMouseY()
-        {
+        public int getMouseY() {
             return mouseY;
         }
     }
