--- ../src-base/minecraft/net/minecraftforge/client/event/InputUpdateEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/InputUpdateEvent.java
@@ -27,18 +27,15 @@
  * This event is fired after player movement inputs are updated.<br>
  * Handlers can freely manipulate {@link MovementInput} to cancel movement.<br>
  */
-public class InputUpdateEvent extends PlayerEvent
-{
+public class InputUpdateEvent extends PlayerEvent {
     private final MovementInput movementInput;
 
-    public InputUpdateEvent(EntityPlayer player, MovementInput movementInput)
-    {
+    public InputUpdateEvent(EntityPlayer player, MovementInput movementInput) {
         super(player);
         this.movementInput = movementInput;
     }
 
-    public MovementInput getMovementInput()
-    {
+    public MovementInput getMovementInput() {
         return movementInput;
     }
 
