--- ../src-base/minecraft/net/minecraftforge/client/event/ModelRegistryEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/ModelRegistryEvent.java
@@ -25,6 +25,5 @@
 /**
  * Fired when the {@link net.minecraftforge.client.model.ModelLoader} is ready to receive registrations
  */
-public class ModelRegistryEvent extends Event implements IContextSetter
-{
+public class ModelRegistryEvent extends Event implements IContextSetter {
 }
