--- ../src-base/minecraft/net/minecraftforge/client/event/MouseEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/MouseEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,10 +19,9 @@
 
 package net.minecraftforge.client.event;
 
-import org.lwjgl.input.Mouse;
-
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
+import org.lwjgl.input.Mouse;
 
 /**
  * Author: MachineMuse (Claire Semple)
