--- ../src-base/minecraft/net/minecraftforge/client/event/RenderItemInFrameEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/RenderItemInFrameEvent.java
@@ -33,32 +33,27 @@
  * You can set canceled to do no further vanilla processing.
  */
 @Cancelable
-public class RenderItemInFrameEvent extends Event
-{
+public class RenderItemInFrameEvent extends Event {
     private final ItemStack item;
     private final EntityItemFrame entityItemFrame;
     private final RenderItemFrame renderer;
 
-    public RenderItemInFrameEvent(EntityItemFrame itemFrame, RenderItemFrame renderItemFrame)
-    {
+    public RenderItemInFrameEvent(EntityItemFrame itemFrame, RenderItemFrame renderItemFrame) {
         item = itemFrame.getDisplayedItem();
         entityItemFrame = itemFrame;
         renderer = renderItemFrame;
     }
 
     @Nonnull
-    public ItemStack getItem()
-    {
+    public ItemStack getItem() {
         return item;
     }
 
-    public EntityItemFrame getEntityItemFrame()
-    {
+    public EntityItemFrame getEntityItemFrame() {
         return entityItemFrame;
     }
 
-    public RenderItemFrame getRenderer()
-    {
+    public RenderItemFrame getRenderer() {
         return renderer;
     }
 }
