--- ../src-base/minecraft/net/minecraft/init/Enchantments.java
+++ ../src-work/minecraft/net/minecraft/init/Enchantments.java
@@ -1,9 +1,10 @@
 package net.minecraft.init;
 
-import javax.annotation.Nullable;
 import net.minecraft.enchantment.Enchantment;
 import net.minecraft.util.ResourceLocation;
 
+import javax.annotation.Nullable;
+
 public class Enchantments
 {
     public static final Enchantment PROTECTION = getRegisteredEnchantment("protection");
