--- ../src-base/minecraft/net/minecraft/init/PotionTypes.java
+++ ../src-work/minecraft/net/minecraft/init/PotionTypes.java
@@ -1,13 +1,12 @@
 package net.minecraft.init;
 
 import com.google.common.collect.Sets;
-import java.util.Set;
 import net.minecraft.potion.PotionType;
 import net.minecraft.util.ResourceLocation;
 
-public class PotionTypes
-{
-    private static final Set<PotionType> CACHE;
+import java.util.Set;
+
+public class PotionTypes {
     public static final PotionType EMPTY;
     public static final PotionType WATER;
     public static final PotionType MUNDANE;
@@ -44,29 +43,12 @@
     public static final PotionType STRONG_STRENGTH;
     public static final PotionType WEAKNESS;
     public static final PotionType LONG_WEAKNESS;
+    private static final Set<PotionType> CACHE;
 
-    private static PotionType getRegisteredPotionType(String id)
-    {
-        PotionType potiontype = PotionType.REGISTRY.getObject(new ResourceLocation(id));
-
-        if (!CACHE.add(potiontype))
-        {
-            throw new IllegalStateException("Invalid Potion requested: " + id);
-        }
-        else
-        {
-            return potiontype;
-        }
-    }
-
-    static
-    {
-        if (!Bootstrap.isRegistered())
-        {
+    static {
+        if (!Bootstrap.isRegistered()) {
             throw new RuntimeException("Accessed Potions before Bootstrap!");
-        }
-        else
-        {
+        } else {
             CACHE = Sets.<PotionType>newHashSet();
             EMPTY = getRegisteredPotionType("empty");
             WATER = getRegisteredPotionType("water");
@@ -107,4 +89,14 @@
             CACHE.clear();
         }
     }
+
+    private static PotionType getRegisteredPotionType(String id) {
+        PotionType potiontype = PotionType.REGISTRY.getObject(new ResourceLocation(id));
+
+        if (!CACHE.add(potiontype)) {
+            throw new IllegalStateException("Invalid Potion requested: " + id);
+        } else {
+            return potiontype;
+        }
+    }
 }
