--- ../src-base/minecraft/net/minecraftforge/common/brewing/BrewingRecipe.java
+++ ../src-work/minecraft/net/minecraftforge/common/brewing/BrewingRecipe.java
@@ -26,14 +26,12 @@
 
 public class BrewingRecipe extends AbstractBrewingRecipe<ItemStack> {
 
-    public BrewingRecipe(@Nonnull ItemStack input, @Nonnull ItemStack ingredient, @Nonnull ItemStack output)
-    {
+    public BrewingRecipe(@Nonnull ItemStack input, @Nonnull ItemStack ingredient, @Nonnull ItemStack output) {
         super(input, ingredient, output);
     }
 
     @Override
-    public boolean isIngredient(@Nonnull ItemStack stack)
-    {
+    public boolean isIngredient(@Nonnull ItemStack stack) {
         return OreDictionary.itemMatches(this.getIngredient(), stack, false);
     }
 }
