--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/ContainerType.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/ContainerType.java
@@ -19,31 +19,25 @@
 
 package net.minecraftforge.fml.common.discovery;
 
-import java.util.List;
-
 import net.minecraftforge.fml.common.ModContainer;
 
-public enum ContainerType
-{
+import java.util.List;
+
+public enum ContainerType {
     JAR(JarDiscoverer.class),
     DIR(DirectoryDiscoverer.class);
 
     private ITypeDiscoverer discoverer;
 
-    private ContainerType(Class<? extends ITypeDiscoverer> discovererClass)
-    {
-        try
-        {
+    private ContainerType(Class<? extends ITypeDiscoverer> discovererClass) {
+        try {
             this.discoverer = discovererClass.newInstance();
-        }
-        catch (ReflectiveOperationException e)
-        {
+        } catch (ReflectiveOperationException e) {
             throw new RuntimeException(e);
         }
     }
 
-    public List<ModContainer> findMods(ModCandidate candidate, ASMDataTable table)
-    {
+    public List<ModContainer> findMods(ModCandidate candidate, ASMDataTable table) {
         return discoverer.discover(candidate, table);
     }
 }
