--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/ITypeDiscoverer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/ITypeDiscoverer.java
@@ -19,13 +19,12 @@
 
 package net.minecraftforge.fml.common.discovery;
 
+import net.minecraftforge.fml.common.ModContainer;
+
 import java.util.List;
 import java.util.regex.Pattern;
 
-import net.minecraftforge.fml.common.ModContainer;
-
-public interface ITypeDiscoverer
-{
+public interface ITypeDiscoverer {
     // main class part, followed by an optional $ and an "inner class" part. $ cannot be last, otherwise scala breaks
     public static Pattern classFile = Pattern.compile("[^\\s\\$]+(\\$[^\\s]+)?\\.class$");
 
