--- ../src-base/minecraft/net/minecraftforge/common/network/DimensionMessageHandler.java
+++ ../src-work/minecraft/net/minecraftforge/common/network/DimensionMessageHandler.java
@@ -24,21 +24,18 @@
 import net.minecraft.world.DimensionType;
 import net.minecraftforge.common.DimensionManager;
 import net.minecraftforge.common.network.ForgeMessage.DimensionRegisterMessage;
-import org.apache.logging.log4j.Level;
 import net.minecraftforge.fml.common.FMLLog;
 
-public class DimensionMessageHandler extends SimpleChannelInboundHandler<ForgeMessage.DimensionRegisterMessage>{
+public class DimensionMessageHandler extends SimpleChannelInboundHandler<ForgeMessage.DimensionRegisterMessage> {
     @Override
-    protected void channelRead0(ChannelHandlerContext ctx, DimensionRegisterMessage msg) throws Exception
-    {
-        if (!DimensionManager.isDimensionRegistered(msg.dimensionId))
-        {
+    protected void channelRead0(ChannelHandlerContext ctx, DimensionRegisterMessage msg) throws Exception {
+        if (!DimensionManager.isDimensionRegistered(msg.dimensionId)) {
             DimensionManager.registerDimension(msg.dimensionId, DimensionType.valueOf(msg.providerId));
         }
     }
+
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception
-    {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         FMLLog.log.error("DimensionMessageHandler exception", cause);
         super.exceptionCaught(ctx, cause);
     }
