--- ../src-base/minecraft/net/minecraftforge/common/network/ForgeMessage.java
+++ ../src-work/minecraft/net/minecraftforge/common/network/ForgeMessage.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,21 +19,19 @@
 
 package net.minecraftforge.common.network;
 
-import java.nio.charset.StandardCharsets;
-import java.util.Map;
-import java.util.Set;
-
-import net.minecraftforge.fluids.Fluid;
-import net.minecraftforge.fluids.FluidRegistry;
-
+import cn.pfcraft.server.Mohist;
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
 import com.google.common.collect.Sets;
-
-import net.minecraftforge.fml.common.FMLLog;
-import net.minecraftforge.fml.common.network.ByteBufUtils;
 import io.netty.buffer.ByteBuf;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
 
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Set;
+
 public abstract class ForgeMessage {
     public static class DimensionRegisterMessage extends ForgeMessage {
         /** The dimension ID to register on client */
@@ -108,7 +106,7 @@
             }
             else
             {
-                FMLLog.log.info("Legacy server message contains no default fluid list - there may be problems with fluids");
+                Mohist.LOGGER.info("Legacy server message contains no default fluid list - there may be problems with fluids");
                 defaultFluids.clear();
             }
         }
