--- ../src-base/minecraft/net/minecraft/client/renderer/color/IBlockColor.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/color/IBlockColor.java
@@ -1,14 +1,14 @@
 package net.minecraft.client.renderer.color;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
-public interface IBlockColor
-{
+public interface IBlockColor {
     int colorMultiplier(IBlockState state, @Nullable IBlockAccess worldIn, @Nullable BlockPos pos, int tintIndex);
 }
