--- ../src-base/minecraft/net/minecraft/client/gui/spectator/BaseSpectatorGroup.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/BaseSpectatorGroup.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.gui.spectator;
 
 import com.google.common.collect.Lists;
-import java.util.List;
 import net.minecraft.client.gui.spectator.categories.TeleportToPlayer;
 import net.minecraft.client.gui.spectator.categories.TeleportToTeam;
 import net.minecraft.util.text.ITextComponent;
@@ -9,24 +8,22 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
-public class BaseSpectatorGroup implements ISpectatorMenuView
-{
+public class BaseSpectatorGroup implements ISpectatorMenuView {
     private final List<ISpectatorMenuObject> items = Lists.<ISpectatorMenuObject>newArrayList();
 
-    public BaseSpectatorGroup()
-    {
+    public BaseSpectatorGroup() {
         this.items.add(new TeleportToPlayer());
         this.items.add(new TeleportToTeam());
     }
 
-    public List<ISpectatorMenuObject> getItems()
-    {
+    public List<ISpectatorMenuObject> getItems() {
         return this.items;
     }
 
-    public ITextComponent getPrompt()
-    {
+    public ITextComponent getPrompt() {
         return new TextComponentTranslation("spectatorMenu.root.prompt", new Object[0]);
     }
 }
