--- ../src-base/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuRecipient.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuRecipient.java
@@ -4,7 +4,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ISpectatorMenuRecipient
-{
+public interface ISpectatorMenuRecipient {
     void onSpectatorMenuClosed(SpectatorMenu menu);
 }
