--- ../src-base/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuObject.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuObject.java
@@ -5,8 +5,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ISpectatorMenuObject
-{
+public interface ISpectatorMenuObject {
     void selectItem(SpectatorMenu menu);
 
     ITextComponent getSpectatorName();
