--- ../src-base/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuView.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/ISpectatorMenuView.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.gui.spectator;
 
-import java.util.List;
 import net.minecraft.util.text.ITextComponent;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public interface ISpectatorMenuView
 {
