--- ../src-base/minecraft/net/minecraft/client/gui/spectator/PlayerMenuObject.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/PlayerMenuObject.java
@@ -13,38 +13,32 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class PlayerMenuObject implements ISpectatorMenuObject
-{
+public class PlayerMenuObject implements ISpectatorMenuObject {
     private final GameProfile profile;
     private final ResourceLocation resourceLocation;
 
-    public PlayerMenuObject(GameProfile profileIn)
-    {
+    public PlayerMenuObject(GameProfile profileIn) {
         this.profile = profileIn;
         this.resourceLocation = AbstractClientPlayer.getLocationSkin(profileIn.getName());
         AbstractClientPlayer.getDownloadImageSkin(this.resourceLocation, profileIn.getName());
     }
 
-    public void selectItem(SpectatorMenu menu)
-    {
+    public void selectItem(SpectatorMenu menu) {
         Minecraft.getMinecraft().getConnection().sendPacket(new CPacketSpectate(this.profile.getId()));
     }
 
-    public ITextComponent getSpectatorName()
-    {
+    public ITextComponent getSpectatorName() {
         return new TextComponentString(this.profile.getName());
     }
 
-    public void renderIcon(float brightness, int alpha)
-    {
+    public void renderIcon(float brightness, int alpha) {
         Minecraft.getMinecraft().getTextureManager().bindTexture(this.resourceLocation);
-        GlStateManager.color(1.0F, 1.0F, 1.0F, (float)alpha / 255.0F);
+        GlStateManager.color(1.0F, 1.0F, 1.0F, (float) alpha / 255.0F);
         Gui.drawScaledCustomSizeModalRect(2, 2, 8.0F, 8.0F, 8, 8, 12, 12, 64.0F, 64.0F);
         Gui.drawScaledCustomSizeModalRect(2, 2, 40.0F, 8.0F, 8, 8, 12, 12, 64.0F, 64.0F);
     }
 
-    public boolean isEnabled()
-    {
+    public boolean isEnabled() {
         return true;
     }
 }
