--- ../src-base/minecraft/net/minecraft/client/gui/spectator/SpectatorMenu.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/SpectatorMenu.java
@@ -2,7 +2,6 @@
 
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.Lists;
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.Gui;
 import net.minecraft.client.gui.GuiSpectator;
@@ -13,13 +12,15 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class SpectatorMenu
 {
-    private static final ISpectatorMenuObject CLOSE_ITEM = new SpectatorMenu.EndSpectatorObject();
-    private static final ISpectatorMenuObject SCROLL_LEFT = new SpectatorMenu.MoveMenuObject(-1, true);
-    private static final ISpectatorMenuObject SCROLL_RIGHT_ENABLED = new SpectatorMenu.MoveMenuObject(1, true);
-    private static final ISpectatorMenuObject SCROLL_RIGHT_DISABLED = new SpectatorMenu.MoveMenuObject(1, false);
+    private static final ISpectatorMenuObject CLOSE_ITEM = new EndSpectatorObject();
+    private static final ISpectatorMenuObject SCROLL_LEFT = new MoveMenuObject(-1, true);
+    private static final ISpectatorMenuObject SCROLL_RIGHT_ENABLED = new MoveMenuObject(1, true);
+    private static final ISpectatorMenuObject SCROLL_RIGHT_DISABLED = new MoveMenuObject(1, false);
     public static final ISpectatorMenuObject EMPTY_SLOT = new ISpectatorMenuObject()
     {
         public void selectItem(SpectatorMenu menu)
