--- ../src-base/minecraft/net/minecraftforge/fml/common/event/FMLEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/event/FMLEvent.java
@@ -25,17 +25,16 @@
  * Parent type to all FML events. This is based on Guava EventBus. Event Subscription isn't using the Guava annotation
  * however, it's using a custom annotation specific to FML {@link net.minecraftforge.fml.common.Mod.EventHandler}
  */
-public class FMLEvent
-{
-    public final String getEventType()
-    {
+public class FMLEvent {
+    public final String getEventType() {
         return getClass().getSimpleName();
     }
-    public final String description()
-    {
-       String cn = getClass().getName();
-       return cn.substring(cn.lastIndexOf('.')+4,cn.length()-5);
+
+    public final String description() {
+        String cn = getClass().getName();
+        return cn.substring(cn.lastIndexOf('.') + 4, cn.length() - 5);
     }
+
     public void applyModContainer(ModContainer activeContainer) {
         // NO OP
     }
