--- ../src-base/minecraft/net/minecraftforge/fml/common/event/FMLPostInitializationEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/event/FMLPostInitializationEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,15 +19,14 @@
 
 package net.minecraftforge.fml.common.event;
 
-import java.util.Arrays;
-import java.util.Objects;
-import java.util.Optional;
-
-import net.minecraftforge.fml.common.FMLLog;
+import cn.pfcraft.server.Mohist;
 import net.minecraftforge.fml.common.Loader;
 import net.minecraftforge.fml.common.LoaderState.ModState;
 
 import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
 
 /**
  * Called after {@link FMLInitializationEvent} has been dispatched on every mod. This is the third and last
@@ -73,7 +72,7 @@
             }
             catch (Exception e)
             {
-                FMLLog.log.info("An error occurred trying to build a soft depend proxy", e);
+                Mohist.LOGGER.info("An error occurred trying to build a soft depend proxy", e);
                 return Optional.empty();
             }
         }
