--- ../src-base/minecraft/net/minecraftforge/fml/common/event/FMLServerStartedEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/event/FMLServerStartedEvent.java
@@ -27,17 +27,14 @@
  * @see net.minecraftforge.fml.common.Mod.EventHandler for how to subscribe to this event
  * @author cpw
  */
-public class FMLServerStartedEvent extends FMLStateEvent
-{
+public class FMLServerStartedEvent extends FMLStateEvent {
 
-    public FMLServerStartedEvent(Object... data)
-    {
+    public FMLServerStartedEvent(Object... data) {
         super(data);
     }
-    
+
     @Override
-    public ModState getModState()
-    {
+    public ModState getModState() {
         return ModState.AVAILABLE;
     }
 
