--- ../src-base/minecraft/net/minecraftforge/event/world/ChunkWatchEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/world/ChunkWatchEvent.java
@@ -19,15 +19,15 @@
 
 package net.minecraftforge.event.world;
 
+import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.management.PlayerChunkMapEntry;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.chunk.Chunk;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
+
 import javax.annotation.Nullable;
-import net.minecraft.entity.player.EntityPlayerMP;
-import net.minecraft.util.math.ChunkPos;
-import net.minecraft.world.World;
-import net.minecraft.world.chunk.Chunk;
 
 /**
  * ChunkWatchEvent is fired when an event involving a chunk being watched occurs.<br>
@@ -51,28 +51,24 @@
     private final Chunk chunkInstance;
 
     @Deprecated //TODO: Remove in 1.13
-    public ChunkWatchEvent(ChunkPos chunk, EntityPlayerMP player)
-    {
+    public ChunkWatchEvent(ChunkPos chunk, EntityPlayerMP player) {
         this.chunk = chunk;
         this.player = player;
         this.chunkInstance = null;
     }
 
-    public ChunkWatchEvent(Chunk chunk, EntityPlayerMP player)
-    {
+    public ChunkWatchEvent(Chunk chunk, EntityPlayerMP player) {
         this.chunk = chunk.getPos();
         this.player = player;
         this.chunkInstance = chunk;
     }
 
     @Deprecated //TODO: Remove in 1.13
-    public ChunkPos getChunk()
-    {
+    public ChunkPos getChunk() {
         return chunk;
     }
 
-    public EntityPlayerMP getPlayer()
-    {
+    public EntityPlayerMP getPlayer() {
         return player;
     }
 
@@ -81,8 +77,7 @@
      * @return The affected chunk.
      */
     @Nullable
-    public Chunk getChunkInstance()
-    {
+    public Chunk getChunkInstance() {
         return chunkInstance;
     }
 
@@ -97,12 +92,15 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
      **/
-    public static class Watch extends ChunkWatchEvent
-    {
+    public static class Watch extends ChunkWatchEvent {
         @Deprecated //TODO: Remove in 1.13
-        public Watch(ChunkPos chunk, EntityPlayerMP player) { super(chunk, player); }
+        public Watch(ChunkPos chunk, EntityPlayerMP player) {
+            super(chunk, player);
+        }
 
-        public Watch(Chunk chunk, EntityPlayerMP player) { super(chunk, player); }
+        public Watch(Chunk chunk, EntityPlayerMP player) {
+            super(chunk, player);
+        }
     }
 
     /**
@@ -116,11 +114,14 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
      **/
-    public static class UnWatch extends ChunkWatchEvent
-    {
+    public static class UnWatch extends ChunkWatchEvent {
         @Deprecated //TODO: Remove in 1.13
-        public UnWatch(ChunkPos chunkLocation, EntityPlayerMP player) { super(chunkLocation, player); }
+        public UnWatch(ChunkPos chunkLocation, EntityPlayerMP player) {
+            super(chunkLocation, player);
+        }
 
-        public UnWatch(Chunk chunk, EntityPlayerMP player) { super(chunk, player); }
+        public UnWatch(Chunk chunk, EntityPlayerMP player) {
+            super(chunk, player);
+        }
     }
 }
