--- ../src-base/minecraft/net/minecraftforge/event/world/ChunkDataEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/world/ChunkDataEvent.java
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.event.world;
 
+import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.Chunk;
-import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.world.chunk.storage.AnvilChunkLoader;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
@@ -36,21 +36,18 @@
  * <br>
  * All children of this event are fired on the {@link MinecraftForge#EVENT_BUS}.<br>
  **/
-public class ChunkDataEvent extends ChunkEvent
-{
+public class ChunkDataEvent extends ChunkEvent {
     private final NBTTagCompound data;
 
-    public ChunkDataEvent(Chunk chunk, NBTTagCompound data)
-    {
+    public ChunkDataEvent(Chunk chunk, NBTTagCompound data) {
         super(chunk);
         this.data = data;
     }
-    
-    public NBTTagCompound getData()
-    {
+
+    public NBTTagCompound getData() {
         return data;
     }
-    
+
     /**
      * ChunkDataEvent.Load is fired when vanilla Minecraft attempts to load Chunk data.<br>
      * This event is fired during chunk loading in
@@ -62,14 +59,12 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
      **/
-    public static class Load extends ChunkDataEvent
-    {
-        public Load(Chunk chunk, NBTTagCompound data)
-        {
+    public static class Load extends ChunkDataEvent {
+        public Load(Chunk chunk, NBTTagCompound data) {
             super(chunk, data);
         }
     }
-    
+
     /**
      * ChunkDataEvent.Save is fired when vanilla Minecraft attempts to save Chunk data.<br>
      * This event is fired during chunk saving in 
@@ -81,10 +76,8 @@
      * <br>
      * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
      **/
-    public static class Save extends ChunkDataEvent
-    {
-        public Save(Chunk chunk, NBTTagCompound data)
-        {
+    public static class Save extends ChunkDataEvent {
+        public Save(Chunk chunk, NBTTagCompound data) {
             super(chunk, data);
         }
     }
