--- ../src-base/minecraft/net/minecraftforge/event/world/WorldEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/world/WorldEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,10 +19,6 @@
 
 package net.minecraftforge.event.world;
 
-import java.util.ArrayList;
-import java.util.Hashtable;
-import java.util.List;
-
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.multiplayer.WorldClient;
 import net.minecraft.client.network.NetHandlerPlayClient;
@@ -40,6 +36,10 @@
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
 
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
 /**
  * WorldEvent is fired when an event involving the world occurs.<br>
  * If a method utilizes this {@link Event} as its parameter, the method will
