--- ../src-base/minecraft/net/minecraft/client/renderer/vertex/VertexBuffer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/vertex/VertexBuffer.java
@@ -1,11 +1,12 @@
 package net.minecraft.client.renderer.vertex;
 
-import java.nio.ByteBuffer;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.OpenGlHelper;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.ByteBuffer;
+
 @SideOnly(Side.CLIENT)
 public class VertexBuffer
 {
