--- ../src-base/minecraft/net/minecraftforge/client/event/sound/PlaySoundEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/sound/PlaySoundEvent.java
@@ -28,37 +28,31 @@
  * If you return null from this function it will prevent the sound from being played,
  * you can return a different entry if you want to change the sound being played.
  */
-public class PlaySoundEvent extends SoundEvent
-{
+public class PlaySoundEvent extends SoundEvent {
     private final String name;
     private final ISound sound;
     private ISound result;
 
-    public PlaySoundEvent(SoundManager manager, ISound sound)
-    {
+    public PlaySoundEvent(SoundManager manager, ISound sound) {
         super(manager);
         this.sound = sound;
         this.name = sound.getSoundLocation().getResourcePath();
         this.setResultSound(sound);
     }
 
-    public String getName()
-    {
+    public String getName() {
         return name;
     }
 
-    public ISound getSound()
-    {
+    public ISound getSound() {
         return sound;
     }
 
-    public ISound getResultSound()
-    {
+    public ISound getResultSound() {
         return result;
     }
 
-    public void setResultSound(ISound result)
-    {
+    public void setResultSound(ISound result) {
         this.result = result;
     }
 }
