--- ../src-base/minecraft/net/minecraftforge/client/event/sound/PlayStreamingSourceEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/sound/PlayStreamingSourceEvent.java
@@ -23,10 +23,8 @@
 import net.minecraft.client.audio.SoundManager;
 import net.minecraftforge.client.event.sound.SoundEvent.SoundSourceEvent;
 
-public class PlayStreamingSourceEvent extends SoundSourceEvent
-{
-    public PlayStreamingSourceEvent(SoundManager manager, ISound sound, String uuid)
-    {
+public class PlayStreamingSourceEvent extends SoundSourceEvent {
+    public PlayStreamingSourceEvent(SoundManager manager, ISound sound, String uuid) {
         super(manager, sound, uuid);
     }
 }
