--- ../src-base/minecraft/net/minecraftforge/client/event/sound/SoundLoadEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/sound/SoundLoadEvent.java
@@ -25,10 +25,8 @@
  * Raised by the SoundManager.loadSoundSettings, this would be a good place for
  * adding your custom sounds to the SoundPool.
  */
-public class SoundLoadEvent extends SoundEvent
-{
-    public SoundLoadEvent(SoundManager manager)
-    {
+public class SoundLoadEvent extends SoundEvent {
+    public SoundLoadEvent(SoundManager manager) {
         super(manager);
     }
 }
