--- ../src-base/minecraft/net/minecraftforge/client/event/sound/SoundEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/sound/SoundEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.client.event.sound;
 
-import net.minecraftforge.fml.common.eventhandler.Event;
 import net.minecraft.client.audio.ISound;
 import net.minecraft.client.audio.SoundManager;
+import net.minecraftforge.fml.common.eventhandler.Event;
 
 public class SoundEvent extends Event
 {
