--- ../src-base/minecraft/net/minecraft/world/end/DragonSpawnManager.java
+++ ../src-work/minecraft/net/minecraft/world/end/DragonSpawnManager.java
@@ -1,7 +1,5 @@
 package net.minecraft.world.end;
 
-import java.util.List;
-import java.util.Random;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityEnderCrystal;
 import net.minecraft.util.math.BlockPos;
@@ -9,6 +7,9 @@
 import net.minecraft.world.biome.BiomeEndDecorator;
 import net.minecraft.world.gen.feature.WorldGenSpikes;
 
+import java.util.List;
+import java.util.Random;
+
 public enum DragonSpawnManager
 {
     START {
