--- ../src-base/minecraft/net/minecraft/client/AnvilConverterException.java
+++ ../src-work/minecraft/net/minecraft/client/AnvilConverterException.java
@@ -4,10 +4,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class AnvilConverterException extends Exception
-{
-    public AnvilConverterException(String exceptionMessage)
-    {
+public class AnvilConverterException extends Exception {
+    public AnvilConverterException(String exceptionMessage) {
         super(exceptionMessage);
     }
 }
