--- ../src-base/minecraft/net/minecraft/client/ClientBrandRetriever.java
+++ ../src-work/minecraft/net/minecraft/client/ClientBrandRetriever.java
@@ -4,10 +4,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ClientBrandRetriever
-{
-    public static String getClientModName()
-    {
+public class ClientBrandRetriever {
+    public static String getClientModName() {
         return net.minecraftforge.fml.common.FMLCommonHandler.instance().getModName();
     }
 }
