--- ../src-base/minecraft/net/minecraftforge/common/animation/Event.java
+++ ../src-work/minecraft/net/minecraftforge/common/animation/Event.java
@@ -25,13 +25,11 @@
 /**
  * Event stored in the clip
  */
-public final class Event implements Comparable<Event>
-{
+public final class Event implements Comparable<Event> {
     private final String event;
     private final float offset;
 
-    public Event(String event, float offset)
-    {
+    public Event(String event, float offset) {
         this.event = event;
         this.offset = offset;
     }
@@ -39,28 +37,24 @@
     /**
      * @return the name of the event.
      */
-    public String event()
-    {
+    public String event() {
         return event;
     }
 
     /**
      * @return how long ago the event happened, relative to the next event / first query time
      */
-    public float offset()
-    {
+    public float offset() {
         return offset;
     }
 
     @Override
-    public int compareTo(Event event)
-    {
+    public int compareTo(Event event) {
         return new Float(offset).compareTo(event.offset);
     }
 
     @Override
-    public String toString()
-    {
+    public String toString() {
         return MoreObjects.toStringHelper(getClass()).add("event", event).add("offset", offset).toString();
     }
 }
