--- ../src-base/minecraft/net/minecraftforge/common/animation/IEventHandler.java
+++ ../src-work/minecraft/net/minecraftforge/common/animation/IEventHandler.java
@@ -20,11 +20,9 @@
 package net.minecraftforge.common.animation;
 
 
-
 /**
  * Handler for animation events;
  */
-public interface IEventHandler<T>
-{
+public interface IEventHandler<T> {
     void handleEvents(T instance, float time, Iterable<Event> pastEvents);
 }
