--- ../src-base/minecraft/net/minecraftforge/common/animation/TimeValues.java
+++ ../src-work/minecraft/net/minecraftforge/common/animation/TimeValues.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,6 @@
 
 package net.minecraftforge.common.animation;
 
-import java.io.IOException;
-import java.util.regex.Pattern;
-
-import net.minecraft.util.IStringSerializable;
-
-import java.util.function.Function;
 import com.google.common.base.Objects;
 import com.google.common.collect.ImmutableList;
 import com.google.gson.Gson;
@@ -32,10 +26,13 @@
 import com.google.gson.TypeAdapterFactory;
 import com.google.gson.reflect.TypeToken;
 import com.google.gson.stream.JsonReader;
-import com.google.gson.stream.JsonToken;
 import com.google.gson.stream.JsonWriter;
+import net.minecraft.util.IStringSerializable;
 
 import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.function.Function;
+import java.util.regex.Pattern;
 
 /**
  * Various implementations of ITimeValue.
