--- ../src-base/minecraft/net/minecraft/network/status/server/SPacketPong.java
+++ ../src-work/minecraft/net/minecraft/network/status/server/SPacketPong.java
@@ -1,35 +1,30 @@
 package net.minecraft.network.status.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusClient;
 
-public class SPacketPong implements Packet<INetHandlerStatusClient>
-{
+import java.io.IOException;
+
+public class SPacketPong implements Packet<INetHandlerStatusClient> {
     private long clientTime;
 
-    public SPacketPong()
-    {
+    public SPacketPong() {
     }
 
-    public SPacketPong(long clientTimeIn)
-    {
+    public SPacketPong(long clientTimeIn) {
         this.clientTime = clientTimeIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.clientTime = buf.readLong();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeLong(this.clientTime);
     }
 
-    public void processPacket(INetHandlerStatusClient handler)
-    {
+    public void processPacket(INetHandlerStatusClient handler) {
         handler.handlePong(this);
     }
 }
