--- ../src-base/minecraft/net/minecraft/network/status/server/SPacketPong.java
+++ ../src-work/minecraft/net/minecraft/network/status/server/SPacketPong.java
@@ -1,10 +1,11 @@
 package net.minecraft.network.status.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusClient;
 
+import java.io.IOException;
+
 public class SPacketPong implements Packet<INetHandlerStatusClient>
 {
     private long clientTime;
