--- ../src-base/minecraft/net/minecraft/world/chunk/storage/AnvilSaveHandler.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/AnvilSaveHandler.java
@@ -1,16 +1,15 @@
 package net.minecraft.world.chunk.storage;
 
-import java.io.File;
-import javax.annotation.Nullable;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.DataFixer;
 import net.minecraft.world.WorldProvider;
-import net.minecraft.world.WorldProviderEnd;
-import net.minecraft.world.WorldProviderHell;
 import net.minecraft.world.storage.SaveHandler;
 import net.minecraft.world.storage.ThreadedFileIOBase;
 import net.minecraft.world.storage.WorldInfo;
 
+import javax.annotation.Nullable;
+import java.io.File;
+
 public class AnvilSaveHandler extends SaveHandler
 {
     public AnvilSaveHandler(File p_i46650_1_, String saveDirectoryName, boolean p_i46650_3_, DataFixer dataFixerIn)
@@ -21,17 +20,7 @@
     public IChunkLoader getChunkLoader(WorldProvider provider)
     {
         File file1 = this.getWorldDirectory();
-
-        if (provider.getSaveFolder() != null)
-        {
-            File file3 = new File(file1, provider.getSaveFolder());
-            file3.mkdirs();
-            return new AnvilChunkLoader(file3, this.dataFixer);
-        }
-        else
-        {
-            return new AnvilChunkLoader(file1, this.dataFixer);
-        }
+        return new AnvilChunkLoader(file1,this.dataFixer);
     }
 
     public void saveWorldInfoWithPlayer(WorldInfo worldInformation, @Nullable NBTTagCompound tagCompound)
