--- ../src-base/minecraft/net/minecraft/world/chunk/storage/IChunkLoader.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/IChunkLoader.java
@@ -1,11 +1,12 @@
 package net.minecraft.world.chunk.storage;
 
-import java.io.IOException;
-import javax.annotation.Nullable;
 import net.minecraft.world.MinecraftException;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.Chunk;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+
 public interface IChunkLoader
 {
     @Nullable
