--- ../src-base/minecraft/net/minecraft/world/chunk/storage/NibbleArrayReader.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/NibbleArrayReader.java
@@ -1,20 +1,17 @@
 package net.minecraft.world.chunk.storage;
 
-public class NibbleArrayReader
-{
+public class NibbleArrayReader {
     public final byte[] data;
     private final int depthBits;
     private final int depthBitsPlusFour;
 
-    public NibbleArrayReader(byte[] dataIn, int depthBitsIn)
-    {
+    public NibbleArrayReader(byte[] dataIn, int depthBitsIn) {
         this.data = dataIn;
         this.depthBits = depthBitsIn;
         this.depthBitsPlusFour = depthBitsIn + 4;
     }
 
-    public int get(int x, int y, int z)
-    {
+    public int get(int x, int y, int z) {
         int i = x << this.depthBitsPlusFour | z << this.depthBits | y;
         int j = i >> 1;
         int k = i & 1;
