--- ../src-base/minecraft/net/minecraft/dispenser/PositionImpl.java
+++ ../src-work/minecraft/net/minecraft/dispenser/PositionImpl.java
@@ -1,30 +1,25 @@
 package net.minecraft.dispenser;
 
-public class PositionImpl implements IPosition
-{
+public class PositionImpl implements IPosition {
     protected final double x;
     protected final double y;
     protected final double z;
 
-    public PositionImpl(double xCoord, double yCoord, double zCoord)
-    {
+    public PositionImpl(double xCoord, double yCoord, double zCoord) {
         this.x = xCoord;
         this.y = yCoord;
         this.z = zCoord;
     }
 
-    public double getX()
-    {
+    public double getX() {
         return this.x;
     }
 
-    public double getY()
-    {
+    public double getY() {
         return this.y;
     }
 
-    public double getZ()
-    {
+    public double getZ() {
         return this.z;
     }
 }
