--- ../src-base/minecraft/net/minecraft/nbt/NBTTagList.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagList.java
@@ -4,6 +4,8 @@
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Objects;
 import org.apache.logging.log4j.LogManager;
@@ -12,10 +14,14 @@
 public class NBTTagList extends NBTBase implements java.lang.Iterable<NBTBase>
 {
     private static final Logger LOGGER = LogManager.getLogger();
-    private List<NBTBase> tagList = Lists.<NBTBase>newArrayList();
+    public List<NBTBase> tagList = Lists.<NBTBase>newArrayList(); // Paper
     private byte tagType = 0;
 
-    void write(DataOutput output) throws IOException
+    public void sort(final Comparator<? extends NBTBase> comparator) {
+        Collections.sort(this.tagList, (Comparator<? super NBTBase>)comparator);
+    }
+
+    public void write(DataOutput output) throws IOException
     {
         if (this.tagList.isEmpty())
         {
