--- ../src-base/minecraft/net/minecraft/nbt/NBTTagByte.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagByte.java
@@ -4,82 +4,66 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagByte extends NBTPrimitive
-{
+public class NBTTagByte extends NBTPrimitive {
     private byte data;
 
-    NBTTagByte()
-    {
+    NBTTagByte() {
     }
 
-    public NBTTagByte(byte data)
-    {
+    public NBTTagByte(byte data) {
         this.data = data;
     }
 
-    void write(DataOutput output) throws IOException
-    {
+    public void write(DataOutput output) throws IOException {
         output.writeByte(this.data);
     }
 
-    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException
-    {
+    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException {
         sizeTracker.read(72L);
         this.data = input.readByte();
     }
 
-    public byte getId()
-    {
+    public byte getId() {
         return 1;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return this.data + "b";
     }
 
-    public NBTTagByte copy()
-    {
+    public NBTTagByte copy() {
         return new NBTTagByte(this.data);
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        return super.equals(p_equals_1_) && this.data == ((NBTTagByte)p_equals_1_).data;
+    public boolean equals(Object p_equals_1_) {
+        return super.equals(p_equals_1_) && this.data == ((NBTTagByte) p_equals_1_).data;
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return super.hashCode() ^ this.data;
     }
 
-    public long getLong()
-    {
-        return (long)this.data;
+    public long getLong() {
+        return (long) this.data;
     }
 
-    public int getInt()
-    {
+    public int getInt() {
         return this.data;
     }
 
-    public short getShort()
-    {
-        return (short)this.data;
+    public short getShort() {
+        return (short) this.data;
     }
 
-    public byte getByte()
-    {
+    public byte getByte() {
         return this.data;
     }
 
-    public double getDouble()
-    {
-        return (double)this.data;
+    public double getDouble() {
+        return (double) this.data;
     }
 
-    public float getFloat()
-    {
-        return (float)this.data;
+    public float getFloat() {
+        return (float) this.data;
     }
 }
