--- ../src-base/minecraft/net/minecraft/nbt/NBTTagEnd.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagEnd.java
@@ -4,29 +4,23 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagEnd extends NBTBase
-{
-    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException
-    {
+public class NBTTagEnd extends NBTBase {
+    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException {
         sizeTracker.read(64L);
     }
 
-    void write(DataOutput output) throws IOException
-    {
+    public void write(DataOutput output) throws IOException {
     }
 
-    public byte getId()
-    {
+    public byte getId() {
         return 0;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return "END";
     }
 
-    public NBTTagEnd copy()
-    {
+    public NBTTagEnd copy() {
         return new NBTTagEnd();
     }
 }
