--- ../src-base/minecraft/net/minecraft/nbt/NBTTagShort.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagShort.java
@@ -4,82 +4,66 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagShort extends NBTPrimitive
-{
+public class NBTTagShort extends NBTPrimitive {
     private short data;
 
-    public NBTTagShort()
-    {
+    public NBTTagShort() {
     }
 
-    public NBTTagShort(short data)
-    {
+    public NBTTagShort(short data) {
         this.data = data;
     }
 
-    void write(DataOutput output) throws IOException
-    {
+    public void write(DataOutput output) throws IOException {
         output.writeShort(this.data);
     }
 
-    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException
-    {
+    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException {
         sizeTracker.read(80L);
         this.data = input.readShort();
     }
 
-    public byte getId()
-    {
+    public byte getId() {
         return 2;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return this.data + "s";
     }
 
-    public NBTTagShort copy()
-    {
+    public NBTTagShort copy() {
         return new NBTTagShort(this.data);
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        return super.equals(p_equals_1_) && this.data == ((NBTTagShort)p_equals_1_).data;
+    public boolean equals(Object p_equals_1_) {
+        return super.equals(p_equals_1_) && this.data == ((NBTTagShort) p_equals_1_).data;
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return super.hashCode() ^ this.data;
     }
 
-    public long getLong()
-    {
-        return (long)this.data;
+    public long getLong() {
+        return (long) this.data;
     }
 
-    public int getInt()
-    {
+    public int getInt() {
         return this.data;
     }
 
-    public short getShort()
-    {
+    public short getShort() {
         return this.data;
     }
 
-    public byte getByte()
-    {
-        return (byte)(this.data & 255);
+    public byte getByte() {
+        return (byte) (this.data & 255);
     }
 
-    public double getDouble()
-    {
-        return (double)this.data;
+    public double getDouble() {
+        return (double) this.data;
     }
 
-    public float getFloat()
-    {
-        return (float)this.data;
+    public float getFloat() {
+        return (float) this.data;
     }
 }
