--- ../src-base/minecraft/net/minecraftforge/event/entity/item/ItemTossEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/item/ItemTossEvent.java
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.event.entity.item;
 
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
 /**
  * Event that is fired whenever a player tosses (Q) an item or drag-n-drops a
@@ -30,19 +30,17 @@
  * removed from the inventory - and thus removed from the system.
  */
 @Cancelable
-public class ItemTossEvent extends ItemEvent
-{
+public class ItemTossEvent extends ItemEvent {
 
     private final EntityPlayer player;
 
     /**
      * Creates a new event for EntityItems tossed by a player.
-     * 
+     *
      * @param entityItem The EntityItem being tossed.
      * @param player The player tossing the item.
      */
-    public ItemTossEvent(EntityItem entityItem, EntityPlayer player)
-    {
+    public ItemTossEvent(EntityItem entityItem, EntityPlayer player) {
         super(entityItem);
         this.player = player;
     }
@@ -50,8 +48,7 @@
     /**
      * The player tossing the item.
      */
-    public EntityPlayer getPlayer()
-    {
+    public EntityPlayer getPlayer() {
         return player;
     }
 }
