--- ../src-base/minecraft/net/minecraftforge/common/model/HiddenModelPart.java
+++ ../src-work/minecraft/net/minecraftforge/common/model/HiddenModelPart.java
@@ -21,17 +21,14 @@
 
 import com.google.common.collect.ImmutableList;
 
-final class HiddenModelPart implements IModelPart
-{
+final class HiddenModelPart implements IModelPart {
     private final ImmutableList<String> path;
 
-    HiddenModelPart(ImmutableList<String> path)
-    {
+    HiddenModelPart(ImmutableList<String> path) {
         this.path = path;
     }
 
-    ImmutableList<String> getPath()
-    {
+    ImmutableList<String> getPath() {
         return path;
     }
 }
