--- ../src-base/minecraft/net/minecraftforge/common/model/IModelState.java
+++ ../src-work/minecraft/net/minecraftforge/common/model/IModelState.java
@@ -25,8 +25,7 @@
  * Represents the dynamic information associated with the model.
  * Common use case is (possibly interpolated) animation frame.
  */
-public interface IModelState
-{
+public interface IModelState {
     /*
      * Returns the transformation that needs to be applied to the specific part of the model.
      * Coordinate system is determined by the part type.
