--- ../src-base/minecraft/net/minecraftforge/common/model/Models.java
+++ ../src-work/minecraft/net/minecraftforge/common/model/Models.java
@@ -21,22 +21,17 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
-import com.google.common.collect.Iterators;
 import com.google.common.collect.UnmodifiableIterator;
 
-public enum Models
-{
+public enum Models {
     ;
 
-    public static IModelPart getHiddenModelPart(ImmutableList<String> path)
-    {
+    public static IModelPart getHiddenModelPart(ImmutableList<String> path) {
         return new HiddenModelPart(path);
     }
 
-    public static UnmodifiableIterator<String> getParts(IModelPart part)
-    {
-        if(part instanceof HiddenModelPart)
-        {
+    public static UnmodifiableIterator<String> getParts(IModelPart part) {
+        if (part instanceof HiddenModelPart) {
             return ((HiddenModelPart) part).getPath().iterator();
         }
         ImmutableSet<String> ret = ImmutableSet.of();
