--- ../src-base/minecraft/net/minecraft/network/login/server/SPacketLoginSuccess.java
+++ ../src-work/minecraft/net/minecraft/network/login/server/SPacketLoginSuccess.java
@@ -1,14 +1,15 @@
 package net.minecraft.network.login.server;
 
 import com.mojang.authlib.GameProfile;
-import java.io.IOException;
-import java.util.UUID;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginClient;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.UUID;
+
 public class SPacketLoginSuccess implements Packet<INetHandlerLoginClient>
 {
     private GameProfile profile;
