--- ../src-base/minecraft/net/minecraft/network/login/server/SPacketEncryptionRequest.java
+++ ../src-work/minecraft/net/minecraft/network/login/server/SPacketEncryptionRequest.java
@@ -1,7 +1,5 @@
 package net.minecraft.network.login.server;
 
-import java.io.IOException;
-import java.security.PublicKey;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginClient;
@@ -9,6 +7,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.security.PublicKey;
+
 public class SPacketEncryptionRequest implements Packet<INetHandlerLoginClient>
 {
     private String hashedServerId;
