--- ../src-base/minecraft/net/minecraft/network/login/server/SPacketEnableCompression.java
+++ ../src-work/minecraft/net/minecraft/network/login/server/SPacketEnableCompression.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.login.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginClient;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketEnableCompression implements Packet<INetHandlerLoginClient>
 {
     private int compressionThreshold;
