--- ../src-base/minecraft/net/minecraft/network/login/server/SPacketDisconnect.java
+++ ../src-work/minecraft/net/minecraft/network/login/server/SPacketDisconnect.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.login.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginClient;
@@ -8,37 +7,32 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketDisconnect implements Packet<INetHandlerLoginClient>
-{
+import java.io.IOException;
+
+public class SPacketDisconnect implements Packet<INetHandlerLoginClient> {
     private ITextComponent reason;
 
-    public SPacketDisconnect()
-    {
+    public SPacketDisconnect() {
     }
 
-    public SPacketDisconnect(ITextComponent p_i46853_1_)
-    {
+    public SPacketDisconnect(ITextComponent p_i46853_1_) {
         this.reason = p_i46853_1_;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.reason = ITextComponent.Serializer.fromJsonLenient(buf.readString(32767));
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeTextComponent(this.reason);
     }
 
-    public void processPacket(INetHandlerLoginClient handler)
-    {
+    public void processPacket(INetHandlerLoginClient handler) {
         handler.handleDisconnect(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public ITextComponent getReason()
-    {
+    public ITextComponent getReason() {
         return this.reason;
     }
 }
