--- ../src-base/minecraft/net/minecraftforge/client/model/ICustomModelLoader.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/ICustomModelLoader.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,15 +19,13 @@
 
 package net.minecraftforge.client.model;
 
-import java.util.function.Predicate;
-
+import net.minecraft.client.resources.IResourceManager;
+import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.client.resource.IResourceType;
 import net.minecraftforge.client.resource.ISelectiveResourceReloadListener;
 import net.minecraftforge.client.resource.VanillaResourceType;
 
-import net.minecraft.client.resources.IResourceManager;
-import net.minecraft.client.resources.IResourceManagerReloadListener;
-import net.minecraft.util.ResourceLocation;
+import java.util.function.Predicate;
 
 public interface ICustomModelLoader extends ISelectiveResourceReloadListener
 {
@@ -42,7 +40,6 @@
             onResourceManagerReload(resourceManager);
         }
     }
-
     /*
      * Checks if given model should be loaded by this loader.
      * Reading file contents is inadvisable, if possible decision should be made based on the location alone.
