--- ../src-base/minecraft/net/minecraftforge/client/model/ICustomModelLoader.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/ICustomModelLoader.java
@@ -19,26 +19,21 @@
 
 package net.minecraftforge.client.model;
 
-import java.util.function.Predicate;
-
+import net.minecraft.client.resources.IResourceManager;
+import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.client.resource.IResourceType;
 import net.minecraftforge.client.resource.ISelectiveResourceReloadListener;
 import net.minecraftforge.client.resource.VanillaResourceType;
 
-import net.minecraft.client.resources.IResourceManager;
-import net.minecraft.client.resources.IResourceManagerReloadListener;
-import net.minecraft.util.ResourceLocation;
+import java.util.function.Predicate;
 
-public interface ICustomModelLoader extends ISelectiveResourceReloadListener
-{
+public interface ICustomModelLoader extends ISelectiveResourceReloadListener {
     @Override
     void onResourceManagerReload(IResourceManager resourceManager);
 
     @Override
-    default void onResourceManagerReload(IResourceManager resourceManager, Predicate<IResourceType> resourcePredicate)
-    {
-        if (resourcePredicate.test(VanillaResourceType.MODELS))
-        {
+    default void onResourceManagerReload(IResourceManager resourceManager, Predicate<IResourceType> resourcePredicate) {
+        if (resourcePredicate.test(VanillaResourceType.MODELS)) {
             onResourceManagerReload(resourceManager);
         }
     }
