--- ../src-base/minecraft/net/minecraftforge/client/model/IModel.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/IModel.java
@@ -19,9 +19,6 @@
 
 package net.minecraftforge.client.model;
 
-import java.util.Collection;
-
-import java.util.Optional;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import net.minecraft.client.renderer.block.model.IBakedModel;
@@ -30,17 +27,18 @@
 import net.minecraft.client.renderer.vertex.VertexFormat;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.common.model.IModelState;
-
-import java.util.function.Function;
 import net.minecraftforge.common.model.TRSRTransformation;
 import net.minecraftforge.common.model.animation.IClip;
 
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Function;
+
 /*
  * Interface for models that can be baked
  * (possibly to different vertex formats and with different state).
  */
-public interface IModel
-{
+public interface IModel {
     /*
      * Returns all model locations that this model depends on.
      * Assume that returned collection is immutable.
