--- ../src-base/minecraft/net/minecraftforge/client/model/ModelLoader.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/ModelLoader.java
@@ -293,13 +293,13 @@
                 Exception exception = null;
                 try
                 {
-                    model = ModelLoaderRegistry.getModel(memory);
+ 					model = ModelLoaderRegistry.getModel(memory);
                 }
-                catch (Exception blockstateException)
+                catch(Exception blockstateException)
                 {
                     try
                     {
-                        model = ModelLoaderRegistry.getModel(file);
+                    	model = ModelLoaderRegistry.getModel(file);
                         ModelLoaderRegistry.addAlias(memory, file);
                     }
                     catch (Exception normalException)
@@ -307,7 +307,7 @@
                         exception = new ItemLoadingException("Could not load item model either from the normal location " + file + " or from the blockstate", normalException, blockstateException);
                     }
                 }
-                if (exception != null)
+                if(exception != null)
                 {
                     storeException(memory, exception);
                     model = ModelLoaderRegistry.getMissingModel(memory, exception);
@@ -1169,7 +1169,6 @@
         {
             ModelResourceLocation variant = (ModelResourceLocation) modelLocation;
             ModelBlockDefinition definition = loader.getModelBlockDefinition(variant);
-
             try
             {
                 VariantList variants = definition.getVariant(variant.getVariant());
@@ -1177,7 +1176,7 @@
             }
             catch (MissingVariantException e)
             {
-                if (definition.equals(loader.multipartDefinitions.get(variant)))
+                if(definition.equals(loader.multipartDefinitions.get(variant)))
                 {
                     IModel model = loader.multipartModels.get(definition);
                     if (model == null)
