--- ../src-base/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleIndexedCodec.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleIndexedCodec.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.fml.common.network.simpleimpl;
 
-import net.minecraftforge.fml.common.network.FMLIndexedMessageToMessageCodec;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
+import net.minecraftforge.fml.common.network.FMLIndexedMessageToMessageCodec;
 
 public class SimpleIndexedCodec extends FMLIndexedMessageToMessageCodec<IMessage> {
     @Override
