--- ../src-base/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleNetworkWrapper.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -20,19 +20,14 @@
 package net.minecraftforge.fml.common.network.simpleimpl;
 
 import io.netty.channel.ChannelFutureListener;
-
-import java.lang.reflect.Method;
-import java.util.EnumMap;
-
-import net.minecraft.entity.Entity;
-import net.minecraft.util.IThreadListener;
-
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelPipeline;
+import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.IThreadListener;
 import net.minecraftforge.fml.common.event.FMLInitializationEvent;
 import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
 import net.minecraftforge.fml.common.network.FMLEmbeddedChannel;
@@ -41,6 +36,9 @@
 import net.minecraftforge.fml.common.network.NetworkRegistry.TargetPoint;
 import net.minecraftforge.fml.relauncher.Side;
 
+import java.lang.reflect.Method;
+import java.util.EnumMap;
+
 /**
  * This class is a simplified netty wrapper for those not wishing to deal with the full power of netty.
  * It provides a simple message driven system, based on a discriminator byte over the custom packet channel.
