--- ../src-base/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleChannelHandlerWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/simpleimpl/SimpleChannelHandlerWrapper.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,20 +19,16 @@
 
 package net.minecraftforge.fml.common.network.simpleimpl;
 
-import org.apache.logging.log4j.Level;
-
+import cn.pfcraft.server.Mohist;
+import com.google.common.base.Preconditions;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
 import net.minecraft.network.INetHandler;
-import net.minecraftforge.fml.common.FMLLog;
 import net.minecraftforge.fml.common.network.FMLOutboundHandler;
 import net.minecraftforge.fml.common.network.NetworkRegistry;
 import net.minecraftforge.fml.relauncher.Side;
 
-import com.google.common.base.Preconditions;
-
-import io.netty.channel.ChannelFutureListener;
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.channel.SimpleChannelInboundHandler;
-
 public class SimpleChannelHandlerWrapper<REQ extends IMessage, REPLY extends IMessage> extends SimpleChannelInboundHandler<REQ> {
     private final IMessageHandler<? super REQ, ? extends REPLY> messageHandler;
     private final Side side;
@@ -64,7 +60,7 @@
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception
     {
-        FMLLog.log.error("SimpleChannelHandlerWrapper exception", cause);
+        Mohist.LOGGER.error("SimpleChannelHandlerWrapper exception", cause);
         super.exceptionCaught(ctx, cause);
     }
 }
