--- ../src-base/minecraft/net/minecraft/world/biome/BiomeHills.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeHills.java
@@ -1,6 +1,5 @@
 package net.minecraft.world.biome;
 
-import java.util.Random;
 import net.minecraft.block.BlockSilverfish;
 import net.minecraft.entity.passive.EntityLlama;
 import net.minecraft.init.Blocks;
@@ -12,22 +11,24 @@
 import net.minecraft.world.gen.feature.WorldGenTaiga2;
 import net.minecraft.world.gen.feature.WorldGenerator;
 
+import java.util.Random;
+
 public class BiomeHills extends Biome
 {
     private final WorldGenerator silverfishSpawner = new WorldGenMinable(Blocks.MONSTER_EGG.getDefaultState().withProperty(BlockSilverfish.VARIANT, BlockSilverfish.EnumType.STONE), 9);
     private final WorldGenTaiga2 spruceGenerator = new WorldGenTaiga2(false);
-    private final BiomeHills.Type type;
+    private final Type type;
 
-    public BiomeHills(BiomeHills.Type p_i46710_1_, Biome.BiomeProperties properties)
+    public BiomeHills(Type p_i46710_1_, BiomeProperties properties)
     {
         super(properties);
 
-        if (p_i46710_1_ == BiomeHills.Type.EXTRA_TREES)
+        if (p_i46710_1_ == Type.EXTRA_TREES)
         {
             this.decorator.treesPerChunk = 3;
         }
 
-        this.spawnableCreatureList.add(new Biome.SpawnListEntry(EntityLlama.class, 5, 4, 6));
+        this.spawnableCreatureList.add(new SpawnListEntry(EntityLlama.class, 5, 4, 6));
         this.type = p_i46710_1_;
     }
 
@@ -61,12 +62,12 @@
         this.topBlock = Blocks.GRASS.getDefaultState();
         this.fillerBlock = Blocks.DIRT.getDefaultState();
 
-        if ((noiseVal < -1.0D || noiseVal > 2.0D) && this.type == BiomeHills.Type.MUTATED)
+        if ((noiseVal < -1.0D || noiseVal > 2.0D) && this.type == Type.MUTATED)
         {
             this.topBlock = Blocks.GRAVEL.getDefaultState();
             this.fillerBlock = Blocks.GRAVEL.getDefaultState();
         }
-        else if (noiseVal > 1.0D && this.type != BiomeHills.Type.EXTRA_TREES)
+        else if (noiseVal > 1.0D && this.type != Type.EXTRA_TREES)
         {
             this.topBlock = Blocks.STONE.getDefaultState();
             this.fillerBlock = Blocks.STONE.getDefaultState();
