--- ../src-base/minecraft/net/minecraft/world/biome/BiomeEnd.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeEnd.java
@@ -5,10 +5,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BiomeEnd extends Biome
-{
-    public BiomeEnd(Biome.BiomeProperties properties)
-    {
+public class BiomeEnd extends Biome {
+    public BiomeEnd(Biome.BiomeProperties properties) {
         super(properties);
         this.spawnableMonsterList.clear();
         this.spawnableCreatureList.clear();
@@ -21,8 +19,7 @@
     }
 
     @SideOnly(Side.CLIENT)
-    public int getSkyColorByTemp(float currentTemperature)
-    {
+    public int getSkyColorByTemp(float currentTemperature) {
         return 0;
     }
 }
