--- ../src-base/minecraft/net/minecraft/world/biome/BiomeDesert.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeDesert.java
@@ -1,7 +1,5 @@
 package net.minecraft.world.biome;
 
-import java.util.Iterator;
-import java.util.Random;
 import net.minecraft.entity.monster.EntityHusk;
 import net.minecraft.entity.monster.EntityZombie;
 import net.minecraft.entity.monster.EntityZombieVillager;
@@ -12,9 +10,12 @@
 import net.minecraft.world.gen.feature.WorldGenDesertWells;
 import net.minecraft.world.gen.feature.WorldGenFossils;
 
+import java.util.Iterator;
+import java.util.Random;
+
 public class BiomeDesert extends Biome
 {
-    public BiomeDesert(Biome.BiomeProperties properties)
+    public BiomeDesert(BiomeProperties properties)
     {
         super(properties);
         this.spawnableCreatureList.clear();
@@ -25,12 +26,12 @@
         this.decorator.reedsPerChunk = 50;
         this.decorator.cactiPerChunk = 10;
         this.spawnableCreatureList.clear();
-        this.spawnableCreatureList.add(new Biome.SpawnListEntry(EntityRabbit.class, 4, 2, 3));
-        Iterator<Biome.SpawnListEntry> iterator = this.spawnableMonsterList.iterator();
+        this.spawnableCreatureList.add(new SpawnListEntry(EntityRabbit.class, 4, 2, 3));
+        Iterator<SpawnListEntry> iterator = this.spawnableMonsterList.iterator();
 
         while (iterator.hasNext())
         {
-            Biome.SpawnListEntry biome$spawnlistentry = iterator.next();
+            SpawnListEntry biome$spawnlistentry = iterator.next();
 
             if (biome$spawnlistentry.entityClass == EntityZombie.class || biome$spawnlistentry.entityClass == EntityZombieVillager.class)
             {
@@ -38,9 +39,9 @@
             }
         }
 
-        this.spawnableMonsterList.add(new Biome.SpawnListEntry(EntityZombie.class, 19, 4, 4));
-        this.spawnableMonsterList.add(new Biome.SpawnListEntry(EntityZombieVillager.class, 1, 1, 1));
-        this.spawnableMonsterList.add(new Biome.SpawnListEntry(EntityHusk.class, 80, 4, 4));
+        this.spawnableMonsterList.add(new SpawnListEntry(EntityZombie.class, 19, 4, 4));
+        this.spawnableMonsterList.add(new SpawnListEntry(EntityZombieVillager.class, 1, 1, 1));
+        this.spawnableMonsterList.add(new SpawnListEntry(EntityHusk.class, 80, 4, 4));
     }
 
     public void decorate(World worldIn, Random rand, BlockPos pos)
