--- ../src-base/minecraft/net/minecraft/world/biome/BiomeForestMutated.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeForestMutated.java
@@ -1,17 +1,15 @@
 package net.minecraft.world.biome;
 
-import java.util.Random;
 import net.minecraft.world.gen.feature.WorldGenAbstractTree;
 
-public class BiomeForestMutated extends BiomeForest
-{
-    public BiomeForestMutated(Biome.BiomeProperties properties)
-    {
+import java.util.Random;
+
+public class BiomeForestMutated extends BiomeForest {
+    public BiomeForestMutated(Biome.BiomeProperties properties) {
         super(BiomeForest.Type.BIRCH, properties);
     }
 
-    public WorldGenAbstractTree getRandomTreeFeature(Random rand)
-    {
+    public WorldGenAbstractTree getRandomTreeFeature(Random rand) {
         return rand.nextBoolean() ? BiomeForest.SUPER_BIRCH_TREE : BiomeForest.BIRCH_TREE;
     }
 }
