--- ../src-base/minecraft/net/minecraft/world/biome/BiomeEndDecorator.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeEndDecorator.java
@@ -7,17 +7,18 @@
 import com.google.common.collect.DiscreteDomain;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Range;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.gen.feature.WorldGenSpikes;
+
 import java.util.Collections;
 import java.util.List;
 import java.util.Random;
 import java.util.concurrent.TimeUnit;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.world.World;
-import net.minecraft.world.gen.feature.WorldGenSpikes;
 
 public class BiomeEndDecorator extends BiomeDecorator
 {
-    private static final LoadingCache<Long, WorldGenSpikes.EndSpike[]> SPIKE_CACHE = CacheBuilder.newBuilder().expireAfterWrite(5L, TimeUnit.MINUTES).<Long, WorldGenSpikes.EndSpike[]>build(new BiomeEndDecorator.SpikeCacheLoader());
+    private static final LoadingCache<Long, WorldGenSpikes.EndSpike[]> SPIKE_CACHE = CacheBuilder.newBuilder().expireAfterWrite(5L, TimeUnit.MINUTES).<Long, WorldGenSpikes.EndSpike[]>build(new SpikeCacheLoader());
     private final WorldGenSpikes spikeGen = new WorldGenSpikes();
 
     protected void genDecorations(Biome biomeIn, World worldIn, Random random)
