--- ../src-base/minecraft/net/minecraft/world/biome/BiomeSnow.java
+++ ../src-work/minecraft/net/minecraft/world/biome/BiomeSnow.java
@@ -1,7 +1,5 @@
 package net.minecraft.world.biome;
 
-import java.util.Iterator;
-import java.util.Random;
 import net.minecraft.entity.monster.EntityPolarBear;
 import net.minecraft.entity.monster.EntitySkeleton;
 import net.minecraft.entity.monster.EntityStray;
@@ -14,13 +12,16 @@
 import net.minecraft.world.gen.feature.WorldGenIceSpike;
 import net.minecraft.world.gen.feature.WorldGenTaiga2;
 
+import java.util.Iterator;
+import java.util.Random;
+
 public class BiomeSnow extends Biome
 {
     private final boolean superIcy;
     private final WorldGenIceSpike iceSpike = new WorldGenIceSpike();
     private final WorldGenIcePath icePatch = new WorldGenIcePath(4);
 
-    public BiomeSnow(boolean superIcyIn, Biome.BiomeProperties properties)
+    public BiomeSnow(boolean superIcyIn, BiomeProperties properties)
     {
         super(properties);
         this.superIcy = superIcyIn;
@@ -31,13 +32,13 @@
         }
 
         this.spawnableCreatureList.clear();
-        this.spawnableCreatureList.add(new Biome.SpawnListEntry(EntityRabbit.class, 10, 2, 3));
-        this.spawnableCreatureList.add(new Biome.SpawnListEntry(EntityPolarBear.class, 1, 1, 2));
-        Iterator<Biome.SpawnListEntry> iterator = this.spawnableMonsterList.iterator();
+        this.spawnableCreatureList.add(new SpawnListEntry(EntityRabbit.class, 10, 2, 3));
+        this.spawnableCreatureList.add(new SpawnListEntry(EntityPolarBear.class, 1, 1, 2));
+        Iterator<SpawnListEntry> iterator = this.spawnableMonsterList.iterator();
 
         while (iterator.hasNext())
         {
-            Biome.SpawnListEntry biome$spawnlistentry = iterator.next();
+            SpawnListEntry biome$spawnlistentry = iterator.next();
 
             if (biome$spawnlistentry.entityClass == EntitySkeleton.class)
             {
@@ -45,8 +46,8 @@
             }
         }
 
-        this.spawnableMonsterList.add(new Biome.SpawnListEntry(EntitySkeleton.class, 20, 4, 4));
-        this.spawnableMonsterList.add(new Biome.SpawnListEntry(EntityStray.class, 80, 4, 4));
+        this.spawnableMonsterList.add(new SpawnListEntry(EntitySkeleton.class, 20, 4, 4));
+        this.spawnableMonsterList.add(new SpawnListEntry(EntityStray.class, 80, 4, 4));
     }
 
     public float getSpawningChance()
