--- ../src-base/minecraft/net/minecraftforge/server/console/ConsoleCommandCompleter.java
+++ ../src-work/minecraft/net/minecraftforge/server/console/ConsoleCommandCompleter.java
@@ -19,21 +19,20 @@
 
 package net.minecraftforge.server.console;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import java.util.List;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
-
+import net.minecraft.server.dedicated.DedicatedServer;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-
-import net.minecraft.server.dedicated.DedicatedServer;
 import org.jline.reader.Candidate;
 import org.jline.reader.Completer;
 import org.jline.reader.LineReader;
 import org.jline.reader.ParsedLine;
 
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
 final class ConsoleCommandCompleter implements Completer
 {
     private static final Logger logger = LogManager.getLogger();
