--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/deobf/LZMAInputSupplier.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/deobf/LZMAInputSupplier.java
@@ -19,24 +19,21 @@
 
 package net.minecraftforge.fml.common.asm.transformers.deobf;
 
-import java.io.IOException;
-import java.io.InputStream;
-
 import LZMA.LzmaInputStream;
-
 import com.google.common.io.ByteSource;
 
+import java.io.IOException;
+import java.io.InputStream;
+
 public class LZMAInputSupplier extends ByteSource {
     private InputStream compressedData;
 
-    public LZMAInputSupplier(InputStream compressedData)
-    {
+    public LZMAInputSupplier(InputStream compressedData) {
         this.compressedData = compressedData;
     }
 
     @Override
-    public InputStream openStream() throws IOException
-    {
+    public InputStream openStream() throws IOException {
         return new LzmaInputStream(this.compressedData);
     }
 
