--- ../src-base/minecraft/net/minecraft/entity/passive/EntityWaterMob.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityWaterMob.java
@@ -5,67 +5,53 @@
 import net.minecraft.util.DamageSource;
 import net.minecraft.world.World;
 
-public abstract class EntityWaterMob extends EntityLiving implements IAnimals
-{
-    public EntityWaterMob(World worldIn)
-    {
+public abstract class EntityWaterMob extends EntityLiving implements IAnimals {
+    public EntityWaterMob(World worldIn) {
         super(worldIn);
     }
 
-    public boolean canBreatheUnderwater()
-    {
+    public boolean canBreatheUnderwater() {
         return true;
     }
 
-    public boolean getCanSpawnHere()
-    {
+    public boolean getCanSpawnHere() {
         return true;
     }
 
-    public boolean isNotColliding()
-    {
+    public boolean isNotColliding() {
         return this.world.checkNoEntityCollision(this.getEntityBoundingBox(), this);
     }
 
-    public int getTalkInterval()
-    {
+    public int getTalkInterval() {
         return 120;
     }
 
-    protected boolean canDespawn()
-    {
+    public boolean canDespawn() {
         return true;
     }
 
-    protected int getExperiencePoints(EntityPlayer player)
-    {
+    protected int getExperiencePoints(EntityPlayer player) {
         return 1 + this.world.rand.nextInt(3);
     }
 
-    public void onEntityUpdate()
-    {
+    public void onEntityUpdate() {
         int i = this.getAir();
         super.onEntityUpdate();
 
-        if (this.isEntityAlive() && !this.isInWater())
-        {
+        if (this.isEntityAlive() && !this.isInWater()) {
             --i;
             this.setAir(i);
 
-            if (this.getAir() == -20)
-            {
+            if (this.getAir() == -20) {
                 this.setAir(0);
                 this.attackEntityFrom(DamageSource.DROWN, 2.0F);
             }
-        }
-        else
-        {
+        } else {
             this.setAir(300);
         }
     }
 
-    public boolean isPushedByWater()
-    {
+    public boolean isPushedByWater() {
         return false;
     }
 }
