--- ../src-base/minecraft/net/minecraft/entity/passive/EntityTameable.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityTameable.java
@@ -1,8 +1,6 @@
 package net.minecraft.entity.passive;
 
 import com.google.common.base.Optional;
-import java.util.UUID;
-import javax.annotation.Nullable;
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
@@ -22,6 +20,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.UUID;
+
 public abstract class EntityTameable extends EntityAnimal implements IEntityOwnable
 {
     protected static final DataParameter<Byte> TAMED = EntityDataManager.<Byte>createKey(EntityTameable.class, DataSerializers.BYTE);
@@ -223,6 +224,11 @@
         return this.aiSit;
     }
 
+    public void setAISit(EntityAISit aiSit)
+    {
+        this.aiSit = aiSit;
+    }
+
     public boolean shouldAttackEntity(EntityLivingBase target, EntityLivingBase owner)
     {
         return true;
