--- ../src-base/minecraft/net/minecraft/entity/passive/EntityZombieHorse.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityZombieHorse.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.passive;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EnumCreatureAttribute;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.player.EntityPlayer;
@@ -15,6 +14,8 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.loot.LootTableList;
 
+import javax.annotation.Nullable;
+
 public class EntityZombieHorse extends AbstractHorse
 {
     public EntityZombieHorse(World worldIn)
