--- ../src-base/minecraft/net/minecraft/entity/passive/EntityHorse.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityHorse.java
@@ -1,7 +1,5 @@
 package net.minecraft.entity.passive;
 
-import java.util.UUID;
-import javax.annotation.Nullable;
 import net.minecraft.block.SoundType;
 import net.minecraft.entity.EntityAgeable;
 import net.minecraft.entity.IEntityLivingData;
@@ -29,6 +27,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.UUID;
+
 public class EntityHorse extends AbstractHorse
 {
     private static final UUID ARMOR_MODIFIER_UUID = UUID.fromString("556E1665-8B10-40C8-8F9D-CF9B1667F295");
@@ -398,14 +399,14 @@
         livingdata = super.onInitialSpawn(difficulty, livingdata);
         int i;
 
-        if (livingdata instanceof EntityHorse.GroupData)
+        if (livingdata instanceof GroupData)
         {
-            i = ((EntityHorse.GroupData)livingdata).variant;
+            i = ((GroupData)livingdata).variant;
         }
         else
         {
             i = this.rand.nextInt(7);
-            livingdata = new EntityHorse.GroupData(i);
+            livingdata = new GroupData(i);
         }
 
         this.setHorseVariant(i | this.rand.nextInt(5) << 8);
