--- ../src-base/minecraft/net/minecraft/entity/passive/EntitySkeletonHorse.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntitySkeletonHorse.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.passive;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EnumCreatureAttribute;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.EntityAISkeletonRiders;
@@ -17,6 +16,8 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.loot.LootTableList;
 
+import javax.annotation.Nullable;
+
 public class EntitySkeletonHorse extends AbstractHorse
 {
     private final EntityAISkeletonRiders skeletonTrapAI = new EntityAISkeletonRiders(this);
