--- ../src-base/minecraft/net/minecraft/client/tutorial/PunchTreeStep.java
+++ ../src-work/minecraft/net/minecraft/client/tutorial/PunchTreeStep.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.tutorial;
 
 import com.google.common.collect.Sets;
-import java.util.Set;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.entity.EntityPlayerSP;
@@ -17,6 +16,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public class PunchTreeStep implements ITutorialStep
 {
