--- ../src-base/minecraft/net/minecraft/client/tutorial/TutorialSteps.java
+++ ../src-work/minecraft/net/minecraft/client/tutorial/TutorialSteps.java
@@ -1,12 +1,12 @@
 package net.minecraft.client.tutorial;
 
-import java.util.function.Function;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.function.Function;
+
 @SideOnly(Side.CLIENT)
-public enum TutorialSteps
-{
+public enum TutorialSteps {
     MOVEMENT("movement", MovementStep::new),
     FIND_TREE("find_tree", FindTreeStep::new),
     PUNCH_TREE("punch_tree", PunchTreeStep::new),
@@ -15,34 +15,28 @@
     NONE("none", CompletedTutorialStep::new);
 
     private final String name;
-    private final Function < Tutorial, ? extends ITutorialStep > tutorial;
+    private final Function<Tutorial, ? extends ITutorialStep> tutorial;
 
-    private <T extends ITutorialStep> TutorialSteps(String nameIn, Function<Tutorial, T> constructor)
-    {
+    private <T extends ITutorialStep> TutorialSteps(String nameIn, Function<Tutorial, T> constructor) {
         this.name = nameIn;
         this.tutorial = constructor;
     }
 
-    public ITutorialStep create(Tutorial tutorial)
-    {
-        return this.tutorial.apply(tutorial);
-    }
-
-    public String getName()
-    {
-        return this.name;
-    }
-
-    public static TutorialSteps getTutorial(String tutorialName)
-    {
-        for (TutorialSteps tutorialsteps : values())
-        {
-            if (tutorialsteps.name.equals(tutorialName))
-            {
+    public static TutorialSteps getTutorial(String tutorialName) {
+        for (TutorialSteps tutorialsteps : values()) {
+            if (tutorialsteps.name.equals(tutorialName)) {
                 return tutorialsteps;
             }
         }
 
         return NONE;
     }
+
+    public ITutorialStep create(Tutorial tutorial) {
+        return this.tutorial.apply(tutorial);
+    }
+
+    public String getName() {
+        return this.name;
+    }
 }
