--- ../src-base/minecraft/net/minecraftforge/common/chunkio/ChunkIOProvider.java
+++ ../src-work/minecraft/net/minecraftforge/common/chunkio/ChunkIOProvider.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,11 +19,11 @@
 
 package net.minecraftforge.common.chunkio;
 
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.ChunkPos;
 import net.minecraft.world.chunk.Chunk;
 import net.minecraft.world.chunk.storage.AnvilChunkLoader;
 import net.minecraft.world.gen.ChunkProviderServer;
-import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.util.math.ChunkPos;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.world.ChunkDataEvent;
 
@@ -73,14 +73,14 @@
                 {
                     throw new RuntimeException(e); // Allow exception to bubble up to afterExecute
                 }
-    
+
                 if (data != null)
                 {
                     this.nbt   = (NBTTagCompound)data[1];
                     this.chunk = (Chunk)data[0];
                 }
             }
-            finally 
+            finally
             {
                 this.ran = true;
                 this.notifyAll();
@@ -107,7 +107,7 @@
 
         provider.id2ChunkMap.put(ChunkPos.asLong(this.chunkInfo.x, this.chunkInfo.z), this.chunk);
         this.chunk.onLoad();
-        this.chunk.populate(provider, provider.chunkGenerator);
+        this.chunk.populateCB(provider, provider.chunkGenerator, false);
 
         this.runCallbacks();
     }
@@ -137,8 +137,8 @@
         this.callbacks.clear();
     }
 
-    public QueuedChunk getChunkInfo() 
+    public QueuedChunk getChunkInfo()
     {
-    	return chunkInfo;
+        return chunkInfo;
     }
 }
