--- ../src-base/minecraft/net/minecraft/client/main/GameConfiguration.java
+++ ../src-work/minecraft/net/minecraft/client/main/GameConfiguration.java
@@ -1,25 +1,26 @@
 package net.minecraft.client.main;
 
 import com.mojang.authlib.properties.PropertyMap;
-import java.io.File;
-import java.net.Proxy;
-import javax.annotation.Nullable;
 import net.minecraft.client.resources.ResourceIndex;
 import net.minecraft.client.resources.ResourceIndexFolder;
 import net.minecraft.util.Session;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.io.File;
+import java.net.Proxy;
+
 @SideOnly(Side.CLIENT)
 public class GameConfiguration
 {
-    public final GameConfiguration.UserInformation userInfo;
-    public final GameConfiguration.DisplayInformation displayInfo;
-    public final GameConfiguration.FolderInformation folderInfo;
-    public final GameConfiguration.GameInformation gameInfo;
-    public final GameConfiguration.ServerInformation serverInfo;
+    public final UserInformation userInfo;
+    public final DisplayInformation displayInfo;
+    public final FolderInformation folderInfo;
+    public final GameInformation gameInfo;
+    public final ServerInformation serverInfo;
 
-    public GameConfiguration(GameConfiguration.UserInformation userInfoIn, GameConfiguration.DisplayInformation displayInfoIn, GameConfiguration.FolderInformation folderInfoIn, GameConfiguration.GameInformation gameInfoIn, GameConfiguration.ServerInformation serverInfoIn)
+    public GameConfiguration(UserInformation userInfoIn, DisplayInformation displayInfoIn, FolderInformation folderInfoIn, GameInformation gameInfoIn, ServerInformation serverInfoIn)
     {
         this.userInfo = userInfoIn;
         this.displayInfo = displayInfoIn;
