--- ../src-base/minecraft/net/minecraft/creativetab/CreativeTabs.java
+++ ../src-work/minecraft/net/minecraft/creativetab/CreativeTabs.java
@@ -1,6 +1,5 @@
 package net.minecraft.creativetab;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.enchantment.EnumEnchantmentType;
 import net.minecraft.init.Blocks;
@@ -13,6 +12,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public abstract class CreativeTabs
 {
     public static CreativeTabs[] CREATIVE_TAB_ARRAY = new CreativeTabs[12];
