--- ../src-base/minecraft/net/minecraftforge/event/entity/player/AttackEntityEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/AttackEntityEvent.java
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.event.entity.player;
 
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
 /**
  * AttackEntityEvent is fired when a player attacks an Entity.<br>
@@ -39,17 +39,15 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class AttackEntityEvent extends PlayerEvent
-{
+public class AttackEntityEvent extends PlayerEvent {
     private final Entity target;
-    public AttackEntityEvent(EntityPlayer player, Entity target)
-    {
+
+    public AttackEntityEvent(EntityPlayer player, Entity target) {
         super(player);
         this.target = target;
     }
 
-    public Entity getTarget()
-    {
+    public Entity getTarget() {
         return target;
     }
 }
