--- ../src-base/minecraft/net/minecraftforge/event/entity/player/ItemTooltipEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/ItemTooltipEvent.java
@@ -19,8 +19,6 @@
 
 package net.minecraftforge.event.entity.player;
 
-import java.util.List;
-
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.entity.player.EntityPlayer;
@@ -28,9 +26,9 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.util.List;
 
-public class ItemTooltipEvent extends PlayerEvent
-{
+public class ItemTooltipEvent extends PlayerEvent {
     private final ITooltipFlag flags;
     @Nonnull
     private final ItemStack itemStack;
@@ -40,8 +38,7 @@
      * This event is fired in {@link ItemStack#getTooltip(EntityPlayer, ITooltipFlag)}, which in turn is called from it's respective GUIContainer.
      * Tooltips are also gathered with a null entityPlayer during startup by {@link Minecraft#populateSearchTreeManager()}.
      */
-    public ItemTooltipEvent(@Nonnull ItemStack itemStack, @Nullable EntityPlayer entityPlayer, List<String> toolTip, ITooltipFlag flags)
-    {
+    public ItemTooltipEvent(@Nonnull ItemStack itemStack, @Nullable EntityPlayer entityPlayer, List<String> toolTip, ITooltipFlag flags) {
         super(entityPlayer);
         this.itemStack = itemStack;
         this.toolTip = toolTip;
@@ -51,8 +48,7 @@
     /**
      * Use to determine if the advanced information on item tooltips is being shown, toggled by F3+H.
      */
-    public ITooltipFlag getFlags()
-    {
+    public ITooltipFlag getFlags() {
         return flags;
     }
 
@@ -60,16 +56,14 @@
      * The {@link ItemStack} with the tooltip.
      */
     @Nonnull
-    public ItemStack getItemStack()
-    {
+    public ItemStack getItemStack() {
         return itemStack;
     }
 
     /**
      * The {@link ItemStack} tooltip.
      */
-    public List<String> getToolTip()
-    {
+    public List<String> getToolTip() {
         return toolTip;
     }
 
@@ -78,8 +72,7 @@
      */
     @Override
     @Nullable
-    public EntityPlayer getEntityPlayer()
-    {
+    public EntityPlayer getEntityPlayer() {
         return super.getEntityPlayer();
     }
 }
