--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerDropsEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerDropsEvent.java
@@ -19,24 +19,22 @@
 
 package net.minecraftforge.event.entity.player;
 
-import java.util.List;
-
-import net.minecraftforge.common.ForgeHooks;
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
-
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.DamageSource;
+import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.event.entity.living.LivingDropsEvent;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
+import java.util.List;
+
 /**
  * Child class of LivingDropEvent that is fired specifically when a
  * player dies.  Canceling the event will prevent ALL drops from entering the
  * world.
  */
 @Cancelable
-public class PlayerDropsEvent extends LivingDropsEvent
-{
+public class PlayerDropsEvent extends LivingDropsEvent {
     private final EntityPlayer entityPlayer;
 
     /**
@@ -46,15 +44,13 @@
      * @param source The source of the damage which is killing the player.
      * @param drops List of all drops entering the world.
      */
-    public PlayerDropsEvent(EntityPlayer entity, DamageSource source, List<EntityItem> drops, boolean recentlyHit)
-    {
+    public PlayerDropsEvent(EntityPlayer entity, DamageSource source, List<EntityItem> drops, boolean recentlyHit) {
         super(entity, source, drops, ForgeHooks.getLootingLevel(entity, source.getTrueSource(), source), recentlyHit);
 
         this.entityPlayer = entity;
     }
 
-    public EntityPlayer getEntityPlayer()
-    {
+    public EntityPlayer getEntityPlayer() {
         return entityPlayer;
     }
 }
