--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerContainerEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerContainerEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
