--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerInteractEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerInteractEvent.java
@@ -50,6 +50,7 @@
     private final BlockPos pos;
     @Nullable
     private final EnumFacing face;
+    public org.bukkit.event.player.PlayerInteractEvent cb;
     private EnumActionResult cancellationResult = EnumActionResult.PASS;
 
     private PlayerInteractEvent(EntityPlayer player, EnumHand hand, BlockPos pos, @Nullable EnumFacing face)
