--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerPickupXpEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerPickupXpEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.event.entity.player;
 
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraft.entity.item.EntityXPOrb;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
 /**
  * This event is called when a player collides with a EntityXPOrb on the ground.
