--- ../src-base/minecraft/net/minecraftforge/energy/EnergyStorage.java
+++ ../src-work/minecraft/net/minecraftforge/energy/EnergyStorage.java
@@ -25,39 +25,33 @@
  * Derived from the Redstone Flux power system designed by King Lemming and originally utilized in Thermal Expansion and related mods.
  * Created with consent and permission of King Lemming and Team CoFH. Released with permission under LGPL 2.1 when bundled with Forge.
  */
-public class EnergyStorage implements IEnergyStorage
-{
+public class EnergyStorage implements IEnergyStorage {
     protected int energy;
     protected int capacity;
     protected int maxReceive;
     protected int maxExtract;
 
-    public EnergyStorage(int capacity)
-    {
+    public EnergyStorage(int capacity) {
         this(capacity, capacity, capacity, 0);
     }
 
-    public EnergyStorage(int capacity, int maxTransfer)
-    {
+    public EnergyStorage(int capacity, int maxTransfer) {
         this(capacity, maxTransfer, maxTransfer, 0);
     }
 
-    public EnergyStorage(int capacity, int maxReceive, int maxExtract)
-    {
+    public EnergyStorage(int capacity, int maxReceive, int maxExtract) {
         this(capacity, maxReceive, maxExtract, 0);
     }
 
-    public EnergyStorage(int capacity, int maxReceive, int maxExtract, int energy)
-    {
+    public EnergyStorage(int capacity, int maxReceive, int maxExtract, int energy) {
         this.capacity = capacity;
         this.maxReceive = maxReceive;
         this.maxExtract = maxExtract;
-        this.energy = Math.max(0 , Math.min(capacity, energy));
+        this.energy = Math.max(0, Math.min(capacity, energy));
     }
 
     @Override
-    public int receiveEnergy(int maxReceive, boolean simulate)
-    {
+    public int receiveEnergy(int maxReceive, boolean simulate) {
         if (!canReceive())
             return 0;
 
@@ -68,8 +62,7 @@
     }
 
     @Override
-    public int extractEnergy(int maxExtract, boolean simulate)
-    {
+    public int extractEnergy(int maxExtract, boolean simulate) {
         if (!canExtract())
             return 0;
 
@@ -80,26 +73,22 @@
     }
 
     @Override
-    public int getEnergyStored()
-    {
+    public int getEnergyStored() {
         return energy;
     }
 
     @Override
-    public int getMaxEnergyStored()
-    {
+    public int getMaxEnergyStored() {
         return capacity;
     }
 
     @Override
-    public boolean canExtract()
-    {
+    public boolean canExtract() {
         return this.maxExtract > 0;
     }
 
     @Override
-    public boolean canReceive()
-    {
+    public boolean canReceive() {
         return this.maxReceive > 0;
     }
 }
