--- ../src-base/minecraft/net/minecraftforge/server/command/CommandEntity.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/CommandEntity.java
@@ -25,6 +25,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Queue;
 import java.util.Set;
 import java.util.stream.Collectors;
 
@@ -112,7 +113,7 @@
                 throw new WrongUsageException("commands.forge.entity.list.invalidworld", dim);
 
             Map<ResourceLocation, MutablePair<Integer, Map<ChunkPos, Integer>>> list = Maps.newHashMap();
-            List<Entity> entities = world.loadedEntityList;
+            Queue<Entity> entities = world.loadedEntityList;
             entities.forEach(e -> {
                 ResourceLocation key = EntityList.getKey(e);
 
