--- ../src-base/minecraft/net/minecraftforge/server/command/CommandDimensions.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/CommandDimensions.java
@@ -30,38 +30,31 @@
 
 import java.util.Map;
 
-public class CommandDimensions extends CommandBase
-{
+public class CommandDimensions extends CommandBase {
     @Override
-    public String getName()
-    {
+    public String getName() {
         return "dimensions";
     }
 
     @Override
-    public String getUsage(ICommandSender sender)
-    {
+    public String getUsage(ICommandSender sender) {
         return "commands.forge.dimensions.usage";
     }
 
     @Override
-    public int getRequiredPermissionLevel()
-    {
+    public int getRequiredPermissionLevel() {
         return 0;
     }
 
     @Override
-    public boolean checkPermission(MinecraftServer server, ICommandSender sender)
-    {
+    public boolean checkPermission(MinecraftServer server, ICommandSender sender) {
         return true;
     }
 
     @Override
-    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
-    {
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
         sender.sendMessage(TextComponentHelper.createComponentTranslation(sender, "commands.forge.dimensions.list"));
-        for (Map.Entry<DimensionType, IntSortedSet> entry : DimensionManager.getRegisteredDimensions().entrySet())
-        {
+        for (Map.Entry<DimensionType, IntSortedSet> entry : DimensionManager.getRegisteredDimensions().entrySet()) {
             sender.sendMessage(new TextComponentString(entry.getKey().getName() + ": " + entry.getValue()));
         }
     }
