--- ../src-base/minecraft/net/minecraftforge/server/command/CommandGenerate.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/CommandGenerate.java
@@ -19,10 +19,6 @@
 
 package net.minecraftforge.server.command;
 
-import javax.annotation.Nullable;
-import java.util.Collections;
-import java.util.List;
-
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
@@ -31,38 +27,35 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraftforge.common.WorldWorkerManager;
 
-class CommandGenerate extends CommandBase
-{
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+class CommandGenerate extends CommandBase {
     @Override
-    public String getName()
-    {
+    public String getName() {
         return "generate";
     }
 
     @Override
-    public List<String> getAliases()
-    {
+    public List<String> getAliases() {
         return Collections.singletonList("gen");
     }
 
     @Override
-    public String getUsage(ICommandSender sender)
-    {
+    public String getUsage(ICommandSender sender) {
         return "commands.forge.gen.usage";
     }
 
     @Override
-    public int getRequiredPermissionLevel()
-    {
+    public int getRequiredPermissionLevel() {
         return 4;
     }
 
     @Override
-    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
-    {
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
         // x y z chunkCount [dim] [interval]
-        if (args.length < 4)
-        {
+        if (args.length < 4) {
             throw new WrongUsageException("commands.forge.gen.usage");
         }
 
@@ -78,10 +71,8 @@
     }
 
     @Override
-    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos)
-    {
-        if (args.length < 4)
-        {
+    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos) {
+        if (args.length < 4) {
             return getTabCompletionCoordinate(args, 0, targetPos);
         }
         // Chunk Count? No completion
