--- ../src-base/minecraft/net/minecraftforge/server/command/CommandGenerate.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/CommandGenerate.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -16,13 +16,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
-
 package net.minecraftforge.server.command;
 
-import javax.annotation.Nullable;
-import java.util.Collections;
-import java.util.List;
-
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
@@ -31,6 +26,10 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraftforge.common.WorldWorkerManager;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 class CommandGenerate extends CommandBase
 {
     @Override
