--- ../src-base/minecraft/net/minecraftforge/server/command/ForgeCommand.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/ForgeCommand.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -20,7 +20,6 @@
 package net.minecraftforge.server.command;
 
 import net.minecraft.command.ICommand;
-
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
 
