--- ../src-base/minecraft/net/minecraftforge/server/command/package-info.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/package-info.java
@@ -21,6 +21,6 @@
 @MethodsReturnNonnullByDefault
 package net.minecraftforge.server.command;
 
-import javax.annotation.ParametersAreNonnullByDefault;
-
 import mcp.MethodsReturnNonnullByDefault;
+
+import javax.annotation.ParametersAreNonnullByDefault;
