--- ../src-base/minecraft/net/minecraft/client/renderer/tileentity/TileEntityEndPortalRenderer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/tileentity/TileEntityEndPortalRenderer.java
@@ -1,7 +1,5 @@
 package net.minecraft.client.renderer.tileentity;
 
-import java.nio.FloatBuffer;
-import java.util.Random;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GLAllocation;
@@ -14,6 +12,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.FloatBuffer;
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class TileEntityEndPortalRenderer extends TileEntitySpecialRenderer<TileEntityEndPortal>
 {
