--- ../src-base/minecraft/ibxm/FastTracker2.java
+++ ../src-work/minecraft/ibxm/FastTracker2.java
@@ -1,7 +1,10 @@
 
 package ibxm;
 
-import java.io.*;
+import java.io.DataInput;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
 
 public class FastTracker2 {
 	public static boolean is_xm( byte[] header_60_bytes ) {
@@ -254,4 +257,3 @@
 		return string;
 	}
 }
-
