--- ../src-base/minecraft/net/minecraft/world/gen/structure/template/BlockRotationProcessor.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/template/BlockRotationProcessor.java
@@ -1,10 +1,11 @@
 package net.minecraft.world.gen.structure.template;
 
-import java.util.Random;
-import javax.annotation.Nullable;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+import java.util.Random;
+
 public class BlockRotationProcessor implements ITemplateProcessor
 {
     private final float chance;
