--- ../src-base/minecraft/net/minecraft/world/gen/structure/template/ITemplateProcessor.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/template/ITemplateProcessor.java
@@ -1,11 +1,11 @@
 package net.minecraft.world.gen.structure.template;
 
-import javax.annotation.Nullable;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
-public interface ITemplateProcessor
-{
+import javax.annotation.Nullable;
+
+public interface ITemplateProcessor {
     @Nullable
     Template.BlockInfo processBlock(World worldIn, BlockPos pos, Template.BlockInfo blockInfoIn);
 }
