--- ../src-base/minecraft/net/minecraft/world/gen/structure/template/PlacementSettings.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/template/PlacementSettings.java
@@ -1,7 +1,5 @@
 package net.minecraft.world.gen.structure.template;
 
-import java.util.Random;
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.util.Mirror;
 import net.minecraft.util.Rotation;
@@ -9,6 +7,9 @@
 import net.minecraft.util.math.ChunkPos;
 import net.minecraft.world.gen.structure.StructureBoundingBox;
 
+import javax.annotation.Nullable;
+import java.util.Random;
+
 public class PlacementSettings
 {
     private Mirror mirror = Mirror.NONE;
