--- ../src-base/minecraft/net/minecraftforge/fml/common/launcher/FMLServerTweaker.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/launcher/FMLServerTweaker.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,14 +19,13 @@
 
 package net.minecraftforge.fml.common.launcher;
 
-import java.io.File;
-import java.util.List;
-
+import net.minecraft.launchwrapper.LaunchClassLoader;
+import net.minecraftforge.fml.relauncher.FMLLaunchHandler;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.core.LoggerContext;
 
-import net.minecraft.launchwrapper.LaunchClassLoader;
-import net.minecraftforge.fml.relauncher.FMLLaunchHandler;
+import java.io.File;
+import java.util.List;
 
 public class FMLServerTweaker extends FMLTweaker {
 
@@ -37,7 +36,7 @@
 
         if (System.getProperty("log4j.configurationFile") == null)
         {
-            System.setProperty("log4j.configurationFile", "log4j2_server.xml");
+            System.setProperty("log4j.configurationFile", "log4j2.xml");
             ((LoggerContext) LogManager.getContext(false)).reconfigure();
         }
     }
@@ -55,9 +54,9 @@
         // and deobfuscated parts of the code. Without, the UI won't show anything
         classLoader.addClassLoaderExclusion("com.mojang.util.QueueLogAppender");
 
-        classLoader.addClassLoaderExclusion("org.jline.");
-        classLoader.addClassLoaderExclusion("com.sun.jna.");
-        classLoader.addClassLoaderExclusion("net.minecraftforge.server.terminalconsole.");
+        classLoader.addClassLoaderExclusion("jline.");
+        classLoader.addClassLoaderExclusion("org.fusesource.");
+        classLoader.addClassLoaderExclusion("net.minecraftforge.server.console.log4j.TerminalConsoleAppender");
 
         FMLLaunchHandler.configureForServerLaunch(classLoader, this);
         FMLLaunchHandler.appendCoreMods();
