--- ../src-base/minecraft/net/minecraftforge/fml/common/launcher/FMLServerTweaker.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/launcher/FMLServerTweaker.java
@@ -37,7 +37,7 @@
 
         if (System.getProperty("log4j.configurationFile") == null)
         {
-            System.setProperty("log4j.configurationFile", "log4j2_server.xml");
+            System.setProperty("log4j.configurationFile", "log4j2_mohist.xml");
             ((LoggerContext) LogManager.getContext(false)).reconfigure();
         }
     }
@@ -61,5 +61,10 @@
 
         FMLLaunchHandler.configureForServerLaunch(classLoader, this);
         FMLLaunchHandler.appendCoreMods();
+
+//        Mohist 把字节码转换委托出去
+//        Mohist start
+        classLoader.registerTransformer("red.mohist.common.asm.CompositeClassTransformer");
+//        Mohist end
     }
 }
