--- ../src-base/minecraft/net/minecraftforge/fml/common/launcher/FMLTweaker.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/launcher/FMLTweaker.java
@@ -51,7 +51,9 @@
         }
         try
         {
-            System.setSecurityManager(new FMLSecurityManager());
+            FMLSecurityManager sm = new FMLSecurityManager();
+            sm.defman = System.getSecurityManager();
+            System.setSecurityManager(sm);
         }
         catch (SecurityException se)
         {
