--- ../src-base/minecraft/net/minecraftforge/fml/common/launcher/Yggdrasil.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/launcher/Yggdrasil.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,16 +19,15 @@
 
 package net.minecraftforge.fml.common.launcher;
 
-import java.net.Proxy;
-import java.util.Map;
-
-import org.apache.logging.log4j.LogManager;
-
 import com.mojang.authlib.Agent;
 import com.mojang.authlib.exceptions.AuthenticationException;
 import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
 import com.mojang.authlib.yggdrasil.YggdrasilUserAuthentication;
+import org.apache.logging.log4j.LogManager;
 
+import java.net.Proxy;
+import java.util.Map;
+
 /**
  * Basic implementation of Mojang's 'Yggdrasil' login system, purely intended as a dev time bare bones login.
  * Login errors are not handled.
