--- ../src-base/minecraft/net/minecraft/tileentity/BannerPattern.java
+++ ../src-work/minecraft/net/minecraft/tileentity/BannerPattern.java
@@ -1,6 +1,5 @@
 package net.minecraft.tileentity;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.BlockFlower;
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
@@ -8,6 +7,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public enum BannerPattern
 {
     BASE("base", "b"),
