--- ../src-base/minecraft/net/minecraft/tileentity/IHopper.java
+++ ../src-work/minecraft/net/minecraft/tileentity/IHopper.java
@@ -9,7 +9,13 @@
 
     double getXPos();
 
+    default double getX() { return getXPos(); } // Paper - OBFHELPER
+
     double getYPos();
 
+    default double getY() { return getYPos(); } // Paper - OBFHELPER
+
     double getZPos();
+
+    default double getZ() { return getZPos(); } // Paper - OBFHELPER
 }
