--- ../src-base/minecraft/net/minecraft/tileentity/TileEntityComparator.java
+++ ../src-work/minecraft/net/minecraft/tileentity/TileEntityComparator.java
@@ -2,30 +2,25 @@
 
 import net.minecraft.nbt.NBTTagCompound;
 
-public class TileEntityComparator extends TileEntity
-{
+public class TileEntityComparator extends TileEntity {
     private int outputSignal;
 
-    public NBTTagCompound writeToNBT(NBTTagCompound compound)
-    {
+    public NBTTagCompound writeToNBT(NBTTagCompound compound) {
         super.writeToNBT(compound);
         compound.setInteger("OutputSignal", this.outputSignal);
         return compound;
     }
 
-    public void readFromNBT(NBTTagCompound compound)
-    {
+    public void readFromNBT(NBTTagCompound compound) {
         super.readFromNBT(compound);
         this.outputSignal = compound.getInteger("OutputSignal");
     }
 
-    public int getOutputSignal()
-    {
+    public int getOutputSignal() {
         return this.outputSignal;
     }
 
-    public void setOutputSignal(int outputSignalIn)
-    {
+    public void setOutputSignal(int outputSignalIn) {
         this.outputSignal = outputSignalIn;
     }
 }
