--- ../src-base/minecraft/net/minecraft/tileentity/TileEntityMobSpawner.java
+++ ../src-work/minecraft/net/minecraft/tileentity/TileEntityMobSpawner.java
@@ -1,6 +1,5 @@
 package net.minecraft.tileentity;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Blocks;
 import net.minecraft.nbt.NBTTagCompound;
@@ -16,6 +15,8 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+
 public class TileEntityMobSpawner extends TileEntity implements ITickable
 {
     private final MobSpawnerBaseLogic spawnerLogic = new MobSpawnerBaseLogic()
