--- ../src-base/minecraft/net/minecraftforge/fml/common/network/internal/FMLMessage.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/internal/FMLMessage.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,18 +19,13 @@
 
 package net.minecraftforge.fml.common.network.internal;
 
+import cn.pfcraft.server.Mohist;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
-
-import java.io.IOException;
-import java.util.List;
-import java.util.UUID;
-
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.datasync.EntityDataManager;
-import net.minecraftforge.fml.common.FMLLog;
 import net.minecraftforge.fml.common.ModContainer;
 import net.minecraftforge.fml.common.network.ByteBufUtils;
 import net.minecraftforge.fml.common.registry.EntityRegistry.EntityRegistration;
@@ -38,6 +33,10 @@
 import net.minecraftforge.fml.common.registry.IThrowableEntity;
 import net.minecraftforge.fml.relauncher.Side;
 
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
 public abstract class FMLMessage {
     public static class CompleteHandshake extends FMLMessage {
         Side target;
@@ -180,7 +179,7 @@
             }
             catch (IOException e)
             {
-                FMLLog.log.fatal("Encountered fatal exception trying to send entity spawn data watchers", e);
+                Mohist.LOGGER.fatal("Encountered fatal exception trying to send entity spawn data watchers", e);
                 throw new RuntimeException(e);
             }
             buf.writeBytes(tmpBuf);
@@ -233,7 +232,7 @@
             }
             catch (IOException e)
             {
-                FMLLog.log.fatal("There was a critical error decoding the datawatcher stream for a mod entity.", e);
+                Mohist.LOGGER.fatal("There was a critical error decoding the datawatcher stream for a mod entity.", e);
                 throw new RuntimeException(e);
             }
 
