--- ../src-base/minecraft/net/minecraftforge/fml/common/network/internal/HandshakeCompletionHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/internal/HandshakeCompletionHandler.java
@@ -19,26 +19,23 @@
 
 package net.minecraftforge.fml.common.network.internal;
 
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
 import net.minecraftforge.fml.common.FMLLog;
 import net.minecraftforge.fml.common.network.handshake.NetworkDispatcher;
 import net.minecraftforge.fml.common.network.internal.FMLMessage.CompleteHandshake;
 
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.channel.SimpleChannelInboundHandler;
-import io.netty.channel.ChannelHandler.Sharable;
-
 @Sharable
 public class HandshakeCompletionHandler extends SimpleChannelInboundHandler<FMLMessage.CompleteHandshake> {
     @Override
-    protected void channelRead0(ChannelHandlerContext ctx, CompleteHandshake msg) throws Exception
-    {
+    protected void channelRead0(ChannelHandlerContext ctx, CompleteHandshake msg) throws Exception {
         NetworkDispatcher dispatcher = ctx.channel().attr(NetworkDispatcher.FML_DISPATCHER).getAndSet(null);
         dispatcher.completeHandshake(msg.target);
     }
 
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception
-    {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         FMLLog.log.error("HandshakeCompletionHandler exception", cause);
         super.exceptionCaught(ctx, cause);
     }
