--- ../src-base/minecraft/net/minecraftforge/fml/common/versioning/InvalidVersionSpecificationException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/versioning/InvalidVersionSpecificationException.java
@@ -27,12 +27,10 @@
  *
  * @author <a href="mailto:brett@apache.org">Brett Porter</a>
  */
-public class InvalidVersionSpecificationException extends Exception
-{
+public class InvalidVersionSpecificationException extends Exception {
     private static final long serialVersionUID = 1L;
 
-    public InvalidVersionSpecificationException( String message )
-    {
-        super( message );
+    public InvalidVersionSpecificationException(String message) {
+        super(message);
     }
 }
