--- ../src-base/minecraft/net/minecraft/client/resources/DefaultResourcePack.java
+++ ../src-work/minecraft/net/minecraft/client/resources/DefaultResourcePack.java
@@ -1,15 +1,6 @@
 package net.minecraft.client.resources;
 
 import com.google.common.collect.ImmutableSet;
-import java.awt.image.BufferedImage;
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URL;
-import java.util.Set;
-import javax.annotation.Nullable;
 import net.minecraft.client.renderer.texture.TextureUtil;
 import net.minecraft.client.resources.data.IMetadataSection;
 import net.minecraft.client.resources.data.MetadataSerializer;
@@ -17,6 +8,12 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.net.URL;
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public class DefaultResourcePack implements IResourcePack
 {
