--- ../src-base/minecraft/net/minecraft/client/resources/FoliageColorReloadListener.java
+++ ../src-work/minecraft/net/minecraft/client/resources/FoliageColorReloadListener.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.resources;
 
-import java.io.IOException;
 import net.minecraft.client.renderer.texture.TextureUtil;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.world.ColorizerFoliage;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class FoliageColorReloadListener implements IResourceManagerReloadListener
 {
