--- ../src-base/minecraft/net/minecraft/client/resources/IResource.java
+++ ../src-work/minecraft/net/minecraft/client/resources/IResource.java
@@ -1,16 +1,16 @@
 package net.minecraft.client.resources;
 
-import java.io.Closeable;
-import java.io.InputStream;
-import javax.annotation.Nullable;
 import net.minecraft.client.resources.data.IMetadataSection;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.io.InputStream;
+
 @SideOnly(Side.CLIENT)
-public interface IResource extends Closeable
-{
+public interface IResource extends Closeable {
     ResourceLocation getResourceLocation();
 
     InputStream getInputStream();
