--- ../src-base/minecraft/net/minecraft/client/resources/IResourceManager.java
+++ ../src-work/minecraft/net/minecraft/client/resources/IResourceManager.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.resources;
 
-import java.io.IOException;
-import java.util.List;
-import java.util.Set;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public interface IResourceManager
 {
