--- ../src-base/minecraft/net/minecraft/client/resources/LegacyV2Adapter.java
+++ ../src-work/minecraft/net/minecraft/client/resources/LegacyV2Adapter.java
@@ -1,16 +1,17 @@
 package net.minecraft.client.resources;
 
-import java.awt.image.BufferedImage;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Set;
-import javax.annotation.Nullable;
 import net.minecraft.client.resources.data.IMetadataSection;
 import net.minecraft.client.resources.data.MetadataSerializer;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public class LegacyV2Adapter implements IResourcePack
 {
