--- ../src-base/minecraft/net/minecraft/client/resources/ResourceIndexFolder.java
+++ ../src-work/minecraft/net/minecraft/client/resources/ResourceIndexFolder.java
@@ -1,27 +1,24 @@
 package net.minecraft.client.resources;
 
-import java.io.File;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.File;
+
 @SideOnly(Side.CLIENT)
-public class ResourceIndexFolder extends ResourceIndex
-{
+public class ResourceIndexFolder extends ResourceIndex {
     private final File baseDir;
 
-    public ResourceIndexFolder(File folder)
-    {
+    public ResourceIndexFolder(File folder) {
         this.baseDir = folder;
     }
 
-    public File getFile(ResourceLocation location)
-    {
+    public File getFile(ResourceLocation location) {
         return new File(this.baseDir, location.toString().replace(':', '/'));
     }
 
-    public File getPackMcmeta()
-    {
+    public File getPackMcmeta() {
         return new File(this.baseDir, "pack.mcmeta");
     }
 }
