--- ../src-base/minecraft/net/minecraft/client/resources/ResourcePackListEntryServer.java
+++ ../src-work/minecraft/net/minecraft/client/resources/ResourcePackListEntryServer.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.resources;
 
 import com.google.gson.JsonParseException;
-import java.io.IOException;
 import net.minecraft.client.gui.GuiScreenResourcePacks;
 import net.minecraft.client.renderer.texture.DynamicTexture;
 import net.minecraft.client.renderer.texture.TextureUtil;
@@ -13,6 +12,8 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class ResourcePackListEntryServer extends ResourcePackListEntry
 {
