--- ../src-base/minecraft/net/minecraft/client/resources/ResourceIndexFolder.java
+++ ../src-work/minecraft/net/minecraft/client/resources/ResourceIndexFolder.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.resources;
 
-import java.io.File;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.File;
+
 @SideOnly(Side.CLIENT)
 public class ResourceIndexFolder extends ResourceIndex
 {
