--- ../src-base/minecraft/net/minecraft/client/resources/ResourcePackListEntryDefault.java
+++ ../src-work/minecraft/net/minecraft/client/resources/ResourcePackListEntryDefault.java
@@ -6,20 +6,16 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ResourcePackListEntryDefault extends ResourcePackListEntryServer
-{
-    public ResourcePackListEntryDefault(GuiScreenResourcePacks resourcePacksGUIIn)
-    {
+public class ResourcePackListEntryDefault extends ResourcePackListEntryServer {
+    public ResourcePackListEntryDefault(GuiScreenResourcePacks resourcePacksGUIIn) {
         super(resourcePacksGUIIn, Minecraft.getMinecraft().getResourcePackRepository().rprDefaultResourcePack);
     }
 
-    protected String getResourcePackName()
-    {
+    protected String getResourcePackName() {
         return "Default";
     }
 
-    public boolean isServerPack()
-    {
+    public boolean isServerPack() {
         return false;
     }
 }
