--- ../src-base/minecraft/net/minecraft/client/resources/SimpleResource.java
+++ ../src-work/minecraft/net/minecraft/client/resources/SimpleResource.java
@@ -3,14 +3,6 @@
 import com.google.common.collect.Maps;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.io.Reader;
-import java.nio.charset.StandardCharsets;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.client.resources.data.IMetadataSection;
 import net.minecraft.client.resources.data.MetadataSerializer;
 import net.minecraft.util.ResourceLocation;
@@ -18,9 +10,13 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.io.IOUtils;
 
+import javax.annotation.Nullable;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
-public class SimpleResource implements IResource
-{
+public class SimpleResource implements IResource {
     private final Map<String, IMetadataSection> mapMetadataSections = Maps.<String, IMetadataSection>newHashMap();
     private final String resourcePackName;
     private final ResourceLocation srResourceLocation;
@@ -30,8 +26,7 @@
     private boolean mcmetaJsonChecked;
     private JsonObject mcmetaJson;
 
-    public SimpleResource(String resourcePackNameIn, ResourceLocation srResourceLocationIn, InputStream resourceInputStreamIn, InputStream mcmetaInputStreamIn, MetadataSerializer srMetadataSerializerIn)
-    {
+    public SimpleResource(String resourcePackNameIn, ResourceLocation srResourceLocationIn, InputStream resourceInputStreamIn, InputStream mcmetaInputStreamIn, MetadataSerializer srMetadataSerializerIn) {
         this.resourcePackName = resourcePackNameIn;
         this.srResourceLocation = srResourceLocationIn;
         this.resourceInputStream = resourceInputStreamIn;
@@ -39,50 +34,38 @@
         this.srMetadataSerializer = srMetadataSerializerIn;
     }
 
-    public ResourceLocation getResourceLocation()
-    {
+    public ResourceLocation getResourceLocation() {
         return this.srResourceLocation;
     }
 
-    public InputStream getInputStream()
-    {
+    public InputStream getInputStream() {
         return this.resourceInputStream;
     }
 
-    public boolean hasMetadata()
-    {
+    public boolean hasMetadata() {
         return this.mcmetaInputStream != null;
     }
 
     @Nullable
-    public <T extends IMetadataSection> T getMetadata(String sectionName)
-    {
-        if (!this.hasMetadata())
-        {
-            return (T)null;
-        }
-        else
-        {
-            if (this.mcmetaJson == null && !this.mcmetaJsonChecked)
-            {
+    public <T extends IMetadataSection> T getMetadata(String sectionName) {
+        if (!this.hasMetadata()) {
+            return (T) null;
+        } else {
+            if (this.mcmetaJson == null && !this.mcmetaJsonChecked) {
                 this.mcmetaJsonChecked = true;
                 BufferedReader bufferedreader = null;
 
-                try
-                {
+                try {
                     bufferedreader = new BufferedReader(new InputStreamReader(this.mcmetaInputStream, StandardCharsets.UTF_8));
                     this.mcmetaJson = (new JsonParser()).parse(bufferedreader).getAsJsonObject();
+                } finally {
+                    IOUtils.closeQuietly((Reader) bufferedreader);
                 }
-                finally
-                {
-                    IOUtils.closeQuietly((Reader)bufferedreader);
-                }
             }
 
-            T t = (T)this.mapMetadataSections.get(sectionName);
+            T t = (T) this.mapMetadataSections.get(sectionName);
 
-            if (t == null)
-            {
+            if (t == null) {
                 t = this.srMetadataSerializer.parseMetadataSection(sectionName, this.mcmetaJson);
             }
 
@@ -90,46 +73,31 @@
         }
     }
 
-    public String getResourcePackName()
-    {
+    public String getResourcePackName() {
         return this.resourcePackName;
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        if (this == p_equals_1_)
-        {
+    public boolean equals(Object p_equals_1_) {
+        if (this == p_equals_1_) {
             return true;
-        }
-        else if (!(p_equals_1_ instanceof SimpleResource))
-        {
+        } else if (!(p_equals_1_ instanceof SimpleResource)) {
             return false;
-        }
-        else
-        {
-            SimpleResource simpleresource = (SimpleResource)p_equals_1_;
+        } else {
+            SimpleResource simpleresource = (SimpleResource) p_equals_1_;
 
-            if (this.srResourceLocation != null)
-            {
-                if (!this.srResourceLocation.equals(simpleresource.srResourceLocation))
-                {
+            if (this.srResourceLocation != null) {
+                if (!this.srResourceLocation.equals(simpleresource.srResourceLocation)) {
                     return false;
                 }
-            }
-            else if (simpleresource.srResourceLocation != null)
-            {
+            } else if (simpleresource.srResourceLocation != null) {
                 return false;
             }
 
-            if (this.resourcePackName != null)
-            {
-                if (!this.resourcePackName.equals(simpleresource.resourcePackName))
-                {
+            if (this.resourcePackName != null) {
+                if (!this.resourcePackName.equals(simpleresource.resourcePackName)) {
                     return false;
                 }
-            }
-            else if (simpleresource.resourcePackName != null)
-            {
+            } else if (simpleresource.resourcePackName != null) {
                 return false;
             }
 
@@ -137,19 +105,16 @@
         }
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         int i = this.resourcePackName != null ? this.resourcePackName.hashCode() : 0;
         i = 31 * i + (this.srResourceLocation != null ? this.srResourceLocation.hashCode() : 0);
         return i;
     }
 
-    public void close() throws IOException
-    {
+    public void close() throws IOException {
         this.resourceInputStream.close();
 
-        if (this.mcmetaInputStream != null)
-        {
+        if (this.mcmetaInputStream != null) {
             this.mcmetaInputStream.close();
         }
     }
