--- ../src-base/minecraft/net/minecraft/client/renderer/debug/DebugRendererCollisionBox.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/debug/DebugRendererCollisionBox.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer.debug;
 
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.RenderGlobal;
@@ -10,6 +9,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class DebugRendererCollisionBox implements DebugRenderer.IDebugRenderer
 {
