--- ../src-base/minecraft/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.renderer.debug;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GlStateManager;
@@ -16,6 +15,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
 public class DebugRendererPathfinding implements DebugRenderer.IDebugRenderer
 {
