--- ../src-base/minecraft/net/minecraft/entity/IMerchant.java
+++ ../src-work/minecraft/net/minecraft/entity/IMerchant.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.BlockPos;
@@ -11,13 +10,14 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public interface IMerchant
-{
-    void setCustomer(@Nullable EntityPlayer player);
+import javax.annotation.Nullable;
 
+public interface IMerchant {
     @Nullable
     EntityPlayer getCustomer();
 
+    void setCustomer(@Nullable EntityPlayer player);
+
     @Nullable
     MerchantRecipeList getRecipes(EntityPlayer player);
 
