--- ../src-base/minecraft/net/minecraft/entity/IRangedAttackMob.java
+++ ../src-work/minecraft/net/minecraft/entity/IRangedAttackMob.java
@@ -4,5 +4,9 @@
 {
     void attackEntityWithRangedAttack(EntityLivingBase target, float distanceFactor);
 
+    default void rangedAttack(EntityLivingBase entityliving, float f) { attackEntityWithRangedAttack(entityliving, f); } // Paper OBF HELPER
+
     void setSwingingArms(boolean swingingArms);
+
+    default void setChargingAttack(boolean flag) { setSwingingArms(flag); } // Paper OBF HELPER
 }
