--- ../src-base/minecraft/net/minecraftforge/client/settings/KeyConflictContext.java
+++ ../src-work/minecraft/net/minecraftforge/client/settings/KeyConflictContext.java
@@ -22,22 +22,19 @@
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
 
-public enum KeyConflictContext implements IKeyConflictContext
-{
+public enum KeyConflictContext implements IKeyConflictContext {
     /**
      * Universal key bindings are used in every context and will conflict with any other context.
      * Key Bindings are universal by default.
      */
     UNIVERSAL {
         @Override
-        public boolean isActive()
-        {
+        public boolean isActive() {
             return true;
         }
 
         @Override
-        public boolean conflicts(IKeyConflictContext other)
-        {
+        public boolean conflicts(IKeyConflictContext other) {
             return true;
         }
     },
@@ -47,14 +44,12 @@
      */
     GUI {
         @Override
-        public boolean isActive()
-        {
+        public boolean isActive() {
             return Minecraft.getMinecraft().currentScreen != null;
         }
 
         @Override
-        public boolean conflicts(IKeyConflictContext other)
-        {
+        public boolean conflicts(IKeyConflictContext other) {
             return this == other;
         }
     },
@@ -64,14 +59,12 @@
      */
     IN_GAME {
         @Override
-        public boolean isActive()
-        {
+        public boolean isActive() {
             return !GUI.isActive();
         }
 
         @Override
-        public boolean conflicts(IKeyConflictContext other)
-        {
+        public boolean conflicts(IKeyConflictContext other) {
             return this == other;
         }
     }
