--- ../src-base/minecraft/net/minecraftforge/client/settings/KeyModifier.java
+++ ../src-work/minecraft/net/minecraftforge/client/settings/KeyModifier.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,14 +19,14 @@
 
 package net.minecraftforge.client.settings;
 
-import javax.annotation.Nullable;
-
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.client.settings.GameSettings;
 import org.lwjgl.input.Keyboard;
 
+import javax.annotation.Nullable;
+
 public enum KeyModifier {
     CONTROL {
         @Override
