--- ../src-base/minecraft/net/minecraftforge/common/util/BlockSnapshot.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/BlockSnapshot.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,9 +19,6 @@
 
 package net.minecraftforge.common.util;
 
-import java.lang.ref.WeakReference;
-import java.util.Objects;
-
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.tileentity.TileEntity;
@@ -32,6 +29,8 @@
 import net.minecraftforge.fml.common.registry.ForgeRegistries;
 
 import javax.annotation.Nullable;
+import java.lang.ref.WeakReference;
+import java.util.Objects;
 
 /**
  * Represents a captured snapshot of a block which will not change
