--- ../src-base/minecraft/net/minecraftforge/common/util/EnumHelper.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/EnumHelper.java
@@ -52,6 +52,7 @@
 import net.minecraft.world.gen.structure.StructureStrongholdPieces.Stronghold.Door;
 import net.minecraftforge.classloading.FMLForgePlugin;
 import org.apache.commons.lang3.ArrayUtils;
+import org.bukkit.World;
 
 import javax.annotation.Nullable;
 
@@ -423,4 +424,11 @@
             setup();
         }
     }
+	
+	public static World.Environment addBukkitEnvironment(int id, String name) {
+        if(!isSetup){
+            setup();
+        }
+        return (World.Environment)addEnum(World.Environment.class,name,new Class[]{Integer.TYPE}, new Object[]{Integer.valueOf(id)});
+    }
 }
