--- ../src-base/minecraft/net/minecraftforge/common/util/EnumHelper.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/EnumHelper.java
@@ -423,4 +423,11 @@
             setup();
         }
     }
+	
+	public static World.Environment addBukkitEnvironment(int id, String name) {
+        if(!isSetup){
+            setup();
+        }
+        return (World.Environment)addEnum(World.Environment.class,name,new Class[]{Integer.TYPE}, new Object[]{Integer.valueOf(id)});
+    }
 }
