--- ../src-base/minecraft/net/minecraftforge/common/util/INBTSerializable.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/INBTSerializable.java
@@ -25,8 +25,8 @@
  * An interface designed to unify various things in the Minecraft
  * code base that can be serialized to and from a NBT tag.
  */
-public interface INBTSerializable<T extends NBTBase>
-{
+public interface INBTSerializable<T extends NBTBase> {
     T serializeNBT();
+
     void deserializeNBT(T nbt);
 }
