--- ../src-base/minecraft/net/minecraftforge/common/util/ITeleporter.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/ITeleporter.java
@@ -33,8 +33,7 @@
  * See the {@link net.minecraft.world.Teleporter} class, which has
  * been patched to implement this interface, for a vanilla example.
  */
-public interface ITeleporter
-{
+public interface ITeleporter {
     /**
      * Called to handle placing the entity in the new world.
      *
@@ -53,8 +52,7 @@
     void placeEntity(World world, Entity entity, float yaw);
 
     // used internally to handle vanilla hardcoding
-    default boolean isVanilla()
-    {
+    default boolean isVanilla() {
         return getClass() == Teleporter.class;
     }
 }
