--- ../src-base/minecraft/net/minecraftforge/common/util/PacketUtil.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/PacketUtil.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -16,7 +16,6 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
-
 package net.minecraftforge.common.util;
 
 import net.minecraft.item.Item;
