--- ../src-base/minecraft/net/minecraftforge/common/util/RecipeMatcher.java
+++ ../src-work/minecraft/net/minecraftforge/common/util/RecipeMatcher.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -16,16 +16,15 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
-
 package net.minecraftforge.common.util;
 
+import com.google.common.base.Predicate;
+
 import java.util.BitSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Queue;
 
-import com.google.common.base.Predicate;
-
 public class RecipeMatcher
 {
     /**
