--- ../src-base/minecraft/net/minecraft/client/network/LanServerInfo.java
+++ ../src-work/minecraft/net/minecraft/client/network/LanServerInfo.java
@@ -5,31 +5,26 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class LanServerInfo
-{
+public class LanServerInfo {
     private final String lanServerMotd;
     private final String lanServerIpPort;
     private long timeLastSeen;
 
-    public LanServerInfo(String p_i47130_1_, String p_i47130_2_)
-    {
+    public LanServerInfo(String p_i47130_1_, String p_i47130_2_) {
         this.lanServerMotd = p_i47130_1_;
         this.lanServerIpPort = p_i47130_2_;
         this.timeLastSeen = Minecraft.getSystemTime();
     }
 
-    public String getServerMotd()
-    {
+    public String getServerMotd() {
         return this.lanServerMotd;
     }
 
-    public String getServerIpPort()
-    {
+    public String getServerIpPort() {
         return this.lanServerIpPort;
     }
 
-    public void updateLastSeen()
-    {
+    public void updateLastSeen() {
         this.timeLastSeen = Minecraft.getSystemTime();
     }
 }
