--- ../src-base/minecraft/net/minecraft/client/network/NetworkPlayerInfo.java
+++ ../src-work/minecraft/net/minecraft/client/network/NetworkPlayerInfo.java
@@ -5,8 +5,6 @@
 import com.mojang.authlib.GameProfile;
 import com.mojang.authlib.minecraft.MinecraftProfileTexture;
 import com.mojang.authlib.minecraft.MinecraftProfileTexture.Type;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.resources.DefaultPlayerSkin;
 import net.minecraft.client.resources.SkinManager;
@@ -18,9 +16,11 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
-public class NetworkPlayerInfo
-{
+public class NetworkPlayerInfo {
     private final GameProfile gameProfile;
     Map<Type, ResourceLocation> playerTextures = Maps.newEnumMap(Type.class);
     private GameType gameType;
@@ -34,99 +34,79 @@
     private long healthBlinkTime;
     private long renderVisibilityId;
 
-    public NetworkPlayerInfo(GameProfile profile)
-    {
+    public NetworkPlayerInfo(GameProfile profile) {
         this.gameProfile = profile;
     }
 
-    public NetworkPlayerInfo(SPacketPlayerListItem.AddPlayerData entry)
-    {
+    public NetworkPlayerInfo(SPacketPlayerListItem.AddPlayerData entry) {
         this.gameProfile = entry.getProfile();
         this.gameType = entry.getGameMode();
         this.responseTime = entry.getPing();
         this.displayName = entry.getDisplayName();
     }
 
-    public GameProfile getGameProfile()
-    {
+    public GameProfile getGameProfile() {
         return this.gameProfile;
     }
 
-    public GameType getGameType()
-    {
+    public GameType getGameType() {
         return this.gameType;
     }
 
-    protected void setGameType(GameType gameMode)
-    {
+    protected void setGameType(GameType gameMode) {
         this.gameType = gameMode;
     }
 
-    public int getResponseTime()
-    {
+    public int getResponseTime() {
         return this.responseTime;
     }
 
-    protected void setResponseTime(int latency)
-    {
+    protected void setResponseTime(int latency) {
         this.responseTime = latency;
     }
 
-    public boolean hasLocationSkin()
-    {
+    public boolean hasLocationSkin() {
         return this.getLocationSkin() != null;
     }
 
-    public String getSkinType()
-    {
+    public String getSkinType() {
         return this.skinType == null ? DefaultPlayerSkin.getSkinType(this.gameProfile.getId()) : this.skinType;
     }
 
-    public ResourceLocation getLocationSkin()
-    {
+    public ResourceLocation getLocationSkin() {
         this.loadPlayerTextures();
-        return (ResourceLocation)MoreObjects.firstNonNull(this.playerTextures.get(Type.SKIN), DefaultPlayerSkin.getDefaultSkin(this.gameProfile.getId()));
+        return (ResourceLocation) MoreObjects.firstNonNull(this.playerTextures.get(Type.SKIN), DefaultPlayerSkin.getDefaultSkin(this.gameProfile.getId()));
     }
 
     @Nullable
-    public ResourceLocation getLocationCape()
-    {
+    public ResourceLocation getLocationCape() {
         this.loadPlayerTextures();
         return this.playerTextures.get(Type.CAPE);
     }
 
     @Nullable
-    public ResourceLocation getLocationElytra()
-    {
+    public ResourceLocation getLocationElytra() {
         this.loadPlayerTextures();
         return this.playerTextures.get(Type.ELYTRA);
     }
 
     @Nullable
-    public ScorePlayerTeam getPlayerTeam()
-    {
+    public ScorePlayerTeam getPlayerTeam() {
         return Minecraft.getMinecraft().world.getScoreboard().getPlayersTeam(this.getGameProfile().getName());
     }
 
-    protected void loadPlayerTextures()
-    {
-        synchronized (this)
-        {
-            if (!this.playerTexturesLoaded)
-            {
+    protected void loadPlayerTextures() {
+        synchronized (this) {
+            if (!this.playerTexturesLoaded) {
                 this.playerTexturesLoaded = true;
-                Minecraft.getMinecraft().getSkinManager().loadProfileTextures(this.gameProfile, new SkinManager.SkinAvailableCallback()
-                {
-                    public void skinAvailable(Type typeIn, ResourceLocation location, MinecraftProfileTexture profileTexture)
-                    {
-                        switch (typeIn)
-                        {
+                Minecraft.getMinecraft().getSkinManager().loadProfileTextures(this.gameProfile, new SkinManager.SkinAvailableCallback() {
+                    public void skinAvailable(Type typeIn, ResourceLocation location, MinecraftProfileTexture profileTexture) {
+                        switch (typeIn) {
                             case SKIN:
                                 NetworkPlayerInfo.this.playerTextures.put(Type.SKIN, location);
                                 NetworkPlayerInfo.this.skinType = profileTexture.getMetadata("model");
 
-                                if (NetworkPlayerInfo.this.skinType == null)
-                                {
+                                if (NetworkPlayerInfo.this.skinType == null) {
                                     NetworkPlayerInfo.this.skinType = "default";
                                 }
 
@@ -143,64 +123,52 @@
         }
     }
 
-    public void setDisplayName(@Nullable ITextComponent displayNameIn)
-    {
-        this.displayName = displayNameIn;
-    }
-
     @Nullable
-    public ITextComponent getDisplayName()
-    {
+    public ITextComponent getDisplayName() {
         return this.displayName;
     }
 
-    public int getLastHealth()
-    {
+    public void setDisplayName(@Nullable ITextComponent displayNameIn) {
+        this.displayName = displayNameIn;
+    }
+
+    public int getLastHealth() {
         return this.lastHealth;
     }
 
-    public void setLastHealth(int p_178836_1_)
-    {
+    public void setLastHealth(int p_178836_1_) {
         this.lastHealth = p_178836_1_;
     }
 
-    public int getDisplayHealth()
-    {
+    public int getDisplayHealth() {
         return this.displayHealth;
     }
 
-    public void setDisplayHealth(int p_178857_1_)
-    {
+    public void setDisplayHealth(int p_178857_1_) {
         this.displayHealth = p_178857_1_;
     }
 
-    public long getLastHealthTime()
-    {
+    public long getLastHealthTime() {
         return this.lastHealthTime;
     }
 
-    public void setLastHealthTime(long p_178846_1_)
-    {
+    public void setLastHealthTime(long p_178846_1_) {
         this.lastHealthTime = p_178846_1_;
     }
 
-    public long getHealthBlinkTime()
-    {
+    public long getHealthBlinkTime() {
         return this.healthBlinkTime;
     }
 
-    public void setHealthBlinkTime(long p_178844_1_)
-    {
+    public void setHealthBlinkTime(long p_178844_1_) {
         this.healthBlinkTime = p_178844_1_;
     }
 
-    public long getRenderVisibilityId()
-    {
+    public long getRenderVisibilityId() {
         return this.renderVisibilityId;
     }
 
-    public void setRenderVisibilityId(long p_178843_1_)
-    {
+    public void setRenderVisibilityId(long p_178843_1_) {
         this.renderVisibilityId = p_178843_1_;
     }
 }
