--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/asm/ModClassVisitor.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/asm/ModClassVisitor.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,11 +19,7 @@
 
 package net.minecraftforge.fml.common.discovery.asm;
 
-import org.objectweb.asm.AnnotationVisitor;
-import org.objectweb.asm.ClassVisitor;
-import org.objectweb.asm.FieldVisitor;
-import org.objectweb.asm.MethodVisitor;
-import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.*;
 
 public class ModClassVisitor extends ClassVisitor
 {
