--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/asm/ModFieldVisitor.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/asm/ModFieldVisitor.java
@@ -23,22 +23,19 @@
 import org.objectweb.asm.FieldVisitor;
 import org.objectweb.asm.Opcodes;
 
-public class ModFieldVisitor extends FieldVisitor
-{
+public class ModFieldVisitor extends FieldVisitor {
 
     private String fieldName;
     private ASMModParser discoverer;
 
-    public ModFieldVisitor(String name, ASMModParser discoverer)
-    {
+    public ModFieldVisitor(String name, ASMModParser discoverer) {
         super(Opcodes.ASM5);
         this.fieldName = name;
         this.discoverer = discoverer;
     }
-    
+
     @Override
-    public AnnotationVisitor visitAnnotation(String annotationName, boolean runtimeVisible)
-    {
+    public AnnotationVisitor visitAnnotation(String annotationName, boolean runtimeVisible) {
         discoverer.startFieldAnnotation(fieldName, annotationName);
         return new ModAnnotationVisitor(discoverer);
     }
