--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/asm/ModMethodVisitor.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/asm/ModMethodVisitor.java
@@ -29,8 +29,7 @@
     private String methodDescriptor;
     private ASMModParser discoverer;
 
-    public ModMethodVisitor(String name, String desc, ASMModParser discoverer)
-    {
+    public ModMethodVisitor(String name, String desc, ASMModParser discoverer) {
         super(Opcodes.ASM5);
         this.methodName = name;
         this.methodDescriptor = desc;
@@ -38,8 +37,7 @@
     }
 
     @Override
-    public AnnotationVisitor visitAnnotation(String annotationName, boolean runtimeVisible)
-    {
+    public AnnotationVisitor visitAnnotation(String annotationName, boolean runtimeVisible) {
         discoverer.startMethodAnnotation(methodName, methodDescriptor, annotationName);
         return new ModAnnotationVisitor(discoverer);
     }
