--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketClientStatus.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketClientStatus.java
@@ -1,45 +1,38 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
-public class CPacketClientStatus implements Packet<INetHandlerPlayServer>
-{
+import java.io.IOException;
+
+public class CPacketClientStatus implements Packet<INetHandlerPlayServer> {
     private CPacketClientStatus.State status;
 
-    public CPacketClientStatus()
-    {
+    public CPacketClientStatus() {
     }
 
-    public CPacketClientStatus(CPacketClientStatus.State p_i46886_1_)
-    {
+    public CPacketClientStatus(CPacketClientStatus.State p_i46886_1_) {
         this.status = p_i46886_1_;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
-        this.status = (CPacketClientStatus.State)buf.readEnumValue(CPacketClientStatus.State.class);
+    public void readPacketData(PacketBuffer buf) throws IOException {
+        this.status = (CPacketClientStatus.State) buf.readEnumValue(CPacketClientStatus.State.class);
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeEnumValue(this.status);
     }
 
-    public void processPacket(INetHandlerPlayServer handler)
-    {
+    public void processPacket(INetHandlerPlayServer handler) {
         handler.processClientStatus(this);
     }
 
-    public CPacketClientStatus.State getStatus()
-    {
+    public CPacketClientStatus.State getStatus() {
         return this.status;
     }
 
-    public static enum State
-    {
+    public static enum State {
         PERFORM_RESPAWN,
         REQUEST_STATS;
     }
