--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketPlayerTryUseItem.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketPlayerTryUseItem.java
@@ -1,41 +1,35 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.EnumHand;
 
-public class CPacketPlayerTryUseItem implements Packet<INetHandlerPlayServer>
-{
+import java.io.IOException;
+
+public class CPacketPlayerTryUseItem implements Packet<INetHandlerPlayServer> {
     private EnumHand hand;
 
-    public CPacketPlayerTryUseItem()
-    {
+    public CPacketPlayerTryUseItem() {
     }
 
-    public CPacketPlayerTryUseItem(EnumHand handIn)
-    {
+    public CPacketPlayerTryUseItem(EnumHand handIn) {
         this.hand = handIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
-        this.hand = (EnumHand)buf.readEnumValue(EnumHand.class);
+    public void readPacketData(PacketBuffer buf) throws IOException {
+        this.hand = (EnumHand) buf.readEnumValue(EnumHand.class);
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeEnumValue(this.hand);
     }
 
-    public void processPacket(INetHandlerPlayServer handler)
-    {
+    public void processPacket(INetHandlerPlayServer handler) {
         handler.processTryUseItem(this);
     }
 
-    public EnumHand getHand()
-    {
+    public EnumHand getHand() {
         return this.hand;
     }
 }
