--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketVehicleMove.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketVehicleMove.java
@@ -1,11 +1,12 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.entity.Entity;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
+import java.io.IOException;
+
 public class CPacketVehicleMove implements Packet<INetHandlerPlayServer>
 {
     private double x;
