--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketAnimation.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketAnimation.java
@@ -1,11 +1,12 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.EnumHand;
 
+import java.io.IOException;
+
 public class CPacketAnimation implements Packet<INetHandlerPlayServer>
 {
     private EnumHand hand;
