--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketSteerBoat.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketSteerBoat.java
@@ -1,10 +1,11 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
+import java.io.IOException;
+
 public class CPacketSteerBoat implements Packet<INetHandlerPlayServer>
 {
     private boolean left;
