--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketPlayerDigging.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketPlayerDigging.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
@@ -9,18 +8,20 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketPlayerDigging implements Packet<INetHandlerPlayServer>
 {
     private BlockPos position;
     private EnumFacing facing;
-    private CPacketPlayerDigging.Action action;
+    private Action action;
 
     public CPacketPlayerDigging()
     {
     }
 
     @SideOnly(Side.CLIENT)
-    public CPacketPlayerDigging(CPacketPlayerDigging.Action actionIn, BlockPos posIn, EnumFacing facingIn)
+    public CPacketPlayerDigging(Action actionIn, BlockPos posIn, EnumFacing facingIn)
     {
         this.action = actionIn;
         this.position = posIn;
@@ -29,7 +30,7 @@
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.action = (CPacketPlayerDigging.Action)buf.readEnumValue(CPacketPlayerDigging.Action.class);
+        this.action = (Action)buf.readEnumValue(Action.class);
         this.position = buf.readBlockPos();
         this.facing = EnumFacing.getFront(buf.readUnsignedByte());
     }
@@ -56,7 +57,7 @@
         return this.facing;
     }
 
-    public CPacketPlayerDigging.Action getAction()
+    public Action getAction()
     {
         return this.action;
     }
