--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketUpdateSign.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketUpdateSign.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketUpdateSign implements Packet<INetHandlerPlayServer>
 {
     private BlockPos pos;
