--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketUseEntity.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketUseEntity.java
@@ -1,7 +1,5 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
-import javax.annotation.Nullable;
 import net.minecraft.entity.Entity;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -12,10 +10,13 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+
 public class CPacketUseEntity implements Packet<INetHandlerPlayServer>
 {
     private int entityId;
-    private CPacketUseEntity.Action action;
+    private Action action;
     private Vec3d hitVec;
     private EnumHand hand;
 
@@ -26,14 +27,14 @@
     public CPacketUseEntity(Entity entityIn)
     {
         this.entityId = entityIn.getEntityId();
-        this.action = CPacketUseEntity.Action.ATTACK;
+        this.action = Action.ATTACK;
     }
 
     @SideOnly(Side.CLIENT)
     public CPacketUseEntity(Entity entityIn, EnumHand handIn)
     {
         this.entityId = entityIn.getEntityId();
-        this.action = CPacketUseEntity.Action.INTERACT;
+        this.action = Action.INTERACT;
         this.hand = handIn;
     }
 
@@ -41,7 +42,7 @@
     public CPacketUseEntity(Entity entityIn, EnumHand handIn, Vec3d hitVecIn)
     {
         this.entityId = entityIn.getEntityId();
-        this.action = CPacketUseEntity.Action.INTERACT_AT;
+        this.action = Action.INTERACT_AT;
         this.hand = handIn;
         this.hitVec = hitVecIn;
     }
@@ -49,14 +50,14 @@
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.entityId = buf.readVarInt();
-        this.action = (CPacketUseEntity.Action)buf.readEnumValue(CPacketUseEntity.Action.class);
+        this.action = (Action)buf.readEnumValue(Action.class);
 
-        if (this.action == CPacketUseEntity.Action.INTERACT_AT)
+        if (this.action == Action.INTERACT_AT)
         {
             this.hitVec = new Vec3d((double)buf.readFloat(), (double)buf.readFloat(), (double)buf.readFloat());
         }
 
-        if (this.action == CPacketUseEntity.Action.INTERACT || this.action == CPacketUseEntity.Action.INTERACT_AT)
+        if (this.action == Action.INTERACT || this.action == Action.INTERACT_AT)
         {
             this.hand = (EnumHand)buf.readEnumValue(EnumHand.class);
         }
@@ -67,14 +68,14 @@
         buf.writeVarInt(this.entityId);
         buf.writeEnumValue(this.action);
 
-        if (this.action == CPacketUseEntity.Action.INTERACT_AT)
+        if (this.action == Action.INTERACT_AT)
         {
             buf.writeFloat((float)this.hitVec.x);
             buf.writeFloat((float)this.hitVec.y);
             buf.writeFloat((float)this.hitVec.z);
         }
 
-        if (this.action == CPacketUseEntity.Action.INTERACT || this.action == CPacketUseEntity.Action.INTERACT_AT)
+        if (this.action == Action.INTERACT || this.action == Action.INTERACT_AT)
         {
             buf.writeEnumValue(this.hand);
         }
@@ -91,7 +92,7 @@
         return worldIn.getEntityByID(this.entityId);
     }
 
-    public CPacketUseEntity.Action getAction()
+    public Action getAction()
     {
         return this.action;
     }
