--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketPlayer.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketPlayer.java
@@ -1,22 +1,23 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketPlayer implements Packet<INetHandlerPlayServer>
 {
-    protected double x;
-    protected double y;
-    protected double z;
-    protected float yaw;
-    protected float pitch;
+    public double x;
+    public double y;
+    public double z;
+    public float yaw;
+    public float pitch;
     protected boolean onGround;
-    protected boolean moving;
-    protected boolean rotating;
+    public boolean moving;
+    public boolean rotating;
 
     public CPacketPlayer()
     {
