--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketKeepAlive.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketKeepAlive.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketKeepAlive implements Packet<INetHandlerPlayServer>
 {
     private long key;
