--- ../src-base/minecraft/net/minecraft/network/play/client/CPacketCloseWindow.java
+++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketCloseWindow.java
@@ -1,12 +1,11 @@
 package net.minecraft.network.play.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketCloseWindow implements Packet<INetHandlerPlayServer>
 {
     private int windowId;
@@ -15,7 +14,6 @@
     {
     }
 
-    @SideOnly(Side.CLIENT)
     public CPacketCloseWindow(int windowIdIn)
     {
         this.windowId = windowIdIn;
