--- ../src-base/minecraft/net/minecraft/world/storage/loot/conditions/EntityHasProperty.java
+++ ../src-work/minecraft/net/minecraft/world/storage/loot/conditions/EntityHasProperty.java
@@ -4,9 +4,6 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
-import java.util.Random;
-import java.util.Set;
-import java.util.Map.Entry;
 import net.minecraft.entity.Entity;
 import net.minecraft.util.JsonUtils;
 import net.minecraft.util.ResourceLocation;
@@ -14,6 +11,10 @@
 import net.minecraft.world.storage.loot.properties.EntityProperty;
 import net.minecraft.world.storage.loot.properties.EntityPropertyManager;
 
+import java.util.Map.Entry;
+import java.util.Random;
+import java.util.Set;
+
 public class EntityHasProperty implements LootCondition
 {
     private final EntityProperty[] properties;
