--- ../src-base/minecraft/net/minecraftforge/event/entity/minecart/MinecartInteractEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/minecart/MinecartInteractEvent.java
@@ -19,16 +19,12 @@
 
 package net.minecraftforge.event.entity.minecart;
 
-import net.minecraft.entity.item.EntityMinecartContainer;
-import net.minecraft.entity.item.EntityMinecartEmpty;
-import net.minecraft.entity.item.EntityMinecartFurnace;
-import net.minecraft.entity.item.EntityMinecartHopper;
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraft.entity.item.EntityMinecart;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.EnumHand;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
 import javax.annotation.Nonnull;
 
@@ -48,20 +44,26 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
 @Cancelable
-public class MinecartInteractEvent extends MinecartEvent
-{
+public class MinecartInteractEvent extends MinecartEvent {
     private final EntityPlayer player;
     private final EnumHand hand;
 
-    public MinecartInteractEvent(EntityMinecart minecart, EntityPlayer player, EnumHand hand)
-    {
+    public MinecartInteractEvent(EntityMinecart minecart, EntityPlayer player, EnumHand hand) {
         super(minecart);
         this.player = player;
         this.hand = hand;
     }
 
-    public EntityPlayer getPlayer() { return player; }
+    public EntityPlayer getPlayer() {
+        return player;
+    }
+
     @Nonnull
-    public ItemStack getItem() { return player.getHeldItem(hand); }
-    public EnumHand getHand() { return hand; }
+    public ItemStack getItem() {
+        return player.getHeldItem(hand);
+    }
+
+    public EnumHand getHand() {
+        return hand;
+    }
 }
