--- ../src-base/minecraft/net/minecraftforge/common/model/animation/IJoint.java
+++ ../src-work/minecraft/net/minecraftforge/common/model/animation/IJoint.java
@@ -19,16 +19,15 @@
 
 package net.minecraftforge.common.model.animation;
 
-import java.util.Optional;
-
 import net.minecraftforge.common.model.IModelPart;
 import net.minecraftforge.common.model.TRSRTransformation;
 
+import java.util.Optional;
+
 /**
  * Model part that's a part of the hierarchical skeleton.
  */
-public interface IJoint extends IModelPart
-{
+public interface IJoint extends IModelPart {
     TRSRTransformation getInvBindPose();
 
     Optional<? extends IJoint> getParent();
