--- ../src-base/minecraft/net/minecraft/inventory/IContainerListener.java
+++ ../src-work/minecraft/net/minecraft/inventory/IContainerListener.java
@@ -3,8 +3,7 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.NonNullList;
 
-public interface IContainerListener
-{
+public interface IContainerListener {
     void sendAllContents(Container containerToSend, NonNullList<ItemStack> itemsList);
 
     void sendSlotContents(Container containerToSend, int slotInd, ItemStack stack);
