--- ../src-base/minecraft/net/minecraft/inventory/ItemStackHelper.java
+++ ../src-work/minecraft/net/minecraft/inventory/ItemStackHelper.java
@@ -1,11 +1,12 @@
 package net.minecraft.inventory;
 
-import java.util.List;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.util.NonNullList;
 
+import java.util.List;
+
 public class ItemStackHelper
 {
     public static ItemStack getAndSplit(List<ItemStack> stacks, int index, int amount)
