--- ../src-base/minecraft/net/minecraft/village/VillageCollection.java
+++ ../src-work/minecraft/net/minecraft/village/VillageCollection.java
@@ -3,7 +3,10 @@
 import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Queue;
 import javax.annotation.Nullable;
+
+import com.google.common.collect.Queues;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockDoor;
 import net.minecraft.block.material.Material;
@@ -19,7 +22,7 @@
 public class VillageCollection extends WorldSavedData
 {
     private World world;
-    private final List<BlockPos> villagerPositionsList = Lists.<BlockPos>newArrayList();
+    private final Queue<BlockPos> villagerPositionsList = Queues.newConcurrentLinkedQueue();//hose
     private final List<VillageDoorInfo> newDoors = Lists.<VillageDoorInfo>newArrayList();
     private final List<Village> villageList = Lists.<Village>newArrayList();
     private int tickCounter;
@@ -125,7 +128,7 @@
     {
         if (!this.villagerPositionsList.isEmpty())
         {
-            this.addDoorsAround(this.villagerPositionsList.remove(0));
+            this.addDoorsAround(this.villagerPositionsList.poll());
         }
     }
 
