--- ../src-base/minecraft/net/minecraft/village/VillageSiege.java
+++ ../src-work/minecraft/net/minecraft/village/VillageSiege.java
@@ -1,10 +1,7 @@
 package net.minecraft.village;
 
-import java.util.Iterator;
-import java.util.List;
-import javax.annotation.Nullable;
+import com.destroystokyo.paper.exception.ServerInternalException;
 import net.minecraft.entity.EntityLiving;
-import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.monster.EntityZombie;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.math.BlockPos;
@@ -13,6 +10,10 @@
 import net.minecraft.world.World;
 import net.minecraft.world.WorldEntitySpawner;
 
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.List;
+
 public class VillageSiege
 {
     private final World world;
@@ -172,16 +173,17 @@
             try
             {
                 entityzombie = new EntityZombie(this.world);
-                entityzombie.onInitialSpawn(this.world.getDifficultyForLocation(new BlockPos(entityzombie)), (IEntityLivingData)null);
+                entityzombie.onInitialSpawn(this.world.getDifficultyForLocation(new BlockPos(entityzombie)), null);
             }
             catch (Exception exception)
             {
                 exception.printStackTrace();
+                ServerInternalException.reportInternalException(exception); // Paper
                 return false;
             }
 
             entityzombie.setLocationAndAngles(vec3d.x, vec3d.y, vec3d.z, this.world.rand.nextFloat() * 360.0F, 0.0F);
-            this.world.spawnEntity(entityzombie);
+            this.world.addEntity(entityzombie, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.VILLAGE_INVASION);
             BlockPos blockpos = this.village.getCenter();
             entityzombie.setHomePosAndDistance(blockpos, this.village.getVillageRadius());
             return true;
