--- ../src-base/minecraft/net/minecraftforge/common/crafting/IConditionFactory.java
+++ ../src-work/minecraft/net/minecraftforge/common/crafting/IConditionFactory.java
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.common.crafting;
 
-import java.util.function.BooleanSupplier;
-
 import com.google.gson.JsonObject;
 
+import java.util.function.BooleanSupplier;
+
 public interface IConditionFactory {
     BooleanSupplier parse(JsonContext context, JsonObject json);
 }
