--- ../src-base/minecraft/net/minecraftforge/common/crafting/IngredientNBT.java
+++ ../src-work/minecraft/net/minecraftforge/common/crafting/IngredientNBT.java
@@ -19,23 +19,21 @@
 
 package net.minecraftforge.common.crafting;
 
-import javax.annotation.Nullable;
-
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.crafting.Ingredient;
 
-public class IngredientNBT extends Ingredient
-{
+import javax.annotation.Nullable;
+
+public class IngredientNBT extends Ingredient {
     private final ItemStack stack;
-    protected IngredientNBT(ItemStack stack)
-    {
+
+    protected IngredientNBT(ItemStack stack) {
         super(stack);
         this.stack = stack;
     }
 
     @Override
-    public boolean apply(@Nullable ItemStack input)
-    {
+    public boolean apply(@Nullable ItemStack input) {
         if (input == null)
             return false;
         //Can't use areItemStacksEqualUsingNBTShareTag because it compares stack size as well
@@ -43,8 +41,7 @@
     }
 
     @Override
-    public boolean isSimple()
-    {
+    public boolean isSimple() {
         return false;
     }
 }
