--- ../src-base/minecraft/net/minecraftforge/common/crafting/IRecipeContainer.java
+++ ../src-work/minecraft/net/minecraftforge/common/crafting/IRecipeContainer.java
@@ -29,8 +29,7 @@
  * For GUIs with recipe books, this allows their containers to have
  * recipe completion and ghost recipes in their craft matrices.
  */
-public interface IRecipeContainer
-{
+public interface IRecipeContainer {
     /**
      * The crafting result slot of your container, where you take out the crafted item.
      * The equivalent for {@link ContainerWorkbench} is {@link ContainerWorkbench#craftResult}.
