--- ../src-base/minecraft/net/minecraftforge/common/crafting/IShapedRecipe.java
+++ ../src-work/minecraft/net/minecraftforge/common/crafting/IShapedRecipe.java
@@ -26,8 +26,8 @@
  * book and auto crafting picks the correct shape.
  * Note: These methods can't be named 'getHeight' or 'getWidth' due to obfusication issues.
  */
-public interface IShapedRecipe extends IRecipe
-{
+public interface IShapedRecipe extends IRecipe {
     int getRecipeWidth();
+
     int getRecipeHeight();
 }
