--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/ChannelRegistrationHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/ChannelRegistrationHandler.java
@@ -48,6 +48,16 @@
             Set<String> channelSet = ImmutableSet.copyOf(split);
             FMLCommonHandler.instance().fireNetRegistrationEvent(manager, channelSet, msg.channel(), side);
             msg.payload().release();
+            // Cauldron start - register bukkit channels for players
+            org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer player = ((net.minecraft.network.NetHandlerPlayServer) ctx.attr(NetworkDispatcher.FML_DISPATCHER).get().getNetHandler()).getPlayer();
+            if (msg.channel().equals("REGISTER")) {
+                for (String channel : channelSet)
+                    player.addChannel(channel);
+            } else {
+                for (String channel : channelSet)
+                    player.removeChannel(channel);
+            }
+            // Cauldron end
         }
         else
         {
