--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/HandshakeInjector.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/HandshakeInjector.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.fml.common.network.handshake;
 
-import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelOutboundHandlerAdapter;
 import io.netty.channel.ChannelPromise;
+import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
 
 public class HandshakeInjector extends ChannelOutboundHandlerAdapter {
 
