--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/IHandshakeState.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/IHandshakeState.java
@@ -21,10 +21,9 @@
 
 import io.netty.channel.ChannelHandlerContext;
 
+import javax.annotation.Nullable;
 import java.util.function.Consumer;
 
-import javax.annotation.Nullable;
-
 public interface IHandshakeState<S> {
     /**
      * Accepts FML handshake message for this state, and if needed - switches to another handshake state
