--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/IHandshakeState.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/IHandshakeState.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -21,10 +21,9 @@
 
 import io.netty.channel.ChannelHandlerContext;
 
+import javax.annotation.Nullable;
 import java.util.function.Consumer;
 
-import javax.annotation.Nullable;
-
 public interface IHandshakeState<S> {
     /**
      * Accepts FML handshake message for this state, and if needed - switches to another handshake state
