--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java
@@ -103,7 +103,7 @@
     public static final AttributeKey<Map<ResourceLocation, ForgeRegistry.Snapshot>> FML_GAMEDATA_SNAPSHOT = AttributeKey.valueOf("fml:gameDataSnapshot");
     public final NetworkManager manager;
     private final PlayerList scm;
-    private EntityPlayerMP player;
+    public EntityPlayerMP player;
     private ConnectionState state;
     private ConnectionType connectionType;
     private final Side side;
@@ -113,6 +113,11 @@
     private Map<String,String> modList = Collections.emptyMap();
     private int overrideLoginDim;
 
+    public EntityPlayerMP getPlayer()
+    {
+        return player;
+    }
+
     public NetworkDispatcher(NetworkManager manager)
     {
         super(false);
@@ -427,6 +432,7 @@
             }
         }
         String channelName = msg.getChannelName();
+        player.getBukkitEntity().addChannel(channelName); // Cauldron - register channel for bukkit player
         if ("FML|HS".equals(channelName) || "REGISTER".equals(channelName) || "UNREGISTER".equals(channelName))
         {
             FMLProxyPacket proxy = new FMLProxyPacket(msg);
@@ -597,7 +603,7 @@
             }
             else
             {
-                FMLLog.log.error("NetworkDispatcher exception", cause);
+                FMLLog.log.debug("NetworkDispatcher exception", cause);
             }
         }
         super.exceptionCaught(ctx, cause);
