--- ../src-base/minecraft/net/minecraft/network/rcon/RConUtils.java
+++ ../src-work/minecraft/net/minecraft/network/rcon/RConUtils.java
@@ -1,9 +1,10 @@
 package net.minecraft.network.rcon;
 
-import java.nio.charset.StandardCharsets;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.charset.StandardCharsets;
+
 @SideOnly(Side.SERVER)
 public class RConUtils
 {
