--- ../src-base/minecraft/net/minecraft/pathfinding/PathNodeType.java
+++ ../src-work/minecraft/net/minecraft/pathfinding/PathNodeType.java
@@ -1,7 +1,6 @@
 package net.minecraft.pathfinding;
 
-public enum PathNodeType
-{
+public enum PathNodeType {
     BLOCKED(-1.0F),
     OPEN(0.0F),
     WALKABLE(0.0F),
@@ -22,13 +21,11 @@
 
     private final float priority;
 
-    private PathNodeType(float priorityIn)
-    {
+    private PathNodeType(float priorityIn) {
         this.priority = priorityIn;
     }
 
-    public float getPriority()
-    {
+    public float getPriority() {
         return this.priority;
     }
 }
