--- ../src-base/minecraft/net/minecraft/pathfinding/PathNavigate.java
+++ ../src-work/minecraft/net/minecraft/pathfinding/PathNavigate.java
@@ -1,6 +1,7 @@
 package net.minecraft.pathfinding;
 
-import javax.annotation.Nullable;
+import com.destroystokyo.paper.MCUtil;
+import com.destroystokyo.paper.event.entity.EntityPathfindEvent;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
@@ -15,18 +16,21 @@
 import net.minecraft.world.ChunkCache;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+
 public abstract class PathNavigate
 {
     protected EntityLiving entity;
+    public Entity getEntity() { return entity; } // Paper - OBFHELPER
     protected World world;
     @Nullable
     protected Path currentPath;
     protected double speed;
-    private final IAttributeInstance pathSearchRange;
+    public final IAttributeInstance pathSearchRange;
     protected int totalTicks;
     private int ticksAtLastPos;
-    private Vec3d lastPosCheck = Vec3d.ZERO;
-    private Vec3d timeoutCachedNode = Vec3d.ZERO;
+    public Vec3d lastPosCheck = Vec3d.ZERO;
+    public Vec3d timeoutCachedNode = Vec3d.ZERO;
     private long timeoutTimer;
     private long lastTimeoutCheck;
     private double timeoutLimit;
@@ -99,6 +103,9 @@
         }
         else
         {
+            if (!new EntityPathfindEvent(this.getEntity().getBukkitEntity(), MCUtil.toLocation(this.getEntity().world, pos), null).callEvent()) {
+                return null;
+            }
             this.targetPos = pos;
             float f = this.getPathSearchRange();
             this.world.profiler.startSection("pathfind");
@@ -128,6 +135,9 @@
             }
             else
             {
+                if (!new EntityPathfindEvent(this.getEntity().getBukkitEntity(), MCUtil.toLocation(entity.world, blockpos), entity.getBukkitEntity()).callEvent()) {
+                    return null;
+                }
                 this.targetPos = blockpos;
                 float f = this.getPathSearchRange();
                 this.world.profiler.startSection("pathfind");
