--- ../src-base/minecraft/net/minecraft/pathfinding/SwimNodeProcessor.java
+++ ../src-work/minecraft/net/minecraft/pathfinding/SwimNodeProcessor.java
@@ -1,6 +1,5 @@
 package net.minecraft.pathfinding;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.EntityLiving;
@@ -9,6 +8,8 @@
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.world.IBlockAccess;
 
+import javax.annotation.Nullable;
+
 public class SwimNodeProcessor extends NodeProcessor
 {
     public PathPoint getStart()
