--- ../src-base/minecraft/net/minecraftforge/event/DifficultyChangeEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/DifficultyChangeEvent.java
@@ -19,14 +19,11 @@
 
 package net.minecraftforge.event;
 
-import javax.annotation.Nonnull;
-
 import net.minecraft.world.EnumDifficulty;
 import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 import net.minecraftforge.fml.common.eventhandler.Event;
-import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
 
 /**
  * DifficultyChangeEvent is fired when difficulty is changing. <br>
@@ -39,24 +36,20 @@
  * <br>
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  **/
-public class DifficultyChangeEvent extends Event
-{
+public class DifficultyChangeEvent extends Event {
     private final EnumDifficulty difficulty;
     private final EnumDifficulty oldDifficulty;
 
-    public DifficultyChangeEvent(EnumDifficulty difficulty, EnumDifficulty oldDifficulty)
-    {
+    public DifficultyChangeEvent(EnumDifficulty difficulty, EnumDifficulty oldDifficulty) {
         this.difficulty = difficulty;
         this.oldDifficulty = oldDifficulty;
     }
 
-    public EnumDifficulty getDifficulty()
-    {
+    public EnumDifficulty getDifficulty() {
         return difficulty;
     }
 
-    public EnumDifficulty getOldDifficulty()
-    {
+    public EnumDifficulty getOldDifficulty() {
         return oldDifficulty;
     }
 }
