--- ../src-base/minecraft/net/minecraftforge/event/LootTableLoadEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/LootTableLoadEvent.java
@@ -35,36 +35,30 @@
  *
  */
 @Cancelable
-public class LootTableLoadEvent extends Event
-{
+public class LootTableLoadEvent extends Event {
     private final ResourceLocation name;
     private LootTable table;
     private LootTableManager lootTableManager;
 
-    public LootTableLoadEvent(ResourceLocation name, LootTable table, LootTableManager lootTableManager)
-    {
+    public LootTableLoadEvent(ResourceLocation name, LootTable table, LootTableManager lootTableManager) {
         this.name = name;
         this.table = table;
         this.lootTableManager = lootTableManager;
     }
 
-    public ResourceLocation getName()
-    {
+    public ResourceLocation getName() {
         return this.name;
     }
 
-    public LootTable getTable()
-    {
+    public LootTable getTable() {
         return this.table;
     }
 
-    public LootTableManager getLootTableManager()
-    {
-        return this.lootTableManager;
+    public void setTable(LootTable table) {
+        this.table = table;
     }
 
-    public void setTable(LootTable table)
-    {
-        this.table = table;
+    public LootTableManager getLootTableManager() {
+        return this.lootTableManager;
     }
 }
