--- ../src-base/minecraft/net/minecraftforge/event/RegistryEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/RegistryEvent.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,11 +19,6 @@
 
 package net.minecraftforge.event;
 
-import java.util.Collection;
-import java.util.stream.Collectors;
-
-import org.apache.commons.lang3.Validate;
-
 import com.google.common.collect.ImmutableList;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.common.ModContainer;
@@ -32,8 +27,12 @@
 import net.minecraftforge.fml.common.eventhandler.IContextSetter;
 import net.minecraftforge.registries.IForgeRegistry;
 import net.minecraftforge.registries.IForgeRegistryEntry;
+import org.apache.commons.lang3.Validate;
 
+import java.util.Collection;
+import java.util.stream.Collectors;
 
+
 /**
  * RegistryEvent supertype.
  */
