--- ../src-base/minecraft/net/minecraftforge/fml/client/config/GuiButtonExt.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/config/GuiButtonExt.java
@@ -32,15 +32,12 @@
  *
  * @author bspkrs
  */
-public class GuiButtonExt extends GuiButton
-{
-    public GuiButtonExt(int id, int xPos, int yPos, String displayString)
-    {
+public class GuiButtonExt extends GuiButton {
+    public GuiButtonExt(int id, int xPos, int yPos, String displayString) {
         super(id, xPos, yPos, displayString);
     }
 
-    public GuiButtonExt(int id, int xPos, int yPos, int width, int height, String displayString)
-    {
+    public GuiButtonExt(int id, int xPos, int yPos, int width, int height, String displayString) {
         super(id, xPos, yPos, width, height, displayString);
     }
 
@@ -48,26 +45,19 @@
      * Draws this button to the screen.
      */
     @Override
-    public void drawButton(Minecraft mc, int mouseX, int mouseY, float partial)
-    {
-        if (this.visible)
-        {
+    public void drawButton(Minecraft mc, int mouseX, int mouseY, float partial) {
+        if (this.visible) {
             this.hovered = mouseX >= this.x && mouseY >= this.y && mouseX < this.x + this.width && mouseY < this.y + this.height;
             int k = this.getHoverState(this.hovered);
             GuiUtils.drawContinuousTexturedBox(BUTTON_TEXTURES, this.x, this.y, 0, 46 + k * 20, this.width, this.height, 200, 20, 2, 3, 2, 2, this.zLevel);
             this.mouseDragged(mc, mouseX, mouseY);
             int color = 14737632;
 
-            if (packedFGColour != 0)
-            {
+            if (packedFGColour != 0) {
                 color = packedFGColour;
-            }
-            else if (!this.enabled)
-            {
+            } else if (!this.enabled) {
                 color = 10526880;
-            }
-            else if (this.hovered)
-            {
+            } else if (this.hovered) {
                 color = 16777120;
             }
 
