--- ../src-base/minecraft/net/minecraftforge/fml/client/config/GuiMessageDialog.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/config/GuiMessageDialog.java
@@ -26,19 +26,16 @@
 
 import javax.annotation.Nullable;
 
-public class GuiMessageDialog extends GuiDisconnected
-{
+public class GuiMessageDialog extends GuiDisconnected {
     protected String buttonText;
 
-    public GuiMessageDialog(@Nullable GuiScreen nextScreen, String title, ITextComponent message, String buttonText)
-    {
+    public GuiMessageDialog(@Nullable GuiScreen nextScreen, String title, ITextComponent message, String buttonText) {
         super(nextScreen, title, message);
         this.buttonText = buttonText;
     }
 
     @Override
-    public void initGui()
-    {
+    public void initGui() {
         super.initGui();
         buttonList.get(0).displayString = I18n.format(buttonText);
     }
