--- ../src-base/minecraft/net/minecraftforge/fml/client/config/GuiSelectStringEntries.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/config/GuiSelectStringEntries.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,18 +19,13 @@
 
 package net.minecraftforge.fml.client.config;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Map.Entry;
-
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiListExtended;
 import net.minecraftforge.fml.client.config.GuiConfigEntries.SelectValueEntry;
 
+import java.util.*;
+import java.util.Map.Entry;
+
 /**
  * This class implements the scrolling list functionality of the GuiSelectString screen.
  *
