--- ../src-base/minecraft/net/minecraftforge/fml/client/config/IConfigElement.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/config/IConfigElement.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,12 @@
 
 package net.minecraftforge.fml.client.config;
 
-import java.util.List;
-import java.util.regex.Pattern;
-
 import net.minecraftforge.fml.client.config.GuiConfigEntries.IConfigEntry;
 import net.minecraftforge.fml.client.config.GuiEditArrayEntries.IArrayEntry;
 
+import java.util.List;
+import java.util.regex.Pattern;
+
 /**
  * This interface provides the information needed by GuiConfig and GuiConfigEntries to display config elements for editing.
  * 
@@ -178,7 +178,7 @@
      */
     default String[] getValidValuesDisplay()
     {
-    	return null;
+        return null;
     }
 
     /**
