--- ../src-base/minecraft/net/minecraft/world/storage/IPlayerFileData.java
+++ ../src-work/minecraft/net/minecraft/world/storage/IPlayerFileData.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.storage;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.NBTTagCompound;
 
+import javax.annotation.Nullable;
+
 public interface IPlayerFileData
 {
     void writePlayerData(EntityPlayer player);
