--- ../src-base/minecraft/net/minecraft/world/storage/SaveDataMemoryStorage.java
+++ ../src-work/minecraft/net/minecraft/world/storage/SaveDataMemoryStorage.java
@@ -1,34 +1,29 @@
 package net.minecraft.world.storage;
 
-import javax.annotation.Nullable;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
-public class SaveDataMemoryStorage extends MapStorage
-{
-    public SaveDataMemoryStorage()
-    {
-        super((ISaveHandler)null);
+public class SaveDataMemoryStorage extends MapStorage {
+    public SaveDataMemoryStorage() {
+        super((ISaveHandler) null);
     }
 
     @Nullable
-    public WorldSavedData getOrLoadData(Class <? extends WorldSavedData > clazz, String dataIdentifier)
-    {
+    public WorldSavedData getOrLoadData(Class<? extends WorldSavedData> clazz, String dataIdentifier) {
         return this.loadedDataMap.get(dataIdentifier);
     }
 
-    public void setData(String dataIdentifier, WorldSavedData data)
-    {
+    public void setData(String dataIdentifier, WorldSavedData data) {
         this.loadedDataMap.put(dataIdentifier, data);
     }
 
-    public void saveAllData()
-    {
+    public void saveAllData() {
     }
 
-    public int getUniqueDataId(String key)
-    {
+    public int getUniqueDataId(String key) {
         return 0;
     }
 }
