--- ../src-base/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
+++ ../src-work/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
@@ -1,6 +1,7 @@
 package net.minecraft.world.storage;
 
 import com.google.common.collect.Lists;
+
 import java.util.Collections;
 import java.util.List;
 
@@ -45,14 +46,15 @@
                 ++this.savedIOCounter;
             }
 
-            try
-            {
-                Thread.sleep(this.isThreadWaiting ? 0L : 10L);
+            // Paper start - Add toggle
+            if (com.destroystokyo.paper.PaperConfig.enableFileIOThreadSleep) {
+                try {
+                    Thread.sleep(this.isThreadWaiting ? 0L : 2L);
+                } catch (InterruptedException interruptedexception) {
+                    interruptedexception.printStackTrace();
+                }
             }
-            catch (InterruptedException interruptedexception1)
-            {
-                interruptedexception1.printStackTrace();
-            }
+            // Paper end
         }
 
         if (this.threadedIOQueue.isEmpty())
