--- ../src-base/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
+++ ../src-work/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
@@ -45,14 +45,15 @@
                 ++this.savedIOCounter;
             }
 
+            // Spigot start - don't sleep in between chunks so we unload faster.
             try
-            {
-                Thread.sleep(this.isThreadWaiting ? 0L : 10L);
+			{
+            	Thread.sleep(this.isThreadWaiting ? 0L : 10L);
             }
-            catch (InterruptedException interruptedexception1)
-            {
-                interruptedexception1.printStackTrace();
-            }
+			catch (InterruptedException interruptedexception1)
+			{
+            	interruptedexception1.printStackTrace();
+            } // Spigot end
         }
 
         if (this.threadedIOQueue.isEmpty())
