--- ../src-base/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
+++ ../src-work/minecraft/net/minecraft/world/storage/ThreadedFileIOBase.java
@@ -46,12 +46,12 @@
             }
 
             try
-            {
-                Thread.sleep(this.isThreadWaiting ? 0L : 10L);
+			{
+            	Thread.sleep(this.isThreadWaiting ? 0L : 10L);
             }
-            catch (InterruptedException interruptedexception1)
-            {
-                interruptedexception1.printStackTrace();
+			catch (InterruptedException interruptedexception1)
+			{
+            	interruptedexception1.printStackTrace();
             }
         }
 
