--- ../src-base/minecraft/net/minecraft/world/storage/WorldSavedData.java
+++ ../src-work/minecraft/net/minecraft/world/storage/WorldSavedData.java
@@ -2,13 +2,11 @@
 
 import net.minecraft.nbt.NBTTagCompound;
 
-public abstract class WorldSavedData implements net.minecraftforge.common.util.INBTSerializable<NBTTagCompound>
-{
+public abstract class WorldSavedData implements net.minecraftforge.common.util.INBTSerializable<NBTTagCompound> {
     public final String mapName;
     private boolean dirty;
 
-    public WorldSavedData(String name)
-    {
+    public WorldSavedData(String name) {
         this.mapName = name;
     }
 
@@ -16,28 +14,23 @@
 
     public abstract NBTTagCompound writeToNBT(NBTTagCompound compound);
 
-    public void markDirty()
-    {
+    public void markDirty() {
         this.setDirty(true);
     }
 
-    public void setDirty(boolean isDirty)
-    {
-        this.dirty = isDirty;
+    public boolean isDirty() {
+        return this.dirty;
     }
 
-    public boolean isDirty()
-    {
-        return this.dirty;
+    public void setDirty(boolean isDirty) {
+        this.dirty = isDirty;
     }
 
-    public void deserializeNBT(NBTTagCompound nbt)
-    {
+    public void deserializeNBT(NBTTagCompound nbt) {
         this.readFromNBT(nbt);
     }
 
-    public NBTTagCompound serializeNBT()
-    {
+    public NBTTagCompound serializeNBT() {
         return this.writeToNBT(new NBTTagCompound());
     }
 }
