--- ../src-base/minecraft/net/minecraft/world/storage/WorldSavedDataCallableSave.java
+++ ../src-work/minecraft/net/minecraft/world/storage/WorldSavedDataCallableSave.java
@@ -1,16 +1,13 @@
 package net.minecraft.world.storage;
 
-public class WorldSavedDataCallableSave implements Runnable
-{
+public class WorldSavedDataCallableSave implements Runnable {
     private final WorldSavedData data;
 
-    public WorldSavedDataCallableSave(WorldSavedData dataIn)
-    {
+    public WorldSavedDataCallableSave(WorldSavedData dataIn) {
         this.data = dataIn;
     }
 
-    public void run()
-    {
+    public void run() {
         this.data.markDirty();
     }
 }
