--- ../src-base/minecraft/net/minecraft/network/login/client/CPacketLoginStart.java
+++ ../src-work/minecraft/net/minecraft/network/login/client/CPacketLoginStart.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.login.client;
 
 import com.mojang.authlib.GameProfile;
-import java.io.IOException;
-import java.util.UUID;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginServer;
 
+import java.io.IOException;
+import java.util.UUID;
+
 public class CPacketLoginStart implements Packet<INetHandlerLoginServer>
 {
     private GameProfile profile;
