--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAILookAtVillager.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAILookAtVillager.java
@@ -3,54 +3,42 @@
 import net.minecraft.entity.monster.EntityIronGolem;
 import net.minecraft.entity.passive.EntityVillager;
 
-public class EntityAILookAtVillager extends EntityAIBase
-{
+public class EntityAILookAtVillager extends EntityAIBase {
     private final EntityIronGolem ironGolem;
     private EntityVillager villager;
     private int lookTime;
 
-    public EntityAILookAtVillager(EntityIronGolem ironGolemIn)
-    {
+    public EntityAILookAtVillager(EntityIronGolem ironGolemIn) {
         this.ironGolem = ironGolemIn;
         this.setMutexBits(3);
     }
 
-    public boolean shouldExecute()
-    {
-        if (!this.ironGolem.world.isDaytime())
-        {
+    public boolean shouldExecute() {
+        if (!this.ironGolem.world.isDaytime()) {
             return false;
-        }
-        else if (this.ironGolem.getRNG().nextInt(8000) != 0)
-        {
+        } else if (this.ironGolem.getRNG().nextInt(8000) != 0) {
             return false;
-        }
-        else
-        {
-            this.villager = (EntityVillager)this.ironGolem.world.findNearestEntityWithinAABB(EntityVillager.class, this.ironGolem.getEntityBoundingBox().grow(6.0D, 2.0D, 6.0D), this.ironGolem);
+        } else {
+            this.villager = (EntityVillager) this.ironGolem.world.findNearestEntityWithinAABB(EntityVillager.class, this.ironGolem.getEntityBoundingBox().grow(6.0D, 2.0D, 6.0D), this.ironGolem);
             return this.villager != null;
         }
     }
 
-    public boolean shouldContinueExecuting()
-    {
+    public boolean shouldContinueExecuting() {
         return this.lookTime > 0;
     }
 
-    public void startExecuting()
-    {
+    public void startExecuting() {
         this.lookTime = 400;
         this.ironGolem.setHoldingRose(true);
     }
 
-    public void resetTask()
-    {
+    public void resetTask() {
         this.ironGolem.setHoldingRose(false);
         this.villager = null;
     }
 
-    public void updateTask()
-    {
+    public void updateTask() {
         this.ironGolem.getLookHelper().setLookPositionWithEntity(this.villager, 30.0F, 30.0F);
         --this.lookTime;
     }
