--- ../src-base/minecraft/net/minecraft/entity/ai/EntityLookHelper.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityLookHelper.java
@@ -5,8 +5,7 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.math.MathHelper;
 
-public class EntityLookHelper
-{
+public class EntityLookHelper {
     private final EntityLiving entity;
     private float deltaLookYaw;
     private float deltaLookPitch;
@@ -15,21 +14,16 @@
     private double posY;
     private double posZ;
 
-    public EntityLookHelper(EntityLiving entitylivingIn)
-    {
+    public EntityLookHelper(EntityLiving entitylivingIn) {
         this.entity = entitylivingIn;
     }
 
-    public void setLookPositionWithEntity(Entity entityIn, float deltaYaw, float deltaPitch)
-    {
+    public void setLookPositionWithEntity(Entity entityIn, float deltaYaw, float deltaPitch) {
         this.posX = entityIn.posX;
 
-        if (entityIn instanceof EntityLivingBase)
-        {
-            this.posY = entityIn.posY + (double)entityIn.getEyeHeight();
-        }
-        else
-        {
+        if (entityIn instanceof EntityLivingBase) {
+            this.posY = entityIn.posY + (double) entityIn.getEyeHeight();
+        } else {
             this.posY = (entityIn.getEntityBoundingBox().minY + entityIn.getEntityBoundingBox().maxY) / 2.0D;
         }
 
@@ -39,8 +33,7 @@
         this.isLooking = true;
     }
 
-    public void setLookPosition(double x, double y, double z, float deltaYaw, float deltaPitch)
-    {
+    public void setLookPosition(double x, double y, double z, float deltaYaw, float deltaPitch) {
         this.posX = x;
         this.posY = y;
         this.posZ = z;
@@ -49,77 +42,63 @@
         this.isLooking = true;
     }
 
-    public void onUpdateLook()
-    {
+    public void onUpdateLook() {
         this.entity.rotationPitch = 0.0F;
 
-        if (this.isLooking)
-        {
+        if (this.isLooking) {
             this.isLooking = false;
             double d0 = this.posX - this.entity.posX;
-            double d1 = this.posY - (this.entity.posY + (double)this.entity.getEyeHeight());
+            double d1 = this.posY - (this.entity.posY + (double) this.entity.getEyeHeight());
             double d2 = this.posZ - this.entity.posZ;
-            double d3 = (double)MathHelper.sqrt(d0 * d0 + d2 * d2);
-            float f = (float)(MathHelper.atan2(d2, d0) * (180D / Math.PI)) - 90.0F;
-            float f1 = (float)(-(MathHelper.atan2(d1, d3) * (180D / Math.PI)));
+            double d3 = (double) MathHelper.sqrt(d0 * d0 + d2 * d2);
+            float f = (float) (MathHelper.atan2(d2, d0) * (180D / Math.PI)) - 90.0F;
+            float f1 = (float) (-(MathHelper.atan2(d1, d3) * (180D / Math.PI)));
             this.entity.rotationPitch = this.updateRotation(this.entity.rotationPitch, f1, this.deltaLookPitch);
             this.entity.rotationYawHead = this.updateRotation(this.entity.rotationYawHead, f, this.deltaLookYaw);
-        }
-        else
-        {
+        } else {
             this.entity.rotationYawHead = this.updateRotation(this.entity.rotationYawHead, this.entity.renderYawOffset, 10.0F);
         }
 
         float f2 = MathHelper.wrapDegrees(this.entity.rotationYawHead - this.entity.renderYawOffset);
 
-        if (!this.entity.getNavigator().noPath())
-        {
-            if (f2 < -75.0F)
-            {
+        if (!this.entity.getNavigator().noPath()) {
+            if (f2 < -75.0F) {
                 this.entity.rotationYawHead = this.entity.renderYawOffset - 75.0F;
             }
 
-            if (f2 > 75.0F)
-            {
+            if (f2 > 75.0F) {
                 this.entity.rotationYawHead = this.entity.renderYawOffset + 75.0F;
             }
         }
     }
 
-    private float updateRotation(float p_75652_1_, float p_75652_2_, float p_75652_3_)
-    {
+    private float updateRotation(float p_75652_1_, float p_75652_2_, float p_75652_3_) {
         float f = MathHelper.wrapDegrees(p_75652_2_ - p_75652_1_);
 
-        if (f > p_75652_3_)
-        {
+        if (f > p_75652_3_) {
             f = p_75652_3_;
         }
 
-        if (f < -p_75652_3_)
-        {
+        if (f < -p_75652_3_) {
             f = -p_75652_3_;
         }
 
         return p_75652_1_ + f;
     }
 
-    public boolean getIsLooking()
-    {
+    public boolean getIsLooking() {
         return this.isLooking;
     }
 
-    public double getLookPosX()
-    {
+    public double getLookPosX() {
         return this.posX;
     }
 
-    public double getLookPosY()
-    {
+    public double getLookPosY() {
         return this.posY;
     }
 
-    public double getLookPosZ()
-    {
+    public double getLookPosZ() {
         return this.posZ;
     }
 }
