--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAISwimming.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAISwimming.java
@@ -11,6 +11,7 @@
     public EntityAISwimming(EntityLiving entityIn)
     {
         this.entity = entityIn;
+        if ( entityIn.world.paperConfig.nerfedMobsShouldJump)  entityIn.goalFloat = this; // Paper
         this.setMutexBits(4);
 
         if (entityIn.getNavigator() instanceof PathNavigateGround)
