--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIPanic.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIPanic.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.ai;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
@@ -9,6 +8,8 @@
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+
 public class EntityAIPanic extends EntityAIBase
 {
     protected final EntityCreature creature;
@@ -73,6 +74,12 @@
 
     public boolean shouldContinueExecuting()
     {
+        // CraftBukkit start - introduce a temporary timeout hack until this is fixed properly
+        if ((this.creature.ticksExisted - this.creature.revengeTimer) > 100) {
+            this.creature.onKillEntity(null);
+            return false;
+        }
+        // CraftBukkit end
         return !this.creature.getNavigator().noPath();
     }
 
