--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIWanderAvoidWaterFlying.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIWanderAvoidWaterFlying.java
@@ -1,7 +1,5 @@
 package net.minecraft.entity.ai;
 
-import java.util.Iterator;
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockLeaves;
 import net.minecraft.entity.EntityCreature;
@@ -11,25 +9,23 @@
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.math.Vec3d;
 
-public class EntityAIWanderAvoidWaterFlying extends EntityAIWanderAvoidWater
-{
-    public EntityAIWanderAvoidWaterFlying(EntityCreature p_i47413_1_, double p_i47413_2_)
-    {
+import javax.annotation.Nullable;
+import java.util.Iterator;
+
+public class EntityAIWanderAvoidWaterFlying extends EntityAIWanderAvoidWater {
+    public EntityAIWanderAvoidWaterFlying(EntityCreature p_i47413_1_, double p_i47413_2_) {
         super(p_i47413_1_, p_i47413_2_);
     }
 
     @Nullable
-    protected Vec3d getPosition()
-    {
+    protected Vec3d getPosition() {
         Vec3d vec3d = null;
 
-        if (this.entity.isInWater() || this.entity.isOverWater())
-        {
+        if (this.entity.isInWater() || this.entity.isOverWater()) {
             vec3d = RandomPositionGenerator.getLandPos(this.entity, 15, 15);
         }
 
-        if (this.entity.getRNG().nextFloat() >= this.probability)
-        {
+        if (this.entity.getRNG().nextFloat() >= this.probability) {
             vec3d = this.getTreePos();
         }
 
@@ -37,8 +33,7 @@
     }
 
     @Nullable
-    private Vec3d getTreePos()
-    {
+    private Vec3d getTreePos() {
         BlockPos blockpos = new BlockPos(this.entity);
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
         BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
@@ -46,27 +41,23 @@
         Iterator iterator = iterable.iterator();
         BlockPos blockpos1;
 
-        while (true)
-        {
-            if (!iterator.hasNext())
-            {
+        while (true) {
+            if (!iterator.hasNext()) {
                 return null;
             }
 
-            blockpos1 = (BlockPos)iterator.next();
+            blockpos1 = (BlockPos) iterator.next();
 
-            if (!blockpos.equals(blockpos1))
-            {
+            if (!blockpos.equals(blockpos1)) {
                 Block block = this.entity.world.getBlockState(blockpos$mutableblockpos1.setPos(blockpos1).move(EnumFacing.DOWN)).getBlock();
                 boolean flag = block instanceof BlockLeaves || block == Blocks.LOG || block == Blocks.LOG2;
 
-                if (flag && this.entity.world.isAirBlock(blockpos1) && this.entity.world.isAirBlock(blockpos$mutableblockpos.setPos(blockpos1).move(EnumFacing.UP)))
-                {
+                if (flag && this.entity.world.isAirBlock(blockpos1) && this.entity.world.isAirBlock(blockpos$mutableblockpos.setPos(blockpos1).move(EnumFacing.UP))) {
                     break;
                 }
             }
         }
 
-        return new Vec3d((double)blockpos1.getX(), (double)blockpos1.getY(), (double)blockpos1.getZ());
+        return new Vec3d((double) blockpos1.getX(), (double) blockpos1.getY(), (double) blockpos1.getZ());
     }
 }
