--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAITargetNonTamed.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAITargetNonTamed.java
@@ -4,18 +4,15 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityTameable;
 
-public class EntityAITargetNonTamed<T extends EntityLivingBase> extends EntityAINearestAttackableTarget<T>
-{
+public class EntityAITargetNonTamed<T extends EntityLivingBase> extends EntityAINearestAttackableTarget<T> {
     private final EntityTameable tameable;
 
-    public EntityAITargetNonTamed(EntityTameable entityIn, Class<T> classTarget, boolean checkSight, Predicate <? super T > targetSelector)
-    {
+    public EntityAITargetNonTamed(EntityTameable entityIn, Class<T> classTarget, boolean checkSight, Predicate<? super T> targetSelector) {
         super(entityIn, classTarget, 10, checkSight, false, targetSelector);
         this.tameable = entityIn;
     }
 
-    public boolean shouldExecute()
-    {
+    public boolean shouldExecute() {
         return !this.tameable.isTamed() && super.shouldExecute();
     }
 }
