--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIWanderAvoidWater.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIWanderAvoidWater.java
@@ -1,9 +1,10 @@
 package net.minecraft.entity.ai;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.util.math.Vec3d;
 
+import javax.annotation.Nullable;
+
 public class EntityAIWanderAvoidWater extends EntityAIWander
 {
     protected final float probability;
