--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIRestrictSun.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIRestrictSun.java
@@ -4,27 +4,22 @@
 import net.minecraft.inventory.EntityEquipmentSlot;
 import net.minecraft.pathfinding.PathNavigateGround;
 
-public class EntityAIRestrictSun extends EntityAIBase
-{
+public class EntityAIRestrictSun extends EntityAIBase {
     private final EntityCreature entity;
 
-    public EntityAIRestrictSun(EntityCreature creature)
-    {
+    public EntityAIRestrictSun(EntityCreature creature) {
         this.entity = creature;
     }
 
-    public boolean shouldExecute()
-    {
+    public boolean shouldExecute() {
         return this.entity.world.isDaytime() && this.entity.getItemStackFromSlot(EntityEquipmentSlot.HEAD).isEmpty();
     }
 
-    public void startExecuting()
-    {
-        ((PathNavigateGround)this.entity.getNavigator()).setAvoidSun(true);
+    public void startExecuting() {
+        ((PathNavigateGround) this.entity.getNavigator()).setAvoidSun(true);
     }
 
-    public void resetTask()
-    {
-        ((PathNavigateGround)this.entity.getNavigator()).setAvoidSun(false);
+    public void resetTask() {
+        ((PathNavigateGround) this.entity.getNavigator()).setAvoidSun(false);
     }
 }
