--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIZombieAttack.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIZombieAttack.java
@@ -2,40 +2,32 @@
 
 import net.minecraft.entity.monster.EntityZombie;
 
-public class EntityAIZombieAttack extends EntityAIAttackMelee
-{
+public class EntityAIZombieAttack extends EntityAIAttackMelee {
     private final EntityZombie zombie;
     private int raiseArmTicks;
 
-    public EntityAIZombieAttack(EntityZombie zombieIn, double speedIn, boolean longMemoryIn)
-    {
+    public EntityAIZombieAttack(EntityZombie zombieIn, double speedIn, boolean longMemoryIn) {
         super(zombieIn, speedIn, longMemoryIn);
         this.zombie = zombieIn;
     }
 
-    public void startExecuting()
-    {
+    public void startExecuting() {
         super.startExecuting();
         this.raiseArmTicks = 0;
     }
 
-    public void resetTask()
-    {
+    public void resetTask() {
         super.resetTask();
         this.zombie.setArmsRaised(false);
     }
 
-    public void updateTask()
-    {
+    public void updateTask() {
         super.updateTask();
         ++this.raiseArmTicks;
 
-        if (this.raiseArmTicks >= 5 && this.attackTick < 10)
-        {
+        if (this.raiseArmTicks >= 5 && this.attackTick < 10) {
             this.zombie.setArmsRaised(true);
-        }
-        else
-        {
+        } else {
             this.zombie.setArmsRaised(false);
         }
     }
