--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIAttackRangedBow.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIAttackRangedBow.java
@@ -3,7 +3,6 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.IRangedAttackMob;
 import net.minecraft.entity.monster.EntityMob;
-import net.minecraft.init.Items;
 import net.minecraft.item.ItemBow;
 import net.minecraft.util.EnumHand;
 
@@ -40,7 +39,7 @@
 
     protected boolean isBowInMainhand()
     {
-        return !this.entity.getHeldItemMainhand().isEmpty() && this.entity.getHeldItemMainhand().getItem() == Items.BOW;
+        return !this.entity.getHeldItemMainhand().isEmpty() && this.entity.getHeldItemMainhand().getItem() instanceof ItemBow;
     }
 
     public boolean shouldContinueExecuting()
@@ -51,13 +50,13 @@
     public void startExecuting()
     {
         super.startExecuting();
-        ((IRangedAttackMob)this.entity).setSwingingArms(true);
+        this.entity.setSwingingArms(true);
     }
 
     public void resetTask()
     {
         super.resetTask();
-        ((IRangedAttackMob)this.entity).setSwingingArms(false);
+        this.entity.setSwingingArms(false);
         this.seeTime = 0;
         this.attackTime = -1;
         this.entity.resetActiveHand();
