--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAITradePlayer.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAITradePlayer.java
@@ -3,60 +3,41 @@
 import net.minecraft.entity.passive.EntityVillager;
 import net.minecraft.entity.player.EntityPlayer;
 
-public class EntityAITradePlayer extends EntityAIBase
-{
+public class EntityAITradePlayer extends EntityAIBase {
     private final EntityVillager villager;
 
-    public EntityAITradePlayer(EntityVillager villagerIn)
-    {
+    public EntityAITradePlayer(EntityVillager villagerIn) {
         this.villager = villagerIn;
         this.setMutexBits(5);
     }
 
-    public boolean shouldExecute()
-    {
-        if (!this.villager.isEntityAlive())
-        {
+    public boolean shouldExecute() {
+        if (!this.villager.isEntityAlive()) {
             return false;
-        }
-        else if (this.villager.isInWater())
-        {
+        } else if (this.villager.isInWater()) {
             return false;
-        }
-        else if (!this.villager.onGround)
-        {
+        } else if (!this.villager.onGround) {
             return false;
-        }
-        else if (this.villager.velocityChanged)
-        {
+        } else if (this.villager.velocityChanged) {
             return false;
-        }
-        else
-        {
+        } else {
             EntityPlayer entityplayer = this.villager.getCustomer();
 
-            if (entityplayer == null)
-            {
+            if (entityplayer == null) {
                 return false;
-            }
-            else if (this.villager.getDistanceSq(entityplayer) > 16.0D)
-            {
+            } else if (this.villager.getDistanceSq(entityplayer) > 16.0D) {
                 return false;
-            }
-            else
-            {
+            } else {
                 return entityplayer.openContainer != null;
             }
         }
     }
 
-    public void startExecuting()
-    {
+    public void startExecuting() {
         this.villager.getNavigator().clearPath();
     }
 
-    public void resetTask()
-    {
-        this.villager.setCustomer((EntityPlayer)null);
+    public void resetTask() {
+        this.villager.setCustomer((EntityPlayer) null);
     }
 }
