--- ../src-base/minecraft/net/minecraft/client/particle/IParticleFactory.java
+++ ../src-work/minecraft/net/minecraft/client/particle/IParticleFactory.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.particle;
 
-import javax.annotation.Nullable;
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public interface IParticleFactory
 {
