--- ../src-base/minecraft/net/minecraft/client/particle/ParticleFallingDust.java
+++ ../src-work/minecraft/net/minecraft/client/particle/ParticleFallingDust.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.particle;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockFalling;
 import net.minecraft.block.state.IBlockState;
@@ -15,6 +14,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class ParticleFallingDust extends Particle
 {
