--- ../src-base/minecraft/net/minecraft/client/particle/ParticleSpell.java
+++ ../src-work/minecraft/net/minecraft/client/particle/ParticleSpell.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.particle;
 
-import java.util.Random;
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class ParticleSpell extends Particle
 {
