--- ../src-base/minecraft/net/minecraft/client/particle/Particle.java
+++ ../src-work/minecraft/net/minecraft/client/particle/Particle.java
@@ -1,7 +1,5 @@
 package net.minecraft.client.particle;
 
-import java.util.List;
-import java.util.Random;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.entity.Entity;
@@ -13,6 +11,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class Particle
 {
