--- ../src-base/minecraft/net/minecraft/client/particle/ParticleBlockDust.java
+++ ../src-work/minecraft/net/minecraft/client/particle/ParticleBlockDust.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.particle;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.EnumBlockRenderType;
@@ -8,6 +7,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class ParticleBlockDust extends ParticleDigging
 {
