--- ../src-base/minecraft/net/minecraft/client/gui/chat/NormalChatListener.java
+++ ../src-work/minecraft/net/minecraft/client/gui/chat/NormalChatListener.java
@@ -7,17 +7,14 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class NormalChatListener implements IChatListener
-{
+public class NormalChatListener implements IChatListener {
     private final Minecraft mc;
 
-    public NormalChatListener(Minecraft p_i47393_1_)
-    {
+    public NormalChatListener(Minecraft p_i47393_1_) {
         this.mc = p_i47393_1_;
     }
 
-    public void say(ChatType chatTypeIn, ITextComponent message)
-    {
+    public void say(ChatType chatTypeIn, ITextComponent message) {
         this.mc.ingameGUI.getChatGUI().printChatMessage(message);
     }
 }
