--- ../src-base/minecraft/net/minecraft/client/gui/chat/OverlayChatListener.java
+++ ../src-work/minecraft/net/minecraft/client/gui/chat/OverlayChatListener.java
@@ -7,17 +7,14 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class OverlayChatListener implements IChatListener
-{
+public class OverlayChatListener implements IChatListener {
     private final Minecraft mc;
 
-    public OverlayChatListener(Minecraft minecraftIn)
-    {
+    public OverlayChatListener(Minecraft minecraftIn) {
         this.mc = minecraftIn;
     }
 
-    public void say(ChatType chatTypeIn, ITextComponent message)
-    {
+    public void say(ChatType chatTypeIn, ITextComponent message) {
         this.mc.ingameGUI.setOverlayMessage(message, false);
     }
 }
