--- ../src-base/minecraft/net/minecraft/client/multiplayer/ServerAddress.java
+++ ../src-work/minecraft/net/minecraft/client/multiplayer/ServerAddress.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.multiplayer;
 
-import java.net.IDN;
-import java.util.Hashtable;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
 import javax.naming.directory.Attributes;
 import javax.naming.directory.DirContext;
 import javax.naming.directory.InitialDirContext;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+import java.net.IDN;
+import java.util.Hashtable;
 
 @SideOnly(Side.CLIENT)
 public class ServerAddress
