--- ../src-base/minecraft/net/minecraft/client/multiplayer/WorldClient.java
+++ ../src-work/minecraft/net/minecraft/client/multiplayer/WorldClient.java
@@ -115,7 +115,7 @@
         return this.clientChunkProvider;
     }
 
-    protected boolean isChunkLoaded(int x, int z, boolean allowEmpty)
+    public boolean isChunkLoaded(int x, int z, boolean allowEmpty)
     {
         return allowEmpty || !this.getChunkProvider().provideChunk(x, z).isEmpty();
     }
@@ -384,9 +384,8 @@
 
         this.unloadedEntityList.clear();
 
-        for (int j1 = 0; j1 < this.loadedEntityList.size(); ++j1)
+        for (Entity entity1 : this.loadedEntityList)
         {
-            Entity entity1 = this.loadedEntityList.get(j1);
             Entity entity2 = entity1.getRidingEntity();
 
             if (entity2 != null)
@@ -409,7 +408,7 @@
                     this.getChunkFromChunkCoords(k1, l).removeEntity(entity1);
                 }
 
-                this.loadedEntityList.remove(j1--);
+                this.loadedEntityList.remove(entity1);
                 this.onEntityRemoved(entity1);
             }
         }
