--- ../src-base/minecraft/net/minecraft/client/gui/inventory/CreativeCrafting.java
+++ ../src-work/minecraft/net/minecraft/client/gui/inventory/CreativeCrafting.java
@@ -10,29 +10,23 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class CreativeCrafting implements IContainerListener
-{
+public class CreativeCrafting implements IContainerListener {
     private final Minecraft mc;
 
-    public CreativeCrafting(Minecraft mc)
-    {
+    public CreativeCrafting(Minecraft mc) {
         this.mc = mc;
     }
 
-    public void sendAllContents(Container containerToSend, NonNullList<ItemStack> itemsList)
-    {
+    public void sendAllContents(Container containerToSend, NonNullList<ItemStack> itemsList) {
     }
 
-    public void sendSlotContents(Container containerToSend, int slotInd, ItemStack stack)
-    {
+    public void sendSlotContents(Container containerToSend, int slotInd, ItemStack stack) {
         this.mc.playerController.sendSlotPacket(stack, slotInd);
     }
 
-    public void sendWindowProperty(Container containerIn, int varToUpdate, int newValue)
-    {
+    public void sendWindowProperty(Container containerIn, int varToUpdate, int newValue) {
     }
 
-    public void sendAllWindowProperties(Container containerIn, IInventory inventory)
-    {
+    public void sendAllWindowProperties(Container containerIn, IInventory inventory) {
     }
 }
