--- ../src-base/minecraft/net/minecraft/client/gui/inventory/GuiCrafting.java
+++ ../src-work/minecraft/net/minecraft/client/gui/inventory/GuiCrafting.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui.inventory;
 
-import java.io.IOException;
 import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiButtonImage;
 import net.minecraft.client.gui.recipebook.GuiRecipeBook;
@@ -17,6 +16,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiCrafting extends GuiContainer implements IRecipeShownListener
 {
