--- ../src-base/minecraft/net/minecraft/client/gui/inventory/GuiEditCommandBlockMinecart.java
+++ ../src-work/minecraft/net/minecraft/client/gui/inventory/GuiEditCommandBlockMinecart.java
@@ -1,8 +1,6 @@
 package net.minecraft.client.gui.inventory;
 
 import io.netty.buffer.Unpooled;
-import java.io.IOException;
-import javax.annotation.Nullable;
 import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.gui.GuiTextField;
@@ -18,6 +16,9 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.lwjgl.input.Keyboard;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiEditCommandBlockMinecart extends GuiScreen implements ITabCompleter
 {
