--- ../src-base/minecraft/net/minecraft/client/gui/inventory/GuiInventory.java
+++ ../src-work/minecraft/net/minecraft/client/gui/inventory/GuiInventory.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui.inventory;
 
-import java.io.IOException;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiButtonImage;
@@ -20,6 +19,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiInventory extends InventoryEffectRenderer implements IRecipeShownListener
 {
