--- ../src-base/minecraft/net/minecraftforge/fml/client/registry/IRenderFactory.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/registry/IRenderFactory.java
@@ -23,7 +23,6 @@
 import net.minecraft.client.renderer.entity.RenderManager;
 import net.minecraft.entity.Entity;
 
-public interface IRenderFactory<T extends Entity>
-{
+public interface IRenderFactory<T extends Entity> {
     Render<? super T> createRenderFor(RenderManager manager);
 }
