--- ../src-base/minecraft/net/minecraftforge/client/model/b3d/B3DClip.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/b3d/B3DClip.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,6 +19,7 @@
 
 package net.minecraftforge.client.model.b3d;
 
+import com.google.common.collect.ImmutableSet;
 import net.minecraftforge.client.model.b3d.B3DLoader.NodeJoint;
 import net.minecraftforge.client.model.b3d.B3DModel.Key;
 import net.minecraftforge.client.model.b3d.B3DModel.Node;
@@ -29,8 +30,6 @@
 import net.minecraftforge.common.model.animation.IJointClip;
 import net.minecraftforge.common.model.animation.JointClips;
 
-import com.google.common.collect.ImmutableSet;
-
 // FIXME: is this fast enough?
 public enum B3DClip implements IClip
 {
