--- ../src-base/minecraft/net/minecraftforge/fml/common/ClassNameUtils.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/ClassNameUtils.java
@@ -19,11 +19,9 @@
 
 package net.minecraftforge.fml.common;
 
-public class ClassNameUtils
-{
-    public static String shortName(Class<?> clz)
-    {
+public class ClassNameUtils {
+    public static String shortName(Class<?> clz) {
         String nm = clz.getName();
-        return nm.indexOf('.') > -1 ? nm.substring(nm.lastIndexOf('.')+1) : nm;
+        return nm.indexOf('.') > -1 ? nm.substring(nm.lastIndexOf('.') + 1) : nm;
     }
 }
