--- ../src-base/minecraft/net/minecraftforge/fml/common/DuplicateModsFoundException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/DuplicateModsFoundException.java
@@ -19,9 +19,6 @@
 
 package net.minecraftforge.fml.common;
 
-import java.io.File;
-import java.util.Map.Entry;
-
 import com.google.common.collect.SetMultimap;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraftforge.fml.client.GuiDupesFound;
@@ -29,21 +26,21 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class DuplicateModsFoundException extends LoaderException implements IDisplayableError
-{
+import java.io.File;
+import java.util.Map.Entry;
+
+public class DuplicateModsFoundException extends LoaderException implements IDisplayableError {
     private static final long serialVersionUID = 1L;
-    public SetMultimap<ModContainer,File> dupes;
+    public SetMultimap<ModContainer, File> dupes;
 
     public DuplicateModsFoundException(SetMultimap<ModContainer, File> dupes) {
         this.dupes = dupes;
     }
 
     @Override
-    protected void printStackTrace(WrappedPrintStream stream)
-    {
+    protected void printStackTrace(WrappedPrintStream stream) {
         stream.println("Duplicate Mods:");
-        for (Entry<ModContainer, File> e : dupes.entries())
-        {
+        for (Entry<ModContainer, File> e : dupes.entries()) {
             stream.println(String.format("\t%s : %s", e.getKey().getModId(), e.getValue().getAbsolutePath()));
         }
         stream.println("");
@@ -51,8 +48,7 @@
 
     @Override
     @SideOnly(Side.CLIENT)
-    public GuiScreen createGui()
-    {
+    public GuiScreen createGui() {
         return new GuiDupesFound(this);
     }
 }
