--- ../src-base/minecraft/net/minecraftforge/fml/common/FMLContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/FMLContainer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,10 +19,11 @@
 
 package net.minecraftforge.fml.common;
 
-import java.io.File;
-import java.security.cert.Certificate;
-import java.util.Arrays;
-import java.util.Map;
+import cn.pfcraft.server.Mohist;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
@@ -42,16 +43,14 @@
 import net.minecraftforge.registries.ForgeRegistry;
 import net.minecraftforge.registries.GameData;
 import net.minecraftforge.registries.RegistryManager;
-
 import org.apache.logging.log4j.LogManager;
-
-import com.google.common.collect.Maps;
-import com.google.common.collect.Multimap;
-import com.google.common.eventbus.EventBus;
-import com.google.common.eventbus.Subscribe;
 import org.apache.logging.log4j.Logger;
 
 import javax.annotation.Nullable;
+import java.io.File;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+import java.util.Map;
 
 /**
  * @author cpw
@@ -121,7 +120,7 @@
 
         NBTTagCompound registries = new NBTTagCompound();
         fmlData.setTag("Registries", registries);
-        FMLLog.log.debug("Gathering id map for writing to world save {}", info.getWorldName());
+        Mohist.LOGGER.debug("Gathering id map for writing to world save {}", info.getWorldName());
 
         for (Map.Entry<ResourceLocation, ForgeRegistry.Snapshot> e : RegistryManager.ACTIVE.takeSnapshot(true).entrySet())
         {
