--- ../src-base/minecraft/net/minecraftforge/fml/common/InjectedModContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/InjectedModContainer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,6 +19,11 @@
 
 package net.minecraftforge.fml.common;
 
+import com.google.common.eventbus.EventBus;
+import net.minecraftforge.fml.common.versioning.ArtifactVersion;
+import net.minecraftforge.fml.common.versioning.VersionRange;
+
+import javax.annotation.Nullable;
 import java.io.File;
 import java.net.URL;
 import java.security.cert.Certificate;
@@ -26,13 +31,6 @@
 import java.util.Map;
 import java.util.Set;
 
-import net.minecraftforge.fml.common.versioning.ArtifactVersion;
-import net.minecraftforge.fml.common.versioning.VersionRange;
-
-import com.google.common.eventbus.EventBus;
-
-import javax.annotation.Nullable;
-
 public class InjectedModContainer implements ModContainer
 {
     private File source;
