--- ../src-base/minecraft/net/minecraftforge/fml/common/IWorldGenerator.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/IWorldGenerator.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,13 +19,13 @@
 
 package net.minecraftforge.fml.common;
 
-import java.util.Random;
-
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.IChunkProvider;
 import net.minecraft.world.gen.IChunkGenerator;
 
+import java.util.Random;
 
+
 /**
  * This is called back during world generation.
  *
