--- ../src-base/minecraft/net/minecraftforge/fml/common/MCPDummyContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/MCPDummyContainer.java
@@ -22,18 +22,17 @@
 import com.google.common.eventbus.EventBus;
 
 public class MCPDummyContainer extends DummyModContainer {
-	public MCPDummyContainer(ModMetadata metadata) {
-		super(metadata);
-	}
+    public MCPDummyContainer(ModMetadata metadata) {
+        super(metadata);
+    }
 
-	@Override
-	public boolean registerBus(EventBus bus, LoadController controller) {
-		return true;
-	}
+    @Override
+    public boolean registerBus(EventBus bus, LoadController controller) {
+        return true;
+    }
 
-	@Override
-	public Disableable canBeDisabled()
-	{
-	    return Disableable.YES;
-	}
+    @Override
+    public Disableable canBeDisabled() {
+        return Disableable.YES;
+    }
 }
