--- ../src-base/minecraft/net/minecraftforge/fml/common/MissingModsException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/MissingModsException.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,16 +19,7 @@
 
 package net.minecraftforge.fml.common;
 
-import javax.annotation.Nullable;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.stream.Collectors;
-
 import com.google.common.base.Preconditions;
-
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraftforge.fml.client.GuiModsMissing;
 import net.minecraftforge.fml.client.IDisplayableError;
@@ -36,6 +27,10 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
 public class MissingModsException extends EnhancedRuntimeException implements IDisplayableError
 {
     private static final long serialVersionUID = 1L;
