--- ../src-base/minecraft/net/minecraftforge/fml/common/MultipleModsErrored.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/MultipleModsErrored.java
@@ -19,40 +19,35 @@
 
 package net.minecraftforge.fml.common;
 
-import java.util.List;
-
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraftforge.fml.client.GuiMultipleModsErrored;
 import net.minecraftforge.fml.client.IDisplayableError;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class MultipleModsErrored extends EnhancedRuntimeException implements IDisplayableError
-{
+import java.util.List;
+
+public class MultipleModsErrored extends EnhancedRuntimeException implements IDisplayableError {
     public final List<WrongMinecraftVersionException> wrongMinecraftExceptions;
     public final List<MissingModsException> missingModsExceptions;
-    public MultipleModsErrored(List<WrongMinecraftVersionException> wrongMinecraftExceptions, List<MissingModsException> missingModsExceptions)
-    {
+
+    public MultipleModsErrored(List<WrongMinecraftVersionException> wrongMinecraftExceptions, List<MissingModsException> missingModsExceptions) {
         this.wrongMinecraftExceptions = wrongMinecraftExceptions;
         this.missingModsExceptions = missingModsExceptions;
     }
 
     @Override
     @SideOnly(Side.CLIENT)
-    public GuiScreen createGui()
-    {
+    public GuiScreen createGui() {
         return new GuiMultipleModsErrored(this);
     }
 
     @Override
-    protected void printStackTrace(WrappedPrintStream stream)
-    {
-        for (WrongMinecraftVersionException wrongMinecraftVersionException : this.wrongMinecraftExceptions)
-        {
+    protected void printStackTrace(WrappedPrintStream stream) {
+        for (WrongMinecraftVersionException wrongMinecraftVersionException : this.wrongMinecraftExceptions) {
             wrongMinecraftVersionException.printStackTrace(stream);
         }
-        for (MissingModsException missingModsException : this.missingModsExceptions)
-        {
+        for (MissingModsException missingModsException : this.missingModsExceptions) {
             missingModsException.printStackTrace(stream);
         }
     }
