--- ../src-base/minecraft/net/minecraftforge/fml/common/Optional.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/Optional.java
@@ -19,11 +19,7 @@
 
 package net.minecraftforge.fml.common;
 
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Repeatable;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
+import java.lang.annotation.*;
 
 /**
  * Classes annotated with this will have the named interface or method removed from the runtime definition of the class
@@ -36,7 +32,8 @@
     /**
      * Not constructable
      */
-    private Optional() {}
+    private Optional() {
+    }
 
     /**
      * Mark a list of interfaces as removable
@@ -52,6 +49,7 @@
          */
         Interface[] value();
     }
+
     /**
      * Used to remove optional interfaces
      * @author cpw
@@ -80,6 +78,7 @@
          */
         boolean striprefs() default false;
     }
+
     /**
      * Used to remove optional methods
      * @author cpw
