--- ../src-base/minecraft/net/minecraftforge/fml/common/ProgressManager.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/ProgressManager.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,13 @@
 
 package net.minecraftforge.fml.common;
 
+import cn.pfcraft.server.Mohist;
+import com.google.common.base.Joiner;
+
 import java.util.Iterator;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 
-import com.google.common.base.Joiner;
-
 /**
  * Not a fully fleshed out API, may change in future MC versions.
  * However feel free to use and suggest additions.
@@ -73,13 +74,13 @@
             if (bar.timeEachStep)
             {
                 String timeString = String.format("%.3f", ((float) (newTime - bar.lastTime) / 1000000 / 1000));
-                FMLLog.log.debug("Bar Step: {} - {} took {}s", bar.getTitle(), bar.getMessage(), timeString);
+                Mohist.LOGGER.debug("Bar Step: {} - {} took {}s", bar.getTitle(), bar.getMessage(), timeString);
             }
             String timeString = String.format("%.3f", ((float) (newTime - bar.startTime) / 1000000 / 1000));
             if (bar.getSteps() == 1)
-                FMLLog.log.debug("Bar Finished: {} - {} took {}s", bar.getTitle(), bar.getMessage(), timeString);
+                Mohist.LOGGER.debug("Bar Finished: {} - {} took {}s", bar.getTitle(), bar.getMessage(), timeString);
             else
-                FMLLog.log.debug("Bar Finished: {} took {}s", bar.getTitle(), timeString);
+                Mohist.LOGGER.debug("Bar Finished: {} took {}s", bar.getTitle(), timeString);
         }
         FMLCommonHandler.instance().processWindowMessages();
     }
@@ -124,7 +125,7 @@
             if (timeEachStep && step != 0)
             {
                 long newTime = System.nanoTime();
-                FMLLog.log.debug(String.format("Bar Step: %s - %s took %.3fs", getTitle(), getMessage(), ((float)(newTime - lastTime) / 1000000 / 1000)));
+                Mohist.LOGGER.debug(String.format("Bar Step: %s - %s took %.3fs", getTitle(), getMessage(), ((float)(newTime - lastTime) / 1000000 / 1000)));
                 lastTime = newTime;
             }
             step++;
