--- ../src-base/minecraft/net/minecraftforge/fml/common/StartupQuery.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/StartupQuery.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,12 @@
 
 package net.minecraftforge.fml.common;
 
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.atomic.AtomicBoolean;
-
+import cn.pfcraft.server.Mohist;
 import net.minecraft.server.MinecraftServer;
 
 import javax.annotation.Nullable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class StartupQuery {
     // internal class/functionality, do not use
@@ -68,7 +68,7 @@
             }
             catch (InterruptedException e)
             {
-                FMLLog.log.warn("query interrupted");
+                Mohist.LOGGER.warn("query interrupted");
                 abort();
             }
 
@@ -120,7 +120,7 @@
 
         if (result != null && prop != null)
         {
-            FMLLog.log.info("Using fml.queryResult {} to answer the following query:\n{}", prop, text);
+            Mohist.LOGGER.info("Using fml.queryResult {} to answer the following query:\n{}", prop, text);
 
             if (prop.equalsIgnoreCase("confirm"))
             {
@@ -133,7 +133,7 @@
                 return;
             }
 
-            FMLLog.log.warn("Invalid value for fml.queryResult: {}, expected confirm or cancel", prop);
+            Mohist.LOGGER.warn("Invalid value for fml.queryResult: {}, expected confirm or cancel", prop);
         }
 
         synchronous = false;
@@ -156,7 +156,7 @@
         }
         catch (InterruptedException e)
         {
-            FMLLog.log.warn("query interrupted");
+            Mohist.LOGGER.warn("query interrupted");
             abort();
         }
     }
