--- ../src-base/minecraft/net/minecraftforge/fml/common/TracingPrintStream.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/TracingPrintStream.java
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.fml.common;
 
-import java.io.PrintStream;
-
 import org.apache.logging.log4j.Logger;
 
+import java.io.PrintStream;
+
 /**
  * PrintStream which redirects it's output to a given logger.
  *
