--- ../src-base/minecraft/net/minecraftforge/fml/common/WorldAccessContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/WorldAccessContainer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,12 +19,14 @@
 
 package net.minecraftforge.fml.common;
 
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.storage.SaveHandler;
+import net.minecraft.world.storage.WorldInfo;
+
 import java.util.Map;
 
-import net.minecraft.nbt.*;
-import net.minecraft.world.storage.*;
 
-
 public interface WorldAccessContainer
 {
     NBTTagCompound getDataForWriting(SaveHandler handler, WorldInfo info);
