--- ../src-base/minecraft/net/minecraftforge/fml/common/WrongMinecraftVersionException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/WrongMinecraftVersionException.java
@@ -25,14 +25,12 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class WrongMinecraftVersionException extends EnhancedRuntimeException implements IDisplayableError
-{
+public class WrongMinecraftVersionException extends EnhancedRuntimeException implements IDisplayableError {
     private static final long serialVersionUID = 1L;
     public ModContainer mod;
     private String mcVersion;
 
-    public WrongMinecraftVersionException(ModContainer mod, String mcver)
-    {
+    public WrongMinecraftVersionException(ModContainer mod, String mcver) {
         super(String.format("Wrong Minecraft version for %s", mod.getModId()));
         this.mod = mod;
         this.mcVersion = mcver;
@@ -50,8 +48,7 @@
 
     @Override
     @SideOnly(Side.CLIENT)
-    public GuiScreen createGui()
-    {
+    public GuiScreen createGui() {
         return new GuiWrongMinecraft(this);
     }
 }
