--- ../src-base/minecraft/net/minecraft/client/renderer/culling/ClippingHelperImpl.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/culling/ClippingHelperImpl.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.renderer.culling;
 
-import java.nio.FloatBuffer;
 import net.minecraft.client.renderer.GLAllocation;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.FloatBuffer;
+
 @SideOnly(Side.CLIENT)
 public class ClippingHelperImpl extends ClippingHelper
 {
