--- ../src-base/minecraft/net/minecraft/client/renderer/culling/ClippingHelperImpl.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/culling/ClippingHelperImpl.java
@@ -1,28 +1,26 @@
 package net.minecraft.client.renderer.culling;
 
-import java.nio.FloatBuffer;
 import net.minecraft.client.renderer.GLAllocation;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.nio.FloatBuffer;
+
 @SideOnly(Side.CLIENT)
-public class ClippingHelperImpl extends ClippingHelper
-{
+public class ClippingHelperImpl extends ClippingHelper {
     private static final ClippingHelperImpl instance = new ClippingHelperImpl();
     private final FloatBuffer projectionMatrixBuffer = GLAllocation.createDirectFloatBuffer(16);
     private final FloatBuffer modelviewMatrixBuffer = GLAllocation.createDirectFloatBuffer(16);
     private final FloatBuffer floatBuffer16 = GLAllocation.createDirectFloatBuffer(16);
 
-    public static ClippingHelper getInstance()
-    {
+    public static ClippingHelper getInstance() {
         instance.init();
         return instance;
     }
 
-    private void normalize(float[] p_180547_1_)
-    {
+    private void normalize(float[] p_180547_1_) {
         float f = MathHelper.sqrt(p_180547_1_[0] * p_180547_1_[0] + p_180547_1_[1] * p_180547_1_[1] + p_180547_1_[2] * p_180547_1_[2]);
         p_180547_1_[0] /= f;
         p_180547_1_[1] /= f;
@@ -30,8 +28,7 @@
         p_180547_1_[3] /= f;
     }
 
-    public void init()
-    {
+    public void init() {
         this.projectionMatrixBuffer.clear();
         this.modelviewMatrixBuffer.clear();
         this.floatBuffer16.clear();
