--- ../src-base/minecraft/net/minecraft/client/renderer/culling/ICamera.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/culling/ICamera.java
@@ -5,8 +5,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ICamera
-{
+public interface ICamera {
     boolean isBoundingBoxInFrustum(AxisAlignedBB p_78546_1_);
 
     void setPosition(double xIn, double yIn, double zIn);
