--- ../src-base/minecraft/net/minecraft/client/settings/KeyBinding.java
+++ ../src-work/minecraft/net/minecraft/client/settings/KeyBinding.java
@@ -2,15 +2,15 @@
 
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
-import java.util.Map;
-import java.util.Set;
-import java.util.function.Supplier;
 import net.minecraft.client.resources.I18n;
-import net.minecraft.util.IntHashMap;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.lwjgl.input.Keyboard;
 
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
 @SideOnly(Side.CLIENT)
 public class KeyBinding implements Comparable<KeyBinding>
 {
