--- ../src-base/minecraft/net/minecraft/client/renderer/entity/layers/LayerWolfCollar.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/layers/LayerWolfCollar.java
@@ -8,20 +8,16 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class LayerWolfCollar implements LayerRenderer<EntityWolf>
-{
+public class LayerWolfCollar implements LayerRenderer<EntityWolf> {
     private static final ResourceLocation WOLF_COLLAR = new ResourceLocation("textures/entity/wolf/wolf_collar.png");
     private final RenderWolf wolfRenderer;
 
-    public LayerWolfCollar(RenderWolf wolfRendererIn)
-    {
+    public LayerWolfCollar(RenderWolf wolfRendererIn) {
         this.wolfRenderer = wolfRendererIn;
     }
 
-    public void doRenderLayer(EntityWolf entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
-        if (entitylivingbaseIn.isTamed() && !entitylivingbaseIn.isInvisible())
-        {
+    public void doRenderLayer(EntityWolf entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
+        if (entitylivingbaseIn.isTamed() && !entitylivingbaseIn.isInvisible()) {
             this.wolfRenderer.bindTexture(WOLF_COLLAR);
             float[] afloat = entitylivingbaseIn.getCollarColor().getColorComponentValues();
             GlStateManager.color(afloat[0], afloat[1], afloat[2]);
@@ -29,8 +25,7 @@
         }
     }
 
-    public boolean shouldCombineTextures()
-    {
+    public boolean shouldCombineTextures() {
         return true;
     }
 }
