--- ../src-base/minecraft/net/minecraft/client/renderer/entity/layers/LayerEnderDragonDeath.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/layers/LayerEnderDragonDeath.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer.entity.layers;
 
-import java.util.Random;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.RenderHelper;
@@ -10,6 +9,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class LayerEnderDragonDeath implements LayerRenderer<EntityDragon>
 {
