--- ../src-base/minecraft/net/minecraft/client/renderer/entity/layers/LayerArrow.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/layers/LayerArrow.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer.entity.layers;
 
-import java.util.Random;
 import net.minecraft.client.model.ModelBox;
 import net.minecraft.client.model.ModelRenderer;
 import net.minecraft.client.renderer.GlStateManager;
@@ -13,6 +12,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class LayerArrow implements LayerRenderer<EntityLivingBase>
 {
