--- ../src-base/minecraft/net/minecraft/client/renderer/entity/layers/LayerRenderer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/layers/LayerRenderer.java
@@ -5,8 +5,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface LayerRenderer<E extends EntityLivingBase>
-{
+public interface LayerRenderer<E extends EntityLivingBase> {
     void doRenderLayer(E entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale);
 
     boolean shouldCombineTextures();
