--- ../src-base/minecraft/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.renderer.entity.layers;
 
 import com.mojang.authlib.GameProfile;
-import java.util.UUID;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.model.ModelRenderer;
 import net.minecraft.client.renderer.GlStateManager;
@@ -23,6 +22,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.lang3.StringUtils;
 
+import java.util.UUID;
+
 @SideOnly(Side.CLIENT)
 public class LayerCustomHead implements LayerRenderer<EntityLivingBase>
 {
