--- ../src-base/minecraft/net/minecraft/block/state/BlockWorldState.java
+++ ../src-work/minecraft/net/minecraft/block/state/BlockWorldState.java
@@ -1,11 +1,12 @@
 package net.minecraft.block.state;
 
 import com.google.common.base.Predicate;
-import javax.annotation.Nullable;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+
 public class BlockWorldState
 {
     private final World world;
