--- ../src-base/minecraft/net/minecraft/block/state/IBlockState.java
+++ ../src-work/minecraft/net/minecraft/block/state/IBlockState.java
@@ -1,10 +1,11 @@
 package net.minecraft.block.state;
 
 import com.google.common.collect.ImmutableMap;
-import java.util.Collection;
 import net.minecraft.block.Block;
 import net.minecraft.block.properties.IProperty;
 
+import java.util.Collection;
+
 public interface IBlockState extends IBlockBehaviors, IBlockProperties
 {
     Collection < IProperty<? >> getPropertyKeys();
