--- ../src-base/minecraft/net/minecraft/block/state/IBlockBehaviors.java
+++ ../src-work/minecraft/net/minecraft/block/state/IBlockBehaviors.java
@@ -4,8 +4,7 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
-public interface IBlockBehaviors
-{
+public interface IBlockBehaviors {
     boolean onBlockEventReceived(World worldIn, BlockPos pos, int id, int param);
 
     void neighborChanged(World worldIn, BlockPos pos, Block blockIn, BlockPos fromPos);
