--- ../src-base/minecraft/net/minecraft/entity/item/EntityFireworkRocket.java
+++ ../src-work/minecraft/net/minecraft/entity/item/EntityFireworkRocket.java
@@ -1,6 +1,7 @@
 package net.minecraft.entity.item;
 
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.MoverType;
 import net.minecraft.entity.player.EntityPlayer;
@@ -23,14 +24,19 @@
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import org.bukkit.craftbukkit.event.CraftEventFactory;
 
+import java.util.UUID;
+
 public class EntityFireworkRocket extends Entity
 {
-    private static final DataParameter<ItemStack> FIREWORK_ITEM = EntityDataManager.<ItemStack>createKey(EntityFireworkRocket.class, DataSerializers.ITEM_STACK);
+    public static final DataParameter<ItemStack> FIREWORK_ITEM = EntityDataManager.<ItemStack>createKey(EntityFireworkRocket.class, DataSerializers.ITEM_STACK);
     private static final DataParameter<Integer> BOOSTED_ENTITY_ID = EntityDataManager.<Integer>createKey(EntityFireworkRocket.class, DataSerializers.VARINT);
     private int fireworkAge;
-    private int lifetime;
+    public int lifetime;
+    public UUID spawningEntity; // Paper
     private EntityLivingBase boostedEntity;
+    public EntityLivingBase getBoostedEntity() { return boostedEntity; } // Paper - OBFHELPER
 
     public EntityFireworkRocket(World worldIn)
     {
@@ -38,6 +44,14 @@
         this.setSize(0.25F, 0.25F);
     }
 
+    // Spigot Start
+    @Override
+    public void inactiveTick() {
+        this.fireworkAge += 1;
+        super.inactiveTick();
+    }
+    // Spigot End
+
     protected void entityInit()
     {
         this.dataManager.register(FIREWORK_ITEM, ItemStack.EMPTY);
@@ -187,8 +201,10 @@
 
         if (!this.world.isRemote && this.fireworkAge > this.lifetime)
         {
-            this.world.setEntityState(this, (byte)17);
-            this.dealExplosionDamage();
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callFireworkExplodeEvent(this).isCancelled()) {
+                this.world.setEntityState(this, (byte) 17);
+                this.dealExplosionDamage();
+            }
             this.setDead();
         }
     }
@@ -209,7 +225,9 @@
         {
             if (this.boostedEntity != null)
             {
+                CraftEventFactory.entityDamage = this;
                 this.boostedEntity.attackEntityFrom(DamageSource.FIREWORKS, (float)(5 + nbttaglist.tagCount() * 2));
+                CraftEventFactory.entityDamage = null;
             }
 
             double d0 = 5.0D;
@@ -235,7 +253,9 @@
                     if (flag)
                     {
                         float f1 = f * (float)Math.sqrt((5.0D - (double)this.getDistance(entitylivingbase)) / 5.0D);
+                        CraftEventFactory.entityDamage = this;
                         entitylivingbase.attackEntityFrom(DamageSource.FIREWORKS, f1);
+                        CraftEventFactory.entityDamage = null;
                     }
                 }
             }
@@ -275,6 +295,12 @@
         {
             compound.setTag("FireworksItem", itemstack.writeToNBT(new NBTTagCompound()));
         }
+
+        // Paper start
+        if (spawningEntity != null) {
+            compound.setUUID("SpawningEntity", spawningEntity);
+        }
+        // Paper end
     }
 
     public void readEntityFromNBT(NBTTagCompound compound)
@@ -292,6 +318,11 @@
                 this.dataManager.set(FIREWORK_ITEM, itemstack);
             }
         }
+        // Paper start
+        if (nbttagcompound.hasUniqueId("SpawningEntity")) {
+            spawningEntity = nbttagcompound.getUUID("SpawningEntity");
+        }
+        // Paper end
     }
 
     public boolean canBeAttackedWithItem()
