--- ../src-base/minecraft/net/minecraft/entity/item/EntityMinecartEmpty.java
+++ ../src-work/minecraft/net/minecraft/entity/item/EntityMinecartEmpty.java
@@ -5,39 +5,28 @@
 import net.minecraft.util.datafix.DataFixer;
 import net.minecraft.world.World;
 
-public class EntityMinecartEmpty extends EntityMinecart
-{
-    public EntityMinecartEmpty(World worldIn)
-    {
+public class EntityMinecartEmpty extends EntityMinecart {
+    public EntityMinecartEmpty(World worldIn) {
         super(worldIn);
     }
 
-    public EntityMinecartEmpty(World worldIn, double x, double y, double z)
-    {
+    public EntityMinecartEmpty(World worldIn, double x, double y, double z) {
         super(worldIn, x, y, z);
     }
 
-    public static void registerFixesMinecartEmpty(DataFixer fixer)
-    {
+    public static void registerFixesMinecartEmpty(DataFixer fixer) {
         EntityMinecart.registerFixesMinecart(fixer, EntityMinecartEmpty.class);
     }
 
-    public boolean processInitialInteract(EntityPlayer player, EnumHand hand)
-    {
+    public boolean processInitialInteract(EntityPlayer player, EnumHand hand) {
         if (super.processInitialInteract(player, hand)) return true;
 
-        if (player.isSneaking())
-        {
+        if (player.isSneaking()) {
             return false;
-        }
-        else if (this.isBeingRidden())
-        {
+        } else if (this.isBeingRidden()) {
             return true;
-        }
-        else
-        {
-            if (!this.world.isRemote)
-            {
+        } else {
+            if (!this.world.isRemote) {
                 player.startRiding(this);
             }
 
@@ -45,17 +34,13 @@
         }
     }
 
-    public void onActivatorRailPass(int x, int y, int z, boolean receivingPower)
-    {
-        if (receivingPower)
-        {
-            if (this.isBeingRidden())
-            {
+    public void onActivatorRailPass(int x, int y, int z, boolean receivingPower) {
+        if (receivingPower) {
+            if (this.isBeingRidden()) {
                 this.removePassengers();
             }
 
-            if (this.getRollingAmplitude() == 0)
-            {
+            if (this.getRollingAmplitude() == 0) {
                 this.setRollingDirection(-this.getRollingDirection());
                 this.setRollingAmplitude(10);
                 this.setDamage(50.0F);
@@ -64,8 +49,7 @@
         }
     }
 
-    public EntityMinecart.Type getType()
-    {
+    public EntityMinecart.Type getType() {
         return EntityMinecart.Type.RIDEABLE;
     }
 }
