--- ../src-base/minecraft/net/minecraft/entity/item/EntityMinecartFurnace.java
+++ ../src-work/minecraft/net/minecraft/entity/item/EntityMinecartFurnace.java
@@ -41,9 +41,9 @@
         EntityMinecart.registerFixesMinecart(fixer, EntityMinecartFurnace.class);
     }
 
-    public EntityMinecart.Type getType()
+    public Type getType()
     {
-        return EntityMinecart.Type.FURNACE;
+        return Type.FURNACE;
     }
 
     protected void entityInit()
