--- ../src-base/minecraft/net/minecraft/entity/item/EntityMinecartChest.java
+++ ../src-work/minecraft/net/minecraft/entity/item/EntityMinecartChest.java
@@ -13,60 +13,48 @@
 import net.minecraft.util.datafix.DataFixer;
 import net.minecraft.world.World;
 
-public class EntityMinecartChest extends EntityMinecartContainer
-{
-    public EntityMinecartChest(World worldIn)
-    {
+public class EntityMinecartChest extends EntityMinecartContainer {
+    public EntityMinecartChest(World worldIn) {
         super(worldIn);
     }
 
-    public EntityMinecartChest(World worldIn, double x, double y, double z)
-    {
+    public EntityMinecartChest(World worldIn, double x, double y, double z) {
         super(worldIn, x, y, z);
     }
 
-    public static void registerFixesMinecartChest(DataFixer fixer)
-    {
+    public static void registerFixesMinecartChest(DataFixer fixer) {
         EntityMinecartContainer.addDataFixers(fixer, EntityMinecartChest.class);
     }
 
-    public void killMinecart(DamageSource source)
-    {
+    public void killMinecart(DamageSource source) {
         super.killMinecart(source);
 
-        if (this.world.getGameRules().getBoolean("doEntityDrops"))
-        {
+        if (this.world.getGameRules().getBoolean("doEntityDrops")) {
             this.dropItemWithOffset(Item.getItemFromBlock(Blocks.CHEST), 1, 0.0F);
         }
     }
 
-    public int getSizeInventory()
-    {
+    public int getSizeInventory() {
         return 27;
     }
 
-    public EntityMinecart.Type getType()
-    {
+    public EntityMinecart.Type getType() {
         return EntityMinecart.Type.CHEST;
     }
 
-    public IBlockState getDefaultDisplayTile()
-    {
+    public IBlockState getDefaultDisplayTile() {
         return Blocks.CHEST.getDefaultState().withProperty(BlockChest.FACING, EnumFacing.NORTH);
     }
 
-    public int getDefaultDisplayTileOffset()
-    {
+    public int getDefaultDisplayTileOffset() {
         return 8;
     }
 
-    public String getGuiID()
-    {
+    public String getGuiID() {
         return "minecraft:chest";
     }
 
-    public Container createContainer(InventoryPlayer playerInventory, EntityPlayer playerIn)
-    {
+    public Container createContainer(InventoryPlayer playerInventory, EntityPlayer playerIn) {
         this.addLoot(playerIn);
         return new ContainerChest(playerInventory, this, playerIn);
     }
