--- ../src-base/minecraft/net/minecraft/advancements/critereon/AbstractCriterionInstance.java
+++ ../src-work/minecraft/net/minecraft/advancements/critereon/AbstractCriterionInstance.java
@@ -3,22 +3,18 @@
 import net.minecraft.advancements.ICriterionInstance;
 import net.minecraft.util.ResourceLocation;
 
-public class AbstractCriterionInstance implements ICriterionInstance
-{
+public class AbstractCriterionInstance implements ICriterionInstance {
     private final ResourceLocation criterion;
 
-    public AbstractCriterionInstance(ResourceLocation criterionIn)
-    {
+    public AbstractCriterionInstance(ResourceLocation criterionIn) {
         this.criterion = criterionIn;
     }
 
-    public ResourceLocation getId()
-    {
+    public ResourceLocation getId() {
         return this.criterion;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return "AbstractCriterionInstance{criterion=" + this.criterion + '}';
     }
 }
