--- ../src-base/minecraft/net/minecraft/advancements/critereon/DamagePredicate.java
+++ ../src-work/minecraft/net/minecraft/advancements/critereon/DamagePredicate.java
@@ -2,11 +2,12 @@
 
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.JsonUtils;
 
+import javax.annotation.Nullable;
+
 public class DamagePredicate
 {
     public static DamagePredicate ANY = new DamagePredicate();
