--- ../src-base/minecraft/net/minecraft/advancements/critereon/DistancePredicate.java
+++ ../src-work/minecraft/net/minecraft/advancements/critereon/DistancePredicate.java
@@ -2,10 +2,11 @@
 
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
-import javax.annotation.Nullable;
 import net.minecraft.util.JsonUtils;
 import net.minecraft.util.math.MathHelper;
 
+import javax.annotation.Nullable;
+
 public class DistancePredicate
 {
     public static final DistancePredicate ANY = new DistancePredicate(MinMaxBounds.UNBOUNDED, MinMaxBounds.UNBOUNDED, MinMaxBounds.UNBOUNDED, MinMaxBounds.UNBOUNDED, MinMaxBounds.UNBOUNDED);
