--- ../src-base/minecraft/net/minecraft/world/gen/ChunkGeneratorFlat.java
+++ ../src-work/minecraft/net/minecraft/world/gen/ChunkGeneratorFlat.java
@@ -1,10 +1,5 @@
 package net.minecraft.world.gen;
 
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Random;
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.EnumCreatureType;
 import net.minecraft.init.Biomes;
@@ -17,13 +12,14 @@
 import net.minecraft.world.chunk.ChunkPrimer;
 import net.minecraft.world.gen.feature.WorldGenDungeons;
 import net.minecraft.world.gen.feature.WorldGenLakes;
-import net.minecraft.world.gen.structure.MapGenMineshaft;
-import net.minecraft.world.gen.structure.MapGenScatteredFeature;
-import net.minecraft.world.gen.structure.MapGenStronghold;
-import net.minecraft.world.gen.structure.MapGenStructure;
-import net.minecraft.world.gen.structure.MapGenVillage;
-import net.minecraft.world.gen.structure.StructureOceanMonument;
+import net.minecraft.world.gen.structure.*;
 
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
 public class ChunkGeneratorFlat implements IChunkGenerator
 {
     private final World world;
@@ -46,7 +42,7 @@
         {
             Map<String, Map<String, String>> map = this.flatWorldGenInfo.getWorldFeatures();
 
-            if (map.containsKey("village"))
+            if (map.containsKey("village") && world.paperConfig.generateVillage)
             {
                 Map<String, String> map1 = (Map)map.get("village");
 
@@ -58,22 +54,22 @@
                 this.structureGenerators.put("Village", new MapGenVillage(map1));
             }
 
-            if (map.containsKey("biome_1"))
+            if (map.containsKey("biome_1") && world.paperConfig.generateTemple)
             {
                 this.structureGenerators.put("Temple", new MapGenScatteredFeature(map.get("biome_1")));
             }
 
-            if (map.containsKey("mineshaft"))
+            if (map.containsKey("mineshaft") && world.paperConfig.generateMineshaft)
             {
                 this.structureGenerators.put("Mineshaft", new MapGenMineshaft(map.get("mineshaft")));
             }
 
-            if (map.containsKey("stronghold"))
+            if (map.containsKey("stronghold") && world.paperConfig.generateStronghold)
             {
                 this.structureGenerators.put("Stronghold", new MapGenStronghold(map.get("stronghold")));
             }
 
-            if (map.containsKey("oceanmonument"))
+            if (map.containsKey("oceanmonument") && world.paperConfig.generateMonument)
             {
                 this.structureGenerators.put("Monument", new StructureOceanMonument(map.get("oceanmonument")));
             }
@@ -89,7 +85,7 @@
             this.lavaLakeGenerator = new WorldGenLakes(Blocks.LAVA);
         }
 
-        this.hasDungeons = this.flatWorldGenInfo.getWorldFeatures().containsKey("dungeon");
+        this.hasDungeons = world.paperConfig.generateDungeon && this.flatWorldGenInfo.getWorldFeatures().containsKey("dungeon");  // Paper
         int j = 0;
         int k = 0;
         boolean flag = true;
