--- ../src-base/minecraft/net/minecraft/world/gen/IChunkGenerator.java
+++ ../src-work/minecraft/net/minecraft/world/gen/IChunkGenerator.java
@@ -1,13 +1,14 @@
 package net.minecraft.world.gen;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.entity.EnumCreatureType;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.chunk.Chunk;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public interface IChunkGenerator
 {
     Chunk generateChunk(int x, int z);
