--- ../src-base/minecraft/net/minecraft/client/renderer/BufferBuilder.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/BufferBuilder.java
@@ -1,14 +1,6 @@
 package net.minecraft.client.renderer;
 
 import com.google.common.primitives.Floats;
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
-import java.nio.FloatBuffer;
-import java.nio.IntBuffer;
-import java.nio.ShortBuffer;
-import java.util.Arrays;
-import java.util.BitSet;
-import java.util.Comparator;
 import net.minecraft.client.renderer.vertex.VertexFormat;
 import net.minecraft.client.renderer.vertex.VertexFormatElement;
 import net.minecraft.util.math.MathHelper;
@@ -17,6 +9,11 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.nio.*;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.Comparator;
+
 @SideOnly(Side.CLIENT)
 public class BufferBuilder
 {
@@ -127,7 +124,7 @@
         this.rawIntBuffer.position(this.getBufferSize());
     }
 
-    public BufferBuilder.State getVertexState()
+    public State getVertexState()
     {
         this.rawIntBuffer.rewind();
         int i = this.getBufferSize();
@@ -136,7 +133,7 @@
         this.rawIntBuffer.get(aint);
         this.rawIntBuffer.limit(this.rawIntBuffer.capacity());
         this.rawIntBuffer.position(i);
-        return new BufferBuilder.State(aint, new VertexFormat(this.vertexFormat));
+        return new State(aint, new VertexFormat(this.vertexFormat));
     }
 
     private int getBufferSize()
@@ -164,7 +161,7 @@
         return f12 * f12 + f13 * f13 + f14 * f14;
     }
 
-    public void setVertexState(BufferBuilder.State state)
+    public void setVertexState(State state)
     {
         this.rawIntBuffer.clear();
         this.growBuffer(state.getRawBuffer().length * 4);
