--- ../src-base/minecraft/net/minecraft/client/renderer/GLAllocation.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/GLAllocation.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.renderer;
 
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.lwjgl.util.glu.GLU;
+
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.FloatBuffer;
 import java.nio.IntBuffer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
-import org.lwjgl.util.glu.GLU;
 
 @SideOnly(Side.CLIENT)
 public class GLAllocation
