--- ../src-base/minecraft/net/minecraft/client/renderer/IImageBuffer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/IImageBuffer.java
@@ -1,12 +1,12 @@
 package net.minecraft.client.renderer;
 
-import java.awt.image.BufferedImage;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.awt.image.BufferedImage;
+
 @SideOnly(Side.CLIENT)
-public interface IImageBuffer
-{
+public interface IImageBuffer {
     BufferedImage parseUserSkin(BufferedImage image);
 
     void skinAvailable();
