--- ../src-base/minecraft/net/minecraft/client/renderer/Tessellator.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/Tessellator.java
@@ -4,30 +4,25 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class Tessellator
-{
+public class Tessellator {
+    private static final Tessellator INSTANCE = new Tessellator(2097152);
     private final BufferBuilder buffer;
     private final WorldVertexBufferUploader vboUploader = new WorldVertexBufferUploader();
-    private static final Tessellator INSTANCE = new Tessellator(2097152);
 
-    public static Tessellator getInstance()
-    {
-        return INSTANCE;
+    public Tessellator(int bufferSize) {
+        this.buffer = new BufferBuilder(bufferSize);
     }
 
-    public Tessellator(int bufferSize)
-    {
-        this.buffer = new BufferBuilder(bufferSize);
+    public static Tessellator getInstance() {
+        return INSTANCE;
     }
 
-    public void draw()
-    {
+    public void draw() {
         this.buffer.finishDrawing();
         this.vboUploader.draw(this.buffer);
     }
 
-    public BufferBuilder getBuffer()
-    {
+    public BufferBuilder getBuffer() {
         return this.buffer;
     }
 }
