--- ../src-base/minecraft/net/minecraft/client/renderer/ViewFrustum.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/ViewFrustum.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer;
 
-import javax.annotation.Nullable;
 import net.minecraft.client.renderer.chunk.IRenderChunkFactory;
 import net.minecraft.client.renderer.chunk.RenderChunk;
 import net.minecraft.util.math.BlockPos;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class ViewFrustum
 {
