--- ../src-base/minecraft/net/minecraft/world/EnumSkyBlock.java
+++ ../src-work/minecraft/net/minecraft/world/EnumSkyBlock.java
@@ -1,14 +1,12 @@
 package net.minecraft.world;
 
-public enum EnumSkyBlock
-{
+public enum EnumSkyBlock {
     SKY(15),
     BLOCK(0);
 
     public final int defaultLightValue;
 
-    private EnumSkyBlock(int defaultLightValueIn)
-    {
+    private EnumSkyBlock(int defaultLightValueIn) {
         this.defaultLightValue = defaultLightValueIn;
     }
 }
