--- ../src-base/minecraft/net/minecraft/world/GameRules.java
+++ ../src-work/minecraft/net/minecraft/world/GameRules.java
@@ -10,7 +10,6 @@
 
     public GameRules()
     {
-        this.addGameRule("doFireTick", "true", GameRules.ValueType.BOOLEAN_VALUE);
         this.addGameRule("mobGriefing", "true", GameRules.ValueType.BOOLEAN_VALUE);
         this.addGameRule("keepInventory", "false", GameRules.ValueType.BOOLEAN_VALUE);
         this.addGameRule("doMobSpawning", "true", GameRules.ValueType.BOOLEAN_VALUE);
