--- ../src-base/minecraft/net/minecraft/world/WorldProviderSurface.java
+++ ../src-work/minecraft/net/minecraft/world/WorldProviderSurface.java
@@ -1,14 +1,11 @@
 package net.minecraft.world;
 
-public class WorldProviderSurface extends WorldProvider
-{
-    public DimensionType getDimensionType()
-    {
+public class WorldProviderSurface extends WorldProvider {
+    public DimensionType getDimensionType() {
         return DimensionType.OVERWORLD;
     }
 
-    public boolean canDropChunk(int x, int z)
-    {
+    public boolean canDropChunk(int x, int z) {
         return !this.world.isSpawnChunk(x, z) || !this.world.provider.getDimensionType().shouldLoadSpawn();
     }
 }
