--- ../src-base/minecraft/net/minecraft/world/WorldProviderEnd.java
+++ ../src-work/minecraft/net/minecraft/world/WorldProviderEnd.java
@@ -1,6 +1,5 @@
 package net.minecraft.world;
 
-import javax.annotation.Nullable;
 import net.minecraft.init.Biomes;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.math.BlockPos;
@@ -13,6 +12,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public class WorldProviderEnd extends WorldProvider
 {
     private DragonFightManager dragonFightManager;
