--- ../src-base/minecraft/net/minecraft/client/gui/spectator/categories/SpectatorDetails.java
+++ ../src-work/minecraft/net/minecraft/client/gui/spectator/categories/SpectatorDetails.java
@@ -1,13 +1,14 @@
 package net.minecraft.client.gui.spectator.categories;
 
 import com.google.common.base.MoreObjects;
-import java.util.List;
 import net.minecraft.client.gui.spectator.ISpectatorMenuObject;
 import net.minecraft.client.gui.spectator.ISpectatorMenuView;
 import net.minecraft.client.gui.spectator.SpectatorMenu;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class SpectatorDetails
 {
