--- ../src-base/minecraft/net/minecraft/entity/boss/dragon/phase/IPhase.java
+++ ../src-work/minecraft/net/minecraft/entity/boss/dragon/phase/IPhase.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.boss.dragon.phase;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.MultiPartEntityPart;
 import net.minecraft.entity.item.EntityEnderCrystal;
 import net.minecraft.entity.player.EntityPlayer;
@@ -8,6 +7,8 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Vec3d;
 
+import javax.annotation.Nullable;
+
 public interface IPhase
 {
     boolean getIsStationary();
