--- ../src-base/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseChargingPlayer.java
+++ ../src-work/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseChargingPlayer.java
@@ -1,11 +1,12 @@
 package net.minecraft.entity.boss.dragon.phase;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.boss.EntityDragon;
 import net.minecraft.util.math.Vec3d;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+
 public class PhaseChargingPlayer extends PhaseBase
 {
     private static final Logger LOGGER = LogManager.getLogger();
