--- ../src-base/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseSittingBase.java
+++ ../src-work/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseSittingBase.java
@@ -5,27 +5,20 @@
 import net.minecraft.entity.projectile.EntityArrow;
 import net.minecraft.util.DamageSource;
 
-public abstract class PhaseSittingBase extends PhaseBase
-{
-    public PhaseSittingBase(EntityDragon p_i46794_1_)
-    {
+public abstract class PhaseSittingBase extends PhaseBase {
+    public PhaseSittingBase(EntityDragon p_i46794_1_) {
         super(p_i46794_1_);
     }
 
-    public boolean getIsStationary()
-    {
+    public boolean getIsStationary() {
         return true;
     }
 
-    public float getAdjustedDamage(MultiPartEntityPart pt, DamageSource src, float damage)
-    {
-        if (src.getImmediateSource() instanceof EntityArrow)
-        {
+    public float getAdjustedDamage(MultiPartEntityPart pt, DamageSource src, float damage) {
+        if (src.getImmediateSource() instanceof EntityArrow) {
             src.getImmediateSource().setFire(1);
             return 0.0F;
-        }
-        else
-        {
+        } else {
             return super.getAdjustedDamage(pt, src, damage);
         }
     }
