--- ../src-base/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseTakeoff.java
+++ ../src-work/minecraft/net/minecraft/entity/boss/dragon/phase/PhaseTakeoff.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.boss.dragon.phase;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.boss.EntityDragon;
 import net.minecraft.pathfinding.Path;
 import net.minecraft.pathfinding.PathPoint;
@@ -8,6 +7,8 @@
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.gen.feature.WorldGenEndPodium;
 
+import javax.annotation.Nullable;
+
 public class PhaseTakeoff extends PhaseBase
 {
     private boolean firstTick;
