--- ../src-base/minecraft/net/minecraftforge/client/model/pipeline/IVertexConsumer.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/pipeline/IVertexConsumer.java
@@ -28,16 +28,19 @@
  * Also assumes that element index passed will increment from 0 to format.getElementCount() - 1.
  * Normal, Color and UV are assumed to be in 0-1 range.
  */
-public interface IVertexConsumer
-{
+public interface IVertexConsumer {
     /**
      * @return the format that should be used for passed data.
      */
     VertexFormat getVertexFormat();
 
     void setQuadTint(int tint);
+
     void setQuadOrientation(EnumFacing orientation);
+
     void setApplyDiffuseLighting(boolean diffuse);
+
     void setTexture(TextureAtlasSprite texture);
+
     void put(int element, float... data);
 }
