--- ../src-base/minecraft/net/minecraftforge/client/model/pipeline/TransformerConsumer.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/pipeline/TransformerConsumer.java
@@ -24,20 +24,17 @@
 public abstract class TransformerConsumer implements IVertexConsumer {
     private IVertexConsumer parent;
 
-    protected TransformerConsumer(IVertexConsumer parent)
-    {
+    protected TransformerConsumer(IVertexConsumer parent) {
         this.parent = parent;
     }
 
     @Override
-    public VertexFormat getVertexFormat()
-    {
+    public VertexFormat getVertexFormat() {
         return parent.getVertexFormat();
     }
 
     @Override
-    public void put(int element, float... data)
-    {
+    public void put(int element, float... data) {
         float[] newData = transform(element, data);
         parent.put(element, newData);
     }
