--- ../src-base/minecraft/net/minecraftforge/registries/IForgeRegistry.java
+++ ../src-work/minecraft/net/minecraftforge/registries/IForgeRegistry.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,16 +19,15 @@
 
 package net.minecraftforge.registries;
 
+import net.minecraft.util.ResourceLocation;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map.Entry;
 import java.util.Set;
 
-import net.minecraft.util.ResourceLocation;
-
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 /**
  * Main interface for the registry system. Use this to query the registry system.
  *
@@ -94,6 +93,7 @@
         void onCreate(IForgeRegistryInternal<V> owner, RegistryManager stage);
     }
 
+
     /**
      * Callback fired when the registry contents are validated.
      */
