--- ../src-base/minecraft/net/minecraft/network/datasync/DataParameter.java
+++ ../src-work/minecraft/net/minecraft/network/datasync/DataParameter.java
@@ -1,45 +1,34 @@
 package net.minecraft.network.datasync;
 
-public class DataParameter<T>
-{
+public class DataParameter<T> {
     private final int id;
     private final DataSerializer<T> serializer;
 
-    public DataParameter(int idIn, DataSerializer<T> serializerIn)
-    {
+    public DataParameter(int idIn, DataSerializer<T> serializerIn) {
         this.id = idIn;
         this.serializer = serializerIn;
     }
 
-    public int getId()
-    {
+    public int getId() {
         return this.id;
     }
 
-    public DataSerializer<T> getSerializer()
-    {
+    public DataSerializer<T> getSerializer() {
         return this.serializer;
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        if (this == p_equals_1_)
-        {
+    public boolean equals(Object p_equals_1_) {
+        if (this == p_equals_1_) {
             return true;
-        }
-        else if (p_equals_1_ != null && this.getClass() == p_equals_1_.getClass())
-        {
-            DataParameter<?> dataparameter = (DataParameter)p_equals_1_;
+        } else if (p_equals_1_ != null && this.getClass() == p_equals_1_.getClass()) {
+            DataParameter<?> dataparameter = (DataParameter) p_equals_1_;
             return this.id == dataparameter.id;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return this.id;
     }
 }
