--- ../src-base/minecraft/net/minecraft/network/datasync/DataSerializer.java
+++ ../src-work/minecraft/net/minecraft/network/datasync/DataSerializer.java
@@ -1,8 +1,9 @@
 package net.minecraft.network.datasync;
 
-import java.io.IOException;
 import net.minecraft.network.PacketBuffer;
 
+import java.io.IOException;
+
 public interface DataSerializer<T>
 {
     void write(PacketBuffer buf, T value);
