--- ../src-base/minecraft/net/minecraft/client/audio/ISoundEventListener.java
+++ ../src-work/minecraft/net/minecraft/client/audio/ISoundEventListener.java
@@ -4,7 +4,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ISoundEventListener
-{
+public interface ISoundEventListener {
     void soundPlay(ISound soundIn, SoundEventAccessor accessor);
 }
