--- ../src-base/minecraft/net/minecraft/client/audio/MovingSoundMinecart.java
+++ ../src-work/minecraft/net/minecraft/client/audio/MovingSoundMinecart.java
@@ -8,39 +8,30 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class MovingSoundMinecart extends MovingSound
-{
+public class MovingSoundMinecart extends MovingSound {
     private final EntityMinecart minecart;
     private float distance = 0.0F;
 
-    public MovingSoundMinecart(EntityMinecart minecartIn)
-    {
+    public MovingSoundMinecart(EntityMinecart minecartIn) {
         super(SoundEvents.ENTITY_MINECART_RIDING, SoundCategory.NEUTRAL);
         this.minecart = minecartIn;
         this.repeat = true;
         this.repeatDelay = 0;
     }
 
-    public void update()
-    {
-        if (this.minecart.isDead)
-        {
+    public void update() {
+        if (this.minecart.isDead) {
             this.donePlaying = true;
-        }
-        else
-        {
-            this.xPosF = (float)this.minecart.posX;
-            this.yPosF = (float)this.minecart.posY;
-            this.zPosF = (float)this.minecart.posZ;
+        } else {
+            this.xPosF = (float) this.minecart.posX;
+            this.yPosF = (float) this.minecart.posY;
+            this.zPosF = (float) this.minecart.posZ;
             float f = MathHelper.sqrt(this.minecart.motionX * this.minecart.motionX + this.minecart.motionZ * this.minecart.motionZ);
 
-            if ((double)f >= 0.01D)
-            {
+            if ((double) f >= 0.01D) {
                 this.distance = MathHelper.clamp(this.distance + 0.0025F, 0.0F, 1.0F);
                 this.volume = 0.0F + MathHelper.clamp(f, 0.0F, 0.5F) * 0.7F;
-            }
-            else
-            {
+            } else {
                 this.distance = 0.0F;
                 this.volume = 0.0F;
             }
