--- ../src-base/minecraft/net/minecraft/client/audio/PositionedSound.java
+++ ../src-work/minecraft/net/minecraft/client/audio/PositionedSound.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.audio;
 
-import javax.annotation.Nullable;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.SoundCategory;
 import net.minecraft.util.SoundEvent;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public abstract class PositionedSound implements ISound
 {
@@ -22,7 +23,7 @@
     protected float zPosF;
     protected boolean repeat;
     protected int repeatDelay;
-    protected ISound.AttenuationType attenuationType;
+    protected AttenuationType attenuationType;
 
     protected PositionedSound(SoundEvent soundIn, SoundCategory categoryIn)
     {
@@ -33,7 +34,7 @@
     {
         this.volume = 1.0F;
         this.pitch = 1.0F;
-        this.attenuationType = ISound.AttenuationType.LINEAR;
+        this.attenuationType = AttenuationType.LINEAR;
         this.positionedSoundLocation = soundId;
         this.category = categoryIn;
     }
@@ -104,7 +105,7 @@
         return this.zPosF;
     }
 
-    public ISound.AttenuationType getAttenuationType()
+    public AttenuationType getAttenuationType()
     {
         return this.attenuationType;
     }
