--- ../src-base/minecraft/net/minecraft/client/audio/SoundList.java
+++ ../src-work/minecraft/net/minecraft/client/audio/SoundList.java
@@ -1,37 +1,33 @@
 package net.minecraft.client.audio;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
-public class SoundList
-{
+public class SoundList {
     private final List<Sound> sounds;
     private final boolean replaceExisting;
     private final String subtitle;
 
-    public SoundList(List<Sound> soundsIn, boolean replceIn, String subtitleIn)
-    {
+    public SoundList(List<Sound> soundsIn, boolean replceIn, String subtitleIn) {
         this.sounds = soundsIn;
         this.replaceExisting = replceIn;
         this.subtitle = subtitleIn;
     }
 
-    public List<Sound> getSounds()
-    {
+    public List<Sound> getSounds() {
         return this.sounds;
     }
 
-    public boolean canReplaceExisting()
-    {
+    public boolean canReplaceExisting() {
         return this.replaceExisting;
     }
 
     @Nullable
-    public String getSubtitle()
-    {
+    public String getSubtitle() {
         return this.subtitle;
     }
 }
