--- ../src-base/minecraft/net/minecraft/network/login/INetHandlerLoginClient.java
+++ ../src-work/minecraft/net/minecraft/network/login/INetHandlerLoginClient.java
@@ -6,8 +6,7 @@
 import net.minecraft.network.login.server.SPacketEncryptionRequest;
 import net.minecraft.network.login.server.SPacketLoginSuccess;
 
-public interface INetHandlerLoginClient extends INetHandler
-{
+public interface INetHandlerLoginClient extends INetHandler {
     void handleEncryptionRequest(SPacketEncryptionRequest packetIn);
 
     void handleLoginSuccess(SPacketLoginSuccess packetIn);
