--- ../src-base/minecraft/net/minecraftforge/advancements/critereon/ItemPredicates.java
+++ ../src-work/minecraft/net/minecraftforge/advancements/critereon/ItemPredicates.java
@@ -20,7 +20,6 @@
 package net.minecraftforge.advancements.critereon;
 
 import com.google.gson.JsonObject;
-
 import net.minecraft.advancements.critereon.ItemPredicate;
 import net.minecraft.util.ResourceLocation;
 
@@ -29,22 +28,18 @@
 import java.util.Map;
 import java.util.function.Function;
 
-public class ItemPredicates
-{
+public class ItemPredicates {
     private static final Map<ResourceLocation, Function<JsonObject, ItemPredicate>> predicates = new HashMap<>();
 
-    static
-    {
+    static {
         register(new ResourceLocation("forge:ore_dict"), OredictItemPredicate::new);
     }
 
-    public static void register(ResourceLocation rl, Function<JsonObject, ItemPredicate> jsonToPredicate)
-    {
+    public static void register(ResourceLocation rl, Function<JsonObject, ItemPredicate> jsonToPredicate) {
         predicates.put(rl, jsonToPredicate);
     }
 
-    public static Map<ResourceLocation, Function<JsonObject, ItemPredicate>> getPredicates()
-    {
+    public static Map<ResourceLocation, Function<JsonObject, ItemPredicate>> getPredicates() {
         return Collections.unmodifiableMap(predicates);
     }
 }
