--- ../src-base/minecraft/net/minecraftforge/advancements/critereon/OredictItemPredicate.java
+++ ../src-work/minecraft/net/minecraftforge/advancements/critereon/OredictItemPredicate.java
@@ -21,30 +21,28 @@
 
 
 import com.google.gson.JsonObject;
-import net.minecraft.util.JsonUtils;
-import org.apache.commons.lang3.ArrayUtils;
-
 import net.minecraft.advancements.critereon.ItemPredicate;
 import net.minecraft.item.ItemStack;
+import net.minecraft.util.JsonUtils;
 import net.minecraftforge.oredict.OreDictionary;
+import org.apache.commons.lang3.ArrayUtils;
 
 /**
  * An {@link ItemPredicate} that matches oredicts.
  */
-public class OredictItemPredicate extends ItemPredicate
-{
+public class OredictItemPredicate extends ItemPredicate {
     private final String ore;
 
-    public OredictItemPredicate(String ore)
-    {
+    public OredictItemPredicate(String ore) {
         this.ore = ore;
     }
 
-    public OredictItemPredicate(JsonObject jsonObject) { this(JsonUtils.getString(jsonObject, "ore")); }
+    public OredictItemPredicate(JsonObject jsonObject) {
+        this(JsonUtils.getString(jsonObject, "ore"));
+    }
 
     @Override
-    public boolean test(ItemStack stack)
-    {
+    public boolean test(ItemStack stack) {
         return !stack.isEmpty() && ArrayUtils.contains(OreDictionary.getOreIDs(stack), OreDictionary.getOreID(ore));
     }
 }
