--- ../src-base/minecraft/net/minecraftforge/advancements/critereon/OredictItemPredicate.java
+++ ../src-work/minecraft/net/minecraftforge/advancements/critereon/OredictItemPredicate.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -21,12 +21,11 @@
 
 
 import com.google.gson.JsonObject;
-import net.minecraft.util.JsonUtils;
-import org.apache.commons.lang3.ArrayUtils;
-
 import net.minecraft.advancements.critereon.ItemPredicate;
 import net.minecraft.item.ItemStack;
+import net.minecraft.util.JsonUtils;
 import net.minecraftforge.oredict.OreDictionary;
+import org.apache.commons.lang3.ArrayUtils;
 
 /**
  * An {@link ItemPredicate} that matches oredicts.
