--- ../src-base/minecraft/net/minecraftforge/fml/common/thread/SidedThreadGroup.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/thread/SidedThreadGroup.java
@@ -21,19 +21,16 @@
 
 import net.minecraftforge.fml.relauncher.Side;
 
+import javax.annotation.Nonnull;
 import java.util.concurrent.ThreadFactory;
 
-import javax.annotation.Nonnull;
-
 /**
  * A thread group and factory combination which belongs to a {@link Side}.
  */
-public final class SidedThreadGroup extends ThreadGroup implements ThreadFactory
-{
+public final class SidedThreadGroup extends ThreadGroup implements ThreadFactory {
     private final Side side;
 
-    SidedThreadGroup(final Side side)
-    {
+    SidedThreadGroup(final Side side) {
         super(side.name());
         this.side = side;
     }
@@ -43,14 +40,12 @@
      *
      * @return the side
      */
-    public Side getSide()
-    {
+    public Side getSide() {
         return this.side;
     }
 
     @Override
-    public Thread newThread(@Nonnull final Runnable runnable)
-    {
+    public Thread newThread(@Nonnull final Runnable runnable) {
         return new Thread(this, runnable);
     }
 }
