--- ../src-base/minecraft/net/minecraftforge/client/gui/NotificationModUpdateScreen.java
+++ ../src-work/minecraft/net/minecraftforge/client/gui/NotificationModUpdateScreen.java
@@ -34,8 +34,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class NotificationModUpdateScreen extends GuiScreen
-{
+public class NotificationModUpdateScreen extends GuiScreen {
 
     private static final ResourceLocation VERSION_CHECK_ICONS = new ResourceLocation(ForgeVersion.MOD_ID, "textures/gui/version_check_icons.png");
 
@@ -43,23 +42,24 @@
     private Status showNotification = null;
     private boolean hasCheckedForUpdates = false;
 
-    public NotificationModUpdateScreen(GuiButton modButton)
-    {
+    public NotificationModUpdateScreen(GuiButton modButton) {
         this.modButton = modButton;
     }
 
+    public static NotificationModUpdateScreen init(GuiMainMenu guiMainMenu, GuiButton modButton) {
+        NotificationModUpdateScreen notificationModUpdateScreen = new NotificationModUpdateScreen(modButton);
+        notificationModUpdateScreen.setGuiSize(guiMainMenu.width, guiMainMenu.height);
+        notificationModUpdateScreen.initGui();
+        return notificationModUpdateScreen;
+    }
+
     @Override
-    public void initGui()
-    {
-        if (!hasCheckedForUpdates)
-        {
-            if (modButton != null)
-            {
-                for (ModContainer mod : Loader.instance().getModList())
-                {
+    public void initGui() {
+        if (!hasCheckedForUpdates) {
+            if (modButton != null) {
+                for (ModContainer mod : Loader.instance().getModList()) {
                     Status status = ForgeVersion.getResult(mod).status;
-                    if (status == Status.OUTDATED || status == Status.BETA_OUTDATED)
-                    {
+                    if (status == Status.OUTDATED || status == Status.BETA_OUTDATED) {
                         // TODO: Needs better visualization, maybe stacked icons
                         // drawn in a terrace-like pattern?
                         showNotification = Status.OUTDATED;
@@ -71,10 +71,8 @@
     }
 
     @Override
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
-        if (showNotification == null || !showNotification.shouldDraw() || ForgeModContainer.disableVersionCheck)
-        {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+        if (showNotification == null || !showNotification.shouldDraw() || ForgeModContainer.disableVersionCheck) {
             return;
         }
 
@@ -91,12 +89,4 @@
         GlStateManager.popMatrix();
     }
 
-    public static NotificationModUpdateScreen init(GuiMainMenu guiMainMenu, GuiButton modButton)
-    {
-        NotificationModUpdateScreen notificationModUpdateScreen = new NotificationModUpdateScreen(modButton);
-        notificationModUpdateScreen.setGuiSize(guiMainMenu.width, guiMainMenu.height);
-        notificationModUpdateScreen.initGui();
-        return notificationModUpdateScreen;
-    }
-
 }
