--- ../src-base/minecraft/net/minecraft/client/renderer/texture/ITextureObject.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/ITextureObject.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.renderer.texture;
 
-import java.io.IOException;
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public interface ITextureObject
 {
