--- ../src-base/minecraft/net/minecraft/client/renderer/texture/ITextureMapPopulator.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/ITextureMapPopulator.java
@@ -4,7 +4,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ITextureMapPopulator
-{
+public interface ITextureMapPopulator {
     void registerSprites(TextureMap textureMapIn);
 }
