--- ../src-base/minecraft/net/minecraft/client/renderer/texture/ITickableTextureObject.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/ITickableTextureObject.java
@@ -4,6 +4,5 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ITickableTextureObject extends ITextureObject, ITickable
-{
+public interface ITickableTextureObject extends ITextureObject, ITickable {
 }
