--- ../src-base/minecraft/net/minecraft/client/renderer/texture/LayeredColorMaskTexture.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/LayeredColorMaskTexture.java
@@ -1,11 +1,5 @@
 package net.minecraft.client.renderer.texture;
 
-import java.awt.Graphics;
-import java.awt.image.BufferedImage;
-import java.awt.image.ImageObserver;
-import java.io.Closeable;
-import java.io.IOException;
-import java.util.List;
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraft.item.EnumDyeColor;
@@ -17,6 +11,13 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.awt.image.ImageObserver;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class LayeredColorMaskTexture extends AbstractTexture
 {
@@ -71,7 +72,8 @@
 
                         if (s != null)
                         {
-                            BufferedImage bufferedimage2 = net.minecraftforge.client.MinecraftForgeClient.getImageLayer(new ResourceLocation(s), resourceManager);
+                           BufferedImage bufferedimage2 = net.minecraftforge.client.MinecraftForgeClient.getImageLayer(new ResourceLocation(s), resourceManager);
+
                             if (bufferedimage2.getWidth() == bufferedimage.getWidth() && bufferedimage2.getHeight() == bufferedimage.getHeight() && bufferedimage2.getType() == 6)
                             {
                                 for (int l = 0; l < bufferedimage2.getHeight(); ++l)
