--- ../src-base/minecraft/net/minecraft/client/renderer/texture/SimpleTexture.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/SimpleTexture.java
@@ -1,8 +1,5 @@
 package net.minecraft.client.renderer.texture;
 
-import java.awt.image.BufferedImage;
-import java.io.Closeable;
-import java.io.IOException;
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraft.client.resources.data.TextureMetadataSection;
@@ -13,52 +10,45 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.awt.image.BufferedImage;
+import java.io.Closeable;
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
-public class SimpleTexture extends AbstractTexture
-{
+public class SimpleTexture extends AbstractTexture {
     private static final Logger LOGGER = LogManager.getLogger();
     protected final ResourceLocation textureLocation;
 
-    public SimpleTexture(ResourceLocation textureResourceLocation)
-    {
+    public SimpleTexture(ResourceLocation textureResourceLocation) {
         this.textureLocation = textureResourceLocation;
     }
 
-    public void loadTexture(IResourceManager resourceManager) throws IOException
-    {
+    public void loadTexture(IResourceManager resourceManager) throws IOException {
         this.deleteGlTexture();
         IResource iresource = null;
 
-        try
-        {
+        try {
             iresource = resourceManager.getResource(this.textureLocation);
             BufferedImage bufferedimage = TextureUtil.readBufferedImage(iresource.getInputStream());
             boolean flag = false;
             boolean flag1 = false;
 
-            if (iresource.hasMetadata())
-            {
-                try
-                {
-                    TextureMetadataSection texturemetadatasection = (TextureMetadataSection)iresource.getMetadata("texture");
+            if (iresource.hasMetadata()) {
+                try {
+                    TextureMetadataSection texturemetadatasection = (TextureMetadataSection) iresource.getMetadata("texture");
 
-                    if (texturemetadatasection != null)
-                    {
+                    if (texturemetadatasection != null) {
                         flag = texturemetadatasection.getTextureBlur();
                         flag1 = texturemetadatasection.getTextureClamp();
                     }
-                }
-                catch (RuntimeException runtimeexception)
-                {
+                } catch (RuntimeException runtimeexception) {
                     LOGGER.warn("Failed reading metadata of: {}", this.textureLocation, runtimeexception);
                 }
             }
 
             TextureUtil.uploadTextureImageAllocate(this.getGlTextureId(), bufferedimage, flag, flag1);
+        } finally {
+            IOUtils.closeQuietly((Closeable) iresource);
         }
-        finally
-        {
-            IOUtils.closeQuietly((Closeable)iresource);
-        }
     }
 }
