--- ../src-base/minecraft/net/minecraft/client/renderer/texture/SimpleTexture.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/texture/SimpleTexture.java
@@ -1,8 +1,5 @@
 package net.minecraft.client.renderer.texture;
 
-import java.awt.image.BufferedImage;
-import java.io.Closeable;
-import java.io.IOException;
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraft.client.resources.data.TextureMetadataSection;
@@ -13,6 +10,10 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.awt.image.BufferedImage;
+import java.io.Closeable;
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class SimpleTexture extends AbstractTexture
 {
