--- ../src-base/minecraft/paulscode/sound/codecs/CodecIBXM.java
+++ ../src-work/minecraft/paulscode/sound/codecs/CodecIBXM.java
@@ -1,25 +1,20 @@
 package paulscode.sound.codecs;
 
+import ibxm.*;
+import paulscode.sound.ICodec;
+import paulscode.sound.SoundBuffer;
+import paulscode.sound.SoundSystemConfig;
+import paulscode.sound.SoundSystemLogger;
+
+import javax.sound.sampled.AudioFormat;
 import java.io.DataInputStream;
-import java.io.InputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URL;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.ShortBuffer;
 
-import javax.sound.sampled.AudioFormat;
-
-import paulscode.sound.ICodec;
-import paulscode.sound.SoundBuffer;
-import paulscode.sound.SoundSystemConfig;
-import paulscode.sound.SoundSystemLogger;
-import ibxm.FastTracker2;
-import ibxm.IBXM;
-import ibxm.Module;
-import ibxm.ProTracker;
-import ibxm.ScreamTracker3;
-
 /**
  * The CodecIBXM class provides an ICodec interface for reading from MOD/S3M/XM
  * files via the IBXM library.
