--- ../src-base/minecraft/net/minecraft/network/status/client/CPacketPing.java
+++ ../src-work/minecraft/net/minecraft/network/status/client/CPacketPing.java
@@ -1,43 +1,37 @@
 package net.minecraft.network.status.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusServer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class CPacketPing implements Packet<INetHandlerStatusServer>
-{
+import java.io.IOException;
+
+public class CPacketPing implements Packet<INetHandlerStatusServer> {
     private long clientTime;
 
-    public CPacketPing()
-    {
+    public CPacketPing() {
     }
 
     @SideOnly(Side.CLIENT)
-    public CPacketPing(long clientTimeIn)
-    {
+    public CPacketPing(long clientTimeIn) {
         this.clientTime = clientTimeIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.clientTime = buf.readLong();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeLong(this.clientTime);
     }
 
-    public void processPacket(INetHandlerStatusServer handler)
-    {
+    public void processPacket(INetHandlerStatusServer handler) {
         handler.processPing(this);
     }
 
-    public long getClientTime()
-    {
+    public long getClientTime() {
         return this.clientTime;
     }
 }
