--- ../src-base/minecraft/net/minecraft/network/status/client/CPacketPing.java
+++ ../src-work/minecraft/net/minecraft/network/status/client/CPacketPing.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.status.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusServer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class CPacketPing implements Packet<INetHandlerStatusServer>
 {
     private long clientTime;
