--- ../src-base/minecraft/net/minecraft/network/status/client/CPacketServerQuery.java
+++ ../src-work/minecraft/net/minecraft/network/status/client/CPacketServerQuery.java
@@ -1,10 +1,11 @@
 package net.minecraft.network.status.client;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusServer;
 
+import java.io.IOException;
+
 public class CPacketServerQuery implements Packet<INetHandlerStatusServer>
 {
     public void readPacketData(PacketBuffer buf) throws IOException
