--- ../src-base/minecraft/net/minecraftforge/fml/common/toposort/ModSortingException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/toposort/ModSortingException.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,8 +19,6 @@
 
 package net.minecraftforge.fml.common.toposort;
 
-import java.util.Set;
-
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraftforge.fml.client.GuiSortingProblem;
 import net.minecraftforge.fml.client.IDisplayableError;
@@ -29,6 +27,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Set;
+
 public class ModSortingException extends EnhancedRuntimeException implements IDisplayableError
 {
     private static final long serialVersionUID = 1L;
