--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderMooshroom.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderMooshroom.java
@@ -8,23 +8,19 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderMooshroom extends RenderLiving<EntityMooshroom>
-{
+public class RenderMooshroom extends RenderLiving<EntityMooshroom> {
     private static final ResourceLocation MOOSHROOM_TEXTURES = new ResourceLocation("textures/entity/cow/mooshroom.png");
 
-    public RenderMooshroom(RenderManager p_i47200_1_)
-    {
+    public RenderMooshroom(RenderManager p_i47200_1_) {
         super(p_i47200_1_, new ModelCow(), 0.7F);
         this.addLayer(new LayerMooshroomMushroom(this));
     }
 
-    public ModelCow getMainModel()
-    {
-        return (ModelCow)super.getMainModel();
+    public ModelCow getMainModel() {
+        return (ModelCow) super.getMainModel();
     }
 
-    protected ResourceLocation getEntityTexture(EntityMooshroom entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityMooshroom entity) {
         return MOOSHROOM_TEXTURES;
     }
 }
