--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderWither.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderWither.java
@@ -9,31 +9,26 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderWither extends RenderLiving<EntityWither>
-{
+public class RenderWither extends RenderLiving<EntityWither> {
     private static final ResourceLocation INVULNERABLE_WITHER_TEXTURES = new ResourceLocation("textures/entity/wither/wither_invulnerable.png");
     private static final ResourceLocation WITHER_TEXTURES = new ResourceLocation("textures/entity/wither/wither.png");
 
-    public RenderWither(RenderManager renderManagerIn)
-    {
+    public RenderWither(RenderManager renderManagerIn) {
         super(renderManagerIn, new ModelWither(0.0F), 1.0F);
         this.addLayer(new LayerWitherAura(this));
     }
 
-    protected ResourceLocation getEntityTexture(EntityWither entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityWither entity) {
         int i = entity.getInvulTime();
         return i > 0 && (i > 80 || i / 5 % 2 != 1) ? INVULNERABLE_WITHER_TEXTURES : WITHER_TEXTURES;
     }
 
-    protected void preRenderCallback(EntityWither entitylivingbaseIn, float partialTickTime)
-    {
+    protected void preRenderCallback(EntityWither entitylivingbaseIn, float partialTickTime) {
         float f = 2.0F;
         int i = entitylivingbaseIn.getInvulTime();
 
-        if (i > 0)
-        {
-            f -= ((float)i - partialTickTime) / 220.0F * 0.5F;
+        if (i > 0) {
+            f -= ((float) i - partialTickTime) / 220.0F * 0.5F;
         }
 
         GlStateManager.scale(f, f, f);
