--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderSquid.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderSquid.java
@@ -8,22 +8,18 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderSquid extends RenderLiving<EntitySquid>
-{
+public class RenderSquid extends RenderLiving<EntitySquid> {
     private static final ResourceLocation SQUID_TEXTURES = new ResourceLocation("textures/entity/squid.png");
 
-    public RenderSquid(RenderManager p_i47192_1_)
-    {
+    public RenderSquid(RenderManager p_i47192_1_) {
         super(p_i47192_1_, new ModelSquid(), 0.7F);
     }
 
-    protected ResourceLocation getEntityTexture(EntitySquid entity)
-    {
+    protected ResourceLocation getEntityTexture(EntitySquid entity) {
         return SQUID_TEXTURES;
     }
 
-    protected void applyRotations(EntitySquid entityLiving, float p_77043_2_, float rotationYaw, float partialTicks)
-    {
+    protected void applyRotations(EntitySquid entityLiving, float p_77043_2_, float rotationYaw, float partialTicks) {
         float f = entityLiving.prevSquidPitch + (entityLiving.squidPitch - entityLiving.prevSquidPitch) * partialTicks;
         float f1 = entityLiving.prevSquidYaw + (entityLiving.squidYaw - entityLiving.prevSquidYaw) * partialTicks;
         GlStateManager.translate(0.0F, 0.5F, 0.0F);
@@ -33,8 +29,7 @@
         GlStateManager.translate(0.0F, -1.2F, 0.0F);
     }
 
-    protected float handleRotationFloat(EntitySquid livingBase, float partialTicks)
-    {
+    protected float handleRotationFloat(EntitySquid livingBase, float partialTicks) {
         return livingBase.lastTentacleAngle + (livingBase.tentacleAngle - livingBase.lastTentacleAngle) * partialTicks;
     }
 }
