--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderGiantZombie.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderGiantZombie.java
@@ -10,38 +10,31 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderGiantZombie extends RenderLiving<EntityGiantZombie>
-{
+public class RenderGiantZombie extends RenderLiving<EntityGiantZombie> {
     private static final ResourceLocation ZOMBIE_TEXTURES = new ResourceLocation("textures/entity/zombie/zombie.png");
     private final float scale;
 
-    public RenderGiantZombie(RenderManager p_i47206_1_, float p_i47206_2_)
-    {
+    public RenderGiantZombie(RenderManager p_i47206_1_, float p_i47206_2_) {
         super(p_i47206_1_, new ModelZombie(), 0.5F * p_i47206_2_);
         this.scale = p_i47206_2_;
         this.addLayer(new LayerHeldItem(this));
-        this.addLayer(new LayerBipedArmor(this)
-        {
-            protected void initArmor()
-            {
+        this.addLayer(new LayerBipedArmor(this) {
+            protected void initArmor() {
                 this.modelLeggings = new ModelZombie(0.5F, true);
                 this.modelArmor = new ModelZombie(1.0F, true);
             }
         });
     }
 
-    public void transformHeldFull3DItemLayer()
-    {
+    public void transformHeldFull3DItemLayer() {
         GlStateManager.translate(0.0F, 0.1875F, 0.0F);
     }
 
-    protected void preRenderCallback(EntityGiantZombie entitylivingbaseIn, float partialTickTime)
-    {
+    protected void preRenderCallback(EntityGiantZombie entitylivingbaseIn, float partialTickTime) {
         GlStateManager.scale(this.scale, this.scale, this.scale);
     }
 
-    protected ResourceLocation getEntityTexture(EntityGiantZombie entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityGiantZombie entity) {
         return ZOMBIE_TEXTURES;
     }
 }
