--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderAreaEffectCloud.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderAreaEffectCloud.java
@@ -1,11 +1,12 @@
 package net.minecraft.client.renderer.entity;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityAreaEffectCloud;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class RenderAreaEffectCloud extends Render<EntityAreaEffectCloud>
 {
