--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderCow.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderCow.java
@@ -7,17 +7,14 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderCow extends RenderLiving<EntityCow>
-{
+public class RenderCow extends RenderLiving<EntityCow> {
     private static final ResourceLocation COW_TEXTURES = new ResourceLocation("textures/entity/cow/cow.png");
 
-    public RenderCow(RenderManager p_i47210_1_)
-    {
+    public RenderCow(RenderManager p_i47210_1_) {
         super(p_i47210_1_, new ModelCow(), 0.7F);
     }
 
-    protected ResourceLocation getEntityTexture(EntityCow entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityCow entity) {
         return COW_TEXTURES;
     }
 }
