--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderLightningBolt.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderLightningBolt.java
@@ -1,7 +1,5 @@
 package net.minecraft.client.renderer.entity;
 
-import java.util.Random;
-import javax.annotation.Nullable;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.Tessellator;
@@ -11,6 +9,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class RenderLightningBolt extends Render<EntityLightningBolt>
 {
