--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderMinecartMobSpawner.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderMinecartMobSpawner.java
@@ -5,10 +5,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderMinecartMobSpawner extends RenderMinecart<EntityMinecartMobSpawner>
-{
-    public RenderMinecartMobSpawner(RenderManager renderManagerIn)
-    {
+public class RenderMinecartMobSpawner extends RenderMinecart<EntityMinecartMobSpawner> {
+    public RenderMinecartMobSpawner(RenderManager renderManagerIn) {
         super(renderManagerIn);
     }
 }
