--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderFish.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderFish.java
@@ -7,7 +7,6 @@
 import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.projectile.EntityFishHook;
-import net.minecraft.init.Items;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.EnumHandSide;
 import net.minecraft.util.ResourceLocation;
