--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderEntity.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderEntity.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.renderer.entity;
 
-import javax.annotation.Nullable;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.entity.Entity;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class RenderEntity extends Render<Entity>
 {
