--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderWitherSkeleton.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderWitherSkeleton.java
@@ -7,22 +7,18 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderWitherSkeleton extends RenderSkeleton
-{
+public class RenderWitherSkeleton extends RenderSkeleton {
     private static final ResourceLocation WITHER_SKELETON_TEXTURES = new ResourceLocation("textures/entity/skeleton/wither_skeleton.png");
 
-    public RenderWitherSkeleton(RenderManager p_i47188_1_)
-    {
+    public RenderWitherSkeleton(RenderManager p_i47188_1_) {
         super(p_i47188_1_);
     }
 
-    protected ResourceLocation getEntityTexture(AbstractSkeleton entity)
-    {
+    protected ResourceLocation getEntityTexture(AbstractSkeleton entity) {
         return WITHER_SKELETON_TEXTURES;
     }
 
-    protected void preRenderCallback(AbstractSkeleton entitylivingbaseIn, float partialTickTime)
-    {
+    protected void preRenderCallback(AbstractSkeleton entitylivingbaseIn, float partialTickTime) {
         GlStateManager.scale(1.2F, 1.2F, 1.2F);
     }
 }
