--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderEvoker.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderEvoker.java
@@ -12,36 +12,29 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderEvoker extends RenderLiving<EntityMob>
-{
+public class RenderEvoker extends RenderLiving<EntityMob> {
     private static final ResourceLocation EVOKER_ILLAGER = new ResourceLocation("textures/entity/illager/evoker.png");
 
-    public RenderEvoker(RenderManager p_i47207_1_)
-    {
+    public RenderEvoker(RenderManager p_i47207_1_) {
         super(p_i47207_1_, new ModelIllager(0.0F, 0.0F, 64, 64), 0.5F);
-        this.addLayer(new LayerHeldItem(this)
-        {
-            public void doRenderLayer(EntityLivingBase entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-            {
-                if (((EntitySpellcasterIllager)entitylivingbaseIn).isSpellcasting())
-                {
+        this.addLayer(new LayerHeldItem(this) {
+            public void doRenderLayer(EntityLivingBase entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
+                if (((EntitySpellcasterIllager) entitylivingbaseIn).isSpellcasting()) {
                     super.doRenderLayer(entitylivingbaseIn, limbSwing, limbSwingAmount, partialTicks, ageInTicks, netHeadYaw, headPitch, scale);
                 }
             }
-            protected void translateToHand(EnumHandSide p_191361_1_)
-            {
-                ((ModelIllager)this.livingEntityRenderer.getMainModel()).getArm(p_191361_1_).postRender(0.0625F);
+
+            protected void translateToHand(EnumHandSide p_191361_1_) {
+                ((ModelIllager) this.livingEntityRenderer.getMainModel()).getArm(p_191361_1_).postRender(0.0625F);
             }
         });
     }
 
-    protected ResourceLocation getEntityTexture(EntityMob entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityMob entity) {
         return EVOKER_ILLAGER;
     }
 
-    protected void preRenderCallback(EntityMob entitylivingbaseIn, float partialTickTime)
-    {
+    protected void preRenderCallback(EntityMob entitylivingbaseIn, float partialTickTime) {
         float f = 0.9375F;
         GlStateManager.scale(0.9375F, 0.9375F, 0.9375F);
     }
