--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderEnderman.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderEnderman.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer.entity;
 
-import java.util.Random;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.model.ModelEnderman;
 import net.minecraft.client.renderer.entity.layers.LayerEndermanEyes;
@@ -10,6 +9,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class RenderEnderman extends RenderLiving<EntityEnderman>
 {
