--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderLlama.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderLlama.java
@@ -8,18 +8,15 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderLlama extends RenderLiving<EntityLlama>
-{
-    private static final ResourceLocation[] LLAMA_TEXTURES = new ResourceLocation[] {new ResourceLocation("textures/entity/llama/llama_creamy.png"), new ResourceLocation("textures/entity/llama/llama_white.png"), new ResourceLocation("textures/entity/llama/llama_brown.png"), new ResourceLocation("textures/entity/llama/llama_gray.png")};
+public class RenderLlama extends RenderLiving<EntityLlama> {
+    private static final ResourceLocation[] LLAMA_TEXTURES = new ResourceLocation[]{new ResourceLocation("textures/entity/llama/llama_creamy.png"), new ResourceLocation("textures/entity/llama/llama_white.png"), new ResourceLocation("textures/entity/llama/llama_brown.png"), new ResourceLocation("textures/entity/llama/llama_gray.png")};
 
-    public RenderLlama(RenderManager p_i47203_1_)
-    {
+    public RenderLlama(RenderManager p_i47203_1_) {
         super(p_i47203_1_, new ModelLlama(0.0F), 0.7F);
         this.addLayer(new LayerLlamaDecor(this));
     }
 
-    protected ResourceLocation getEntityTexture(EntityLlama entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityLlama entity) {
         return LLAMA_TEXTURES[entity.getVariant()];
     }
 }
