--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderStray.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderStray.java
@@ -7,18 +7,15 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderStray extends RenderSkeleton
-{
+public class RenderStray extends RenderSkeleton {
     private static final ResourceLocation STRAY_SKELETON_TEXTURES = new ResourceLocation("textures/entity/skeleton/stray.png");
 
-    public RenderStray(RenderManager p_i47191_1_)
-    {
+    public RenderStray(RenderManager p_i47191_1_) {
         super(p_i47191_1_);
         this.addLayer(new LayerStrayClothing(this));
     }
 
-    protected ResourceLocation getEntityTexture(AbstractSkeleton entity)
-    {
+    protected ResourceLocation getEntityTexture(AbstractSkeleton entity) {
         return STRAY_SKELETON_TEXTURES;
     }
 }
