--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderZombie.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderZombie.java
@@ -8,17 +8,13 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderZombie extends RenderBiped<EntityZombie>
-{
+public class RenderZombie extends RenderBiped<EntityZombie> {
     private static final ResourceLocation ZOMBIE_TEXTURES = new ResourceLocation("textures/entity/zombie/zombie.png");
 
-    public RenderZombie(RenderManager renderManagerIn)
-    {
+    public RenderZombie(RenderManager renderManagerIn) {
         super(renderManagerIn, new ModelZombie(), 0.5F);
-        LayerBipedArmor layerbipedarmor = new LayerBipedArmor(this)
-        {
-            protected void initArmor()
-            {
+        LayerBipedArmor layerbipedarmor = new LayerBipedArmor(this) {
+            protected void initArmor() {
                 this.modelLeggings = new ModelZombie(0.5F, true);
                 this.modelArmor = new ModelZombie(1.0F, true);
             }
@@ -26,8 +22,7 @@
         this.addLayer(layerbipedarmor);
     }
 
-    protected ResourceLocation getEntityTexture(EntityZombie entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityZombie entity) {
         return ZOMBIE_TEXTURES;
     }
 }
