--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderPig.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderPig.java
@@ -8,18 +8,15 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderPig extends RenderLiving<EntityPig>
-{
+public class RenderPig extends RenderLiving<EntityPig> {
     private static final ResourceLocation PIG_TEXTURES = new ResourceLocation("textures/entity/pig/pig.png");
 
-    public RenderPig(RenderManager p_i47198_1_)
-    {
+    public RenderPig(RenderManager p_i47198_1_) {
         super(p_i47198_1_, new ModelPig(), 0.7F);
         this.addLayer(new LayerSaddle(this));
     }
 
-    protected ResourceLocation getEntityTexture(EntityPig entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityPig entity) {
         return PIG_TEXTURES;
     }
 }
