--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderBat.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderBat.java
@@ -9,33 +9,25 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderBat extends RenderLiving<EntityBat>
-{
+public class RenderBat extends RenderLiving<EntityBat> {
     private static final ResourceLocation BAT_TEXTURES = new ResourceLocation("textures/entity/bat.png");
 
-    public RenderBat(RenderManager renderManagerIn)
-    {
+    public RenderBat(RenderManager renderManagerIn) {
         super(renderManagerIn, new ModelBat(), 0.25F);
     }
 
-    protected ResourceLocation getEntityTexture(EntityBat entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityBat entity) {
         return BAT_TEXTURES;
     }
 
-    protected void preRenderCallback(EntityBat entitylivingbaseIn, float partialTickTime)
-    {
+    protected void preRenderCallback(EntityBat entitylivingbaseIn, float partialTickTime) {
         GlStateManager.scale(0.35F, 0.35F, 0.35F);
     }
 
-    protected void applyRotations(EntityBat entityLiving, float p_77043_2_, float rotationYaw, float partialTicks)
-    {
-        if (entityLiving.getIsBatHanging())
-        {
+    protected void applyRotations(EntityBat entityLiving, float p_77043_2_, float rotationYaw, float partialTicks) {
+        if (entityLiving.getIsBatHanging()) {
             GlStateManager.translate(0.0F, -0.1F, 0.0F);
-        }
-        else
-        {
+        } else {
             GlStateManager.translate(0.0F, MathHelper.cos(p_77043_2_ * 0.3F) * 0.1F, 0.0F);
         }
 
