--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderTippedArrow.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderTippedArrow.java
@@ -6,18 +6,15 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderTippedArrow extends RenderArrow<EntityTippedArrow>
-{
+public class RenderTippedArrow extends RenderArrow<EntityTippedArrow> {
     public static final ResourceLocation RES_ARROW = new ResourceLocation("textures/entity/projectiles/arrow.png");
     public static final ResourceLocation RES_TIPPED_ARROW = new ResourceLocation("textures/entity/projectiles/tipped_arrow.png");
 
-    public RenderTippedArrow(RenderManager manager)
-    {
+    public RenderTippedArrow(RenderManager manager) {
         super(manager);
     }
 
-    protected ResourceLocation getEntityTexture(EntityTippedArrow entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityTippedArrow entity) {
         return entity.getColor() > 0 ? RES_TIPPED_ARROW : RES_ARROW;
     }
 }
