--- ../src-base/minecraft/net/minecraft/client/renderer/entity/RenderPigZombie.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/entity/RenderPigZombie.java
@@ -8,25 +8,20 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class RenderPigZombie extends RenderBiped<EntityPigZombie>
-{
+public class RenderPigZombie extends RenderBiped<EntityPigZombie> {
     private static final ResourceLocation ZOMBIE_PIGMAN_TEXTURE = new ResourceLocation("textures/entity/zombie_pigman.png");
 
-    public RenderPigZombie(RenderManager renderManagerIn)
-    {
+    public RenderPigZombie(RenderManager renderManagerIn) {
         super(renderManagerIn, new ModelZombie(), 0.5F);
-        this.addLayer(new LayerBipedArmor(this)
-        {
-            protected void initArmor()
-            {
+        this.addLayer(new LayerBipedArmor(this) {
+            protected void initArmor() {
                 this.modelLeggings = new ModelZombie(0.5F, true);
                 this.modelArmor = new ModelZombie(1.0F, true);
             }
         });
     }
 
-    protected ResourceLocation getEntityTexture(EntityPigZombie entity)
-    {
+    protected ResourceLocation getEntityTexture(EntityPigZombie entity) {
         return ZOMBIE_PIGMAN_TEXTURE;
     }
 }
