--- ../src-base/minecraft/net/minecraft/util/datafix/walkers/BlockEntityTag.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/walkers/BlockEntityTag.java
@@ -1,8 +1,6 @@
 package net.minecraft.util.datafix.walkers;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.datafix.FixTypes;
@@ -11,60 +9,15 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class BlockEntityTag implements IDataWalker
-{
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class BlockEntityTag implements IDataWalker {
     private static final Logger LOGGER = LogManager.getLogger();
     private static final Map<String, String> NEW_TO_OLD_ID_MAP = Maps.<String, String>newHashMap();
     private static final Map<String, String> ITEM_ID_TO_BLOCK_ENTITY_ID = Maps.<String, String>newHashMap();
 
-    @Nullable
-    private static String getBlockEntityID(int blockID, String p_188267_1_)
-    {
-        return blockID < 515 ? (String)NEW_TO_OLD_ID_MAP.get((new ResourceLocation(p_188267_1_)).toString()) : (String)ITEM_ID_TO_BLOCK_ENTITY_ID.get((new ResourceLocation(p_188267_1_)).toString());
-    }
-
-    public NBTTagCompound process(IDataFixer fixer, NBTTagCompound compound, int versionIn)
-    {
-        if (!compound.hasKey("tag", 10))
-        {
-            return compound;
-        }
-        else
-        {
-            NBTTagCompound nbttagcompound = compound.getCompoundTag("tag");
-
-            if (nbttagcompound.hasKey("BlockEntityTag", 10))
-            {
-                NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("BlockEntityTag");
-                String s = compound.getString("id");
-                String s1 = getBlockEntityID(versionIn, s);
-                boolean flag;
-
-                if (s1 == null)
-                {
-                    LOGGER.warn("Unable to resolve BlockEntity for ItemInstance: {}", (Object)s);
-                    flag = false;
-                }
-                else
-                {
-                    flag = !nbttagcompound1.hasKey("id");
-                    nbttagcompound1.setString("id", s1);
-                }
-
-                fixer.process(FixTypes.BLOCK_ENTITY, nbttagcompound1, versionIn);
-
-                if (flag)
-                {
-                    nbttagcompound1.removeTag("id");
-                }
-            }
-
-            return compound;
-        }
-    }
-
-    static
-    {
+    static {
         Map<String, String> map = NEW_TO_OLD_ID_MAP;
         map.put("minecraft:furnace", "Furnace");
         map.put("minecraft:lit_furnace", "Furnace");
@@ -154,4 +107,41 @@
         map.put("minecraft:end_gateway", "minecraft:end_gateway");
         map.put("minecraft:shield", "minecraft:shield");
     }
+
+    @Nullable
+    private static String getBlockEntityID(int blockID, String p_188267_1_) {
+        return blockID < 515 ? (String) NEW_TO_OLD_ID_MAP.get((new ResourceLocation(p_188267_1_)).toString()) : (String) ITEM_ID_TO_BLOCK_ENTITY_ID.get((new ResourceLocation(p_188267_1_)).toString());
+    }
+
+    public NBTTagCompound process(IDataFixer fixer, NBTTagCompound compound, int versionIn) {
+        if (!compound.hasKey("tag", 10)) {
+            return compound;
+        } else {
+            NBTTagCompound nbttagcompound = compound.getCompoundTag("tag");
+
+            if (nbttagcompound.hasKey("BlockEntityTag", 10)) {
+                NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("BlockEntityTag");
+                String s = compound.getString("id");
+                String s1 = getBlockEntityID(versionIn, s);
+                boolean flag;
+
+                if (s1 == null) {
+                    // CraftBukkit - Remove unnecessary warning (occurs when deserializing a Shulker Box item)
+                    // LOGGER.warn("Unable to resolve BlockEntity for ItemInstance: {}", (Object)s);
+                    flag = false;
+                } else {
+                    flag = !nbttagcompound1.hasKey("id");
+                    nbttagcompound1.setString("id", s1);
+                }
+
+                fixer.process(FixTypes.BLOCK_ENTITY, nbttagcompound1, versionIn);
+
+                if (flag) {
+                    nbttagcompound1.removeTag("id");
+                }
+            }
+
+            return compound;
+        }
+    }
 }
