--- ../src-base/minecraft/net/minecraft/util/datafix/walkers/BlockEntityTag.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/walkers/BlockEntityTag.java
@@ -1,8 +1,6 @@
 package net.minecraft.util.datafix.walkers;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.datafix.FixTypes;
@@ -11,6 +9,9 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+
 public class BlockEntityTag implements IDataWalker
 {
     private static final Logger LOGGER = LogManager.getLogger();
@@ -42,7 +43,8 @@
 
                 if (s1 == null)
                 {
-                    LOGGER.warn("Unable to resolve BlockEntity for ItemInstance: {}", (Object)s);
+                    // CraftBukkit - Remove unnecessary warning (occurs when deserializing a Shulker Box item)
+                    // LOGGER.warn("Unable to resolve BlockEntity for ItemInstance: {}", (Object)s);
                     flag = false;
                 }
                 else
