--- ../src-base/minecraft/net/minecraft/util/datafix/walkers/Filtered.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/walkers/Filtered.java
@@ -8,30 +8,21 @@
 import net.minecraft.util.datafix.IDataFixer;
 import net.minecraft.util.datafix.IDataWalker;
 
-public abstract class Filtered implements IDataWalker
-{
+public abstract class Filtered implements IDataWalker {
     private final ResourceLocation key;
 
-    public Filtered(Class<?> p_i47309_1_)
-    {
-        if (Entity.class.isAssignableFrom(p_i47309_1_))
-        {
-            this.key = EntityList.getKey((Class<Entity>)p_i47309_1_);
-        }
-        else if (TileEntity.class.isAssignableFrom(p_i47309_1_))
-        {
-            this.key = TileEntity.getKey((Class<TileEntity>)p_i47309_1_);
-        }
-        else
-        {
+    public Filtered(Class<?> p_i47309_1_) {
+        if (Entity.class.isAssignableFrom(p_i47309_1_)) {
+            this.key = EntityList.getKey((Class<Entity>) p_i47309_1_);
+        } else if (TileEntity.class.isAssignableFrom(p_i47309_1_)) {
+            this.key = TileEntity.getKey((Class<TileEntity>) p_i47309_1_);
+        } else {
             this.key = null;
         }
     }
 
-    public NBTTagCompound process(IDataFixer fixer, NBTTagCompound compound, int versionIn)
-    {
-        if ((new ResourceLocation(compound.getString("id"))).equals(this.key))
-        {
+    public NBTTagCompound process(IDataFixer fixer, NBTTagCompound compound, int versionIn) {
+        if ((new ResourceLocation(compound.getString("id"))).equals(this.key)) {
             compound = this.filteredProcess(fixer, compound, versionIn);
         }
 
