--- ../src-base/minecraft/net/minecraft/client/gui/recipebook/GuiButtonRecipe.java
+++ ../src-work/minecraft/net/minecraft/client/gui/recipebook/GuiButtonRecipe.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui.recipebook;
 
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiScreen;
@@ -15,6 +14,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class GuiButtonRecipe extends GuiButton
 {
