--- ../src-base/minecraft/net/minecraftforge/fluids/BlockFluidBase.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/BlockFluidBase.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,15 +19,8 @@
 
 package net.minecraftforge.fluids;
 
-import java.util.Map;
-import java.util.Random;
-
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Maps;
-
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockLiquid;
 import net.minecraft.block.BlockStairs;
@@ -58,6 +51,11 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Random;
+
 /**
  * This is a base implementation for Fluid blocks.
  *
@@ -267,9 +265,9 @@
      */
     public boolean canDisplace(IBlockAccess world, BlockPos pos)
     {
+
         IBlockState state = world.getBlockState(pos);
         Block block = state.getBlock();
-
         if (block.isAir(state, world, pos))
         {
             return true;
@@ -634,6 +632,7 @@
         float total = 0;
         int count = 0;
 
+
         for (int i = 0; i < flow.length; i++)
         {
             if (flow[i] >= quantaFraction)
@@ -811,6 +810,6 @@
     {
         float filled = getFilledPercentage(world, pos);
         return filled < 0 ? vec.y > pos.getY() + filled + 1
-                          : vec.y < pos.getY() + filled;
+                : vec.y < pos.getY() + filled;
     }
 }
