--- ../src-base/minecraft/net/minecraftforge/fluids/FluidActionResult.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/FluidActionResult.java
@@ -19,10 +19,10 @@
 
 package net.minecraftforge.fluids;
 
-import javax.annotation.Nonnull;
-
 import net.minecraft.item.ItemStack;
 
+import javax.annotation.Nonnull;
+
 /**
  * Holds the result of a fluid action from {@link FluidUtil}.
  *
@@ -32,33 +32,28 @@
  * Successful actions may have an empty ItemStack result in some cases,
  * for example the action succeeded and the resulting item was consumed.
  */
-public class FluidActionResult
-{
-	public static final FluidActionResult FAILURE = new FluidActionResult(false, ItemStack.EMPTY);
+public class FluidActionResult {
+    public static final FluidActionResult FAILURE = new FluidActionResult(false, ItemStack.EMPTY);
 
-	public final boolean success;
-	@Nonnull
-	public final ItemStack result;
+    public final boolean success;
+    @Nonnull
+    public final ItemStack result;
 
-	public FluidActionResult(@Nonnull ItemStack result)
-	{
-		this(true, result);
-	}
+    public FluidActionResult(@Nonnull ItemStack result) {
+        this(true, result);
+    }
 
-	private FluidActionResult(boolean success, @Nonnull ItemStack result)
-	{
-		this.success = success;
-		this.result = result;
-	}
+    private FluidActionResult(boolean success, @Nonnull ItemStack result) {
+        this.success = success;
+        this.result = result;
+    }
 
-	public boolean isSuccess()
-	{
-		return success;
-	}
+    public boolean isSuccess() {
+        return success;
+    }
 
-	@Nonnull
-	public ItemStack getResult()
-	{
-		return result;
-	}
+    @Nonnull
+    public ItemStack getResult() {
+        return result;
+    }
 }
