--- ../src-base/minecraft/net/minecraft/client/util/ITooltipFlag.java
+++ ../src-work/minecraft/net/minecraft/client/util/ITooltipFlag.java
@@ -4,25 +4,21 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface ITooltipFlag
-{
+public interface ITooltipFlag {
     boolean isAdvanced();
 
     @SideOnly(Side.CLIENT)
-    public static enum TooltipFlags implements ITooltipFlag
-    {
+    public static enum TooltipFlags implements ITooltipFlag {
         NORMAL(false),
         ADVANCED(true);
 
         final boolean isAdvanced;
 
-        private TooltipFlags(boolean advanced)
-        {
+        private TooltipFlags(boolean advanced) {
             this.isAdvanced = advanced;
         }
 
-        public boolean isAdvanced()
-        {
+        public boolean isAdvanced() {
             return this.isAdvanced;
         }
     }
