--- ../src-base/minecraft/net/minecraft/client/util/JsonBlendingMode.java
+++ ../src-work/minecraft/net/minecraft/client/util/JsonBlendingMode.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.util;
 
 import com.google.gson.JsonObject;
-import java.util.Locale;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.util.JsonUtils;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Locale;
+
 @SideOnly(Side.CLIENT)
 public class JsonBlendingMode
 {
