--- ../src-base/minecraft/net/minecraft/client/util/SearchTree.java
+++ ../src-work/minecraft/net/minecraft/client/util/SearchTree.java
@@ -4,14 +4,14 @@
 import com.google.common.collect.Lists;
 import it.unimi.dsi.fastutil.objects.Object2IntMap;
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
-import java.util.function.Consumer;
 import java.util.function.Function;
-import net.minecraft.util.ResourceLocation;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
 public class SearchTree<T> implements ISearchTree<T>
@@ -73,7 +73,7 @@
         else
         {
             List<T> list1 = this.byName.search(searchText);
-            return (List<T>)(list1.isEmpty() ? list : Lists.newArrayList(new SearchTree.MergingIterator(list.iterator(), list1.iterator(), this.numericContents)));
+            return (List<T>)(list1.isEmpty() ? list : Lists.newArrayList(new MergingIterator(list.iterator(), list1.iterator(), this.numericContents)));
         }
     }
 
