--- ../src-base/minecraft/net/minecraft/client/util/SuffixArray.java
+++ ../src-work/minecraft/net/minecraft/client/util/SuffixArray.java
@@ -4,19 +4,16 @@
 import com.google.common.collect.Sets;
 import it.unimi.dsi.fastutil.Arrays;
 import it.unimi.dsi.fastutil.Swapper;
-import it.unimi.dsi.fastutil.ints.IntArrayList;
-import it.unimi.dsi.fastutil.ints.IntComparator;
-import it.unimi.dsi.fastutil.ints.IntList;
-import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
-import it.unimi.dsi.fastutil.ints.IntSet;
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
+import it.unimi.dsi.fastutil.ints.*;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public class SuffixArray<T>
 {
