--- ../src-base/minecraft/net/minecraftforge/fml/common/registry/ExistingSubstitutionException.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/registry/ExistingSubstitutionException.java
@@ -22,9 +22,9 @@
 import net.minecraft.util.ResourceLocation;
 
 public class ExistingSubstitutionException extends Exception {
+    private static final long serialVersionUID = 1L;
+
     public ExistingSubstitutionException(ResourceLocation fromName, Object toReplace) {
     }
 
-    private static final long serialVersionUID = 1L;
-
 }
