--- ../src-base/minecraft/net/minecraftforge/fml/common/registry/IEntityAdditionalSpawnData.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/registry/IEntityAdditionalSpawnData.java
@@ -25,8 +25,7 @@
  * A interface for Entities that need extra information to be communicated
  * between the server and client when they are spawned.
  */
-public interface IEntityAdditionalSpawnData
-{
+public interface IEntityAdditionalSpawnData {
     /**
      * Called by the server when constructing the spawn packet.
      * Data should be added to the provided stream.
