--- ../src-base/minecraft/net/minecraft/server/integrated/IntegratedServerCommandManager.java
+++ ../src-work/minecraft/net/minecraft/server/integrated/IntegratedServerCommandManager.java
@@ -5,10 +5,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class IntegratedServerCommandManager extends ServerCommandManager
-{
-    public IntegratedServerCommandManager(IntegratedServer server)
-    {
+public class IntegratedServerCommandManager extends ServerCommandManager {
+    public IntegratedServerCommandManager(IntegratedServer server) {
         super(server);
     }
 }
