--- ../src-base/minecraft/net/minecraft/server/integrated/IntegratedPlayerList.java
+++ ../src-work/minecraft/net/minecraft/server/integrated/IntegratedPlayerList.java
@@ -1,13 +1,14 @@
 package net.minecraft.server.integrated;
 
 import com.mojang.authlib.GameProfile;
-import java.net.SocketAddress;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.server.management.PlayerList;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.net.SocketAddress;
+
 @SideOnly(Side.CLIENT)
 public class IntegratedPlayerList extends PlayerList
 {
