--- ../src-base/minecraft/net/minecraft/entity/monster/EntityStray.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntityStray.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity.monster;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.projectile.EntityArrow;
 import net.minecraft.entity.projectile.EntityTippedArrow;
@@ -15,6 +14,8 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.loot.LootTableList;
 
+import javax.annotation.Nullable;
+
 public class EntityStray extends AbstractSkeleton
 {
     public EntityStray(World worldIn)
