--- ../src-base/minecraft/net/minecraftforge/fml/common/eventhandler/Cancelable.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/eventhandler/Cancelable.java
@@ -19,12 +19,13 @@
 
 package net.minecraftforge.fml.common.eventhandler;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
 @Retention(value = RUNTIME)
 @Target(value = TYPE)
-public @interface Cancelable{}
+public @interface Cancelable {
+}
