--- ../src-base/minecraft/net/minecraftforge/fml/common/eventhandler/EventPriority.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/eventhandler/EventPriority.java
@@ -19,8 +19,7 @@
 
 package net.minecraftforge.fml.common.eventhandler;
 
-public enum EventPriority implements IEventListener
-{
+public enum EventPriority implements IEventListener {
     /*Priority of event listeners, listeners will be sorted with respect to this priority level.
      *
      * Note:
@@ -32,11 +31,10 @@
     NORMAL,
     LOW,
     LOWEST //Last to execute
-;
+    ;
 
     @Override
-    public void invoke(Event event)
-    {
+    public void invoke(Event event) {
         event.setPhase(this);
     }
 }
