--- ../src-base/minecraft/net/minecraftforge/fml/common/eventhandler/IContextSetter.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/eventhandler/IContextSetter.java
@@ -25,7 +25,9 @@
 //This has a major performance impact so use sparingly.
 //Note: The context IS NOT thread aware as this would be ungodly slow.
 //So This should ONLY be used for Forge mod lifecycle events.
-public interface IContextSetter
-{
-    default void setModContainer(ModContainer mod){};
+public interface IContextSetter {
+    default void setModContainer(ModContainer mod) {
+    }
+
+    ;
 }
