--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentFishingSpeed.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentFishingSpeed.java
@@ -2,26 +2,21 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentFishingSpeed extends Enchantment
-{
-    protected EnchantmentFishingSpeed(Enchantment.Rarity rarityIn, EnumEnchantmentType typeIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentFishingSpeed extends Enchantment {
+    protected EnchantmentFishingSpeed(Enchantment.Rarity rarityIn, EnumEnchantmentType typeIn, EntityEquipmentSlot... slots) {
         super(rarityIn, typeIn, slots);
         this.setName("fishingSpeed");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 15 + (enchantmentLevel - 1) * 9;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return super.getMinEnchantability(enchantmentLevel) + 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 3;
     }
 }
