--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentWaterWorker.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentWaterWorker.java
@@ -2,26 +2,21 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentWaterWorker extends Enchantment
-{
-    public EnchantmentWaterWorker(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentWaterWorker extends Enchantment {
+    public EnchantmentWaterWorker(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots) {
         super(rarityIn, EnumEnchantmentType.ARMOR_HEAD, slots);
         this.setName("waterWorker");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 1;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return this.getMinEnchantability(enchantmentLevel) + 40;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 }
