--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentMending.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentMending.java
@@ -2,31 +2,25 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentMending extends Enchantment
-{
-    public EnchantmentMending(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentMending extends Enchantment {
+    public EnchantmentMending(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots) {
         super(rarityIn, EnumEnchantmentType.BREAKABLE, slots);
         this.setName("mending");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return enchantmentLevel * 25;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return this.getMinEnchantability(enchantmentLevel) + 50;
     }
 
-    public boolean isTreasureEnchantment()
-    {
+    public boolean isTreasureEnchantment() {
         return true;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 }
