--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
@@ -2,12 +2,8 @@
 
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Random;
-import java.util.Map.Entry;
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.EnumCreatureAttribute;
 import net.minecraft.entity.player.EntityPlayer;
@@ -23,12 +19,18 @@
 import net.minecraft.util.WeightedRandom;
 import net.minecraft.util.math.MathHelper;
 
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+
 public class EnchantmentHelper
 {
-    private static final EnchantmentHelper.ModifierDamage ENCHANTMENT_MODIFIER_DAMAGE = new EnchantmentHelper.ModifierDamage();
-    private static final EnchantmentHelper.ModifierLiving ENCHANTMENT_MODIFIER_LIVING = new EnchantmentHelper.ModifierLiving();
-    private static final EnchantmentHelper.HurtIterator ENCHANTMENT_ITERATOR_HURT = new EnchantmentHelper.HurtIterator();
-    private static final EnchantmentHelper.DamageIterator ENCHANTMENT_ITERATOR_DAMAGE = new EnchantmentHelper.DamageIterator();
+    private static final ModifierDamage ENCHANTMENT_MODIFIER_DAMAGE = new ModifierDamage();
+    private static final ModifierLiving ENCHANTMENT_MODIFIER_LIVING = new ModifierLiving();
+    private static final HurtIterator ENCHANTMENT_ITERATOR_HURT = new HurtIterator();
+    private static final DamageIterator ENCHANTMENT_ITERATOR_DAMAGE = new DamageIterator();
 
     public static int getEnchantmentLevel(Enchantment enchID, ItemStack stack)
     {
@@ -108,7 +110,7 @@
         }
     }
 
-    private static void applyEnchantmentModifier(EnchantmentHelper.IModifier modifier, ItemStack stack)
+    private static void applyEnchantmentModifier(IModifier modifier, ItemStack stack)
     {
         if (!stack.isEmpty())
         {
@@ -127,7 +129,7 @@
         }
     }
 
-    private static void applyEnchantmentModifierArray(EnchantmentHelper.IModifier modifier, Iterable<ItemStack> stacks)
+    private static void applyEnchantmentModifierArray(IModifier modifier, Iterable<ItemStack> stacks)
     {
         for (ItemStack itemstack : stacks)
         {
@@ -192,8 +194,8 @@
             applyEnchantmentModifier(ENCHANTMENT_ITERATOR_DAMAGE, p_151385_0_.getHeldItemMainhand());
         }
 
-        ENCHANTMENT_ITERATOR_DAMAGE.user = null; //Forge Fix memory leaks: https://bugs.mojang.com/browse/MC-128547
-        ENCHANTMENT_ITERATOR_DAMAGE.target = null;
+        ENCHANTMENT_ITERATOR_DAMAGE.target = null; //Forge Fix memory leaks: https://bugs.mojang.com/browse/MC-128547
+        ENCHANTMENT_ITERATOR_DAMAGE.user = null;
     }
 
     public static int getMaxEnchantmentLevel(Enchantment p_185284_0_, EntityLivingBase p_185284_1_)
@@ -282,6 +284,7 @@
         return getEnchantmentLevel(Enchantments.VANISHING_CURSE, p_190939_0_) > 0;
     }
 
+    public static ItemStack getRandomEquippedItemWithEnchant(Enchantment enchantment, EntityLivingBase entityliving) { return getEnchantedItem(enchantment, entityliving); } // Paper - OBFHELPER
     public static ItemStack getEnchantedItem(Enchantment p_92099_0_, EntityLivingBase p_92099_1_)
     {
         List<ItemStack> list = p_92099_0_.getEntityEquipment(p_92099_1_);
@@ -436,7 +439,7 @@
         return list;
     }
 
-    static final class DamageIterator implements EnchantmentHelper.IModifier
+    static final class DamageIterator implements IModifier
         {
             public EntityLivingBase user;
             public Entity target;
@@ -451,7 +454,7 @@
             }
         }
 
-    static final class HurtIterator implements EnchantmentHelper.IModifier
+    static final class HurtIterator implements IModifier
         {
             public EntityLivingBase user;
             public Entity attacker;
@@ -471,7 +474,7 @@
         void calculateModifier(Enchantment enchantmentIn, int enchantmentLevel);
     }
 
-    static final class ModifierDamage implements EnchantmentHelper.IModifier
+    static final class ModifierDamage implements IModifier
         {
             public int damageModifier;
             public DamageSource source;
@@ -486,7 +489,7 @@
             }
         }
 
-    static final class ModifierLiving implements EnchantmentHelper.IModifier
+    static final class ModifierLiving implements IModifier
         {
             public float livingModifier;
             public EnumCreatureAttribute entityLiving;
