--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentBindingCurse.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentBindingCurse.java
@@ -2,36 +2,29 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentBindingCurse extends Enchantment
-{
-    public EnchantmentBindingCurse(Enchantment.Rarity p_i47254_1_, EntityEquipmentSlot... p_i47254_2_)
-    {
+public class EnchantmentBindingCurse extends Enchantment {
+    public EnchantmentBindingCurse(Enchantment.Rarity p_i47254_1_, EntityEquipmentSlot... p_i47254_2_) {
         super(p_i47254_1_, EnumEnchantmentType.WEARABLE, p_i47254_2_);
         this.setName("binding_curse");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 25;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 
-    public boolean isTreasureEnchantment()
-    {
+    public boolean isTreasureEnchantment() {
         return true;
     }
 
-    public boolean isCurse()
-    {
+    public boolean isCurse() {
         return true;
     }
 }
