--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentUntouching.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentUntouching.java
@@ -3,31 +3,25 @@
 import net.minecraft.init.Enchantments;
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentUntouching extends Enchantment
-{
-    protected EnchantmentUntouching(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentUntouching extends Enchantment {
+    protected EnchantmentUntouching(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots) {
         super(rarityIn, EnumEnchantmentType.DIGGER, slots);
         this.setName("untouching");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 15;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return super.getMinEnchantability(enchantmentLevel) + 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 
-    public boolean canApplyTogether(Enchantment ench)
-    {
+    public boolean canApplyTogether(Enchantment ench) {
         return super.canApplyTogether(ench) && ench != Enchantments.FORTUNE;
     }
 }
