--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentFireAspect.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentFireAspect.java
@@ -2,26 +2,21 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentFireAspect extends Enchantment
-{
-    protected EnchantmentFireAspect(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentFireAspect extends Enchantment {
+    protected EnchantmentFireAspect(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots) {
         super(rarityIn, EnumEnchantmentType.WEAPON, slots);
         this.setName("fire");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 10 + 20 * (enchantmentLevel - 1);
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return super.getMinEnchantability(enchantmentLevel) + 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 2;
     }
 }
