--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentArrowInfinite.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentArrowInfinite.java
@@ -2,31 +2,25 @@
 
 import net.minecraft.inventory.EntityEquipmentSlot;
 
-public class EnchantmentArrowInfinite extends Enchantment
-{
-    public EnchantmentArrowInfinite(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots)
-    {
+public class EnchantmentArrowInfinite extends Enchantment {
+    public EnchantmentArrowInfinite(Enchantment.Rarity rarityIn, EntityEquipmentSlot... slots) {
         super(rarityIn, EnumEnchantmentType.BOW, slots);
         this.setName("arrowInfinite");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 20;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 
-    public boolean canApplyTogether(Enchantment ench)
-    {
+    public boolean canApplyTogether(Enchantment ench) {
         return ench instanceof EnchantmentMending ? false : super.canApplyTogether(ench);
     }
 }
