--- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentData.java
+++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentData.java
@@ -2,13 +2,11 @@
 
 import net.minecraft.util.WeightedRandom;
 
-public class EnchantmentData extends WeightedRandom.Item
-{
+public class EnchantmentData extends WeightedRandom.Item {
     public final Enchantment enchantment;
     public final int enchantmentLevel;
 
-    public EnchantmentData(Enchantment enchantmentObj, int enchLevel)
-    {
+    public EnchantmentData(Enchantment enchantmentObj, int enchLevel) {
         super(enchantmentObj.getRarity().getWeight());
         this.enchantment = enchantmentObj;
         this.enchantmentLevel = enchLevel;
