--- ../src-base/minecraft/net/minecraft/block/BlockFalling.java
+++ ../src-work/minecraft/net/minecraft/block/BlockFalling.java
@@ -1,6 +1,5 @@
 package net.minecraft.block;
 
-import java.util.Random;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
@@ -12,6 +11,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 public class BlockFalling extends Block
 {
     public static boolean fallInstantly;
@@ -88,6 +89,12 @@
         return 2;
     }
 
+    // Paper start - OBFHELPER
+    public static boolean canMoveThrough(IBlockState blockdata) {
+        return BlockFalling.canFallThrough(blockdata);
+    }
+   // Paper end
+
     public static boolean canFallThrough(IBlockState state)
     {
         Block block = state.getBlock();
