--- ../src-base/minecraft/net/minecraft/block/BlockRedFlower.java
+++ ../src-work/minecraft/net/minecraft/block/BlockRedFlower.java
@@ -1,9 +1,7 @@
 package net.minecraft.block;
 
-public class BlockRedFlower extends BlockFlower
-{
-    public BlockFlower.EnumFlowerColor getBlockType()
-    {
+public class BlockRedFlower extends BlockFlower {
+    public BlockFlower.EnumFlowerColor getBlockType() {
         return BlockFlower.EnumFlowerColor.RED;
     }
 }
