--- ../src-base/minecraft/net/minecraft/block/BlockEmptyDrops.java
+++ ../src-work/minecraft/net/minecraft/block/BlockEmptyDrops.java
@@ -1,11 +1,12 @@
 package net.minecraft.block;
 
-import java.util.Random;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
 
+import java.util.Random;
+
 public class BlockEmptyDrops extends Block
 {
     public BlockEmptyDrops(Material materialIn)
