--- ../src-base/minecraft/net/minecraft/block/BlockChest.java
+++ ../src-work/minecraft/net/minecraft/block/BlockChest.java
@@ -1,6 +1,5 @@
 package net.minecraft.block;
 
-import javax.annotation.Nullable;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyDirection;
@@ -20,11 +19,7 @@
 import net.minecraft.stats.StatList;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.tileentity.TileEntityChest;
-import net.minecraft.util.EnumBlockRenderType;
-import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumHand;
-import net.minecraft.util.Mirror;
-import net.minecraft.util.Rotation;
+import net.minecraft.util.*;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
@@ -34,6 +29,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public class BlockChest extends BlockContainer
 {
     public static final PropertyDirection FACING = BlockHorizontal.FACING;
@@ -270,7 +267,6 @@
         for (EnumFacing enumfacing1 : EnumFacing.Plane.HORIZONTAL)
         {
             IBlockState iblockstate = worldIn.getBlockState(pos.offset(enumfacing1));
-
             if (iblockstate.getBlock() == this)
             {
                 return state;
@@ -470,7 +466,7 @@
 
                     if (block == this)
                     {
-                        if (!allowBlocking && this.isBlocked(worldIn, blockpos)) // Forge: fix MC-99321
+                        if (!allowBlocking && this.isBlocked(worldIn, blockpos)) // Forge and Paper: fix MC-99321
                         {
                             return null;
                         }
