--- ../src-base/minecraft/net/minecraft/block/BlockBed.java
+++ ../src-work/minecraft/net/minecraft/block/BlockBed.java
@@ -1,7 +1,5 @@
 package net.minecraft.block;
 
-import java.util.Random;
-import javax.annotation.Nullable;
 import net.minecraft.block.material.EnumPushReaction;
 import net.minecraft.block.material.MapColor;
 import net.minecraft.block.material.Material;
@@ -14,7 +12,6 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayer;
-import net.minecraft.init.Biomes;
 import net.minecraft.init.Items;
 import net.minecraft.item.EnumDyeColor;
 import net.minecraft.item.Item;
@@ -36,22 +33,25 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Random;
+
 public class BlockBed extends BlockHorizontal implements ITileEntityProvider
 {
-    public static final PropertyEnum<BlockBed.EnumPartType> PART = PropertyEnum.<BlockBed.EnumPartType>create("part", BlockBed.EnumPartType.class);
+    public static final PropertyEnum<EnumPartType> PART = PropertyEnum.<EnumPartType>create("part", EnumPartType.class);
     public static final PropertyBool OCCUPIED = PropertyBool.create("occupied");
     protected static final AxisAlignedBB BED_AABB = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.5625D, 1.0D);
 
     public BlockBed()
     {
         super(Material.CLOTH);
-        this.setDefaultState(this.blockState.getBaseState().withProperty(PART, BlockBed.EnumPartType.FOOT).withProperty(OCCUPIED, Boolean.valueOf(false)));
+        this.setDefaultState(this.blockState.getBaseState().withProperty(PART, EnumPartType.FOOT).withProperty(OCCUPIED, Boolean.valueOf(false)));
         this.hasTileEntity = true;
     }
 
     public MapColor getMapColor(IBlockState state, IBlockAccess worldIn, BlockPos pos)
     {
-        if (state.getValue(PART) == BlockBed.EnumPartType.FOOT)
+        if (state.getValue(PART) == EnumPartType.FOOT)
         {
             TileEntity tileentity = worldIn.getTileEntity(pos);
 
@@ -73,7 +73,7 @@
         }
         else
         {
-            if (state.getValue(PART) != BlockBed.EnumPartType.HEAD)
+            if (state.getValue(PART) != EnumPartType.HEAD)
             {
                 pos = pos.offset((EnumFacing)state.getValue(FACING));
                 state = worldIn.getBlockState(pos);
@@ -194,7 +194,7 @@
     {
         EnumFacing enumfacing = (EnumFacing)state.getValue(FACING);
 
-        if (state.getValue(PART) == BlockBed.EnumPartType.FOOT)
+        if (state.getValue(PART) == EnumPartType.FOOT)
         {
             if (worldIn.getBlockState(pos.offset(enumfacing)).getBlock() != this)
             {
@@ -214,7 +214,7 @@
 
     public Item getItemDropped(IBlockState state, Random rand, int fortune)
     {
-        return state.getValue(PART) == BlockBed.EnumPartType.FOOT ? Items.AIR : Items.BED;
+        return state.getValue(PART) == EnumPartType.FOOT ? Items.AIR : Items.BED;
     }
 
     public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
@@ -229,9 +229,51 @@
     }
 
     @Nullable
-    public static BlockPos getSafeExitLocation(World worldIn, BlockPos pos, int tries)
+    public static BlockPos getSafeExitLocation(World worldIn, BlockPos pos, int i)
     {
         EnumFacing enumfacing = (EnumFacing)worldIn.getBlockState(pos).getValue(FACING);
+        // Paper - replace whole method
+        int radius = worldIn.paperConfig.bedSearchRadius;
+        for (int r = 1; r <= radius; r++) {
+            int x = -r;
+            int z = r;
+            // Iterates the edge of half of the box; then negates for other half.
+            while (x <= r && z > -r) {
+                for (int y = -1; y <= 1; y++) {
+                    BlockPos pos1 = pos.add(x, y, z);
+                    if (isSafeRespawn(worldIn, pos1)) {
+                        if (i-- <= 0) {
+                            return pos1;
+                        }
+                    }
+                    pos1 = pos.add(-x, y, -z);
+                    if (isSafeRespawn(worldIn, pos1)) {
+                        if (i-- <= 0) {
+                            return pos1;
+                        }
+                    }
+                    pos1 = pos.add(enumfacing.getFrontOffsetX() + x, y, enumfacing.getFrontOffsetZ() + z);
+                    if (isSafeRespawn(worldIn, pos1)) {
+                        if (i-- <= 0) {
+                            return pos1;
+                        }
+                    }
+                    pos1 = pos.add(enumfacing.getFrontOffsetX() - x, y, enumfacing.getFrontOffsetZ() - z);
+                    if (isSafeRespawn(worldIn, pos1)) {
+                        if (i-- <= 0) {
+                            return pos1;
+                        }
+                     }
+                }
+                if (x < r) {
+                    x++;
+                } else {
+                    z--;
+                }
+            }
+        }
+
+        return null; /* // Paper comment out
         int i = pos.getX();
         int j = pos.getY();
         int k = pos.getZ();
@@ -262,9 +304,13 @@
             }
         }
 
-        return null;
+        return null;*/ // Paper
     }
 
+    protected static boolean isSafeRespawn(World world, BlockPos blockposition) { // Paper - OBFHELPER + behavior improvement
+        return hasRoomForPlayer(world, blockposition) && world.getBlockState(blockposition.down()).getMaterial().isBuildable(); // Paper - ensure solid block
+    }
+
     protected static boolean hasRoomForPlayer(World worldIn, BlockPos pos)
     {
         return worldIn.getBlockState(pos.down()).isTopSolid() && !worldIn.getBlockState(pos).getMaterial().isSolid() && !worldIn.getBlockState(pos.up()).getMaterial().isSolid();
@@ -272,7 +318,7 @@
 
     public void dropBlockAsItemWithChance(World worldIn, BlockPos pos, IBlockState state, float chance, int fortune)
     {
-        if (state.getValue(PART) == BlockBed.EnumPartType.HEAD)
+        if (state.getValue(PART) == EnumPartType.HEAD)
         {
             TileEntity tileentity = worldIn.getTileEntity(pos);
             EnumDyeColor enumdyecolor = tileentity instanceof TileEntityBed ? ((TileEntityBed)tileentity).getColor() : EnumDyeColor.RED;
@@ -300,7 +346,7 @@
     {
         BlockPos blockpos = pos;
 
-        if (state.getValue(PART) == BlockBed.EnumPartType.FOOT)
+        if (state.getValue(PART) == EnumPartType.FOOT)
         {
             blockpos = pos.offset((EnumFacing)state.getValue(FACING));
         }
@@ -312,7 +358,7 @@
 
     public void onBlockHarvested(World worldIn, BlockPos pos, IBlockState state, EntityPlayer player)
     {
-        if (player.capabilities.isCreativeMode && state.getValue(PART) == BlockBed.EnumPartType.FOOT)
+        if (player.capabilities.isCreativeMode && state.getValue(PART) == EnumPartType.FOOT)
         {
             BlockPos blockpos = pos.offset((EnumFacing)state.getValue(FACING));
 
@@ -325,7 +371,7 @@
 
     public void harvestBlock(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
     {
-        if (state.getValue(PART) == BlockBed.EnumPartType.HEAD && te instanceof TileEntityBed)
+        if (state.getValue(PART) == EnumPartType.HEAD && te instanceof TileEntityBed)
         {
             TileEntityBed tileentitybed = (TileEntityBed)te;
             ItemStack itemstack = tileentitybed.getItemStack();
@@ -346,12 +392,12 @@
     public IBlockState getStateFromMeta(int meta)
     {
         EnumFacing enumfacing = EnumFacing.getHorizontal(meta);
-        return (meta & 8) > 0 ? this.getDefaultState().withProperty(PART, BlockBed.EnumPartType.HEAD).withProperty(FACING, enumfacing).withProperty(OCCUPIED, Boolean.valueOf((meta & 4) > 0)) : this.getDefaultState().withProperty(PART, BlockBed.EnumPartType.FOOT).withProperty(FACING, enumfacing);
+        return (meta & 8) > 0 ? this.getDefaultState().withProperty(PART, EnumPartType.HEAD).withProperty(FACING, enumfacing).withProperty(OCCUPIED, Boolean.valueOf((meta & 4) > 0)) : this.getDefaultState().withProperty(PART, EnumPartType.FOOT).withProperty(FACING, enumfacing);
     }
 
     public IBlockState getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos)
     {
-        if (state.getValue(PART) == BlockBed.EnumPartType.FOOT)
+        if (state.getValue(PART) == EnumPartType.FOOT)
         {
             IBlockState iblockstate = worldIn.getBlockState(pos.offset((EnumFacing)state.getValue(FACING)));
 
@@ -379,7 +425,7 @@
         int i = 0;
         i = i | ((EnumFacing)state.getValue(FACING)).getHorizontalIndex();
 
-        if (state.getValue(PART) == BlockBed.EnumPartType.HEAD)
+        if (state.getValue(PART) == EnumPartType.HEAD)
         {
             i |= 8;
 
