--- ../src-base/minecraft/net/minecraft/block/BlockGravel.java
+++ ../src-work/minecraft/net/minecraft/block/BlockGravel.java
@@ -1,6 +1,5 @@
 package net.minecraft.block;
 
-import java.util.Random;
 import net.minecraft.block.material.MapColor;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Items;
@@ -10,26 +9,23 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BlockGravel extends BlockFalling
-{
-    public Item getItemDropped(IBlockState state, Random rand, int fortune)
-    {
-        if (fortune > 3)
-        {
+import java.util.Random;
+
+public class BlockGravel extends BlockFalling {
+    public Item getItemDropped(IBlockState state, Random rand, int fortune) {
+        if (fortune > 3) {
             fortune = 3;
         }
 
         return rand.nextInt(10 - fortune * 3) == 0 ? Items.FLINT : super.getItemDropped(state, rand, fortune);
     }
 
-    public MapColor getMapColor(IBlockState state, IBlockAccess worldIn, BlockPos pos)
-    {
+    public MapColor getMapColor(IBlockState state, IBlockAccess worldIn, BlockPos pos) {
         return MapColor.STONE;
     }
 
     @SideOnly(Side.CLIENT)
-    public int getDustColor(IBlockState state)
-    {
+    public int getDustColor(IBlockState state) {
         return -8356741;
     }
 }
