--- ../src-base/minecraft/net/minecraft/block/BlockGlass.java
+++ ../src-work/minecraft/net/minecraft/block/BlockGlass.java
@@ -1,6 +1,5 @@
 package net.minecraft.block;
 
-import java.util.Random;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
@@ -8,32 +7,28 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BlockGlass extends BlockBreakable
-{
-    public BlockGlass(Material materialIn, boolean ignoreSimilarity)
-    {
+import java.util.Random;
+
+public class BlockGlass extends BlockBreakable {
+    public BlockGlass(Material materialIn, boolean ignoreSimilarity) {
         super(materialIn, ignoreSimilarity);
         this.setCreativeTab(CreativeTabs.BUILDING_BLOCKS);
     }
 
-    public int quantityDropped(Random random)
-    {
+    public int quantityDropped(Random random) {
         return 0;
     }
 
     @SideOnly(Side.CLIENT)
-    public BlockRenderLayer getBlockLayer()
-    {
+    public BlockRenderLayer getBlockLayer() {
         return BlockRenderLayer.CUTOUT;
     }
 
-    public boolean isFullCube(IBlockState state)
-    {
+    public boolean isFullCube(IBlockState state) {
         return false;
     }
 
-    protected boolean canSilkHarvest()
-    {
+    protected boolean canSilkHarvest() {
         return true;
     }
 }
