--- ../src-base/minecraft/net/minecraft/block/BlockButtonStone.java
+++ ../src-work/minecraft/net/minecraft/block/BlockButtonStone.java
@@ -1,12 +1,13 @@
 package net.minecraft.block;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.util.SoundCategory;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+
 public class BlockButtonStone extends BlockButton
 {
     protected BlockButtonStone()
