--- ../src-base/minecraft/net/minecraft/block/BlockHalfStoneSlab.java
+++ ../src-work/minecraft/net/minecraft/block/BlockHalfStoneSlab.java
@@ -1,9 +1,7 @@
 package net.minecraft.block;
 
-public class BlockHalfStoneSlab extends BlockStoneSlab
-{
-    public boolean isDouble()
-    {
+public class BlockHalfStoneSlab extends BlockStoneSlab {
+    public boolean isDouble() {
         return false;
     }
 }
