--- ../src-base/minecraft/net/minecraft/potion/PotionHealth.java
+++ ../src-work/minecraft/net/minecraft/potion/PotionHealth.java
@@ -1,19 +1,15 @@
 package net.minecraft.potion;
 
-public class PotionHealth extends Potion
-{
-    public PotionHealth(boolean isBadEffectIn, int liquidColorIn)
-    {
+public class PotionHealth extends Potion {
+    public PotionHealth(boolean isBadEffectIn, int liquidColorIn) {
         super(isBadEffectIn, liquidColorIn);
     }
 
-    public boolean isInstant()
-    {
+    public boolean isInstant() {
         return true;
     }
 
-    public boolean isReady(int duration, int amplifier)
-    {
+    public boolean isReady(int duration, int amplifier) {
         return duration >= 1;
     }
 }
