--- ../src-base/minecraft/net/minecraft/potion/PotionType.java
+++ ../src-work/minecraft/net/minecraft/potion/PotionType.java
@@ -2,12 +2,13 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.UnmodifiableIterator;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.init.MobEffects;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.registry.RegistryNamespacedDefaultedByKey;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class PotionType extends net.minecraftforge.registries.IForgeRegistryEntry.Impl<PotionType>
 {
     @Deprecated // unused
