--- ../src-base/minecraft/net/minecraft/potion/PotionHealthBoost.java
+++ ../src-work/minecraft/net/minecraft/potion/PotionHealthBoost.java
@@ -3,19 +3,15 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.ai.attributes.AbstractAttributeMap;
 
-public class PotionHealthBoost extends Potion
-{
-    public PotionHealthBoost(boolean isBadEffectIn, int liquidColorIn)
-    {
+public class PotionHealthBoost extends Potion {
+    public PotionHealthBoost(boolean isBadEffectIn, int liquidColorIn) {
         super(isBadEffectIn, liquidColorIn);
     }
 
-    public void removeAttributesModifiersFromEntity(EntityLivingBase entityLivingBaseIn, AbstractAttributeMap attributeMapIn, int amplifier)
-    {
+    public void removeAttributesModifiersFromEntity(EntityLivingBase entityLivingBaseIn, AbstractAttributeMap attributeMapIn, int amplifier) {
         super.removeAttributesModifiersFromEntity(entityLivingBaseIn, attributeMapIn, amplifier);
 
-        if (entityLivingBaseIn.getHealth() > entityLivingBaseIn.getMaxHealth())
-        {
+        if (entityLivingBaseIn.getHealth() > entityLivingBaseIn.getMaxHealth()) {
             entityLivingBaseIn.setHealth(entityLivingBaseIn.getMaxHealth());
         }
     }
