--- ../src-base/minecraft/net/minecraftforge/items/wrapper/EntityArmorInvWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/items/wrapper/EntityArmorInvWrapper.java
@@ -27,10 +27,8 @@
  * Exposes the armor inventory of an {@link EntityLivingBase} as an {@link IItemHandler} using {@link EntityLivingBase#getItemStackFromSlot} and
  * {@link EntityLivingBase#setItemStackToSlot}.
  */
-public class EntityArmorInvWrapper extends EntityEquipmentInvWrapper
-{
-    public EntityArmorInvWrapper(final EntityLivingBase entity)
-    {
+public class EntityArmorInvWrapper extends EntityEquipmentInvWrapper {
+    public EntityArmorInvWrapper(final EntityLivingBase entity) {
         super(entity, EntityEquipmentSlot.Type.ARMOR);
     }
 }
