--- ../src-base/minecraft/net/minecraftforge/items/wrapper/EntityHandsInvWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/items/wrapper/EntityHandsInvWrapper.java
@@ -27,10 +27,8 @@
  * Exposes the hands inventory of an {@link EntityLivingBase} as an {@link IItemHandler} using {@link EntityLivingBase#getItemStackFromSlot} and
  * {@link EntityLivingBase#setItemStackToSlot}.
  */
-public class EntityHandsInvWrapper extends EntityEquipmentInvWrapper
-{
-    public EntityHandsInvWrapper(EntityLivingBase entity)
-    {
+public class EntityHandsInvWrapper extends EntityEquipmentInvWrapper {
+    public EntityHandsInvWrapper(EntityLivingBase entity) {
         super(entity, EntityEquipmentSlot.Type.HAND);
     }
 }
