--- ../src-base/minecraft/net/minecraftforge/items/wrapper/PlayerInvWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/items/wrapper/PlayerInvWrapper.java
@@ -21,10 +21,8 @@
 
 import net.minecraft.entity.player.InventoryPlayer;
 
-public class PlayerInvWrapper extends CombinedInvWrapper
-{
-    public PlayerInvWrapper(InventoryPlayer inv)
-    {
+public class PlayerInvWrapper extends CombinedInvWrapper {
+    public PlayerInvWrapper(InventoryPlayer inv) {
         super(new PlayerMainInvWrapper(inv), new PlayerArmorInvWrapper(inv), new PlayerOffhandInvWrapper(inv));
     }
 }
