--- ../src-base/minecraft/net/minecraftforge/items/wrapper/PlayerOffhandInvWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/items/wrapper/PlayerOffhandInvWrapper.java
@@ -21,10 +21,8 @@
 
 import net.minecraft.entity.player.InventoryPlayer;
 
-public class PlayerOffhandInvWrapper extends RangedWrapper
-{
-    public PlayerOffhandInvWrapper(InventoryPlayer inv)
-    {
+public class PlayerOffhandInvWrapper extends RangedWrapper {
+    public PlayerOffhandInvWrapper(InventoryPlayer inv) {
         super(new InvWrapper(inv), inv.mainInventory.size() + inv.armorInventory.size(),
                 inv.mainInventory.size() + inv.armorInventory.size() + inv.offHandInventory.size());
     }
