--- ../src-base/minecraft/net/minecraftforge/oredict/OreIngredient.java
+++ ../src-work/minecraft/net/minecraftforge/oredict/OreIngredient.java
@@ -19,9 +19,6 @@
 
 package net.minecraftforge.oredict;
 
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 import it.unimi.dsi.fastutil.ints.IntArrayList;
 import it.unimi.dsi.fastutil.ints.IntComparators;
 import it.unimi.dsi.fastutil.ints.IntList;
@@ -31,28 +28,26 @@
 import net.minecraft.item.crafting.Ingredient;
 import net.minecraft.util.NonNullList;
 
-public class OreIngredient extends Ingredient
-{
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class OreIngredient extends Ingredient {
     private NonNullList<ItemStack> ores;
     private IntList itemIds = null;
     private ItemStack[] array = null;
     private int lastSizeA = -1, lastSizeL = -1;
 
-    public OreIngredient(String ore)
-    {
+    public OreIngredient(String ore) {
         super(0);
         ores = OreDictionary.getOres(ore);
     }
 
     @Override
     @Nonnull
-    public ItemStack[] getMatchingStacks()
-    {
-        if (array == null || this.lastSizeA != ores.size())
-        {
+    public ItemStack[] getMatchingStacks() {
+        if (array == null || this.lastSizeA != ores.size()) {
             NonNullList<ItemStack> lst = NonNullList.create();
-            for (ItemStack itemstack : this.ores)
-            {
+            for (ItemStack itemstack : this.ores) {
                 if (itemstack.getMetadata() == OreDictionary.WILDCARD_VALUE)
                     itemstack.getItem().getSubItems(CreativeTabs.SEARCH, lst);
                 else
@@ -67,23 +62,17 @@
 
     @Override
     @Nonnull
-    public IntList getValidItemStacksPacked()
-    {
-        if (this.itemIds == null || this.lastSizeL != ores.size())
-        {
+    public IntList getValidItemStacksPacked() {
+        if (this.itemIds == null || this.lastSizeL != ores.size()) {
             this.itemIds = new IntArrayList(this.ores.size());
 
-            for (ItemStack itemstack : this.ores)
-            {
-                if (itemstack.getMetadata() == OreDictionary.WILDCARD_VALUE)
-                {
+            for (ItemStack itemstack : this.ores) {
+                if (itemstack.getMetadata() == OreDictionary.WILDCARD_VALUE) {
                     NonNullList<ItemStack> lst = NonNullList.create();
                     itemstack.getItem().getSubItems(CreativeTabs.SEARCH, lst);
                     for (ItemStack item : lst)
                         this.itemIds.add(RecipeItemHelper.pack(item));
-                }
-                else
-                {
+                } else {
                     this.itemIds.add(RecipeItemHelper.pack(itemstack));
                 }
             }
@@ -97,8 +86,7 @@
 
 
     @Override
-    public boolean apply(@Nullable ItemStack input)
-    {
+    public boolean apply(@Nullable ItemStack input) {
         if (input == null)
             return false;
 
@@ -110,15 +98,13 @@
     }
 
     @Override
-    protected void invalidate()
-    {
+    protected void invalidate() {
         this.itemIds = null;
         this.array = null;
     }
 
     @Override
-    public boolean isSimple()
-    {
+    public boolean isSimple() {
         return true;
     }
 }
