--- ../src-base/minecraft/net/minecraft/command/server/CommandListPlayers.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandListPlayers.java
@@ -8,27 +8,22 @@
 import net.minecraft.util.text.TextComponentString;
 import net.minecraft.util.text.TextComponentTranslation;
 
-public class CommandListPlayers extends CommandBase
-{
-    public String getName()
-    {
+public class CommandListPlayers extends CommandBase {
+    public String getName() {
         return "list";
     }
 
-    public int getRequiredPermissionLevel()
-    {
+    public int getRequiredPermissionLevel() {
         return 0;
     }
 
-    public String getUsage(ICommandSender sender)
-    {
+    public String getUsage(ICommandSender sender) {
         return "commands.players.usage";
     }
 
-    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
-    {
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
         int i = server.getCurrentPlayerCount();
-        sender.sendMessage(new TextComponentTranslation("commands.players.list", new Object[] {i, server.getMaxPlayers()}));
+        sender.sendMessage(new TextComponentTranslation("commands.players.list", new Object[]{i, server.getMaxPlayers()}));
         sender.sendMessage(new TextComponentString(server.getPlayerList().getFormattedListOfPlayers(args.length > 0 && "uuids".equalsIgnoreCase(args[0]))));
         sender.setCommandStat(CommandResultStats.Type.QUERY_RESULT, i);
     }
