--- ../src-base/minecraft/net/minecraft/command/server/CommandOp.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandOp.java
@@ -2,9 +2,6 @@
 
 import com.google.common.collect.Lists;
 import com.mojang.authlib.GameProfile;
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
@@ -12,6 +9,10 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandOp extends CommandBase
 {
     public String getName()
