--- ../src-base/minecraft/net/minecraft/command/server/CommandSummon.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandSummon.java
@@ -1,8 +1,5 @@
 package net.minecraft.command.server;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
@@ -22,6 +19,10 @@
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.storage.AnvilChunkLoader;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandSummon extends CommandBase
 {
     public String getName()
