--- ../src-base/minecraft/net/minecraft/command/server/CommandStop.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandStop.java
@@ -5,22 +5,17 @@
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
 
-public class CommandStop extends CommandBase
-{
-    public String getName()
-    {
+public class CommandStop extends CommandBase {
+    public String getName() {
         return "stop";
     }
 
-    public String getUsage(ICommandSender sender)
-    {
+    public String getUsage(ICommandSender sender) {
         return "commands.stop.usage";
     }
 
-    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
-    {
-        if (server.worlds != null)
-        {
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
+        if (server.worlds != null) {
             notifyCommandListener(sender, this, "commands.stop.start", new Object[0]);
         }
 
