--- ../src-base/minecraft/net/minecraft/command/server/CommandEmote.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandEmote.java
@@ -1,7 +1,5 @@
 package net.minecraft.command.server;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
@@ -12,6 +10,9 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentTranslation;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class CommandEmote extends CommandBase
 {
     public String getName()
