--- ../src-base/minecraft/net/minecraft/item/crafting/Ingredient.java
+++ ../src-work/minecraft/net/minecraft/item/crafting/Ingredient.java
@@ -4,11 +4,12 @@
 import it.unimi.dsi.fastutil.ints.IntArrayList;
 import it.unimi.dsi.fastutil.ints.IntComparators;
 import it.unimi.dsi.fastutil.ints.IntList;
-import javax.annotation.Nullable;
 import net.minecraft.client.util.RecipeItemHelper;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 
+import javax.annotation.Nullable;
+
 public class Ingredient implements Predicate<ItemStack>
 {
     //Because Mojang caches things... we need to invalidate them.. so... here we go..
@@ -20,7 +21,7 @@
             return p_apply_1_.isEmpty();
         }
     };
-    private final ItemStack[] matchingStacks;
+    public final ItemStack[] matchingStacks;
     private final ItemStack[] matchingStacksExploded;
     private IntList matchingStacksPacked;
     private final boolean isSimple;
