--- ../src-base/minecraft/net/minecraft/util/BitArray.java
+++ ../src-work/minecraft/net/minecraft/util/BitArray.java
@@ -12,7 +12,6 @@
 
     public BitArray(int bitsPerEntryIn, int arraySizeIn)
     {
-        Validate.inclusiveBetween(1L, 32L, (long)bitsPerEntryIn);
         this.arraySize = arraySizeIn;
         this.bitsPerEntry = bitsPerEntryIn;
         this.maxEntryValue = (1L << bitsPerEntryIn) - 1L;
@@ -21,8 +20,6 @@
 
     public void setAt(int index, int value)
     {
-        Validate.inclusiveBetween(0L, (long)(this.arraySize - 1), (long)index);
-        Validate.inclusiveBetween(0L, this.maxEntryValue, (long)value);
         int i = index * this.bitsPerEntry;
         int j = i / 64;
         int k = ((index + 1) * this.bitsPerEntry - 1) / 64;
@@ -39,7 +36,6 @@
 
     public int getAt(int index)
     {
-        Validate.inclusiveBetween(0L, (long)(this.arraySize - 1), (long)index);
         int i = index * this.bitsPerEntry;
         int j = i / 64;
         int k = ((index + 1) * this.bitsPerEntry - 1) / 64;
