--- ../src-base/minecraft/net/minecraft/util/BlockRenderLayer.java
+++ ../src-work/minecraft/net/minecraft/util/BlockRenderLayer.java
@@ -1,7 +1,6 @@
 package net.minecraft.util;
 
-public enum BlockRenderLayer
-{
+public enum BlockRenderLayer {
     SOLID("Solid"),
     CUTOUT_MIPPED("Mipped Cutout"),
     CUTOUT("Cutout"),
@@ -9,13 +8,11 @@
 
     private final String layerName;
 
-    private BlockRenderLayer(String layerNameIn)
-    {
+    private BlockRenderLayer(String layerNameIn) {
         this.layerName = layerNameIn;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return this.layerName;
     }
 }
