--- ../src-base/minecraft/net/minecraft/util/CombatTracker.java
+++ ../src-work/minecraft/net/minecraft/util/CombatTracker.java
@@ -1,8 +1,6 @@
 package net.minecraft.util;
 
 import com.google.common.collect.Lists;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
@@ -13,6 +11,9 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentTranslation;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class CombatTracker
 {
     private final List<CombatEntry> combatEntries = Lists.<CombatEntry>newArrayList();
