--- ../src-base/minecraft/net/minecraft/util/CombatEntry.java
+++ ../src-work/minecraft/net/minecraft/util/CombatEntry.java
@@ -1,11 +1,11 @@
 package net.minecraft.util;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.text.ITextComponent;
 
-public class CombatEntry
-{
+import javax.annotation.Nullable;
+
+public class CombatEntry {
     private final DamageSource damageSrc;
     private final int time;
     private final float damage;
@@ -13,8 +13,7 @@
     private final String fallSuffix;
     private final float fallDistance;
 
-    public CombatEntry(DamageSource damageSrcIn, int timeIn, float healthAmount, float damageAmount, String fallSuffixIn, float fallDistanceIn)
-    {
+    public CombatEntry(DamageSource damageSrcIn, int timeIn, float healthAmount, float damageAmount, String fallSuffixIn, float fallDistanceIn) {
         this.damageSrc = damageSrcIn;
         this.time = timeIn;
         this.damage = damageAmount;
@@ -23,35 +22,29 @@
         this.fallDistance = fallDistanceIn;
     }
 
-    public DamageSource getDamageSrc()
-    {
+    public DamageSource getDamageSrc() {
         return this.damageSrc;
     }
 
-    public float getDamage()
-    {
+    public float getDamage() {
         return this.damage;
     }
 
-    public boolean isLivingDamageSrc()
-    {
+    public boolean isLivingDamageSrc() {
         return this.damageSrc.getTrueSource() instanceof EntityLivingBase;
     }
 
     @Nullable
-    public String getFallSuffix()
-    {
+    public String getFallSuffix() {
         return this.fallSuffix;
     }
 
     @Nullable
-    public ITextComponent getDamageSrcDisplayName()
-    {
+    public ITextComponent getDamageSrcDisplayName() {
         return this.getDamageSrc().getTrueSource() == null ? null : this.getDamageSrc().getTrueSource().getDisplayName();
     }
 
-    public float getDamageAmount()
-    {
+    public float getDamageAmount() {
         return this.damageSrc == DamageSource.OUT_OF_WORLD ? Float.MAX_VALUE : this.fallDistance;
     }
 }
