--- ../src-base/minecraft/net/minecraft/util/EnumHandSide.java
+++ ../src-work/minecraft/net/minecraft/util/EnumHandSide.java
@@ -5,26 +5,22 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public enum EnumHandSide
-{
+public enum EnumHandSide {
     LEFT(new TextComponentTranslation("options.mainHand.left", new Object[0])),
     RIGHT(new TextComponentTranslation("options.mainHand.right", new Object[0]));
 
     private final ITextComponent handName;
 
-    private EnumHandSide(ITextComponent nameIn)
-    {
+    private EnumHandSide(ITextComponent nameIn) {
         this.handName = nameIn;
     }
 
     @SideOnly(Side.CLIENT)
-    public EnumHandSide opposite()
-    {
+    public EnumHandSide opposite() {
         return this == LEFT ? RIGHT : LEFT;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return this.handName.getUnformattedText();
     }
 }
