--- ../src-base/minecraft/net/minecraft/util/IThreadListener.java
+++ ../src-work/minecraft/net/minecraft/util/IThreadListener.java
@@ -2,8 +2,7 @@
 
 import com.google.common.util.concurrent.ListenableFuture;
 
-public interface IThreadListener
-{
+public interface IThreadListener {
     ListenableFuture<Object> addScheduledTask(Runnable runnableToSchedule);
 
     boolean isCallingFromMinecraftThread();
