--- ../src-base/minecraft/net/minecraft/util/LazyLoadBase.java
+++ ../src-work/minecraft/net/minecraft/util/LazyLoadBase.java
@@ -1,16 +1,13 @@
 package net.minecraft.util;
 
-public abstract class LazyLoadBase<T>
-{
+public abstract class LazyLoadBase<T> {
     private T value;
     private boolean isLoaded;
 
-    public T getValue()
-    {
-        if (!this.isLoaded)
-        {
+    public T getValue() {
+        if (!this.isLoaded) {
             this.isLoaded = true;
-            this.value = (T)this.load();
+            this.value = (T) this.load();
         }
 
         return this.value;
