--- ../src-base/minecraft/net/minecraft/util/LoggingPrintStream.java
+++ ../src-work/minecraft/net/minecraft/util/LoggingPrintStream.java
@@ -1,10 +1,11 @@
 package net.minecraft.util;
 
-import java.io.OutputStream;
-import java.io.PrintStream;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.io.OutputStream;
+import java.io.PrintStream;
+
 public class LoggingPrintStream extends PrintStream
 {
     protected static final Logger LOGGER = LogManager.getLogger();
