--- ../src-base/minecraft/net/minecraft/util/Rotation.java
+++ ../src-work/minecraft/net/minecraft/util/Rotation.java
@@ -1,28 +1,32 @@
 package net.minecraft.util;
 
-public enum Rotation
-{
+public enum Rotation {
     NONE("rotate_0"),
     CLOCKWISE_90("rotate_90"),
     CLOCKWISE_180("rotate_180"),
     COUNTERCLOCKWISE_90("rotate_270");
 
-    private final String name;
     private static final String[] rotationNames = new String[values().length];
 
-    private Rotation(String nameIn)
-    {
+    static {
+        int i = 0;
+
+        for (Rotation rotation : values()) {
+            rotationNames[i++] = rotation.name;
+        }
+    }
+
+    private final String name;
+
+    private Rotation(String nameIn) {
         this.name = nameIn;
     }
 
-    public Rotation add(Rotation rotation)
-    {
-        switch (rotation)
-        {
+    public Rotation add(Rotation rotation) {
+        switch (rotation) {
             case CLOCKWISE_180:
 
-                switch (this)
-                {
+                switch (this) {
                     case NONE:
                         return CLOCKWISE_180;
                     case CLOCKWISE_90:
@@ -35,8 +39,7 @@
 
             case COUNTERCLOCKWISE_90:
 
-                switch (this)
-                {
+                switch (this) {
                     case NONE:
                         return COUNTERCLOCKWISE_90;
                     case CLOCKWISE_90:
@@ -49,8 +52,7 @@
 
             case CLOCKWISE_90:
 
-                switch (this)
-                {
+                switch (this) {
                     case NONE:
                         return CLOCKWISE_90;
                     case CLOCKWISE_90:
@@ -66,16 +68,11 @@
         }
     }
 
-    public EnumFacing rotate(EnumFacing facing)
-    {
-        if (facing.getAxis() == EnumFacing.Axis.Y)
-        {
+    public EnumFacing rotate(EnumFacing facing) {
+        if (facing.getAxis() == EnumFacing.Axis.Y) {
             return facing;
-        }
-        else
-        {
-            switch (this)
-            {
+        } else {
+            switch (this) {
                 case CLOCKWISE_90:
                     return facing.rotateY();
                 case CLOCKWISE_180:
@@ -88,10 +85,8 @@
         }
     }
 
-    public int rotate(int p_185833_1_, int p_185833_2_)
-    {
-        switch (this)
-        {
+    public int rotate(int p_185833_1_, int p_185833_2_) {
+        switch (this) {
             case CLOCKWISE_90:
                 return (p_185833_1_ + p_185833_2_ / 4) % p_185833_2_;
             case CLOCKWISE_180:
@@ -102,14 +97,4 @@
                 return p_185833_1_;
         }
     }
-
-    static
-    {
-        int i = 0;
-
-        for (Rotation rotation : values())
-        {
-            rotationNames[i++] = rotation.name;
-        }
-    }
 }
