--- ../src-base/minecraft/net/minecraft/util/Session.java
+++ ../src-work/minecraft/net/minecraft/util/Session.java
@@ -3,16 +3,16 @@
 import com.google.common.collect.Maps;
 import com.mojang.authlib.GameProfile;
 import com.mojang.util.UUIDTypeAdapter;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
 import java.util.Locale;
 import java.util.Map;
 import java.util.UUID;
-import javax.annotation.Nullable;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class Session
-{
+public class Session {
     private final String username;
     private final String playerID;
     private final String token;
@@ -20,10 +20,8 @@
     /** Forge: Cache of the local session's GameProfile properties. */
     private com.mojang.authlib.properties.PropertyMap properties;
 
-    public Session(String usernameIn, String playerIDIn, String tokenIn, String sessionTypeIn)
-    {
-        if (usernameIn == null || usernameIn.isEmpty())
-        {
+    public Session(String usernameIn, String playerIDIn, String tokenIn, String sessionTypeIn) {
+        if (usernameIn == null || usernameIn.isEmpty()) {
             usernameIn = "MissingName";
             playerIDIn = tokenIn = "NotValid";
             org.apache.logging.log4j.Logger logger = net.minecraftforge.fml.common.FMLLog.log;
@@ -42,80 +40,67 @@
         this.sessionType = Session.Type.setSessionType(sessionTypeIn);
     }
 
-    public String getSessionID()
-    {
+    public String getSessionID() {
         return "token:" + this.token + ":" + this.playerID;
     }
 
-    public String getPlayerID()
-    {
+    public String getPlayerID() {
         return this.playerID;
     }
 
-    public String getUsername()
-    {
+    public String getUsername() {
         return this.username;
     }
 
-    public String getToken()
-    {
+    public String getToken() {
         return this.token;
     }
 
-    public GameProfile getProfile()
-    {
-        try
-        {
+    public GameProfile getProfile() {
+        try {
             UUID uuid = UUIDTypeAdapter.fromString(this.getPlayerID());
             GameProfile ret = new GameProfile(uuid, this.getUsername());    //Forge: Adds cached GameProfile properties to returned GameProfile.
-            if (properties != null) ret.getProperties().putAll(properties); // Helps to cut down on calls to the session service,
+            if (properties != null)
+                ret.getProperties().putAll(properties); // Helps to cut down on calls to the session service,
             return ret;                                                     // which helps to fix MC-52974.
+        } catch (IllegalArgumentException var2) {
+            return new GameProfile(net.minecraft.entity.player.EntityPlayer.getUUID(new GameProfile((UUID) null, this.getUsername())), this.getUsername());
         }
-        catch (IllegalArgumentException var2)
-        {
-            return new GameProfile(net.minecraft.entity.player.EntityPlayer.getUUID(new GameProfile((UUID)null, this.getUsername())), this.getUsername());
-        }
     }
 
     /* ======================================== FORGE START ===================================== */
     //For internal use only. Modders should never need to use this.
-    public void setProperties(com.mojang.authlib.properties.PropertyMap properties)
-    {
-        if(this.properties == null) this.properties = properties;
+    public void setProperties(com.mojang.authlib.properties.PropertyMap properties) {
+        if (this.properties == null) this.properties = properties;
     }
 
-    public boolean hasCachedProperties()
-    {
+    public boolean hasCachedProperties() {
         return properties != null;
     }
     /* ========================================= FORGE END ====================================== */
 
     @SideOnly(Side.CLIENT)
-    public static enum Type
-    {
+    public static enum Type {
         LEGACY("legacy"),
         MOJANG("mojang");
 
         private static final Map<String, Session.Type> SESSION_TYPES = Maps.<String, Session.Type>newHashMap();
+
+        static {
+            for (Session.Type session$type : values()) {
+                SESSION_TYPES.put(session$type.sessionType, session$type);
+            }
+        }
+
         private final String sessionType;
 
-        private Type(String sessionTypeIn)
-        {
+        private Type(String sessionTypeIn) {
             this.sessionType = sessionTypeIn;
         }
 
         @Nullable
-        public static Session.Type setSessionType(String sessionTypeIn)
-        {
+        public static Session.Type setSessionType(String sessionTypeIn) {
             return SESSION_TYPES.get(sessionTypeIn.toLowerCase(Locale.ROOT));
         }
-
-        static
-        {
-            for (Session.Type session$type : values())
-            {
-                SESSION_TYPES.put(session$type.sessionType, session$type);
-            }
-        }
     }
 }
