--- ../src-base/minecraft/net/minecraft/util/Session.java
+++ ../src-work/minecraft/net/minecraft/util/Session.java
@@ -1,14 +1,16 @@
 package net.minecraft.util;
 
+import red.mohist.Mohist;
 import com.google.common.collect.Maps;
 import com.mojang.authlib.GameProfile;
 import com.mojang.util.UUIDTypeAdapter;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
 import java.util.Locale;
 import java.util.Map;
 import java.util.UUID;
-import javax.annotation.Nullable;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
 public class Session
@@ -26,14 +28,13 @@
         {
             usernameIn = "MissingName";
             playerIDIn = tokenIn = "NotValid";
-            org.apache.logging.log4j.Logger logger = net.minecraftforge.fml.common.FMLLog.log;
-            logger.log(org.apache.logging.log4j.Level.WARN, "=========================================================");
-            logger.log(org.apache.logging.log4j.Level.WARN, "WARNING!! the username was not set for this session, typically");
-            logger.log(org.apache.logging.log4j.Level.WARN, "this means you installed Forge incorrectly. We have set your");
-            logger.log(org.apache.logging.log4j.Level.WARN, "name to \"MissingName\" and your session to nothing. Please");
-            logger.log(org.apache.logging.log4j.Level.WARN, "check your installation and post a console log from the launcher");
-            logger.log(org.apache.logging.log4j.Level.WARN, "when asking for help!");
-            logger.log(org.apache.logging.log4j.Level.WARN, "=========================================================");
+            Mohist.LOGGER.warn("=========================================================");
+            Mohist.LOGGER.warn( "WARNING!! the username was not set for this session, typically");
+            Mohist.LOGGER.warn( "this means you installed Forge incorrectly. We have set your");
+            Mohist.LOGGER.warn( "name to \"MissingName\" and your session to nothing. Please");
+            Mohist.LOGGER.warn( "check your installation and post a console log from the launcher");
+            Mohist.LOGGER.warn( "when asking for help!");
+            Mohist.LOGGER.warn( "=========================================================");
         }
 
         this.username = usernameIn;
