--- ../src-base/minecraft/net/minecraft/util/Tuple.java
+++ ../src-work/minecraft/net/minecraft/util/Tuple.java
@@ -1,23 +1,19 @@
 package net.minecraft.util;
 
-public class Tuple<A, B>
-{
+public class Tuple<A, B> {
     private A a;
     private B b;
 
-    public Tuple(A aIn, B bIn)
-    {
+    public Tuple(A aIn, B bIn) {
         this.a = aIn;
         this.b = bIn;
     }
 
-    public A getFirst()
-    {
+    public A getFirst() {
         return this.a;
     }
 
-    public B getSecond()
-    {
+    public B getSecond() {
         return this.b;
     }
 }
