--- ../src-base/minecraft/net/minecraft/network/LegacyPingHandler.java
+++ ../src-work/minecraft/net/minecraft/network/LegacyPingHandler.java
@@ -5,12 +5,13 @@
 import io.netty.channel.ChannelFutureListener;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelInboundHandlerAdapter;
-import java.net.InetSocketAddress;
-import java.nio.charset.StandardCharsets;
 import net.minecraft.server.MinecraftServer;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+
 public class LegacyPingHandler extends ChannelInboundHandlerAdapter
 {
     private static final Logger LOGGER = LogManager.getLogger();
