--- ../src-base/minecraft/net/minecraft/network/NettyCompressionDecoder.java
+++ ../src-work/minecraft/net/minecraft/network/NettyCompressionDecoder.java
@@ -5,6 +5,7 @@
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.ByteToMessageDecoder;
 import io.netty.handler.codec.DecoderException;
+
 import java.util.List;
 import java.util.zip.DataFormatException;
 import java.util.zip.Inflater;
