--- ../src-base/minecraft/net/minecraft/network/NettyEncryptingDecoder.java
+++ ../src-work/minecraft/net/minecraft/network/NettyEncryptingDecoder.java
@@ -3,9 +3,10 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToMessageDecoder;
-import java.util.List;
+
 import javax.crypto.Cipher;
 import javax.crypto.ShortBufferException;
+import java.util.List;
 
 public class NettyEncryptingDecoder extends MessageToMessageDecoder<ByteBuf>
 {
