--- ../src-base/minecraft/net/minecraft/network/NettyPacketDecoder.java
+++ ../src-work/minecraft/net/minecraft/network/NettyPacketDecoder.java
@@ -3,13 +3,14 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.ByteToMessageDecoder;
-import java.io.IOException;
-import java.util.List;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Marker;
 import org.apache.logging.log4j.MarkerManager;
 
+import java.io.IOException;
+import java.util.List;
+
 public class NettyPacketDecoder extends ByteToMessageDecoder
 {
     private static final Logger LOGGER = LogManager.getLogger();
