--- ../src-base/minecraft/net/minecraft/item/EnumRarity.java
+++ ../src-work/minecraft/net/minecraft/item/EnumRarity.java
@@ -2,8 +2,7 @@
 
 import net.minecraft.util.text.TextFormatting;
 
-public enum EnumRarity implements net.minecraftforge.common.IRarity
-{
+public enum EnumRarity implements net.minecraftforge.common.IRarity {
     COMMON(TextFormatting.WHITE, "Common"),
     UNCOMMON(TextFormatting.YELLOW, "Uncommon"),
     RARE(TextFormatting.AQUA, "Rare"),
@@ -12,21 +11,18 @@
     public final TextFormatting rarityColor;
     public final String rarityName;
 
-    private EnumRarity(TextFormatting color, String name)
-    {
+    private EnumRarity(TextFormatting color, String name) {
         this.rarityColor = color;
         this.rarityName = name;
     }
 
     @Override
-    public TextFormatting getColor()
-    {
+    public TextFormatting getColor() {
         return this.rarityColor;
     }
 
     @Override
-    public String getName()
-    {
+    public String getName() {
         return this.rarityName;
     }
 }
