--- ../src-base/minecraft/net/minecraft/item/IItemPropertyGetter.java
+++ ../src-work/minecraft/net/minecraft/item/IItemPropertyGetter.java
@@ -1,11 +1,12 @@
 package net.minecraft.item;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public interface IItemPropertyGetter
 {
     @SideOnly(Side.CLIENT)
