--- ../src-base/minecraft/net/minecraft/item/ItemClock.java
+++ ../src-work/minecraft/net/minecraft/item/ItemClock.java
@@ -1,6 +1,5 @@
 package net.minecraft.item;
 
-import javax.annotation.Nullable;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.ResourceLocation;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public class ItemClock extends Item
 {
     public ItemClock()
