--- ../src-base/minecraft/net/minecraft/item/ItemFireworkCharge.java
+++ ../src-work/minecraft/net/minecraft/item/ItemFireworkCharge.java
@@ -1,7 +1,5 @@
 package net.minecraft.item;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTTagCompound;
@@ -10,6 +8,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class ItemFireworkCharge extends Item
 {
     @SideOnly(Side.CLIENT)
