--- ../src-base/minecraft/net/minecraft/item/ItemBanner.java
+++ ../src-work/minecraft/net/minecraft/item/ItemBanner.java
@@ -1,7 +1,5 @@
 package net.minecraft.item;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.block.BlockStandingSign;
 import net.minecraft.block.BlockWallSign;
@@ -28,6 +26,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class ItemBanner extends ItemBlock
 {
     public ItemBanner()
