--- ../src-base/minecraft/net/minecraft/item/ItemFirework.java
+++ ../src-work/minecraft/net/minecraft/item/ItemFirework.java
@@ -1,8 +1,6 @@
 package net.minecraft.item;
 
 import com.google.common.collect.Lists;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.entity.item.EntityFireworkRocket;
 import net.minecraft.entity.player.EntityPlayer;
@@ -18,6 +16,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class ItemFirework extends Item
 {
     public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
@@ -26,6 +27,7 @@
         {
             ItemStack itemstack = player.getHeldItem(hand);
             EntityFireworkRocket entityfireworkrocket = new EntityFireworkRocket(worldIn, (double)((float)pos.getX() + hitX), (double)((float)pos.getY() + hitY), (double)((float)pos.getZ() + hitZ), itemstack);
+            entityfireworkrocket.spawningEntity = player.getUniqueID(); // Paper
             worldIn.spawnEntity(entityfireworkrocket);
 
             if (!player.capabilities.isCreativeMode)
@@ -46,6 +48,7 @@
             if (!worldIn.isRemote)
             {
                 EntityFireworkRocket entityfireworkrocket = new EntityFireworkRocket(worldIn, itemstack, playerIn);
+                entityfireworkrocket.spawningEntity = playerIn.getUniqueID(); // Paper
                 worldIn.spawnEntity(entityfireworkrocket);
 
                 if (!playerIn.capabilities.isCreativeMode)
