--- ../src-base/minecraft/net/minecraft/item/ItemEndCrystal.java
+++ ../src-work/minecraft/net/minecraft/item/ItemEndCrystal.java
@@ -1,6 +1,5 @@
 package net.minecraft.item;
 
-import java.util.List;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.Entity;
@@ -18,6 +17,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 public class ItemEndCrystal extends Item
 {
     public ItemEndCrystal()
