--- ../src-base/minecraft/net/minecraft/item/ItemTippedArrow.java
+++ ../src-work/minecraft/net/minecraft/item/ItemTippedArrow.java
@@ -1,7 +1,5 @@
 package net.minecraft.item;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
@@ -16,29 +14,25 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class ItemTippedArrow extends ItemArrow
-{
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class ItemTippedArrow extends ItemArrow {
     @SideOnly(Side.CLIENT)
-    public ItemStack getDefaultInstance()
-    {
+    public ItemStack getDefaultInstance() {
         return PotionUtils.addPotionToItemStack(super.getDefaultInstance(), PotionTypes.POISON);
     }
 
-    public EntityArrow createArrow(World worldIn, ItemStack stack, EntityLivingBase shooter)
-    {
+    public EntityArrow createArrow(World worldIn, ItemStack stack, EntityLivingBase shooter) {
         EntityTippedArrow entitytippedarrow = new EntityTippedArrow(worldIn, shooter);
         entitytippedarrow.setPotionEffect(stack);
         return entitytippedarrow;
     }
 
-    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items)
-    {
-        if (this.isInCreativeTab(tab))
-        {
-            for (PotionType potiontype : PotionType.REGISTRY)
-            {
-                if (!potiontype.getEffects().isEmpty())
-                {
+    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items) {
+        if (this.isInCreativeTab(tab)) {
+            for (PotionType potiontype : PotionType.REGISTRY) {
+                if (!potiontype.getEffects().isEmpty()) {
                     items.add(PotionUtils.addPotionToItemStack(new ItemStack(this), potiontype));
                 }
             }
@@ -46,13 +40,11 @@
     }
 
     @SideOnly(Side.CLIENT)
-    public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn)
-    {
+    public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
         PotionUtils.addPotionTooltip(stack, tooltip, 0.125F);
     }
 
-    public String getItemStackDisplayName(ItemStack stack)
-    {
+    public String getItemStackDisplayName(ItemStack stack) {
         return I18n.translateToLocal(PotionUtils.getPotionFromItem(stack).getNamePrefixed("tipped_arrow.effect."));
     }
 }
