--- ../src-base/minecraft/net/minecraft/item/ItemHoe.java
+++ ../src-work/minecraft/net/minecraft/item/ItemHoe.java
@@ -3,7 +3,6 @@
 import com.google.common.collect.Multimap;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockDirt;
-import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
@@ -25,9 +24,9 @@
 public class ItemHoe extends Item
 {
     private final float speed;
-    protected Item.ToolMaterial toolMaterial;
+    protected ToolMaterial toolMaterial;
 
-    public ItemHoe(Item.ToolMaterial material)
+    public ItemHoe(ToolMaterial material)
     {
         this.toolMaterial = material;
         this.maxStackSize = 1;
