--- ../src-base/minecraft/net/minecraft/item/ItemLingeringPotion.java
+++ ../src-work/minecraft/net/minecraft/item/ItemLingeringPotion.java
@@ -1,7 +1,5 @@
 package net.minecraft.item;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.projectile.EntityPotion;
@@ -17,6 +15,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 public class ItemLingeringPotion extends ItemPotion
 {
     public String getItemStackDisplayName(ItemStack stack)
