--- ../src-base/minecraft/net/minecraft/item/ItemPiston.java
+++ ../src-work/minecraft/net/minecraft/item/ItemPiston.java
@@ -2,15 +2,12 @@
 
 import net.minecraft.block.Block;
 
-public class ItemPiston extends ItemBlock
-{
-    public ItemPiston(Block block)
-    {
+public class ItemPiston extends ItemBlock {
+    public ItemPiston(Block block) {
         super(block);
     }
 
-    public int getMetadata(int damage)
-    {
+    public int getMetadata(int damage) {
         return 7;
     }
 }
