--- ../src-base/minecraft/net/minecraft/item/ItemShulkerBox.java
+++ ../src-work/minecraft/net/minecraft/item/ItemShulkerBox.java
@@ -2,10 +2,8 @@
 
 import net.minecraft.block.Block;
 
-public class ItemShulkerBox extends ItemBlock
-{
-    public ItemShulkerBox(Block blockInstance)
-    {
+public class ItemShulkerBox extends ItemBlock {
+    public ItemShulkerBox(Block blockInstance) {
         super(blockInstance);
         this.setMaxStackSize(1);
     }
