--- ../src-base/minecraft/net/minecraft/item/ItemAnvilBlock.java
+++ ../src-work/minecraft/net/minecraft/item/ItemAnvilBlock.java
@@ -2,15 +2,12 @@
 
 import net.minecraft.block.Block;
 
-public class ItemAnvilBlock extends ItemMultiTexture
-{
-    public ItemAnvilBlock(Block block)
-    {
-        super(block, block, new String[] {"intact", "slightlyDamaged", "veryDamaged"});
+public class ItemAnvilBlock extends ItemMultiTexture {
+    public ItemAnvilBlock(Block block) {
+        super(block, block, new String[]{"intact", "slightlyDamaged", "veryDamaged"});
     }
 
-    public int getMetadata(int damage)
-    {
+    public int getMetadata(int damage) {
         return damage << 2;
     }
 }
