--- ../src-base/minecraft/net/minecraft/item/ItemCarrotOnAStick.java
+++ ../src-work/minecraft/net/minecraft/item/ItemCarrotOnAStick.java
@@ -12,47 +12,36 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class ItemCarrotOnAStick extends Item
-{
-    public ItemCarrotOnAStick()
-    {
+public class ItemCarrotOnAStick extends Item {
+    public ItemCarrotOnAStick() {
         this.setCreativeTab(CreativeTabs.TRANSPORTATION);
         this.setMaxStackSize(1);
         this.setMaxDamage(25);
     }
 
     @SideOnly(Side.CLIENT)
-    public boolean isFull3D()
-    {
+    public boolean isFull3D() {
         return true;
     }
 
     @SideOnly(Side.CLIENT)
-    public boolean shouldRotateAroundWhenRendering()
-    {
+    public boolean shouldRotateAroundWhenRendering() {
         return true;
     }
 
-    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer playerIn, EnumHand handIn)
-    {
+    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer playerIn, EnumHand handIn) {
         ItemStack itemstack = playerIn.getHeldItem(handIn);
 
-        if (worldIn.isRemote)
-        {
+        if (worldIn.isRemote) {
             return new ActionResult<ItemStack>(EnumActionResult.PASS, itemstack);
-        }
-        else
-        {
-            if (playerIn.isRiding() && playerIn.getRidingEntity() instanceof EntityPig)
-            {
-                EntityPig entitypig = (EntityPig)playerIn.getRidingEntity();
+        } else {
+            if (playerIn.isRiding() && playerIn.getRidingEntity() instanceof EntityPig) {
+                EntityPig entitypig = (EntityPig) playerIn.getRidingEntity();
 
-                if (itemstack.getMaxDamage() - itemstack.getMetadata() >= 7 && entitypig.boost())
-                {
+                if (itemstack.getMaxDamage() - itemstack.getMetadata() >= 7 && entitypig.boost()) {
                     itemstack.damageItem(7, playerIn);
 
-                    if (itemstack.isEmpty())
-                    {
+                    if (itemstack.isEmpty()) {
                         ItemStack itemstack1 = new ItemStack(Items.FISHING_ROD);
                         itemstack1.setTagCompound(itemstack.getTagCompound());
                         return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, itemstack1);
