--- ../src-base/minecraft/net/minecraft/item/ItemWritableBook.java
+++ ../src-work/minecraft/net/minecraft/item/ItemWritableBook.java
@@ -9,41 +9,23 @@
 import net.minecraft.util.EnumHand;
 import net.minecraft.world.World;
 
-public class ItemWritableBook extends Item
-{
-    public ItemWritableBook()
-    {
+public class ItemWritableBook extends Item {
+    public ItemWritableBook() {
         this.setMaxStackSize(1);
     }
 
-    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer playerIn, EnumHand handIn)
-    {
-        ItemStack itemstack = playerIn.getHeldItem(handIn);
-        playerIn.openBook(itemstack, handIn);
-        playerIn.addStat(StatList.getObjectUseStats(this));
-        return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, itemstack);
-    }
-
-    public static boolean isNBTValid(NBTTagCompound nbt)
-    {
-        if (nbt == null)
-        {
+    public static boolean isNBTValid(NBTTagCompound nbt) {
+        if (nbt == null) {
             return false;
-        }
-        else if (!nbt.hasKey("pages", 9))
-        {
+        } else if (!nbt.hasKey("pages", 9)) {
             return false;
-        }
-        else
-        {
+        } else {
             NBTTagList nbttaglist = nbt.getTagList("pages", 8);
 
-            for (int i = 0; i < nbttaglist.tagCount(); ++i)
-            {
+            for (int i = 0; i < nbttaglist.tagCount(); ++i) {
                 String s = nbttaglist.getStringTagAt(i);
 
-                if (s.length() > 32767)
-                {
+                if (s.length() > 32767) {
                     return false;
                 }
             }
@@ -51,4 +33,11 @@
             return true;
         }
     }
+
+    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer playerIn, EnumHand handIn) {
+        ItemStack itemstack = playerIn.getHeldItem(handIn);
+        playerIn.openBook(itemstack, handIn);
+        playerIn.addStat(StatList.getObjectUseStats(this));
+        return new ActionResult<ItemStack>(EnumActionResult.SUCCESS, itemstack);
+    }
 }
