--- ../src-base/minecraft/net/minecraft/item/ItemSimpleFoiled.java
+++ ../src-work/minecraft/net/minecraft/item/ItemSimpleFoiled.java
@@ -3,11 +3,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class ItemSimpleFoiled extends Item
-{
+public class ItemSimpleFoiled extends Item {
     @SideOnly(Side.CLIENT)
-    public boolean hasEffect(ItemStack stack)
-    {
+    public boolean hasEffect(ItemStack stack) {
         return true;
     }
 }
