--- ../src-base/minecraft/net/minecraft/item/ItemBook.java
+++ ../src-work/minecraft/net/minecraft/item/ItemBook.java
@@ -1,14 +1,11 @@
 package net.minecraft.item;
 
-public class ItemBook extends Item
-{
-    public boolean isEnchantable(ItemStack stack)
-    {
+public class ItemBook extends Item {
+    public boolean isEnchantable(ItemStack stack) {
         return stack.getCount() == 1;
     }
 
-    public int getItemEnchantability()
-    {
+    public int getItemEnchantability() {
         return 1;
     }
 }
