--- ../src-base/minecraft/net/minecraft/item/ItemFireball.java
+++ ../src-work/minecraft/net/minecraft/item/ItemFireball.java
@@ -12,38 +12,33 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
-public class ItemFireball extends Item
-{
-    public ItemFireball()
-    {
+public class ItemFireball extends Item {
+    public ItemFireball() {
         this.setCreativeTab(CreativeTabs.MISC);
     }
 
-    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
-    {
-        if (worldIn.isRemote)
-        {
+    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+        if (worldIn.isRemote) {
             return EnumActionResult.SUCCESS;
-        }
-        else
-        {
+        } else {
             pos = pos.offset(facing);
             ItemStack itemstack = player.getHeldItem(hand);
 
-            if (!player.canPlayerEdit(pos, facing, itemstack))
-            {
+            if (!player.canPlayerEdit(pos, facing, itemstack)) {
                 return EnumActionResult.FAIL;
-            }
-            else
-            {
-                if (worldIn.getBlockState(pos).getMaterial() == Material.AIR)
-                {
-                    worldIn.playSound((EntityPlayer)null, pos, SoundEvents.ITEM_FIRECHARGE_USE, SoundCategory.BLOCKS, 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
+            } else {
+                if (worldIn.getBlockState(pos).getMaterial() == Material.AIR) {
+                    if (org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callBlockIgniteEvent(worldIn, pos.getX(), pos.getY(), pos.getZ(), org.bukkit.event.block.BlockIgniteEvent.IgniteCause.FIREBALL, player).isCancelled()) {
+                        if (!player.capabilities.isCreativeMode) {
+                            itemstack.shrink(1);
+                        }
+                        return EnumActionResult.PASS;
+                    }
+                    worldIn.playSound((EntityPlayer) null, pos, SoundEvents.ITEM_FIRECHARGE_USE, SoundCategory.BLOCKS, 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
                     worldIn.setBlockState(pos, Blocks.FIRE.getDefaultState());
                 }
 
-                if (!player.capabilities.isCreativeMode)
-                {
+                if (!player.capabilities.isCreativeMode) {
                     itemstack.shrink(1);
                 }
 
