--- ../src-base/minecraft/net/minecraft/item/ItemNameTag.java
+++ ../src-work/minecraft/net/minecraft/item/ItemNameTag.java
@@ -6,29 +6,22 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.EnumHand;
 
-public class ItemNameTag extends Item
-{
-    public ItemNameTag()
-    {
+public class ItemNameTag extends Item {
+    public ItemNameTag() {
         this.setCreativeTab(CreativeTabs.TOOLS);
     }
 
-    public boolean itemInteractionForEntity(ItemStack stack, EntityPlayer playerIn, EntityLivingBase target, EnumHand hand)
-    {
-        if (stack.hasDisplayName() && !(target instanceof EntityPlayer))
-        {
+    public boolean itemInteractionForEntity(ItemStack stack, EntityPlayer playerIn, EntityLivingBase target, EnumHand hand) {
+        if (stack.hasDisplayName() && !(target instanceof EntityPlayer)) {
             target.setCustomNameTag(stack.getDisplayName());
 
-            if (target instanceof EntityLiving)
-            {
-                ((EntityLiving)target).enablePersistence();
+            if (target instanceof EntityLiving) {
+                ((EntityLiving) target).enablePersistence();
             }
 
             stack.shrink(1);
             return true;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
