--- ../src-base/minecraft/net/minecraft/item/ItemColored.java
+++ ../src-work/minecraft/net/minecraft/item/ItemColored.java
@@ -2,40 +2,31 @@
 
 import net.minecraft.block.Block;
 
-public class ItemColored extends ItemBlock
-{
+public class ItemColored extends ItemBlock {
     private String[] subtypeNames;
 
-    public ItemColored(Block block, boolean hasSubtypes)
-    {
+    public ItemColored(Block block, boolean hasSubtypes) {
         super(block);
 
-        if (hasSubtypes)
-        {
+        if (hasSubtypes) {
             this.setMaxDamage(0);
             this.setHasSubtypes(true);
         }
     }
 
-    public int getMetadata(int damage)
-    {
+    public int getMetadata(int damage) {
         return damage;
     }
 
-    public ItemColored setSubtypeNames(String[] names)
-    {
+    public ItemColored setSubtypeNames(String[] names) {
         this.subtypeNames = names;
         return this;
     }
 
-    public String getUnlocalizedName(ItemStack stack)
-    {
-        if (this.subtypeNames == null)
-        {
+    public String getUnlocalizedName(ItemStack stack) {
+        if (this.subtypeNames == null) {
             return super.getUnlocalizedName(stack);
-        }
-        else
-        {
+        } else {
             int i = stack.getMetadata();
             return i >= 0 && i < this.subtypeNames.length ? super.getUnlocalizedName(stack) + "." + this.subtypeNames[i] : super.getUnlocalizedName(stack);
         }
