--- ../src-base/minecraft/net/minecraft/item/ItemSaddle.java
+++ ../src-work/minecraft/net/minecraft/item/ItemSaddle.java
@@ -8,31 +8,24 @@
 import net.minecraft.util.EnumHand;
 import net.minecraft.util.SoundCategory;
 
-public class ItemSaddle extends Item
-{
-    public ItemSaddle()
-    {
+public class ItemSaddle extends Item {
+    public ItemSaddle() {
         this.maxStackSize = 1;
         this.setCreativeTab(CreativeTabs.TRANSPORTATION);
     }
 
-    public boolean itemInteractionForEntity(ItemStack stack, EntityPlayer playerIn, EntityLivingBase target, EnumHand hand)
-    {
-        if (target instanceof EntityPig)
-        {
-            EntityPig entitypig = (EntityPig)target;
+    public boolean itemInteractionForEntity(ItemStack stack, EntityPlayer playerIn, EntityLivingBase target, EnumHand hand) {
+        if (target instanceof EntityPig) {
+            EntityPig entitypig = (EntityPig) target;
 
-            if (!entitypig.getSaddled() && !entitypig.isChild())
-            {
+            if (!entitypig.getSaddled() && !entitypig.isChild()) {
                 entitypig.setSaddled(true);
                 entitypig.world.playSound(playerIn, entitypig.posX, entitypig.posY, entitypig.posZ, SoundEvents.ENTITY_PIG_SADDLE, SoundCategory.NEUTRAL, 0.5F, 1.0F);
                 stack.shrink(1);
             }
 
             return true;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
