--- ../src-base/minecraft/net/minecraft/item/ItemSword.java
+++ ../src-work/minecraft/net/minecraft/item/ItemSword.java
@@ -18,9 +18,9 @@
 public class ItemSword extends Item
 {
     private final float attackDamage;
-    private final Item.ToolMaterial material;
+    private final ToolMaterial material;
 
-    public ItemSword(Item.ToolMaterial material)
+    public ItemSword(ToolMaterial material)
     {
         this.material = material;
         this.maxStackSize = 1;
