--- ../src-base/minecraft/net/minecraftforge/event/enchanting/EnchantmentLevelSetEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/enchanting/EnchantmentLevelSetEvent.java
@@ -34,8 +34,7 @@
  * from 0-15 and indicates how many bookshelves surround the enchanting table. The {@link #itemStack} representing the item being
  * enchanted is also available.
  */
-public class EnchantmentLevelSetEvent extends Event
-{
+public class EnchantmentLevelSetEvent extends Event {
     private final World world;
     private final BlockPos pos;
     private final int enchantRow;
@@ -45,8 +44,7 @@
     private final int originalLevel;
     private int level;
 
-    public EnchantmentLevelSetEvent(World world, BlockPos pos, int enchantRow, int power, @Nonnull ItemStack itemStack, int level)
-    {
+    public EnchantmentLevelSetEvent(World world, BlockPos pos, int enchantRow, int power, @Nonnull ItemStack itemStack, int level) {
         this.world = world;
         this.pos = pos;
         this.enchantRow = enchantRow;
@@ -58,82 +56,74 @@
 
     /**
      * Get the world object
-     * 
+     *
      * @return the world object
      */
-    public World getWorld()
-    {
+    public World getWorld() {
         return world;
     }
 
     /**
      * Get the pos of the enchantment table
-     * 
+     *
      * @return the pos of the enchantment table
      */
-    public BlockPos getPos()
-    {
+    public BlockPos getPos() {
         return pos;
     }
 
     /**
      * Get the row for which the enchantment level is being set
-     * 
+     *
      * @return the row for which the enchantment level is being set
      */
-    public int getEnchantRow()
-    {
+    public int getEnchantRow() {
         return enchantRow;
     }
 
     /**
      * Get the power (# of bookshelves) for the enchanting table
-     * 
+     *
      * @return the power (# of bookshelves) for the enchanting table
      */
-    public int getPower()
-    {
+    public int getPower() {
         return power;
     }
 
     /**
      * Get the item being enchanted
-     * 
+     *
      * @return the item being enchanted
      */
     @Nonnull
-    public ItemStack getItem()
-    {
+    public ItemStack getItem() {
         return itemStack;
     }
 
     /**
      * Get the original level of the enchantment for this row (0-30)
-     * 
+     *
      * @return the original level of the enchantment for this row (0-30)
      */
-    public int getOriginalLevel()
-    {
+    public int getOriginalLevel() {
         return originalLevel;
     }
 
     /**
      * Get the level of the enchantment for this row (0-30)
-     * 
+     *
      * @return the level of the enchantment for this row (0-30)
      */
-    public int getLevel()
-    {
+    public int getLevel() {
         return level;
     }
 
     /**
      * Set the new level of the enchantment (0-30)
-     * 
+     *
      * @param level the new level of the enchantment (0-30)
      */
-    public void setLevel(int level)
-    {
+    public void setLevel(int level) {
         this.level = level;
     }
 }
