--- ../src-base/minecraft/net/minecraft/util/math/AxisAlignedBB.java
+++ ../src-work/minecraft/net/minecraft/util/math/AxisAlignedBB.java
@@ -1,11 +1,12 @@
 package net.minecraft.util.math;
 
 import com.google.common.annotations.VisibleForTesting;
-import javax.annotation.Nullable;
 import net.minecraft.util.EnumFacing;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+
 public class AxisAlignedBB
 {
     public final double minX;
