--- ../src-base/minecraft/net/minecraft/util/math/Rotations.java
+++ ../src-work/minecraft/net/minecraft/util/math/Rotations.java
@@ -3,26 +3,22 @@
 import net.minecraft.nbt.NBTTagFloat;
 import net.minecraft.nbt.NBTTagList;
 
-public class Rotations
-{
+public class Rotations {
     protected final float x;
     protected final float y;
     protected final float z;
 
-    public Rotations(float x, float y, float z)
-    {
+    public Rotations(float x, float y, float z) {
         this.x = !Float.isInfinite(x) && !Float.isNaN(x) ? x % 360.0F : 0.0F;
         this.y = !Float.isInfinite(y) && !Float.isNaN(y) ? y % 360.0F : 0.0F;
         this.z = !Float.isInfinite(z) && !Float.isNaN(z) ? z % 360.0F : 0.0F;
     }
 
-    public Rotations(NBTTagList nbt)
-    {
+    public Rotations(NBTTagList nbt) {
         this(nbt.getFloatAt(0), nbt.getFloatAt(1), nbt.getFloatAt(2));
     }
 
-    public NBTTagList writeToNBT()
-    {
+    public NBTTagList writeToNBT() {
         NBTTagList nbttaglist = new NBTTagList();
         nbttaglist.appendTag(new NBTTagFloat(this.x));
         nbttaglist.appendTag(new NBTTagFloat(this.y));
@@ -30,31 +26,24 @@
         return nbttaglist;
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        if (!(p_equals_1_ instanceof Rotations))
-        {
+    public boolean equals(Object p_equals_1_) {
+        if (!(p_equals_1_ instanceof Rotations)) {
             return false;
-        }
-        else
-        {
-            Rotations rotations = (Rotations)p_equals_1_;
+        } else {
+            Rotations rotations = (Rotations) p_equals_1_;
             return this.x == rotations.x && this.y == rotations.y && this.z == rotations.z;
         }
     }
 
-    public float getX()
-    {
+    public float getX() {
         return this.x;
     }
 
-    public float getY()
-    {
+    public float getY() {
         return this.y;
     }
 
-    public float getZ()
-    {
+    public float getZ() {
         return this.z;
     }
 }
