--- ../src-base/minecraft/net/minecraft/util/math/Vec3i.java
+++ ../src-work/minecraft/net/minecraft/util/math/Vec3i.java
@@ -1,6 +1,7 @@
 package net.minecraft.util.math;
 
 import com.google.common.base.MoreObjects;
+
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
@@ -83,6 +84,7 @@
     {
         return this.z;
     }
+    // Paper end
 
     public Vec3i crossProduct(Vec3i vec)
     {
