--- ../src-base/minecraft/net/minecraftforge/event/brewing/PlayerBrewedPotionEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/brewing/PlayerBrewedPotionEvent.java
@@ -28,12 +28,10 @@
 /**
  * This event is called when a player picks up a potion from a brewing stand.
  */
-public class PlayerBrewedPotionEvent extends PlayerEvent
-{
+public class PlayerBrewedPotionEvent extends PlayerEvent {
     private final ItemStack stack;
 
-    public PlayerBrewedPotionEvent(EntityPlayer player, @Nonnull ItemStack stack)
-    {
+    public PlayerBrewedPotionEvent(EntityPlayer player, @Nonnull ItemStack stack) {
         super(player);
         this.stack = stack;
     }
@@ -42,8 +40,7 @@
      * The ItemStack of the potion.
      */
     @Nonnull
-    public ItemStack getStack()
-    {
+    public ItemStack getStack() {
         return stack;
     }
 }
