--- ../src-base/minecraft/net/minecraft/network/status/INetHandlerStatusClient.java
+++ ../src-work/minecraft/net/minecraft/network/status/INetHandlerStatusClient.java
@@ -4,8 +4,7 @@
 import net.minecraft.network.status.server.SPacketPong;
 import net.minecraft.network.status.server.SPacketServerInfo;
 
-public interface INetHandlerStatusClient extends INetHandler
-{
+public interface INetHandlerStatusClient extends INetHandler {
     void handleServerInfo(SPacketServerInfo packetIn);
 
     void handlePong(SPacketPong packetIn);
