--- ../src-base/minecraft/net/minecraft/network/status/INetHandlerStatusServer.java
+++ ../src-work/minecraft/net/minecraft/network/status/INetHandlerStatusServer.java
@@ -4,8 +4,7 @@
 import net.minecraft.network.status.client.CPacketPing;
 import net.minecraft.network.status.client.CPacketServerQuery;
 
-public interface INetHandlerStatusServer extends INetHandler
-{
+public interface INetHandlerStatusServer extends INetHandler {
     void processPing(CPacketPing packetIn);
 
     void processServerQuery(CPacketServerQuery packetIn);
