--- ../src-base/minecraft/net/minecraft/client/resources/data/AnimationFrame.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/AnimationFrame.java
@@ -4,34 +4,28 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class AnimationFrame
-{
+public class AnimationFrame {
     private final int frameIndex;
     private final int frameTime;
 
-    public AnimationFrame(int frameIndexIn)
-    {
+    public AnimationFrame(int frameIndexIn) {
         this(frameIndexIn, -1);
     }
 
-    public AnimationFrame(int frameIndexIn, int frameTimeIn)
-    {
+    public AnimationFrame(int frameIndexIn, int frameTimeIn) {
         this.frameIndex = frameIndexIn;
         this.frameTime = frameTimeIn;
     }
 
-    public boolean hasNoTime()
-    {
+    public boolean hasNoTime() {
         return this.frameTime == -1;
     }
 
-    public int getFrameTime()
-    {
+    public int getFrameTime() {
         return this.frameTime;
     }
 
-    public int getFrameIndex()
-    {
+    public int getFrameIndex() {
         return this.frameIndex;
     }
 }
