--- ../src-base/minecraft/net/minecraft/client/resources/data/BaseMetadataSectionSerializer.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/BaseMetadataSectionSerializer.java
@@ -4,6 +4,5 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public abstract class BaseMetadataSectionSerializer<T extends IMetadataSection> implements IMetadataSectionSerializer<T>
-{
+public abstract class BaseMetadataSectionSerializer<T extends IMetadataSection> implements IMetadataSectionSerializer<T> {
 }
