--- ../src-base/minecraft/net/minecraft/client/resources/data/FontMetadataSection.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/FontMetadataSection.java
@@ -4,14 +4,12 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class FontMetadataSection implements IMetadataSection
-{
+public class FontMetadataSection implements IMetadataSection {
     private final float[] charWidths;
     private final float[] charLefts;
     private final float[] charSpacings;
 
-    public FontMetadataSection(float[] charWidthsIn, float[] charLeftsIn, float[] charSpacingsIn)
-    {
+    public FontMetadataSection(float[] charWidthsIn, float[] charLeftsIn, float[] charSpacingsIn) {
         this.charWidths = charWidthsIn;
         this.charLefts = charLeftsIn;
         this.charSpacings = charSpacingsIn;
