--- ../src-base/minecraft/net/minecraft/client/resources/data/IMetadataSectionSerializer.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/IMetadataSectionSerializer.java
@@ -5,7 +5,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface IMetadataSectionSerializer<T extends IMetadataSection> extends JsonDeserializer<T>
-{
+public interface IMetadataSectionSerializer<T extends IMetadataSection> extends JsonDeserializer<T> {
     String getSectionName();
 }
