--- ../src-base/minecraft/net/minecraft/client/resources/data/LanguageMetadataSection.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/LanguageMetadataSection.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.resources.data;
 
-import java.util.Collection;
 import net.minecraft.client.resources.Language;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Collection;
+
 @SideOnly(Side.CLIENT)
 public class LanguageMetadataSection implements IMetadataSection
 {
