--- ../src-base/minecraft/net/minecraft/client/resources/data/TextureMetadataSection.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/TextureMetadataSection.java
@@ -4,24 +4,20 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class TextureMetadataSection implements IMetadataSection
-{
+public class TextureMetadataSection implements IMetadataSection {
     private final boolean textureBlur;
     private final boolean textureClamp;
 
-    public TextureMetadataSection(boolean textureBlurIn, boolean textureClampIn)
-    {
+    public TextureMetadataSection(boolean textureBlurIn, boolean textureClampIn) {
         this.textureBlur = textureBlurIn;
         this.textureClamp = textureClampIn;
     }
 
-    public boolean getTextureBlur()
-    {
+    public boolean getTextureBlur() {
         return this.textureBlur;
     }
 
-    public boolean getTextureClamp()
-    {
+    public boolean getTextureClamp() {
         return this.textureClamp;
     }
 }
