--- ../src-base/minecraft/net/minecraft/client/resources/data/TextureMetadataSectionSerializer.java
+++ ../src-work/minecraft/net/minecraft/client/resources/data/TextureMetadataSectionSerializer.java
@@ -4,11 +4,12 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
-import java.lang.reflect.Type;
 import net.minecraft.util.JsonUtils;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.lang.reflect.Type;
+
 @SideOnly(Side.CLIENT)
 public class TextureMetadataSectionSerializer extends BaseMetadataSectionSerializer<TextureMetadataSection>
 {
