--- ../src-base/minecraft/net/minecraft/client/model/IMultipassModel.java
+++ ../src-work/minecraft/net/minecraft/client/model/IMultipassModel.java
@@ -5,7 +5,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface IMultipassModel
-{
+public interface IMultipassModel {
     void renderMultipass(Entity p_187054_1_, float p_187054_2_, float p_187054_3_, float p_187054_4_, float p_187054_5_, float p_187054_6_, float scale);
 }
