--- ../src-base/minecraft/net/minecraft/client/model/ModelBanner.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelBanner.java
@@ -4,14 +4,12 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelBanner extends ModelBase
-{
+public class ModelBanner extends ModelBase {
     public ModelRenderer bannerSlate;
     public ModelRenderer bannerStand;
     public ModelRenderer bannerTop;
 
-    public ModelBanner()
-    {
+    public ModelBanner() {
         this.textureWidth = 64;
         this.textureHeight = 64;
         this.bannerSlate = new ModelRenderer(this, 0, 0);
@@ -22,8 +20,7 @@
         this.bannerTop.addBox(-10.0F, -32.0F, -1.0F, 20, 2, 2, 0.0F);
     }
 
-    public void renderBanner()
-    {
+    public void renderBanner() {
         this.bannerSlate.rotationPointY = -32.0F;
         this.bannerSlate.render(0.0625F);
         this.bannerStand.render(0.0625F);
