--- ../src-base/minecraft/net/minecraft/client/model/ModelBat.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelBat.java
@@ -7,8 +7,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelBat extends ModelBase
-{
+public class ModelBat extends ModelBase {
     private final ModelRenderer batHead;
     private final ModelRenderer batBody;
     private final ModelRenderer batRightWing;
@@ -16,8 +15,7 @@
     private final ModelRenderer batOuterRightWing;
     private final ModelRenderer batOuterLeftWing;
 
-    public ModelBat()
-    {
+    public ModelBat() {
         this.textureWidth = 64;
         this.textureHeight = 64;
         this.batHead = new ModelRenderer(this, 0, 0);
@@ -50,42 +48,37 @@
         this.batLeftWing.addChild(this.batOuterLeftWing);
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         this.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, entityIn);
         this.batHead.render(scale);
         this.batBody.render(scale);
     }
 
-    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn)
-    {
-        if (((EntityBat)entityIn).getIsBatHanging())
-        {
+    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn) {
+        if (((EntityBat) entityIn).getIsBatHanging()) {
             this.batHead.rotateAngleX = headPitch * 0.017453292F;
-            this.batHead.rotateAngleY = (float)Math.PI - netHeadYaw * 0.017453292F;
-            this.batHead.rotateAngleZ = (float)Math.PI;
+            this.batHead.rotateAngleY = (float) Math.PI - netHeadYaw * 0.017453292F;
+            this.batHead.rotateAngleZ = (float) Math.PI;
             this.batHead.setRotationPoint(0.0F, -2.0F, 0.0F);
             this.batRightWing.setRotationPoint(-3.0F, 0.0F, 3.0F);
             this.batLeftWing.setRotationPoint(3.0F, 0.0F, 3.0F);
-            this.batBody.rotateAngleX = (float)Math.PI;
+            this.batBody.rotateAngleX = (float) Math.PI;
             this.batRightWing.rotateAngleX = -0.15707964F;
-            this.batRightWing.rotateAngleY = -((float)Math.PI * 2F / 5F);
+            this.batRightWing.rotateAngleY = -((float) Math.PI * 2F / 5F);
             this.batOuterRightWing.rotateAngleY = -1.7278761F;
             this.batLeftWing.rotateAngleX = this.batRightWing.rotateAngleX;
             this.batLeftWing.rotateAngleY = -this.batRightWing.rotateAngleY;
             this.batOuterLeftWing.rotateAngleY = -this.batOuterRightWing.rotateAngleY;
-        }
-        else
-        {
+        } else {
             this.batHead.rotateAngleX = headPitch * 0.017453292F;
             this.batHead.rotateAngleY = netHeadYaw * 0.017453292F;
             this.batHead.rotateAngleZ = 0.0F;
             this.batHead.setRotationPoint(0.0F, 0.0F, 0.0F);
             this.batRightWing.setRotationPoint(0.0F, 0.0F, 0.0F);
             this.batLeftWing.setRotationPoint(0.0F, 0.0F, 0.0F);
-            this.batBody.rotateAngleX = ((float)Math.PI / 4F) + MathHelper.cos(ageInTicks * 0.1F) * 0.15F;
+            this.batBody.rotateAngleX = ((float) Math.PI / 4F) + MathHelper.cos(ageInTicks * 0.1F) * 0.15F;
             this.batBody.rotateAngleY = 0.0F;
-            this.batRightWing.rotateAngleY = MathHelper.cos(ageInTicks * 1.3F) * (float)Math.PI * 0.25F;
+            this.batRightWing.rotateAngleY = MathHelper.cos(ageInTicks * 1.3F) * (float) Math.PI * 0.25F;
             this.batLeftWing.rotateAngleY = -this.batRightWing.rotateAngleY;
             this.batOuterRightWing.rotateAngleY = this.batRightWing.rotateAngleY * 0.5F;
             this.batOuterLeftWing.rotateAngleY = -this.batRightWing.rotateAngleY * 0.5F;
