--- ../src-base/minecraft/net/minecraft/client/model/ModelEnderman.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelEnderman.java
@@ -5,13 +5,11 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelEnderman extends ModelBiped
-{
+public class ModelEnderman extends ModelBiped {
     public boolean isCarrying;
     public boolean isAttacking;
 
-    public ModelEnderman(float scale)
-    {
+    public ModelEnderman(float scale) {
         super(0.0F, -14.0F, 64, 32);
         float f = -14.0F;
         this.bipedHeadwear = new ModelRenderer(this, 0, 16);
@@ -36,8 +34,7 @@
         this.bipedLeftLeg.setRotationPoint(2.0F, -2.0F, 0.0F);
     }
 
-    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn)
-    {
+    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn) {
         super.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scaleFactor, entityIn);
         this.bipedHead.showModel = true;
         float f = -14.0F;
@@ -46,54 +43,45 @@
         this.bipedBody.rotationPointZ = -0.0F;
         this.bipedRightLeg.rotateAngleX -= 0.0F;
         this.bipedLeftLeg.rotateAngleX -= 0.0F;
-        this.bipedRightArm.rotateAngleX = (float)((double)this.bipedRightArm.rotateAngleX * 0.5D);
-        this.bipedLeftArm.rotateAngleX = (float)((double)this.bipedLeftArm.rotateAngleX * 0.5D);
-        this.bipedRightLeg.rotateAngleX = (float)((double)this.bipedRightLeg.rotateAngleX * 0.5D);
-        this.bipedLeftLeg.rotateAngleX = (float)((double)this.bipedLeftLeg.rotateAngleX * 0.5D);
+        this.bipedRightArm.rotateAngleX = (float) ((double) this.bipedRightArm.rotateAngleX * 0.5D);
+        this.bipedLeftArm.rotateAngleX = (float) ((double) this.bipedLeftArm.rotateAngleX * 0.5D);
+        this.bipedRightLeg.rotateAngleX = (float) ((double) this.bipedRightLeg.rotateAngleX * 0.5D);
+        this.bipedLeftLeg.rotateAngleX = (float) ((double) this.bipedLeftLeg.rotateAngleX * 0.5D);
         float f1 = 0.4F;
 
-        if (this.bipedRightArm.rotateAngleX > 0.4F)
-        {
+        if (this.bipedRightArm.rotateAngleX > 0.4F) {
             this.bipedRightArm.rotateAngleX = 0.4F;
         }
 
-        if (this.bipedLeftArm.rotateAngleX > 0.4F)
-        {
+        if (this.bipedLeftArm.rotateAngleX > 0.4F) {
             this.bipedLeftArm.rotateAngleX = 0.4F;
         }
 
-        if (this.bipedRightArm.rotateAngleX < -0.4F)
-        {
+        if (this.bipedRightArm.rotateAngleX < -0.4F) {
             this.bipedRightArm.rotateAngleX = -0.4F;
         }
 
-        if (this.bipedLeftArm.rotateAngleX < -0.4F)
-        {
+        if (this.bipedLeftArm.rotateAngleX < -0.4F) {
             this.bipedLeftArm.rotateAngleX = -0.4F;
         }
 
-        if (this.bipedRightLeg.rotateAngleX > 0.4F)
-        {
+        if (this.bipedRightLeg.rotateAngleX > 0.4F) {
             this.bipedRightLeg.rotateAngleX = 0.4F;
         }
 
-        if (this.bipedLeftLeg.rotateAngleX > 0.4F)
-        {
+        if (this.bipedLeftLeg.rotateAngleX > 0.4F) {
             this.bipedLeftLeg.rotateAngleX = 0.4F;
         }
 
-        if (this.bipedRightLeg.rotateAngleX < -0.4F)
-        {
+        if (this.bipedRightLeg.rotateAngleX < -0.4F) {
             this.bipedRightLeg.rotateAngleX = -0.4F;
         }
 
-        if (this.bipedLeftLeg.rotateAngleX < -0.4F)
-        {
+        if (this.bipedLeftLeg.rotateAngleX < -0.4F) {
             this.bipedLeftLeg.rotateAngleX = -0.4F;
         }
 
-        if (this.isCarrying)
-        {
+        if (this.isCarrying) {
             this.bipedRightArm.rotateAngleX = -0.5F;
             this.bipedLeftArm.rotateAngleX = -0.5F;
             this.bipedRightArm.rotateAngleZ = 0.05F;
@@ -115,8 +103,7 @@
         this.bipedHeadwear.rotateAngleY = this.bipedHead.rotateAngleY;
         this.bipedHeadwear.rotateAngleZ = this.bipedHead.rotateAngleZ;
 
-        if (this.isAttacking)
-        {
+        if (this.isAttacking) {
             float f2 = 1.0F;
             this.bipedHead.rotationPointY -= 5.0F;
         }
