--- ../src-base/minecraft/net/minecraft/client/model/ModelLlamaSpit.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelLlamaSpit.java
@@ -5,17 +5,14 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelLlamaSpit extends ModelBase
-{
+public class ModelLlamaSpit extends ModelBase {
     private final ModelRenderer main;
 
-    public ModelLlamaSpit()
-    {
+    public ModelLlamaSpit() {
         this(0.0F);
     }
 
-    public ModelLlamaSpit(float p_i47225_1_)
-    {
+    public ModelLlamaSpit(float p_i47225_1_) {
         this.main = new ModelRenderer(this);
         int i = 2;
         this.main.setTextureOffset(0, 0).addBox(-4.0F, 0.0F, 0.0F, 2, 2, 2, p_i47225_1_);
@@ -28,8 +25,7 @@
         this.main.setRotationPoint(0.0F, 0.0F, 0.0F);
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         this.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, entityIn);
         this.main.render(scale);
     }
