--- ../src-base/minecraft/net/minecraft/client/model/ModelGhast.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelGhast.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.model;
 
-import java.util.Random;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.entity.Entity;
 import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Random;
+
 @SideOnly(Side.CLIENT)
 public class ModelGhast extends ModelBase
 {
