--- ../src-base/minecraft/net/minecraft/client/model/ModelCow.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelCow.java
@@ -4,10 +4,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelCow extends ModelQuadruped
-{
-    public ModelCow()
-    {
+public class ModelCow extends ModelQuadruped {
+    public ModelCow() {
         super(12, 0.0F);
         this.head = new ModelRenderer(this, 0, 0);
         this.head.addBox(-4.0F, -4.0F, -6.0F, 8, 8, 6, 0.0F);
