--- ../src-base/minecraft/net/minecraft/client/model/ModelEnderCrystal.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelEnderCrystal.java
@@ -6,33 +6,28 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelEnderCrystal extends ModelBase
-{
+public class ModelEnderCrystal extends ModelBase {
     private final ModelRenderer cube;
     private final ModelRenderer glass = new ModelRenderer(this, "glass");
     private ModelRenderer base;
 
-    public ModelEnderCrystal(float p_i1170_1_, boolean renderBase)
-    {
+    public ModelEnderCrystal(float p_i1170_1_, boolean renderBase) {
         this.glass.setTextureOffset(0, 0).addBox(-4.0F, -4.0F, -4.0F, 8, 8, 8);
         this.cube = new ModelRenderer(this, "cube");
         this.cube.setTextureOffset(32, 0).addBox(-4.0F, -4.0F, -4.0F, 8, 8, 8);
 
-        if (renderBase)
-        {
+        if (renderBase) {
             this.base = new ModelRenderer(this, "base");
             this.base.setTextureOffset(0, 16).addBox(-6.0F, 0.0F, -6.0F, 12, 4, 12);
         }
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         GlStateManager.pushMatrix();
         GlStateManager.scale(2.0F, 2.0F, 2.0F);
         GlStateManager.translate(0.0F, -0.5F, 0.0F);
 
-        if (this.base != null)
-        {
+        if (this.base != null) {
             this.base.render(scale);
         }
 
