--- ../src-base/minecraft/net/minecraft/client/model/ModelPig.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelPig.java
@@ -4,15 +4,12 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelPig extends ModelQuadruped
-{
-    public ModelPig()
-    {
+public class ModelPig extends ModelQuadruped {
+    public ModelPig() {
         this(0.0F);
     }
 
-    public ModelPig(float scale)
-    {
+    public ModelPig(float scale) {
         super(6, scale);
         this.head.setTextureOffset(16, 16).addBox(-2.0F, 0.0F, -9.0F, 4, 3, 1, scale);
         this.childYOffset = 4.0F;
