--- ../src-base/minecraft/net/minecraft/client/model/ModelShulkerBullet.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelShulkerBullet.java
@@ -5,12 +5,10 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelShulkerBullet extends ModelBase
-{
+public class ModelShulkerBullet extends ModelBase {
     public ModelRenderer renderer;
 
-    public ModelShulkerBullet()
-    {
+    public ModelShulkerBullet() {
         this.textureWidth = 64;
         this.textureHeight = 32;
         this.renderer = new ModelRenderer(this);
@@ -20,14 +18,12 @@
         this.renderer.setRotationPoint(0.0F, 0.0F, 0.0F);
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         this.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, entityIn);
         this.renderer.render(scale);
     }
 
-    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn)
-    {
+    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn) {
         super.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scaleFactor, entityIn);
         this.renderer.rotateAngleY = netHeadYaw * 0.017453292F;
         this.renderer.rotateAngleX = headPitch * 0.017453292F;
