--- ../src-base/minecraft/net/minecraft/client/model/ModelSkeletonHead.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelSkeletonHead.java
@@ -5,17 +5,14 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelSkeletonHead extends ModelBase
-{
+public class ModelSkeletonHead extends ModelBase {
     public ModelRenderer skeletonHead;
 
-    public ModelSkeletonHead()
-    {
+    public ModelSkeletonHead() {
         this(0, 35, 64, 64);
     }
 
-    public ModelSkeletonHead(int p_i1155_1_, int p_i1155_2_, int p_i1155_3_, int p_i1155_4_)
-    {
+    public ModelSkeletonHead(int p_i1155_1_, int p_i1155_2_, int p_i1155_3_, int p_i1155_4_) {
         this.textureWidth = p_i1155_3_;
         this.textureHeight = p_i1155_4_;
         this.skeletonHead = new ModelRenderer(this, p_i1155_1_, p_i1155_2_);
@@ -23,14 +20,12 @@
         this.skeletonHead.setRotationPoint(0.0F, 0.0F, 0.0F);
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         this.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, entityIn);
         this.skeletonHead.render(scale);
     }
 
-    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn)
-    {
+    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn) {
         super.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scaleFactor, entityIn);
         this.skeletonHead.rotateAngleY = netHeadYaw * 0.017453292F;
         this.skeletonHead.rotateAngleX = headPitch * 0.017453292F;
