--- ../src-base/minecraft/net/minecraft/client/model/ModelArmorStand.java
+++ ../src-work/minecraft/net/minecraft/client/model/ModelArmorStand.java
@@ -8,20 +8,17 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class ModelArmorStand extends ModelArmorStandArmor
-{
+public class ModelArmorStand extends ModelArmorStandArmor {
     public ModelRenderer standRightSide;
     public ModelRenderer standLeftSide;
     public ModelRenderer standWaist;
     public ModelRenderer standBase;
 
-    public ModelArmorStand()
-    {
+    public ModelArmorStand() {
         this(0.0F);
     }
 
-    public ModelArmorStand(float modelSize)
-    {
+    public ModelArmorStand(float modelSize) {
         super(modelSize, 64, 64);
         this.bipedHead = new ModelRenderer(this, 0, 0);
         this.bipedHead.addBox(-1.0F, -7.0F, -1.0F, 2, 7, 2, modelSize);
@@ -59,13 +56,11 @@
         this.bipedHeadwear.showModel = false;
     }
 
-    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn)
-    {
+    public void setRotationAngles(float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scaleFactor, Entity entityIn) {
         super.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scaleFactor, entityIn);
 
-        if (entityIn instanceof EntityArmorStand)
-        {
-            EntityArmorStand entityarmorstand = (EntityArmorStand)entityIn;
+        if (entityIn instanceof EntityArmorStand) {
+            EntityArmorStand entityarmorstand = (EntityArmorStand) entityIn;
             this.bipedLeftArm.showModel = entityarmorstand.getShowArms();
             this.bipedRightArm.showModel = entityarmorstand.getShowArms();
             this.standBase.showModel = !entityarmorstand.hasNoBasePlate();
@@ -86,13 +81,11 @@
         }
     }
 
-    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale)
-    {
+    public void render(Entity entityIn, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch, float scale) {
         super.render(entityIn, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale);
         GlStateManager.pushMatrix();
 
-        if (this.isChild)
-        {
+        if (this.isChild) {
             float f = 2.0F;
             GlStateManager.scale(0.5F, 0.5F, 0.5F);
             GlStateManager.translate(0.0F, 24.0F * scale, 0.0F);
@@ -100,11 +93,8 @@
             this.standLeftSide.render(scale);
             this.standWaist.render(scale);
             this.standBase.render(scale);
-        }
-        else
-        {
-            if (entityIn.isSneaking())
-            {
+        } else {
+            if (entityIn.isSneaking()) {
                 GlStateManager.translate(0.0F, 0.2F, 0.0F);
             }
 
@@ -117,8 +107,7 @@
         GlStateManager.popMatrix();
     }
 
-    public void postRenderArm(float scale, EnumHandSide side)
-    {
+    public void postRenderArm(float scale, EnumHandSide side) {
         ModelRenderer modelrenderer = this.getArmForSide(side);
         boolean flag = modelrenderer.showModel;
         modelrenderer.showModel = true;
