--- ../src-base/minecraft/net/minecraft/client/model/TextureOffset.java
+++ ../src-work/minecraft/net/minecraft/client/model/TextureOffset.java
@@ -4,13 +4,11 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class TextureOffset
-{
+public class TextureOffset {
     public final int textureOffsetX;
     public final int textureOffsetY;
 
-    public TextureOffset(int textureOffsetXIn, int textureOffsetYIn)
-    {
+    public TextureOffset(int textureOffsetXIn, int textureOffsetYIn) {
         this.textureOffsetX = textureOffsetXIn;
         this.textureOffsetY = textureOffsetYIn;
     }
