--- ../src-base/minecraft/net/minecraftforge/event/entity/EntityTravelToDimensionEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/EntityTravelToDimensionEvent.java
@@ -22,7 +22,6 @@
 import net.minecraft.entity.Entity;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
 
 /**
  * EntityTravelToDimensionEvent is fired before an Entity travels to a dimension.<br>
@@ -37,18 +36,15 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
  **/
 @Cancelable
-public class EntityTravelToDimensionEvent extends EntityEvent
-{
+public class EntityTravelToDimensionEvent extends EntityEvent {
     private final int dimension;
 
-    public EntityTravelToDimensionEvent(Entity entity, int dimension)
-    {
+    public EntityTravelToDimensionEvent(Entity entity, int dimension) {
         super(entity);
         this.dimension = dimension;
     }
 
-    public int getDimension()
-    {
+    public int getDimension() {
         return dimension;
     }
 }
