--- ../src-base/minecraft/net/minecraftforge/event/entity/EntityStruckByLightningEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/EntityStruckByLightningEvent.java
@@ -19,11 +19,11 @@
 
 package net.minecraftforge.event.entity;
 
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.effect.EntityLightningBolt;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.ForgeEventFactory;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraft.entity.Entity;
-import net.minecraft.entity.effect.EntityLightningBolt;
 
 /**
  * EntityStruckByLightningEvent is fired when an Entity is about to be struck by lightening.<br>
@@ -40,18 +40,15 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
  **/
 @Cancelable
-public class EntityStruckByLightningEvent extends EntityEvent
-{
+public class EntityStruckByLightningEvent extends EntityEvent {
     private final EntityLightningBolt lightning;
 
-    public EntityStruckByLightningEvent(Entity entity, EntityLightningBolt lightning)
-    {
+    public EntityStruckByLightningEvent(Entity entity, EntityLightningBolt lightning) {
         super(entity);
         this.lightning = lightning;
     }
 
-    public EntityLightningBolt getLightning()
-    {
+    public EntityLightningBolt getLightning() {
         return lightning;
     }
 }
