--- ../src-base/minecraft/net/minecraftforge/event/entity/EntityMountEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/EntityMountEvent.java
@@ -23,7 +23,6 @@
 import net.minecraft.world.World;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
-import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
 
 /**
  * This event gets fired whenever a entity mounts/dismounts another entity.<br>
@@ -36,50 +35,43 @@
  * This event does not have a result. {@link HasResult}<br>
  *<br>
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
- * 
+ *
  */
 
 @Cancelable
-public class EntityMountEvent extends EntityEvent
-{
-    
+public class EntityMountEvent extends EntityEvent {
+
     private final Entity entityMounting;
     private final Entity entityBeingMounted;
     private final World worldObj;
-    
+
     private final boolean isMounting;
 
-    public EntityMountEvent(Entity entityMounting, Entity entityBeingMounted, World entityWorld, boolean isMounting)
-    {
+    public EntityMountEvent(Entity entityMounting, Entity entityBeingMounted, World entityWorld, boolean isMounting) {
         super(entityMounting);
         this.entityMounting = entityMounting;
         this.entityBeingMounted = entityBeingMounted;
         this.worldObj = entityWorld;
         this.isMounting = isMounting;
     }
-    
-    public boolean isMounting()
-    {
+
+    public boolean isMounting() {
         return isMounting;
     }
-    
-    public boolean isDismounting()
-    {
+
+    public boolean isDismounting() {
         return !isMounting;
     }
 
-    public Entity getEntityMounting()
-    {
+    public Entity getEntityMounting() {
         return entityMounting;
     }
 
-    public Entity getEntityBeingMounted()
-    {
+    public Entity getEntityBeingMounted() {
         return entityBeingMounted;
     }
 
-    public World getWorldObj()
-    {
+    public World getWorldObj() {
         return worldObj;
     }
 }
