--- ../src-base/minecraft/net/minecraftforge/event/entity/EntityMobGriefingEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/EntityMobGriefingEvent.java
@@ -22,7 +22,6 @@
 import net.minecraft.entity.Entity;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
-import net.minecraftforge.fml.common.eventhandler.Event.Result;
 
 /**
  * EntityMobGriefingEvent is fired when mob griefing is about to occur and allows an event listener to specify whether it should or not.<br>
@@ -35,10 +34,8 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  */
 @HasResult
-public class EntityMobGriefingEvent extends EntityEvent
-{
-    public EntityMobGriefingEvent(Entity entity)
-    {
+public class EntityMobGriefingEvent extends EntityEvent {
+    public EntityMobGriefingEvent(Entity entity) {
         super(entity);
     }
 }
