--- ../src-base/minecraft/net/minecraft/client/renderer/block/model/multipart/ConditionPropertyValue.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/model/multipart/ConditionPropertyValue.java
@@ -1,20 +1,16 @@
 package net.minecraft.client.renderer.block.model.multipart;
 
-import com.google.common.base.Function;
-import com.google.common.base.MoreObjects;
-import com.google.common.base.Optional;
-import com.google.common.base.Predicate;
-import com.google.common.base.Predicates;
-import com.google.common.base.Splitter;
+import com.google.common.base.*;
 import com.google.common.collect.Iterables;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.state.BlockStateContainer;
 import net.minecraft.block.state.IBlockState;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class ConditionPropertyValue implements ICondition
 {
