--- ../src-base/minecraft/net/minecraft/command/CommandLocate.java
+++ ../src-work/minecraft/net/minecraft/command/CommandLocate.java
@@ -1,12 +1,13 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.text.TextComponentTranslation;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandLocate extends CommandBase
 {
     public String getName()
