--- ../src-base/minecraft/net/minecraft/command/CommandWorldBorder.java
+++ ../src-work/minecraft/net/minecraft/command/CommandWorldBorder.java
@@ -1,14 +1,15 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.world.border.WorldBorder;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandWorldBorder extends CommandBase
 {
     public String getName()
