--- ../src-base/minecraft/net/minecraft/command/CommandBlockData.java
+++ ../src-work/minecraft/net/minecraft/command/CommandBlockData.java
@@ -1,8 +1,5 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.nbt.JsonToNBT;
 import net.minecraft.nbt.NBTException;
@@ -12,6 +9,10 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandBlockData extends CommandBase
 {
     public String getName()
