--- ../src-base/minecraft/net/minecraft/command/CommandEntityData.java
+++ ../src-work/minecraft/net/minecraft/command/CommandEntityData.java
@@ -1,6 +1,5 @@
 package net.minecraft.command;
 
-import java.util.UUID;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.JsonToNBT;
@@ -8,6 +7,8 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.server.MinecraftServer;
 
+import java.util.UUID;
+
 public class CommandEntityData extends CommandBase
 {
     public String getName()
