--- ../src-base/minecraft/net/minecraft/command/CommandTitle.java
+++ ../src-work/minecraft/net/minecraft/command/CommandTitle.java
@@ -1,9 +1,6 @@
 package net.minecraft.command;
 
 import com.google.gson.JsonParseException;
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.network.play.server.SPacketTitle;
 import net.minecraft.server.MinecraftServer;
@@ -13,6 +10,10 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandTitle extends CommandBase
 {
     private static final Logger LOGGER = LogManager.getLogger();
