--- ../src-base/minecraft/net/minecraft/command/CommandSenderWrapper.java
+++ ../src-work/minecraft/net/minecraft/command/CommandSenderWrapper.java
@@ -1,7 +1,5 @@
 package net.minecraft.command;
 
-import java.util.Objects;
-import javax.annotation.Nullable;
 import net.minecraft.entity.Entity;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
@@ -9,9 +7,12 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+import java.util.Objects;
+
 public class CommandSenderWrapper implements ICommandSender
 {
-    private final ICommandSender delegate;
+    public final ICommandSender delegate;
     @Nullable
     private final Vec3d positionVector;
     @Nullable
