--- ../src-base/minecraft/net/minecraft/command/CommandKill.java
+++ ../src-work/minecraft/net/minecraft/command/CommandKill.java
@@ -1,13 +1,14 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandKill extends CommandBase
 {
     public String getName()
