--- ../src-base/minecraft/net/minecraft/command/CommandReplaceItem.java
+++ ../src-work/minecraft/net/minecraft/command/CommandReplaceItem.java
@@ -1,10 +1,6 @@
 package net.minecraft.command;
 
 import com.google.common.collect.Maps;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
@@ -20,6 +16,11 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
 public class CommandReplaceItem extends CommandBase
 {
     private static final Map<String, Integer> SHORTCUTS = Maps.<String, Integer>newHashMap();
