--- ../src-base/minecraft/net/minecraft/command/CommandNotFoundException.java
+++ ../src-work/minecraft/net/minecraft/command/CommandNotFoundException.java
@@ -1,19 +1,15 @@
 package net.minecraft.command;
 
-public class CommandNotFoundException extends CommandException
-{
-    public CommandNotFoundException()
-    {
+public class CommandNotFoundException extends CommandException {
+    public CommandNotFoundException() {
         this("commands.generic.notFound");
     }
 
-    public CommandNotFoundException(String message, Object... args)
-    {
+    public CommandNotFoundException(String message, Object... args) {
         super(message, args);
     }
 
-    public synchronized Throwable fillInStackTrace()
-    {
+    public synchronized Throwable fillInStackTrace() {
         return this;
     }
 }
