--- ../src-base/minecraft/net/minecraft/command/CommandDifficulty.java
+++ ../src-work/minecraft/net/minecraft/command/CommandDifficulty.java
@@ -1,13 +1,14 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.world.EnumDifficulty;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandDifficulty extends CommandBase
 {
     public String getName()
