--- ../src-base/minecraft/net/minecraft/command/CommandTP.java
+++ ../src-work/minecraft/net/minecraft/command/CommandTP.java
@@ -1,10 +1,5 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
-import java.util.Set;
-import javax.annotation.Nullable;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayerMP;
@@ -13,6 +8,12 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
 public class CommandTP extends CommandBase
 {
     public String getName()
@@ -74,25 +75,12 @@
             {
                 Entity entity1 = getEntity(server, sender, args[args.length - 1]);
 
-                if (entity1.world != entity.world)
-                {
-                    throw new CommandException("commands.tp.notSameDimension", new Object[0]);
+                // CraftBukkit Start
+                // Use Bukkit teleport method in all cases. It has cross dimensional handling, events
+                if (entity.getBukkitEntity().teleport(entity1.getBukkitEntity(), org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.COMMAND)) {
+                    notifyCommandListener(sender, (ICommand) this, "commands.tp.success", new Object[] { entity.getName(), entity1.getName()});
+                    // CraftBukkit End
                 }
-                else
-                {
-                    entity.dismountRidingEntity();
-
-                    if (entity instanceof EntityPlayerMP)
-                    {
-                        ((EntityPlayerMP)entity).connection.setPlayerLocation(entity1.posX, entity1.posY, entity1.posZ, entity1.rotationYaw, entity1.rotationPitch);
-                    }
-                    else
-                    {
-                        entity.setLocationAndAngles(entity1.posX, entity1.posY, entity1.posZ, entity1.rotationYaw, entity1.rotationPitch);
-                    }
-
-                    notifyCommandListener(sender, this, "commands.tp.success", new Object[] {entity.getName(), entity1.getName()});
-                }
             }
         }
     }
@@ -143,7 +131,7 @@
             }
 
             teleportingEntity.dismountRidingEntity();
-            ((EntityPlayerMP)teleportingEntity).connection.setPlayerLocation(argX.getAmount(), argY.getAmount(), argZ.getAmount(), f, f1, set);
+            ((EntityPlayerMP)teleportingEntity).connection.setPlayerLocation(argX.getAmount(), argY.getAmount(), argZ.getAmount(), f, f1, set, org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.COMMAND);
             teleportingEntity.setRotationYawHead(f);
         }
         else
