--- ../src-base/minecraft/net/minecraft/command/CommandEffect.java
+++ ../src-work/minecraft/net/minecraft/command/CommandEffect.java
@@ -1,8 +1,5 @@
 package net.minecraft.command;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.potion.Potion;
 import net.minecraft.potion.PotionEffect;
@@ -10,6 +7,10 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.text.TextComponentTranslation;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 public class CommandEffect extends CommandBase
 {
     public String getName()
