--- ../src-base/minecraft/net/minecraft/command/ICommandManager.java
+++ ../src-work/minecraft/net/minecraft/command/ICommandManager.java
@@ -1,12 +1,12 @@
 package net.minecraft.command;
 
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nullable;
 import java.util.List;
 import java.util.Map;
-import javax.annotation.Nullable;
-import net.minecraft.util.math.BlockPos;
 
-public interface ICommandManager
-{
+public interface ICommandManager {
     int executeCommand(ICommandSender sender, String rawCommand);
 
     List<String> getTabCompletions(ICommandSender sender, String input, @Nullable BlockPos pos);
@@ -14,4 +14,6 @@
     List<ICommand> getPossibleCommands(ICommandSender sender);
 
     Map<String, ICommand> getCommands();
+
+    Map<String, ICommand> getCommandMod();
 }
