--- ../src-base/minecraft/net/minecraft/command/InvalidBlockStateException.java
+++ ../src-work/minecraft/net/minecraft/command/InvalidBlockStateException.java
@@ -1,19 +1,15 @@
 package net.minecraft.command;
 
-public class InvalidBlockStateException extends CommandException
-{
-    public InvalidBlockStateException()
-    {
+public class InvalidBlockStateException extends CommandException {
+    public InvalidBlockStateException() {
         this("commands.generic.blockstate.invalid");
     }
 
-    public InvalidBlockStateException(String message, Object... objects)
-    {
+    public InvalidBlockStateException(String message, Object... objects) {
         super(message, objects);
     }
 
-    public synchronized Throwable fillInStackTrace()
-    {
+    public synchronized Throwable fillInStackTrace() {
         return this;
     }
 }
