--- ../src-base/minecraft/net/minecraft/command/PlayerNotFoundException.java
+++ ../src-work/minecraft/net/minecraft/command/PlayerNotFoundException.java
@@ -1,19 +1,15 @@
 package net.minecraft.command;
 
-public class PlayerNotFoundException extends CommandException
-{
-    public PlayerNotFoundException(String message)
-    {
+public class PlayerNotFoundException extends CommandException {
+    public PlayerNotFoundException(String message) {
         super(message);
     }
 
-    public PlayerNotFoundException(String message, Object... replacements)
-    {
+    public PlayerNotFoundException(String message, Object... replacements) {
         super(message, replacements);
     }
 
-    public synchronized Throwable fillInStackTrace()
-    {
+    public synchronized Throwable fillInStackTrace() {
         return this;
     }
 }
