--- ../src-base/minecraft/net/minecraft/command/WrongUsageException.java
+++ ../src-work/minecraft/net/minecraft/command/WrongUsageException.java
@@ -1,14 +1,11 @@
 package net.minecraft.command;
 
-public class WrongUsageException extends SyntaxErrorException
-{
-    public WrongUsageException(String message, Object... replacements)
-    {
+public class WrongUsageException extends SyntaxErrorException {
+    public WrongUsageException(String message, Object... replacements) {
         super(message, replacements);
     }
 
-    public synchronized Throwable fillInStackTrace()
-    {
+    public synchronized Throwable fillInStackTrace() {
         return this;
     }
 }
