--- ../src-base/minecraft/net/minecraftforge/client/resource/ISelectiveResourceReloadListener.java
+++ ../src-work/minecraft/net/minecraftforge/client/resource/ISelectiveResourceReloadListener.java
@@ -19,16 +19,14 @@
 
 package net.minecraftforge.client.resource;
 
-import java.util.function.Predicate;
-
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraft.client.resources.IResourceManagerReloadListener;
 
-public interface ISelectiveResourceReloadListener extends IResourceManagerReloadListener
-{
+import java.util.function.Predicate;
+
+public interface ISelectiveResourceReloadListener extends IResourceManagerReloadListener {
     @Override
-    default void onResourceManagerReload(IResourceManager resourceManager)
-    {
+    default void onResourceManagerReload(IResourceManager resourceManager) {
         // For compatibility, call the selective version from the non-selective function
         onResourceManagerReload(resourceManager, SelectiveReloadStateHandler.INSTANCE.get());
     }
