--- ../src-base/minecraft/net/minecraftforge/client/resource/ReloadRequirements.java
+++ ../src-work/minecraft/net/minecraftforge/client/resource/ReloadRequirements.java
@@ -19,13 +19,12 @@
 
 package net.minecraftforge.client.resource;
 
-import java.util.Set;
-import java.util.function.Predicate;
-
+import com.google.common.collect.Sets;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-import com.google.common.collect.Sets;
+import java.util.Set;
+import java.util.function.Predicate;
 
 /**
  * Holds methods to create standard predicates to select {@link IResourceType}s that should be reloaded.
