--- ../src-base/minecraft/net/minecraftforge/client/resource/ReloadRequirements.java
+++ ../src-work/minecraft/net/minecraftforge/client/resource/ReloadRequirements.java
@@ -19,27 +19,24 @@
 
 package net.minecraftforge.client.resource;
 
-import java.util.Set;
-import java.util.function.Predicate;
-
+import com.google.common.collect.Sets;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-import com.google.common.collect.Sets;
+import java.util.Set;
+import java.util.function.Predicate;
 
 /**
  * Holds methods to create standard predicates to select {@link IResourceType}s that should be reloaded.
  */
 @SideOnly(Side.CLIENT)
-public final class ReloadRequirements
-{
+public final class ReloadRequirements {
     /**
      * Creates a reload predicate accepting all resource types.
      *
      * @return a predicate accepting all types
      */
-    public static Predicate<IResourceType> all()
-    {
+    public static Predicate<IResourceType> all() {
         return type -> true;
     }
 
@@ -49,8 +46,7 @@
      * @param inclusion the set of resource types to be included in the reload
      * @return an inclusion predicate based on the given types
      */
-    public static Predicate<IResourceType> include(IResourceType... inclusion)
-    {
+    public static Predicate<IResourceType> include(IResourceType... inclusion) {
         Set<IResourceType> inclusionSet = Sets.newHashSet(inclusion);
         return inclusionSet::contains;
     }
