--- ../src-base/minecraft/net/minecraftforge/client/resource/SelectiveReloadStateHandler.java
+++ ../src-work/minecraft/net/minecraftforge/client/resource/SelectiveReloadStateHandler.java
@@ -19,11 +19,11 @@
 
 package net.minecraftforge.client.resource;
 
+import net.minecraftforge.common.ForgeModContainer;
+
 import javax.annotation.Nullable;
 import java.util.function.Predicate;
 
-import net.minecraftforge.common.ForgeModContainer;
-
 /**
  * Handles reload parameters for selective loaders.
  */
