--- ../src-base/minecraft/net/minecraftforge/client/ClientCommandHandler.java
+++ ../src-work/minecraft/net/minecraftforge/client/ClientCommandHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,23 +19,19 @@
 
 package net.minecraftforge.client;
 
-import java.util.List;
-
+import cn.pfcraft.server.Mohist;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiChat;
-import net.minecraft.command.CommandException;
-import net.minecraft.command.CommandHandler;
-import net.minecraft.command.ICommand;
-import net.minecraft.command.ICommandSender;
-import net.minecraft.command.WrongUsageException;
+import net.minecraft.command.*;
 import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.TextFormatting;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.CommandEvent;
 import net.minecraftforge.fml.client.FMLClientHandler;
-import net.minecraft.util.text.TextComponentTranslation;
-import net.minecraft.util.text.TextFormatting;
-import net.minecraftforge.fml.common.FMLLog;
 
+import java.util.List;
+
 import static net.minecraft.util.text.TextFormatting.*;
 
 /**
@@ -111,7 +107,7 @@
         catch (Throwable t)
         {
             sender.sendMessage(format(RED, "commands.generic.exception"));
-            FMLLog.log.error("Command '{}' threw an exception:", message, t);
+            Mohist.LOGGER.error("Command '{}' threw an exception:", message, t);
         }
 
         return -1;
