--- ../src-base/minecraft/net/minecraftforge/client/CloudRenderer.java
+++ ../src-work/minecraft/net/minecraftforge/client/CloudRenderer.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,21 +19,8 @@
 
 package net.minecraftforge.client;
 
-import javax.annotation.Nonnull;
-import java.nio.ByteBuffer;
-import java.util.function.Predicate;
-
-import net.minecraftforge.client.resource.IResourceType;
-import net.minecraftforge.client.resource.VanillaResourceType;
-import org.lwjgl.opengl.GL11;
-
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.renderer.BufferBuilder;
-import net.minecraft.client.renderer.EntityRenderer;
-import net.minecraft.client.renderer.GLAllocation;
-import net.minecraft.client.renderer.GlStateManager;
-import net.minecraft.client.renderer.OpenGlHelper;
-import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.*;
 import net.minecraft.client.renderer.texture.DynamicTexture;
 import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
 import net.minecraft.client.renderer.vertex.VertexBuffer;
@@ -44,9 +31,16 @@
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.math.Vec3d;
-import net.minecraftforge.common.ForgeModContainer;
+import net.minecraftforge.client.resource.IResourceType;
 import net.minecraftforge.client.resource.ISelectiveResourceReloadListener;
+import net.minecraftforge.client.resource.VanillaResourceType;
+import net.minecraftforge.common.ForgeModContainer;
+import org.lwjgl.opengl.GL11;
 
+import javax.annotation.Nonnull;
+import java.nio.ByteBuffer;
+import java.util.function.Predicate;
+
 public class CloudRenderer implements ISelectiveResourceReloadListener
 {
     // Shared constants.
