--- ../src-base/minecraft/net/minecraftforge/client/FluidContainerColorer.java
+++ ../src-work/minecraft/net/minecraftforge/client/FluidContainerColorer.java
@@ -26,11 +26,9 @@
 
 import javax.annotation.Nonnull;
 
-public class FluidContainerColorer implements IItemColor
-{
+public class FluidContainerColorer implements IItemColor {
     @Override
-    public int colorMultiplier(@Nonnull ItemStack stack, int tintIndex)
-    {
+    public int colorMultiplier(@Nonnull ItemStack stack, int tintIndex) {
         if (tintIndex != 1) return 0xFFFFFFFF;
         FluidStack fluidStack = FluidUtil.getFluidContained(stack);
         if (fluidStack == null) return 0xFFFFFFFF;
