--- ../src-base/minecraft/net/minecraftforge/client/IClientCommand.java
+++ ../src-work/minecraft/net/minecraftforge/client/IClientCommand.java
@@ -25,8 +25,7 @@
 /**
  * Client-side commands can implement this interface to allow additional control over when the command may be used.
  */
-public interface IClientCommand extends ICommand
-{
+public interface IClientCommand extends ICommand {
 
     /**
      * Determine whether this command can be used without the "/" prefix. By default this is true.
