--- ../src-base/minecraft/net/minecraftforge/client/IRenderContextHandler.java
+++ ../src-work/minecraft/net/minecraftforge/client/IRenderContextHandler.java
@@ -23,8 +23,7 @@
  * @deprecated TODO remove in 1.13. This has never been used
  */
 @Deprecated
-public interface IRenderContextHandler
-{
+public interface IRenderContextHandler {
     /** Run before the specified rendering context.
      */
     void beforeRenderContext();
