--- ../src-base/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderWorker.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderWorker.java
@@ -4,10 +4,6 @@
 import com.google.common.util.concurrent.FutureCallback;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.CancellationException;
-import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.RegionRenderCacheBuilder;
 import net.minecraft.crash.CrashReport;
@@ -21,6 +17,11 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+
 @SideOnly(Side.CLIENT)
 public class ChunkRenderWorker implements Runnable
 {
