--- ../src-base/minecraft/net/minecraft/client/renderer/chunk/IRenderChunkFactory.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/chunk/IRenderChunkFactory.java
@@ -6,7 +6,6 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface IRenderChunkFactory
-{
+public interface IRenderChunkFactory {
     RenderChunk create(World worldIn, RenderGlobal renderGlobalIn, int index);
 }
