--- ../src-base/minecraft/net/minecraft/client/renderer/chunk/SetVisibility.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/chunk/SetVisibility.java
@@ -1,11 +1,12 @@
 package net.minecraft.client.renderer.chunk;
 
-import java.util.BitSet;
-import java.util.Set;
 import net.minecraft.util.EnumFacing;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.BitSet;
+import java.util.Set;
+
 @SideOnly(Side.CLIENT)
 public class SetVisibility
 {
