--- ../src-base/minecraft/net/minecraft/client/renderer/chunk/VboChunkFactory.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/chunk/VboChunkFactory.java
@@ -6,10 +6,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class VboChunkFactory implements IRenderChunkFactory
-{
-    public RenderChunk create(World worldIn, RenderGlobal renderGlobalIn, int index)
-    {
+public class VboChunkFactory implements IRenderChunkFactory {
+    public RenderChunk create(World worldIn, RenderGlobal renderGlobalIn, int index) {
         return new RenderChunk(worldIn, renderGlobalIn, index);
     }
 }
