--- ../src-base/minecraft/net/minecraftforge/common/config/FieldWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/FieldWrapper.java
@@ -1,24 +1,10 @@
-/*
- * Minecraft Forge
- * Copyright (c) 2016-2018.
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation version 2.1
- * of the License.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
- */
-
 package net.minecraftforge.common.config;
 
+import com.google.common.collect.Lists;
+import net.minecraftforge.common.config.Config.RangeDouble;
+import net.minecraftforge.common.config.Config.RangeInt;
+import org.apache.commons.lang3.StringUtils;
+
 import java.lang.reflect.Field;
 import java.lang.reflect.GenericArrayType;
 import java.lang.reflect.ParameterizedType;
@@ -27,12 +13,6 @@
 import java.util.Map.Entry;
 import java.util.regex.Pattern;
 
-import com.google.common.collect.Lists;
-
-import net.minecraftforge.common.config.Config.RangeDouble;
-import net.minecraftforge.common.config.Config.RangeInt;
-import org.apache.commons.lang3.StringUtils;
-
 import static net.minecraftforge.common.config.ConfigManager.*;
 
 public abstract class FieldWrapper implements IFieldWrapper
@@ -205,7 +185,6 @@
         {
             return StringUtils.replaceOnce(key, this.baseName, "");
         }
-
     }
 
     private static class EnumWrapper extends SingleValueFieldWrapper
