--- ../src-base/minecraft/net/minecraftforge/common/config/IFieldWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/IFieldWrapper.java
@@ -24,14 +24,13 @@
  * In general: The key is the fully qualified property name, where each subcategory is appended with a dot.
  * i.e: general.map.isPresent
  */
-public interface IFieldWrapper
-{
+public interface IFieldWrapper {
 
     /**
      * @return The type adapter to serialize the values returned by getValue. Null if non-primitive.
      */
     ITypeAdapter getTypeAdapter();
-    
+
     /**
      * @return a list of fully qualified property keys handled by this field
      */
@@ -77,11 +76,10 @@
      * @param hasSlidingControl true if the property is going to have a slider control attached in the configuration UI;
      *                          works in conjunction with {@link Config.RangeInt} and {@link Config.RangeDouble}
      */
-    default void setupConfiguration(Configuration cfg, String desc, String langKey, boolean reqMCRestart, boolean reqWorldRestart, boolean hasSlidingControl)
-    {
+    default void setupConfiguration(Configuration cfg, String desc, String langKey, boolean reqMCRestart, boolean reqWorldRestart, boolean hasSlidingControl) {
         setupConfiguration(cfg, desc, langKey, reqMCRestart, reqWorldRestart);
     }
-    
+
     /**
      * i.e. general.map in the example above
      * @return the category name in which the entries should be saved. This includes the parent categories
