--- ../src-base/minecraft/net/minecraftforge/common/config/IFieldWrapper.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/IFieldWrapper.java
@@ -31,7 +31,7 @@
      * @return The type adapter to serialize the values returned by getValue. Null if non-primitive.
      */
     ITypeAdapter getTypeAdapter();
-    
+
     /**
      * @return a list of fully qualified property keys handled by this field
      */
@@ -81,7 +81,7 @@
     {
         setupConfiguration(cfg, desc, langKey, reqMCRestart, reqWorldRestart);
     }
-    
+
     /**
      * i.e. general.map in the example above
      * @return the category name in which the entries should be saved. This includes the parent categories
