--- ../src-base/minecraft/net/minecraftforge/common/config/ITypeAdapter.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/ITypeAdapter.java
@@ -24,8 +24,7 @@
 /**
  * Abstracts the types of properties away. Higher level logic must prevent invalid data types.
  */
-interface ITypeAdapter
-{
+interface ITypeAdapter {
     /**
      * Assigns the default value to the property
      * @param property the property whose default value will be assigned
@@ -38,14 +37,14 @@
      * @param property the property whose value will be set
      * @param value the set value
      */
-	void setValue(Property property, Object value);
+    void setValue(Property property, Object value);
 
     /**
      * Retrieves the properties value
      * @param prop the property whose value will be retrieved
      * @return the properties value
      */
-	Object getValue(Property prop);
+    Object getValue(Property prop);
 
     Type getType();
 
