--- ../src-base/minecraft/net/minecraftforge/common/config/Property.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/Property.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,18 +19,14 @@
 
 package net.minecraftforge.common.config;
 
+import net.minecraftforge.fml.client.config.*;
+import net.minecraftforge.fml.client.config.GuiConfigEntries.IConfigEntry;
+import net.minecraftforge.fml.client.config.GuiEditArrayEntries.IArrayEntry;
+
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.regex.Pattern;
 
-import net.minecraftforge.fml.client.config.GuiConfig;
-import net.minecraftforge.fml.client.config.GuiConfigEntries;
-import net.minecraftforge.fml.client.config.GuiConfigEntries.IConfigEntry;
-import net.minecraftforge.fml.client.config.GuiEditArray;
-import net.minecraftforge.fml.client.config.GuiEditArrayEntries;
-import net.minecraftforge.fml.client.config.GuiEditArrayEntries.IArrayEntry;
-import net.minecraftforge.fml.client.config.IConfigElement;
-
 public class Property
 {
     public String getComment()
@@ -82,7 +78,7 @@
     private String langKey;
     private String minValue;
     private String maxValue;
-    /** If true, will have a slider attached automatically in configuration UI */
+
     private boolean hasSlidingControl;
 
     private Class<? extends IConfigEntry> configEntryClass = null;
