--- ../src-base/minecraft/net/minecraftforge/common/config/TypeAdapters.java
+++ ../src-work/minecraft/net/minecraftforge/common/config/TypeAdapters.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -22,17 +22,12 @@
 //=========================================================
 // Run away thar' be dragons!
 //=========================================================
-import java.util.Arrays;
 
-import com.google.common.primitives.Booleans;
-import com.google.common.primitives.Bytes;
-import com.google.common.primitives.Doubles;
-import com.google.common.primitives.Floats;
-import com.google.common.primitives.Ints;
-import com.google.common.primitives.Shorts;
-
+import com.google.common.primitives.*;
 import net.minecraftforge.common.config.Property.Type;
 
+import java.util.Arrays;
+
 class TypeAdapters
 {
    /*
