--- ../src-base/minecraft/net/minecraftforge/server/permission/context/ContextKeys.java
+++ ../src-work/minecraft/net/minecraftforge/server/permission/context/ContextKeys.java
@@ -28,8 +28,7 @@
 /**
  * Some default context keys, for easier compatibility
  */
-public class ContextKeys
-{
+public class ContextKeys {
     /**
      * BlockPos for interacting, breaking and other permissions
      */
