--- ../src-base/minecraft/net/minecraftforge/server/permission/context/IContext.java
+++ ../src-work/minecraft/net/minecraftforge/server/permission/context/IContext.java
@@ -27,8 +27,7 @@
 /**
  * Use {@link BlockPosContext} or {@link PlayerContext} when possible
  */
-public interface IContext
-{
+public interface IContext {
     /**
      * World from where permission is requested. Can be null
      */
