--- ../src-base/minecraft/net/minecraftforge/server/permission/context/TargetContext.java
+++ ../src-work/minecraft/net/minecraftforge/server/permission/context/TargetContext.java
@@ -24,26 +24,22 @@
 
 import javax.annotation.Nullable;
 
-public class TargetContext extends PlayerContext
-{
+public class TargetContext extends PlayerContext {
     private final Entity target;
 
-    public TargetContext(EntityPlayer ep, @Nullable Entity entity)
-    {
+    public TargetContext(EntityPlayer ep, @Nullable Entity entity) {
         super(ep);
         target = entity;
     }
 
     @Override
     @Nullable
-    public <T> T get(ContextKey<T> key)
-    {
+    public <T> T get(ContextKey<T> key) {
         return key.equals(ContextKeys.TARGET) ? (T) target : super.get(key);
     }
 
     @Override
-    protected boolean covers(ContextKey<?> key)
-    {
+    protected boolean covers(ContextKey<?> key) {
         return target != null && key.equals(ContextKeys.TARGET);
     }
 }
