--- ../src-base/minecraft/net/minecraft/client/renderer/block/statemap/IStateMapper.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/statemap/IStateMapper.java
@@ -1,14 +1,14 @@
 package net.minecraft.client.renderer.block.statemap;
 
-import java.util.Map;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.renderer.block.model.ModelResourceLocation;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
-public interface IStateMapper
-{
+public interface IStateMapper {
     Map<IBlockState, ModelResourceLocation> putStateModelLocations(Block blockIn);
 }
