--- ../src-base/minecraft/net/minecraft/network/handshake/INetHandlerHandshakeServer.java
+++ ../src-work/minecraft/net/minecraft/network/handshake/INetHandlerHandshakeServer.java
@@ -3,7 +3,6 @@
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.handshake.client.C00Handshake;
 
-public interface INetHandlerHandshakeServer extends INetHandler
-{
+public interface INetHandlerHandshakeServer extends INetHandler {
     void processHandshake(C00Handshake packetIn);
 }
