--- ../src-base/minecraft/net/minecraftforge/fml/client/GuiCustomModLoadingErrorScreen.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/GuiCustomModLoadingErrorScreen.java
@@ -19,22 +19,21 @@
 
 package net.minecraftforge.fml.client;
 
-public class GuiCustomModLoadingErrorScreen extends GuiErrorBase
-{
+public class GuiCustomModLoadingErrorScreen extends GuiErrorBase {
     private CustomModLoadingErrorDisplayException customException;
-    public GuiCustomModLoadingErrorScreen(CustomModLoadingErrorDisplayException customException)
-    {
+
+    public GuiCustomModLoadingErrorScreen(CustomModLoadingErrorDisplayException customException) {
         this.customException = customException;
     }
+
     @Override
-    public void initGui()
-    {
+    public void initGui() {
         super.initGui();
         this.customException.initGui(this, fontRenderer);
     }
+
     @Override
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         this.drawDefaultBackground();
         super.drawScreen(mouseX, mouseY, partialTicks);
         this.customException.drawScreen(this, fontRenderer, mouseX, mouseY, partialTicks);
