--- ../src-base/minecraft/net/minecraftforge/fml/client/GuiErrorBase.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/GuiErrorBase.java
@@ -28,23 +28,20 @@
 import java.awt.*;
 import java.io.File;
 
-public class GuiErrorBase extends GuiErrorScreen
-{
+public class GuiErrorBase extends GuiErrorScreen {
     static final File minecraftDir = new File(Loader.instance().getConfigDir().getParent());
     static final File logFile = new File(minecraftDir, "logs/latest.log");
-    public GuiErrorBase()
-    {
+
+    public GuiErrorBase() {
         super(null, null);
     }
 
-    private String translateOrDefault(String translateKey, String alternative, Object... format)
-    {
+    private String translateOrDefault(String translateKey, String alternative, Object... format) {
         return I18n.hasKey(translateKey) ? I18n.format(translateKey, format) : String.format(alternative, format); //When throwing a DuplicateModsException, the translation system does not work...
     }
 
     @Override
-    public void initGui()
-    {
+    public void initGui() {
         super.initGui();
         this.buttonList.clear();
         this.buttonList.add(new GuiButton(10, 50, this.height - 38, this.width / 2 - 55, 20, translateOrDefault("fml.button.open.mods.folder", "Open Mods Folder")));
@@ -53,38 +50,26 @@
     }
 
     @Override
-    protected void actionPerformed(GuiButton button)
-    {
-        if (button.id == 10)
-        {
-            try
-            {
+    protected void actionPerformed(GuiButton button) {
+        if (button.id == 10) {
+            try {
                 File modsDir = new File(minecraftDir, "mods");
                 Desktop.getDesktop().open(modsDir);
-            }
-            catch (Exception e)
-            {
+            } catch (Exception e) {
                 FMLLog.log.error("Problem opening mods folder", e);
             }
-        }
-        else if (button.id == 11)
-        {
-            try
-            {
+        } else if (button.id == 11) {
+            try {
                 Desktop.getDesktop().open(logFile);
-            }
-            catch (Exception e)
-            {
+            } catch (Exception e) {
                 FMLLog.log.error("Problem opening log file {}", logFile, e);
             }
         }
     }
 
     @Override
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
-        for(GuiButton button : buttonList)
-        {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+        for (GuiButton button : buttonList) {
             button.drawButton(this.mc, mouseX, mouseY, partialTicks);
         }
     }
