--- ../src-base/minecraft/net/minecraftforge/fml/client/GuiModsMissing.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/GuiModsMissing.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,8 +19,6 @@
 
 package net.minecraftforge.fml.client;
 
-import java.util.List;
-
 import net.minecraft.client.resources.I18n;
 import net.minecraft.util.text.TextFormatting;
 import net.minecraftforge.fml.common.Loader;
@@ -29,6 +27,8 @@
 import net.minecraftforge.fml.common.versioning.ArtifactVersion;
 import net.minecraftforge.fml.common.versioning.DefaultArtifactVersion;
 
+import java.util.List;
+
 public class GuiModsMissing extends GuiErrorBase
 {
     private MissingModsException modsMissing;
