--- ../src-base/minecraft/net/minecraftforge/fml/client/IDisplayableError.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/IDisplayableError.java
@@ -23,8 +23,7 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public interface IDisplayableError
-{
+public interface IDisplayableError {
     @SideOnly(Side.CLIENT)
     GuiScreen createGui();
 }
