--- ../src-base/minecraft/net/minecraftforge/fml/client/IModGuiFactory.java
+++ ../src-work/minecraft/net/minecraftforge/fml/client/IModGuiFactory.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,10 +19,11 @@
 
 package net.minecraftforge.fml.client;
 
-import java.util.Set;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
 
+import java.util.Set;
+
 /**
  * This is the interface you need to implement if you want to provide a customized config screen.
  * {@link DefaultGuiFactory} provides a default implementation of this interface and will be used
