--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketHeldItemChange.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketHeldItemChange.java
@@ -1,43 +1,37 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketHeldItemChange implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+
+public class SPacketHeldItemChange implements Packet<INetHandlerPlayClient> {
     private int heldItemHotbarIndex;
 
-    public SPacketHeldItemChange()
-    {
+    public SPacketHeldItemChange() {
     }
 
-    public SPacketHeldItemChange(int hotbarIndexIn)
-    {
+    public SPacketHeldItemChange(int hotbarIndexIn) {
         this.heldItemHotbarIndex = hotbarIndexIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.heldItemHotbarIndex = buf.readByte();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeByte(this.heldItemHotbarIndex);
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handleHeldItemChange(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public int getHeldItemHotbarIndex()
-    {
+    public int getHeldItemHotbarIndex() {
         return this.heldItemHotbarIndex;
     }
 }
