--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketTitle.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketTitle.java
@@ -17,7 +17,17 @@
     private int fadeInTime;
     private int displayTime;
     private int fadeOutTime;
+    // Paper start
+    public net.md_5.bungee.api.chat.BaseComponent[] components;
 
+    public SPacketTitle(Type action, net.md_5.bungee.api.chat.BaseComponent[] components, int fadeIn, int stay, int fadeOut) {
+        this.type = action;
+        this.components = components;
+        this.fadeInTime = fadeIn;
+        this.displayTime = stay;
+        this.fadeOutTime = fadeOut;
+    }
+    // Paper end
     public SPacketTitle()
     {
     }
@@ -64,7 +74,13 @@
 
         if (this.type == SPacketTitle.Type.TITLE || this.type == SPacketTitle.Type.SUBTITLE || this.type == SPacketTitle.Type.ACTIONBAR)
         {
-            buf.writeTextComponent(this.message);
+            // Paper start
+            if (this.components != null) {
+                buf.writeString(net.md_5.bungee.chat.ComponentSerializer.toString(components));
+            } else {
+                buf.writeTextComponent(this.message);
+            }
+            // Paper end
         }
 
         if (this.type == SPacketTitle.Type.TIMES)
