--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketUpdateTileEntity.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketUpdateTileEntity.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -9,57 +8,50 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketUpdateTileEntity implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+
+public class SPacketUpdateTileEntity implements Packet<INetHandlerPlayClient> {
     private BlockPos blockPos;
     private int tileEntityType;
     private NBTTagCompound nbt;
 
-    public SPacketUpdateTileEntity()
-    {
+    public SPacketUpdateTileEntity() {
     }
 
-    public SPacketUpdateTileEntity(BlockPos blockPosIn, int tileEntityTypeIn, NBTTagCompound compoundIn)
-    {
+    public SPacketUpdateTileEntity(BlockPos blockPosIn, int tileEntityTypeIn, NBTTagCompound compoundIn) {
         this.blockPos = blockPosIn;
         this.tileEntityType = tileEntityTypeIn;
         this.nbt = compoundIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.blockPos = buf.readBlockPos();
         this.tileEntityType = buf.readUnsignedByte();
         this.nbt = buf.readCompoundTag();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeBlockPos(this.blockPos);
-        buf.writeByte((byte)this.tileEntityType);
+        buf.writeByte((byte) this.tileEntityType);
         buf.writeCompoundTag(this.nbt);
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handleUpdateTileEntity(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public BlockPos getPos()
-    {
+    public BlockPos getPos() {
         return this.blockPos;
     }
 
     @SideOnly(Side.CLIENT)
-    public int getTileEntityType()
-    {
+    public int getTileEntityType() {
         return this.tileEntityType;
     }
 
     @SideOnly(Side.CLIENT)
-    public NBTTagCompound getNbtCompound()
-    {
+    public NBTTagCompound getNbtCompound() {
         return this.nbt;
     }
 }
