--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketSpawnGlobalEntity.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketSpawnGlobalEntity.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.effect.EntityLightningBolt;
 import net.minecraft.network.Packet;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketSpawnGlobalEntity implements Packet<INetHandlerPlayClient>
 {
     private int entityId;
