--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketStatistics.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketStatistics.java
@@ -1,9 +1,6 @@
 package net.minecraft.network.play.server;
 
 import com.google.common.collect.Maps;
-import java.io.IOException;
-import java.util.Map;
-import java.util.Map.Entry;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
@@ -12,6 +9,10 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.Map;
+import java.util.Map.Entry;
+
 public class SPacketStatistics implements Packet<INetHandlerPlayClient>
 {
     private Map<StatBase, Integer> statisticMap;
