--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketCloseWindow.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketCloseWindow.java
@@ -1,10 +1,11 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 
+import java.io.IOException;
+
 public class SPacketCloseWindow implements Packet<INetHandlerPlayClient>
 {
     private int windowId;
