--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketSoundEffect.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketSoundEffect.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
@@ -10,6 +9,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.lang3.Validate;
 
+import java.io.IOException;
+
 public class SPacketSoundEffect implements Packet<INetHandlerPlayClient>
 {
     private SoundEvent sound;
