--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketBlockChange.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketBlockChange.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.network.Packet;
@@ -11,6 +10,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketBlockChange implements Packet<INetHandlerPlayClient>
 {
     private BlockPos blockPosition;
