--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketBlockAction.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketBlockAction.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.block.Block;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketBlockAction implements Packet<INetHandlerPlayClient>
 {
     private BlockPos blockPosition;
