--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketSpawnPainting.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketSpawnPainting.java
@@ -1,7 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
-import java.util.UUID;
 import net.minecraft.entity.item.EntityPainting;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -11,20 +9,20 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketSpawnPainting implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+import java.util.UUID;
+
+public class SPacketSpawnPainting implements Packet<INetHandlerPlayClient> {
     private int entityID;
     private UUID uniqueId;
     private BlockPos position;
     private EnumFacing facing;
     private String title;
 
-    public SPacketSpawnPainting()
-    {
+    public SPacketSpawnPainting() {
     }
 
-    public SPacketSpawnPainting(EntityPainting painting)
-    {
+    public SPacketSpawnPainting(EntityPainting painting) {
         this.entityID = painting.getEntityId();
         this.uniqueId = painting.getUniqueID();
         this.position = painting.getHangingPosition();
@@ -32,8 +30,7 @@
         this.title = painting.art.title;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.entityID = buf.readVarInt();
         this.uniqueId = buf.readUniqueId();
         this.title = buf.readString(EntityPainting.EnumArt.MAX_NAME_LENGTH);
@@ -41,8 +38,7 @@
         this.facing = EnumFacing.getHorizontal(buf.readUnsignedByte());
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeVarInt(this.entityID);
         buf.writeUniqueId(this.uniqueId);
         buf.writeString(this.title);
@@ -50,38 +46,32 @@
         buf.writeByte(this.facing.getHorizontalIndex());
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handleSpawnPainting(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public int getEntityID()
-    {
+    public int getEntityID() {
         return this.entityID;
     }
 
     @SideOnly(Side.CLIENT)
-    public UUID getUniqueId()
-    {
+    public UUID getUniqueId() {
         return this.uniqueId;
     }
 
     @SideOnly(Side.CLIENT)
-    public BlockPos getPosition()
-    {
+    public BlockPos getPosition() {
         return this.position;
     }
 
     @SideOnly(Side.CLIENT)
-    public EnumFacing getFacing()
-    {
+    public EnumFacing getFacing() {
         return this.facing;
     }
 
     @SideOnly(Side.CLIENT)
-    public String getTitle()
-    {
+    public String getTitle() {
         return this.title;
     }
 }
