--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketTeams.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketTeams.java
@@ -1,8 +1,6 @@
 package net.minecraft.network.play.server;
 
 import com.google.common.collect.Lists;
-import java.io.IOException;
-import java.util.Collection;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
@@ -11,6 +9,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.Collection;
+
 public class SPacketTeams implements Packet<INetHandlerPlayClient>
 {
     private String name = "";
@@ -120,7 +121,7 @@
             buf.writeString(this.suffix);
             buf.writeByte(this.friendlyFlags);
             buf.writeString(this.nameTagVisibility);
-            buf.writeString(this.collisionRule);
+            buf.writeString(!com.destroystokyo.paper.PaperConfig.enablePlayerCollisions ? "never" : this.collisionRule); // Paper
             buf.writeByte(this.color);
         }
 
