--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketPlaceGhostRecipe.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketPlaceGhostRecipe.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.item.crafting.CraftingManager;
 import net.minecraft.item.crafting.IRecipe;
 import net.minecraft.network.Packet;
@@ -9,6 +8,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketPlaceGhostRecipe implements Packet<INetHandlerPlayClient>
 {
     private int field_194314_a;
