--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketChangeGameState.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketChangeGameState.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketChangeGameState implements Packet<INetHandlerPlayClient>
 {
     public static final String[] MESSAGE_NAMES = new String[] {"tile.bed.notValid"};
