--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketSpawnPlayer.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketSpawnPlayer.java
@@ -1,9 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
-import java.util.List;
-import java.util.UUID;
-import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -12,6 +8,11 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
 public class SPacketSpawnPlayer implements Packet<INetHandlerPlayClient>
 {
     private int entityId;
