--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketPlayerAbilities.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketPlayerAbilities.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.entity.player.PlayerCapabilities;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -8,8 +7,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketPlayerAbilities implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+
+public class SPacketPlayerAbilities implements Packet<INetHandlerPlayClient> {
     private boolean invulnerable;
     private boolean flying;
     private boolean allowFlying;
@@ -17,12 +17,10 @@
     private float flySpeed;
     private float walkSpeed;
 
-    public SPacketPlayerAbilities()
-    {
+    public SPacketPlayerAbilities() {
     }
 
-    public SPacketPlayerAbilities(PlayerCapabilities capabilities)
-    {
+    public SPacketPlayerAbilities(PlayerCapabilities capabilities) {
         this.setInvulnerable(capabilities.disableDamage);
         this.setFlying(capabilities.isFlying);
         this.setAllowFlying(capabilities.allowFlying);
@@ -31,8 +29,7 @@
         this.setWalkSpeed(capabilities.getWalkSpeed());
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         byte b0 = buf.readByte();
         this.setInvulnerable((b0 & 1) > 0);
         this.setFlying((b0 & 2) > 0);
@@ -42,28 +39,23 @@
         this.setWalkSpeed(buf.readFloat());
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         byte b0 = 0;
 
-        if (this.isInvulnerable())
-        {
-            b0 = (byte)(b0 | 1);
+        if (this.isInvulnerable()) {
+            b0 = (byte) (b0 | 1);
         }
 
-        if (this.isFlying())
-        {
-            b0 = (byte)(b0 | 2);
+        if (this.isFlying()) {
+            b0 = (byte) (b0 | 2);
         }
 
-        if (this.isAllowFlying())
-        {
-            b0 = (byte)(b0 | 4);
+        if (this.isAllowFlying()) {
+            b0 = (byte) (b0 | 4);
         }
 
-        if (this.isCreativeMode())
-        {
-            b0 = (byte)(b0 | 8);
+        if (this.isCreativeMode()) {
+            b0 = (byte) (b0 | 8);
         }
 
         buf.writeByte(b0);
@@ -71,70 +63,57 @@
         buf.writeFloat(this.walkSpeed);
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handlePlayerAbilities(this);
     }
 
-    public boolean isInvulnerable()
-    {
+    public boolean isInvulnerable() {
         return this.invulnerable;
     }
 
-    public void setInvulnerable(boolean isInvulnerable)
-    {
+    public void setInvulnerable(boolean isInvulnerable) {
         this.invulnerable = isInvulnerable;
     }
 
-    public boolean isFlying()
-    {
+    public boolean isFlying() {
         return this.flying;
     }
 
-    public void setFlying(boolean isFlying)
-    {
+    public void setFlying(boolean isFlying) {
         this.flying = isFlying;
     }
 
-    public boolean isAllowFlying()
-    {
+    public boolean isAllowFlying() {
         return this.allowFlying;
     }
 
-    public void setAllowFlying(boolean isAllowFlying)
-    {
+    public void setAllowFlying(boolean isAllowFlying) {
         this.allowFlying = isAllowFlying;
     }
 
-    public boolean isCreativeMode()
-    {
+    public boolean isCreativeMode() {
         return this.creativeMode;
     }
 
-    public void setCreativeMode(boolean isCreativeMode)
-    {
+    public void setCreativeMode(boolean isCreativeMode) {
         this.creativeMode = isCreativeMode;
     }
 
     @SideOnly(Side.CLIENT)
-    public float getFlySpeed()
-    {
+    public float getFlySpeed() {
         return this.flySpeed;
     }
 
-    public void setFlySpeed(float flySpeedIn)
-    {
+    public void setFlySpeed(float flySpeedIn) {
         this.flySpeed = flySpeedIn;
     }
 
     @SideOnly(Side.CLIENT)
-    public float getWalkSpeed()
-    {
+    public float getWalkSpeed() {
         return this.walkSpeed;
     }
 
-    public void setWalkSpeed(float walkSpeedIn)
-    {
+    public void setWalkSpeed(float walkSpeedIn) {
         this.walkSpeed = walkSpeedIn;
     }
 }
