--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketMaps.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketMaps.java
@@ -1,7 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
-import java.util.Collection;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
@@ -10,6 +8,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.Collection;
+
 public class SPacketMaps implements Packet<INetHandlerPlayClient>
 {
     private int mapId;
