--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketDisplayObjective.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketDisplayObjective.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
@@ -8,6 +7,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketDisplayObjective implements Packet<INetHandlerPlayClient>
 {
     private int position;
