--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketEntityHeadLook.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketEntityHeadLook.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.entity.Entity;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -9,47 +8,41 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketEntityHeadLook implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+
+public class SPacketEntityHeadLook implements Packet<INetHandlerPlayClient> {
     private int entityId;
     private byte yaw;
 
-    public SPacketEntityHeadLook()
-    {
+    public SPacketEntityHeadLook() {
     }
 
-    public SPacketEntityHeadLook(Entity entityIn, byte yawIn)
-    {
+    public SPacketEntityHeadLook(Entity entityIn, byte yawIn) {
         this.entityId = entityIn.getEntityId();
         this.yaw = yawIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.entityId = buf.readVarInt();
         this.yaw = buf.readByte();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeVarInt(this.entityId);
         buf.writeByte(this.yaw);
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handleEntityHeadLook(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public Entity getEntity(World worldIn)
-    {
+    public Entity getEntity(World worldIn) {
         return worldIn.getEntityByID(this.entityId);
     }
 
     @SideOnly(Side.CLIENT)
-    public byte getYaw()
-    {
+    public byte getYaw() {
         return this.yaw;
     }
 }
