--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketUseBed.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketUseBed.java
@@ -1,6 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -10,47 +9,41 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class SPacketUseBed implements Packet<INetHandlerPlayClient>
-{
+import java.io.IOException;
+
+public class SPacketUseBed implements Packet<INetHandlerPlayClient> {
     private int playerID;
     private BlockPos bedPos;
 
-    public SPacketUseBed()
-    {
+    public SPacketUseBed() {
     }
 
-    public SPacketUseBed(EntityPlayer player, BlockPos posIn)
-    {
+    public SPacketUseBed(EntityPlayer player, BlockPos posIn) {
         this.playerID = player.getEntityId();
         this.bedPos = posIn;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.playerID = buf.readVarInt();
         this.bedPos = buf.readBlockPos();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeVarInt(this.playerID);
         buf.writeBlockPos(this.bedPos);
     }
 
-    public void processPacket(INetHandlerPlayClient handler)
-    {
+    public void processPacket(INetHandlerPlayClient handler) {
         handler.handleUseBed(this);
     }
 
     @SideOnly(Side.CLIENT)
-    public EntityPlayer getPlayer(World worldIn)
-    {
-        return (EntityPlayer)worldIn.getEntityByID(this.playerID);
+    public EntityPlayer getPlayer(World worldIn) {
+        return (EntityPlayer) worldIn.getEntityByID(this.playerID);
     }
 
     @SideOnly(Side.CLIENT)
-    public BlockPos getBedPosition()
-    {
+    public BlockPos getBedPosition() {
         return this.bedPos;
     }
 }
