--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketTimeUpdate.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketTimeUpdate.java
@@ -1,12 +1,13 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 public class SPacketTimeUpdate implements Packet<INetHandlerPlayClient>
 {
     private long totalWorldTime;
@@ -30,6 +31,7 @@
                 this.worldTime = -1L;
             }
         }
+        this.totalWorldTime = this.totalWorldTime % 192000;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
