--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketSpawnPainting.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketSpawnPainting.java
@@ -1,7 +1,5 @@
 package net.minecraft.network.play.server;
 
-import java.io.IOException;
-import java.util.UUID;
 import net.minecraft.entity.item.EntityPainting;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
@@ -11,6 +9,9 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.UUID;
+
 public class SPacketSpawnPainting implements Packet<INetHandlerPlayClient>
 {
     private int entityID;
