--- ../src-base/minecraft/net/minecraftforge/fluids/capability/templates/EmptyFluidHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/capability/templates/EmptyFluidHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,15 +19,15 @@
 
 package net.minecraftforge.fluids.capability.templates;
 
-import javax.annotation.Nullable;
-
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.FluidTankInfo;
-import net.minecraftforge.fluids.capability.FluidTankProperties;
 import net.minecraftforge.fluids.IFluidTank;
-import net.minecraftforge.fluids.capability.IFluidTankProperties;
+import net.minecraftforge.fluids.capability.FluidTankProperties;
 import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
 
+import javax.annotation.Nullable;
+
 public class EmptyFluidHandler implements IFluidHandler, IFluidTank
 {
     public static final EmptyFluidHandler INSTANCE = new EmptyFluidHandler();
