--- ../src-base/minecraft/net/minecraft/client/gui/GuiButton.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiButton.java
@@ -10,8 +10,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class GuiButton extends Gui
-{
+public class GuiButton extends Gui {
     protected static final ResourceLocation BUTTON_TEXTURES = new ResourceLocation("textures/gui/widgets.png");
     public int width;
     public int height;
@@ -21,16 +20,14 @@
     public int id;
     public boolean enabled;
     public boolean visible;
-    protected boolean hovered;
     public int packedFGColour; //FML
+    protected boolean hovered;
 
-    public GuiButton(int buttonId, int x, int y, String buttonText)
-    {
+    public GuiButton(int buttonId, int x, int y, String buttonText) {
         this(buttonId, x, y, 200, 20, buttonText);
     }
 
-    public GuiButton(int buttonId, int x, int y, int widthIn, int heightIn, String buttonText)
-    {
+    public GuiButton(int buttonId, int x, int y, int widthIn, int heightIn, String buttonText) {
         this.width = 200;
         this.height = 20;
         this.enabled = true;
@@ -43,26 +40,20 @@
         this.displayString = buttonText;
     }
 
-    protected int getHoverState(boolean mouseOver)
-    {
+    protected int getHoverState(boolean mouseOver) {
         int i = 1;
 
-        if (!this.enabled)
-        {
+        if (!this.enabled) {
             i = 0;
-        }
-        else if (mouseOver)
-        {
+        } else if (mouseOver) {
             i = 2;
         }
 
         return i;
     }
 
-    public void drawButton(Minecraft mc, int mouseX, int mouseY, float partialTicks)
-    {
-        if (this.visible)
-        {
+    public void drawButton(Minecraft mc, int mouseX, int mouseY, float partialTicks) {
+        if (this.visible) {
             FontRenderer fontrenderer = mc.fontRenderer;
             mc.getTextureManager().bindTexture(BUTTON_TEXTURES);
             GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
@@ -76,17 +67,11 @@
             this.mouseDragged(mc, mouseX, mouseY);
             int j = 14737632;
 
-            if (packedFGColour != 0)
-            {
+            if (packedFGColour != 0) {
                 j = packedFGColour;
-            }
-            else
-            if (!this.enabled)
-            {
+            } else if (!this.enabled) {
                 j = 10526880;
-            }
-            else if (this.hovered)
-            {
+            } else if (this.hovered) {
                 j = 16777120;
             }
 
@@ -94,40 +79,32 @@
         }
     }
 
-    protected void mouseDragged(Minecraft mc, int mouseX, int mouseY)
-    {
+    protected void mouseDragged(Minecraft mc, int mouseX, int mouseY) {
     }
 
-    public void mouseReleased(int mouseX, int mouseY)
-    {
+    public void mouseReleased(int mouseX, int mouseY) {
     }
 
-    public boolean mousePressed(Minecraft mc, int mouseX, int mouseY)
-    {
+    public boolean mousePressed(Minecraft mc, int mouseX, int mouseY) {
         return this.enabled && this.visible && mouseX >= this.x && mouseY >= this.y && mouseX < this.x + this.width && mouseY < this.y + this.height;
     }
 
-    public boolean isMouseOver()
-    {
+    public boolean isMouseOver() {
         return this.hovered;
     }
 
-    public void drawButtonForegroundLayer(int mouseX, int mouseY)
-    {
+    public void drawButtonForegroundLayer(int mouseX, int mouseY) {
     }
 
-    public void playPressSound(SoundHandler soundHandlerIn)
-    {
+    public void playPressSound(SoundHandler soundHandlerIn) {
         soundHandlerIn.playSound(PositionedSoundRecord.getMasterRecord(SoundEvents.UI_BUTTON_CLICK, 1.0F));
     }
 
-    public int getButtonWidth()
-    {
+    public int getButtonWidth() {
         return this.width;
     }
 
-    public void setWidth(int width)
-    {
+    public void setWidth(int width) {
         this.width = width;
     }
 }
