--- ../src-base/minecraft/net/minecraft/client/gui/GuiConfirmOpenLink.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiConfirmOpenLink.java
@@ -1,10 +1,11 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.resources.I18n;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiConfirmOpenLink extends GuiYesNo
 {
