--- ../src-base/minecraft/net/minecraft/client/gui/GuiDisconnected.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiDisconnected.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
-import java.util.List;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.util.text.ITextComponent;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class GuiDisconnected extends GuiScreen
 {
