--- ../src-base/minecraft/net/minecraft/client/gui/GuiDownloadTerrain.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiDownloadTerrain.java
@@ -5,22 +5,18 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class GuiDownloadTerrain extends GuiScreen
-{
-    public void initGui()
-    {
+public class GuiDownloadTerrain extends GuiScreen {
+    public void initGui() {
         this.buttonList.clear();
     }
 
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         this.drawBackground(0);
         this.drawCenteredString(this.fontRenderer, I18n.format("multiplayer.downloadingTerrain"), this.width / 2, this.height / 2 - 50, 16777215);
         super.drawScreen(mouseX, mouseY, partialTicks);
     }
 
-    public boolean doesGuiPauseGame()
-    {
+    public boolean doesGuiPauseGame() {
         return false;
     }
 }
