--- ../src-base/minecraft/net/minecraft/client/gui/GuiHopper.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiHopper.java
@@ -11,14 +11,12 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class GuiHopper extends GuiContainer
-{
+public class GuiHopper extends GuiContainer {
     private static final ResourceLocation HOPPER_GUI_TEXTURE = new ResourceLocation("textures/gui/container/hopper.png");
     private final IInventory playerInventory;
     private final IInventory hopperInventory;
 
-    public GuiHopper(InventoryPlayer playerInv, IInventory hopperInv)
-    {
+    public GuiHopper(InventoryPlayer playerInv, IInventory hopperInv) {
         super(new ContainerHopper(playerInv, hopperInv, Minecraft.getMinecraft().player));
         this.playerInventory = playerInv;
         this.hopperInventory = hopperInv;
@@ -26,21 +24,18 @@
         this.ySize = 133;
     }
 
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         this.drawDefaultBackground();
         super.drawScreen(mouseX, mouseY, partialTicks);
         this.renderHoveredToolTip(mouseX, mouseY);
     }
 
-    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
-    {
+    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
         this.fontRenderer.drawString(this.hopperInventory.getDisplayName().getUnformattedText(), 8, 6, 4210752);
         this.fontRenderer.drawString(this.playerInventory.getDisplayName().getUnformattedText(), 8, this.ySize - 96 + 2, 4210752);
     }
 
-    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY)
-    {
+    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
         GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
         this.mc.getTextureManager().bindTexture(HOPPER_GUI_TEXTURE);
         int i = (this.width - this.xSize) / 2;
