--- ../src-base/minecraft/net/minecraft/client/gui/GuiIngameMenu.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiIngameMenu.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.gui.achievement.GuiStats;
 import net.minecraft.client.gui.advancements.GuiScreenAdvancements;
 import net.minecraft.client.multiplayer.WorldClient;
@@ -9,22 +8,21 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
-public class GuiIngameMenu extends GuiScreen
-{
+public class GuiIngameMenu extends GuiScreen {
     private int saveStep;
     private int visibleTime;
 
-    public void initGui()
-    {
+    public void initGui() {
         this.saveStep = 0;
         this.buttonList.clear();
         int i = -16;
         int j = 98;
         this.buttonList.add(new GuiButton(1, this.width / 2 - 100, this.height / 4 + 120 + -16, I18n.format("menu.returnToMenu")));
 
-        if (!this.mc.isIntegratedServerRunning())
-        {
+        if (!this.mc.isIntegratedServerRunning()) {
             (this.buttonList.get(0)).displayString = I18n.format("menu.disconnect");
         }
 
@@ -37,10 +35,8 @@
         this.buttonList.add(new GuiButton(6, this.width / 2 + 2, this.height / 4 + 48 + -16, 98, 20, I18n.format("gui.stats")));
     }
 
-    protected void actionPerformed(GuiButton button) throws IOException
-    {
-        switch (button.id)
-        {
+    protected void actionPerformed(GuiButton button) throws IOException {
+        switch (button.id) {
             case 0:
                 this.mc.displayGuiScreen(new GuiOptions(this, this.mc.gameSettings));
                 break;
@@ -49,19 +45,14 @@
                 boolean flag1 = this.mc.isConnectedToRealms();
                 button.enabled = false;
                 this.mc.world.sendQuittingDisconnectingPacket();
-                this.mc.loadWorld((WorldClient)null);
+                this.mc.loadWorld((WorldClient) null);
 
-                if (flag)
-                {
+                if (flag) {
                     this.mc.displayGuiScreen(new GuiMainMenu());
-                }
-                else if (flag1)
-                {
+                } else if (flag1) {
                     RealmsBridge realmsbridge = new RealmsBridge();
                     realmsbridge.switchToRealms(new GuiMainMenu());
-                }
-                else
-                {
+                } else {
                     this.mc.displayGuiScreen(new GuiMultiplayer(new GuiMainMenu()));
                 }
 
@@ -70,16 +61,16 @@
             default:
                 break;
             case 4:
-                this.mc.displayGuiScreen((GuiScreen)null);
+                this.mc.displayGuiScreen((GuiScreen) null);
                 this.mc.setIngameFocus();
                 break;
             case 5:
                 if (this.mc.player != null)
-                this.mc.displayGuiScreen(new GuiScreenAdvancements(this.mc.player.connection.getAdvancementManager()));
+                    this.mc.displayGuiScreen(new GuiScreenAdvancements(this.mc.player.connection.getAdvancementManager()));
                 break;
             case 6:
                 if (this.mc.player != null)
-                this.mc.displayGuiScreen(new GuiStats(this, this.mc.player.getStatFileWriter()));
+                    this.mc.displayGuiScreen(new GuiStats(this, this.mc.player.getStatFileWriter()));
                 break;
             case 7:
                 this.mc.displayGuiScreen(new GuiShareToLan(this));
@@ -90,14 +81,12 @@
         }
     }
 
-    public void updateScreen()
-    {
+    public void updateScreen() {
         super.updateScreen();
         ++this.visibleTime;
     }
 
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         this.drawDefaultBackground();
         this.drawCenteredString(this.fontRenderer, I18n.format("menu.game"), this.width / 2, 40, 16777215);
         super.drawScreen(mouseX, mouseY, partialTicks);
