--- ../src-base/minecraft/net/minecraft/client/gui/GuiLabel.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiLabel.java
@@ -1,13 +1,14 @@
 package net.minecraft.client.gui;
 
 import com.google.common.collect.Lists;
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.resources.I18n;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class GuiLabel extends Gui
 {
