--- ../src-base/minecraft/net/minecraft/client/gui/GuiListButton.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiListButton.java
@@ -6,14 +6,12 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public class GuiListButton extends GuiButton
-{
-    private boolean value;
+public class GuiListButton extends GuiButton {
     private final String localizationStr;
     private final GuiPageButtonList.GuiResponder guiResponder;
+    private boolean value;
 
-    public GuiListButton(GuiPageButtonList.GuiResponder responder, int buttonId, int x, int y, String localizationStrIn, boolean valueIn)
-    {
+    public GuiListButton(GuiPageButtonList.GuiResponder responder, int buttonId, int x, int y, String localizationStrIn, boolean valueIn) {
         super(buttonId, x, y, 150, 20, "");
         this.localizationStr = localizationStrIn;
         this.value = valueIn;
@@ -21,29 +19,23 @@
         this.guiResponder = responder;
     }
 
-    private String buildDisplayString()
-    {
+    private String buildDisplayString() {
         return I18n.format(this.localizationStr) + ": " + I18n.format(this.value ? "gui.yes" : "gui.no");
     }
 
-    public void setValue(boolean valueIn)
-    {
+    public void setValue(boolean valueIn) {
         this.value = valueIn;
         this.displayString = this.buildDisplayString();
         this.guiResponder.setEntryValue(this.id, valueIn);
     }
 
-    public boolean mousePressed(Minecraft mc, int mouseX, int mouseY)
-    {
-        if (super.mousePressed(mc, mouseX, mouseY))
-        {
+    public boolean mousePressed(Minecraft mc, int mouseX, int mouseY) {
+        if (super.mousePressed(mc, mouseX, mouseY)) {
             this.value = !this.value;
             this.displayString = this.buildDisplayString();
             this.guiResponder.setEntryValue(this.id, this.value);
             return true;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
