--- ../src-base/minecraft/net/minecraft/client/gui/GuiOptions.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiOptions.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.client.settings.GameSettings;
 import net.minecraft.util.text.ITextComponent;
@@ -10,6 +9,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiOptions extends GuiScreen
 {
