--- ../src-base/minecraft/net/minecraft/client/gui/GuiPlayerTabOverlay.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiPlayerTabOverlay.java
@@ -3,9 +3,6 @@
 import com.google.common.collect.ComparisonChain;
 import com.google.common.collect.Ordering;
 import com.mojang.authlib.GameProfile;
-import java.util.Comparator;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.network.NetHandlerPlayClient;
 import net.minecraft.client.network.NetworkPlayerInfo;
@@ -23,10 +20,14 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Comparator;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class GuiPlayerTabOverlay extends Gui
 {
-    private static final Ordering<NetworkPlayerInfo> ENTRY_ORDERING = Ordering.from(new GuiPlayerTabOverlay.PlayerComparator());
+    private static final Ordering<NetworkPlayerInfo> ENTRY_ORDERING = Ordering.from(new PlayerComparator());
     private final Minecraft mc;
     private final GuiIngame guiIngame;
     private ITextComponent footer;
