--- ../src-base/minecraft/net/minecraft/client/gui/GuiResourcePackSelected.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiResourcePackSelected.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.gui;
 
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.client.resources.ResourcePackListEntry;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class GuiResourcePackSelected extends GuiResourcePackList
 {
