--- ../src-base/minecraft/net/minecraft/client/gui/GuiResourcePackList.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiResourcePackList.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.resources.ResourcePackListEntry;
@@ -8,51 +7,45 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
-public abstract class GuiResourcePackList extends GuiListExtended
-{
+public abstract class GuiResourcePackList extends GuiListExtended {
     protected final Minecraft mc;
     protected final List<ResourcePackListEntry> resourcePackEntries;
 
-    public GuiResourcePackList(Minecraft mcIn, int p_i45055_2_, int p_i45055_3_, List<ResourcePackListEntry> p_i45055_4_)
-    {
+    public GuiResourcePackList(Minecraft mcIn, int p_i45055_2_, int p_i45055_3_, List<ResourcePackListEntry> p_i45055_4_) {
         super(mcIn, p_i45055_2_, p_i45055_3_, 32, p_i45055_3_ - 55 + 4, 36);
         this.mc = mcIn;
         this.resourcePackEntries = p_i45055_4_;
         this.centerListVertically = false;
-        this.setHasListHeader(true, (int)((float)mcIn.fontRenderer.FONT_HEIGHT * 1.5F));
+        this.setHasListHeader(true, (int) ((float) mcIn.fontRenderer.FONT_HEIGHT * 1.5F));
     }
 
-    protected void drawListHeader(int insideLeft, int insideTop, Tessellator tessellatorIn)
-    {
+    protected void drawListHeader(int insideLeft, int insideTop, Tessellator tessellatorIn) {
         String s = TextFormatting.UNDERLINE + "" + TextFormatting.BOLD + this.getListHeader();
         this.mc.fontRenderer.drawString(s, insideLeft + this.width / 2 - this.mc.fontRenderer.getStringWidth(s) / 2, Math.min(this.top + 3, insideTop), 16777215);
     }
 
     protected abstract String getListHeader();
 
-    public List<ResourcePackListEntry> getList()
-    {
+    public List<ResourcePackListEntry> getList() {
         return this.resourcePackEntries;
     }
 
-    protected int getSize()
-    {
+    protected int getSize() {
         return this.getList().size();
     }
 
-    public ResourcePackListEntry getListEntry(int index)
-    {
-        return (ResourcePackListEntry)this.getList().get(index);
+    public ResourcePackListEntry getListEntry(int index) {
+        return (ResourcePackListEntry) this.getList().get(index);
     }
 
-    public int getListWidth()
-    {
+    public int getListWidth() {
         return this.width;
     }
 
-    protected int getScrollBarX()
-    {
+    protected int getScrollBarX() {
         return this.right - 6;
     }
 }
