--- ../src-base/minecraft/net/minecraft/client/gui/GuiResourcePackList.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiResourcePackList.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.resources.ResourcePackListEntry;
@@ -8,6 +7,8 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public abstract class GuiResourcePackList extends GuiListExtended
 {
