--- ../src-base/minecraft/net/minecraft/client/gui/GuiScreenServerList.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiScreenServerList.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.multiplayer.ServerData;
 import net.minecraft.client.resources.I18n;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.lwjgl.input.Keyboard;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiScreenServerList extends GuiScreen
 {
