--- ../src-base/minecraft/net/minecraft/client/gui/GuiScreenDemo.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiScreenDemo.java
@@ -1,7 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
-import java.net.URI;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.client.settings.GameSettings;
@@ -11,47 +9,42 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.io.IOException;
+import java.net.URI;
+
 @SideOnly(Side.CLIENT)
-public class GuiScreenDemo extends GuiScreen
-{
+public class GuiScreenDemo extends GuiScreen {
     private static final Logger LOGGER = LogManager.getLogger();
     private static final ResourceLocation DEMO_BACKGROUND_LOCATION = new ResourceLocation("textures/gui/demo_background.png");
 
-    public void initGui()
-    {
+    public void initGui() {
         this.buttonList.clear();
         int i = -16;
         this.buttonList.add(new GuiButton(1, this.width / 2 - 116, this.height / 2 + 62 + -16, 114, 20, I18n.format("demo.help.buy")));
         this.buttonList.add(new GuiButton(2, this.width / 2 + 2, this.height / 2 + 62 + -16, 114, 20, I18n.format("demo.help.later")));
     }
 
-    protected void actionPerformed(GuiButton button) throws IOException
-    {
-        switch (button.id)
-        {
+    protected void actionPerformed(GuiButton button) throws IOException {
+        switch (button.id) {
             case 1:
                 button.enabled = false;
 
-                try
-                {
+                try {
                     Class<?> oclass = Class.forName("java.awt.Desktop");
-                    Object object = oclass.getMethod("getDesktop").invoke((Object)null);
+                    Object object = oclass.getMethod("getDesktop").invoke((Object) null);
                     oclass.getMethod("browse", URI.class).invoke(object, new URI("http://www.minecraft.net/store?source=demo"));
-                }
-                catch (Throwable throwable)
-                {
+                } catch (Throwable throwable) {
                     LOGGER.error("Couldn't open link", throwable);
                 }
 
                 break;
             case 2:
-                this.mc.displayGuiScreen((GuiScreen)null);
+                this.mc.displayGuiScreen((GuiScreen) null);
                 this.mc.setIngameFocus();
         }
     }
 
-    public void drawDefaultBackground()
-    {
+    public void drawDefaultBackground() {
         super.drawDefaultBackground();
         GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
         this.mc.getTextureManager().bindTexture(DEMO_BACKGROUND_LOCATION);
@@ -60,8 +53,7 @@
         this.drawTexturedModalRect(i, j, 0, 0, 248, 166);
     }
 
-    public void drawScreen(int mouseX, int mouseY, float partialTicks)
-    {
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         this.drawDefaultBackground();
         int i = (this.width - 248) / 2 + 10;
         int j = (this.height - 166) / 2 + 8;
