--- ../src-base/minecraft/net/minecraft/client/gui/GuiWorldEdit.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiWorldEdit.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.renderer.OpenGlHelper;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.world.storage.ISaveFormat;
@@ -10,6 +9,8 @@
 import org.apache.commons.io.FileUtils;
 import org.lwjgl.input.Keyboard;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiWorldEdit extends GuiScreen
 {
