--- ../src-base/minecraft/net/minecraft/client/gui/GuiWorldSelection.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiWorldSelection.java
@@ -2,14 +2,15 @@
 
 import com.google.common.base.Splitter;
 import com.google.common.collect.Lists;
-import java.io.IOException;
-import javax.annotation.Nullable;
 import net.minecraft.client.resources.I18n;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class GuiWorldSelection extends GuiScreen
 {
