--- ../src-base/minecraft/net/minecraft/client/gui/IProgressMeter.java
+++ ../src-work/minecraft/net/minecraft/client/gui/IProgressMeter.java
@@ -4,9 +4,8 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public interface IProgressMeter
-{
-    String[] LOADING_STRINGS = new String[] {"oooooo", "Oooooo", "oOoooo", "ooOooo", "oooOoo", "ooooOo", "oooooO"};
+public interface IProgressMeter {
+    String[] LOADING_STRINGS = new String[]{"oooooo", "Oooooo", "oOoooo", "ooOooo", "oooOoo", "ooooOo", "oooooO"};
 
     void onStatsUpdated();
 }
