--- ../src-base/minecraft/net/minecraft/client/gui/ScreenChatOptions.java
+++ ../src-work/minecraft/net/minecraft/client/gui/ScreenChatOptions.java
@@ -1,12 +1,13 @@
 package net.minecraft.client.gui;
 
-import java.io.IOException;
 import net.minecraft.client.gui.chat.NarratorChatListener;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.client.settings.GameSettings;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import java.io.IOException;
+
 @SideOnly(Side.CLIENT)
 public class ScreenChatOptions extends GuiScreen
 {
