--- ../src-base/minecraft/net/minecraftforge/items/IItemHandlerModifiable.java
+++ ../src-work/minecraft/net/minecraftforge/items/IItemHandlerModifiable.java
@@ -23,8 +23,7 @@
 
 import javax.annotation.Nonnull;
 
-public interface IItemHandlerModifiable extends IItemHandler
-{
+public interface IItemHandlerModifiable extends IItemHandler {
     /**
      * Overrides the stack in the given slot. This method is used by the
      * standard Forge helper methods and classes. It is not intended for
