--- ../src-base/minecraft/net/minecraftforge/items/ItemStackHandler.java
+++ ../src-work/minecraft/net/minecraftforge/items/ItemStackHandler.java
@@ -170,6 +170,7 @@
         return true;
     }
 
+
     @Override
     public NBTTagCompound serializeNBT()
     {
@@ -223,4 +224,11 @@
     {
 
     }
+
+    // Mohist start
+    public NonNullList<ItemStack> getStacksList() {
+        return this.stacks;
+    }
+    // Mohist end
+
 }
