--- ../src-base/minecraft/net/minecraftforge/items/ItemStackHandler.java
+++ ../src-work/minecraft/net/minecraftforge/items/ItemStackHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -170,6 +170,7 @@
         return true;
     }
 
+
     @Override
     public NBTTagCompound serializeNBT()
     {
@@ -223,4 +224,11 @@
     {
 
     }
+
+    // Mohist start
+    public NonNullList<ItemStack> getStacksList() {
+        return this.stacks;
+    }
+    // Mohist end
+
 }
