--- ../src-base/minecraft/net/minecraft/client/renderer/block/model/BuiltInModel.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/model/BuiltInModel.java
@@ -1,14 +1,15 @@
 package net.minecraft.client.renderer.block.model;
 
-import java.util.Collections;
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.util.EnumFacing;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
 @SideOnly(Side.CLIENT)
 public class BuiltInModel implements IBakedModel
 {
