--- ../src-base/minecraft/net/minecraft/client/renderer/block/model/FaceBakery.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/model/FaceBakery.java
@@ -1,6 +1,5 @@
 package net.minecraft.client.renderer.block.model;
 
-import javax.annotation.Nullable;
 import net.minecraft.client.renderer.EnumFaceDirection;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.util.EnumFacing;
@@ -12,34 +11,36 @@
 import org.lwjgl.util.vector.Vector3f;
 import org.lwjgl.util.vector.Vector4f;
 
+import javax.annotation.Nullable;
+
 @SideOnly(Side.CLIENT)
 public class FaceBakery
 {
     private static final float SCALE_ROTATION_22_5 = 1.0F / (float)Math.cos(0.39269909262657166D) - 1.0F;
     private static final float SCALE_ROTATION_GENERAL = 1.0F / (float)Math.cos((Math.PI / 4D)) - 1.0F;
-    private static final FaceBakery.Rotation[] UV_ROTATIONS = new FaceBakery.Rotation[ModelRotation.values().length * EnumFacing.values().length];
-    private static final FaceBakery.Rotation UV_ROTATION_0 = new FaceBakery.Rotation()
+    private static final Rotation[] UV_ROTATIONS = new Rotation[ModelRotation.values().length * EnumFacing.values().length];
+    private static final Rotation UV_ROTATION_0 = new Rotation()
     {
         BlockFaceUV makeRotatedUV(float p_188007_1_, float p_188007_2_, float p_188007_3_, float p_188007_4_)
         {
             return new BlockFaceUV(new float[] {p_188007_1_, p_188007_2_, p_188007_3_, p_188007_4_}, 0);
         }
     };
-    private static final FaceBakery.Rotation UV_ROTATION_270 = new FaceBakery.Rotation()
+    private static final Rotation UV_ROTATION_270 = new Rotation()
     {
         BlockFaceUV makeRotatedUV(float p_188007_1_, float p_188007_2_, float p_188007_3_, float p_188007_4_)
         {
             return new BlockFaceUV(new float[] {p_188007_4_, 16.0F - p_188007_1_, p_188007_2_, 16.0F - p_188007_3_}, 270);
         }
     };
-    private static final FaceBakery.Rotation UV_ROTATION_INVERSE = new FaceBakery.Rotation()
+    private static final Rotation UV_ROTATION_INVERSE = new Rotation()
     {
         BlockFaceUV makeRotatedUV(float p_188007_1_, float p_188007_2_, float p_188007_3_, float p_188007_4_)
         {
             return new BlockFaceUV(new float[] {16.0F - p_188007_1_, 16.0F - p_188007_2_, 16.0F - p_188007_3_, 16.0F - p_188007_4_}, 0);
         }
     };
-    private static final FaceBakery.Rotation UV_ROTATION_90 = new FaceBakery.Rotation()
+    private static final Rotation UV_ROTATION_90 = new Rotation()
     {
         BlockFaceUV makeRotatedUV(float p_188007_1_, float p_188007_2_, float p_188007_3_, float p_188007_4_)
         {
@@ -359,7 +360,7 @@
         }
     }
 
-    private static void addUvRotation(ModelRotation p_188013_0_, EnumFacing p_188013_1_, FaceBakery.Rotation p_188013_2_)
+    private static void addUvRotation(ModelRotation p_188013_0_, EnumFacing p_188013_1_, Rotation p_188013_2_)
     {
         UV_ROTATIONS[getIndex(p_188013_0_, p_188013_1_)] = p_188013_2_;
     }
