--- ../src-base/minecraft/net/minecraft/client/renderer/block/model/ModelRotation.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/model/ModelRotation.java
@@ -1,7 +1,6 @@
 package net.minecraft.client.renderer.block.model;
 
 import com.google.common.collect.Maps;
-import java.util.Map;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.relauncher.Side;
@@ -9,6 +8,8 @@
 import org.lwjgl.util.vector.Matrix4f;
 import org.lwjgl.util.vector.Vector3f;
 
+import java.util.Map;
+
 @SideOnly(Side.CLIENT)
 public enum ModelRotation implements net.minecraftforge.common.model.IModelState, net.minecraftforge.common.model.ITransformation
 {
