--- ../src-base/minecraft/net/minecraft/world/gen/feature/WorldGenPumpkin.java
+++ ../src-work/minecraft/net/minecraft/world/gen/feature/WorldGenPumpkin.java
@@ -1,12 +1,13 @@
 package net.minecraft.world.gen.feature;
 
-import java.util.Random;
 import net.minecraft.block.BlockPumpkin;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import java.util.Random;
+
 public class WorldGenPumpkin extends WorldGenerator
 {
     public boolean generate(World worldIn, Random rand, BlockPos position)
