--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/ElderGuardianSplit.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/ElderGuardianSplit.java
@@ -3,19 +3,14 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.IFixableData;
 
-public class ElderGuardianSplit implements IFixableData
-{
-    public int getFixVersion()
-    {
+public class ElderGuardianSplit implements IFixableData {
+    public int getFixVersion() {
         return 700;
     }
 
-    public NBTTagCompound fixTagCompound(NBTTagCompound compound)
-    {
-        if ("Guardian".equals(compound.getString("id")))
-        {
-            if (compound.getBoolean("Elder"))
-            {
+    public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
+        if ("Guardian".equals(compound.getString("id"))) {
+            if (compound.getBoolean("Elder")) {
                 compound.setString("id", "ElderGuardian");
             }
 
