--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/EntityArmorAndHeld.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/EntityArmorAndHeld.java
@@ -5,27 +5,22 @@
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.util.datafix.IFixableData;
 
-public class EntityArmorAndHeld implements IFixableData
-{
-    public int getFixVersion()
-    {
+public class EntityArmorAndHeld implements IFixableData {
+    public int getFixVersion() {
         return 100;
     }
 
-    public NBTTagCompound fixTagCompound(NBTTagCompound compound)
-    {
+    public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
         NBTTagList nbttaglist = compound.getTagList("Equipment", 10);
 
-        if (!nbttaglist.hasNoTags() && !compound.hasKey("HandItems", 10))
-        {
+        if (!nbttaglist.hasNoTags() && !compound.hasKey("HandItems", 10)) {
             NBTTagList nbttaglist1 = new NBTTagList();
             nbttaglist1.appendTag(nbttaglist.get(0));
             nbttaglist1.appendTag(new NBTTagCompound());
             compound.setTag("HandItems", nbttaglist1);
         }
 
-        if (nbttaglist.tagCount() > 1 && !compound.hasKey("ArmorItem", 10))
-        {
+        if (nbttaglist.tagCount() > 1 && !compound.hasKey("ArmorItem", 10)) {
             NBTTagList nbttaglist3 = new NBTTagList();
             nbttaglist3.appendTag(nbttaglist.getCompoundTagAt(1));
             nbttaglist3.appendTag(nbttaglist.getCompoundTagAt(2));
@@ -36,20 +31,17 @@
 
         compound.removeTag("Equipment");
 
-        if (compound.hasKey("DropChances", 9))
-        {
+        if (compound.hasKey("DropChances", 9)) {
             NBTTagList nbttaglist4 = compound.getTagList("DropChances", 5);
 
-            if (!compound.hasKey("HandDropChances", 10))
-            {
+            if (!compound.hasKey("HandDropChances", 10)) {
                 NBTTagList nbttaglist2 = new NBTTagList();
                 nbttaglist2.appendTag(new NBTTagFloat(nbttaglist4.getFloatAt(0)));
                 nbttaglist2.appendTag(new NBTTagFloat(0.0F));
                 compound.setTag("HandDropChances", nbttaglist2);
             }
 
-            if (!compound.hasKey("ArmorDropChances", 10))
-            {
+            if (!compound.hasKey("ArmorDropChances", 10)) {
                 NBTTagList nbttaglist5 = new NBTTagList();
                 nbttaglist5.appendTag(new NBTTagFloat(nbttaglist4.getFloatAt(1)));
                 nbttaglist5.appendTag(new NBTTagFloat(nbttaglist4.getFloatAt(2)));
