--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/ItemIntIDToString.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/ItemIntIDToString.java
@@ -3,32 +3,10 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.IFixableData;
 
-public class ItemIntIDToString implements IFixableData
-{
+public class ItemIntIDToString implements IFixableData {
     private static final String[] ID_MAP = new String[2268];
 
-    public int getFixVersion()
-    {
-        return 102;
-    }
-
-    public NBTTagCompound fixTagCompound(NBTTagCompound compound)
-    {
-        if (compound.hasKey("id", 99))
-        {
-            short short1 = compound.getShort("id");
-
-            if (short1 > 0 && short1 < ID_MAP.length && ID_MAP[short1] != null)
-            {
-                compound.setString("id", ID_MAP[short1]);
-            }
-        }
-
-        return compound;
-    }
-
-    static
-    {
+    static {
         ID_MAP[1] = "minecraft:stone";
         ID_MAP[2] = "minecraft:grass";
         ID_MAP[3] = "minecraft:dirt";
@@ -345,4 +323,20 @@
         ID_MAP[2266] = "minecraft:record_11";
         ID_MAP[2267] = "minecraft:record_wait";
     }
+
+    public int getFixVersion() {
+        return 102;
+    }
+
+    public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
+        if (compound.hasKey("id", 99)) {
+            short short1 = compound.getShort("id");
+
+            if (short1 > 0 && short1 < ID_MAP.length && ID_MAP[short1] != null) {
+                compound.setString("id", ID_MAP[short1]);
+            }
+        }
+
+        return compound;
+    }
 }
