--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguage.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguage.java
@@ -1,20 +1,17 @@
 package net.minecraft.util.datafix.fixes;
 
-import java.util.Locale;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.IFixableData;
 
-public class OptionsLowerCaseLanguage implements IFixableData
-{
-    public int getFixVersion()
-    {
+import java.util.Locale;
+
+public class OptionsLowerCaseLanguage implements IFixableData {
+    public int getFixVersion() {
         return 816;
     }
 
-    public NBTTagCompound fixTagCompound(NBTTagCompound compound)
-    {
-        if (compound.hasKey("lang", 8))
-        {
+    public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
+        if (compound.hasKey("lang", 8)) {
             compound.setString("lang", compound.getString("lang").toLowerCase(Locale.ROOT));
         }
 
