--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/ShulkerBoxEntityColor.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/ShulkerBoxEntityColor.java
@@ -3,18 +3,14 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.IFixableData;
 
-public class ShulkerBoxEntityColor implements IFixableData
-{
-    public int getFixVersion()
-    {
+public class ShulkerBoxEntityColor implements IFixableData {
+    public int getFixVersion() {
         return 808;
     }
 
-    public NBTTagCompound fixTagCompound(NBTTagCompound compound)
-    {
-        if ("minecraft:shulker".equals(compound.getString("id")) && !compound.hasKey("Color", 99))
-        {
-            compound.setByte("Color", (byte)10);
+    public NBTTagCompound fixTagCompound(NBTTagCompound compound) {
+        if ("minecraft:shulker".equals(compound.getString("id")) && !compound.hasKey("Color", 99)) {
+            compound.setByte("Color", (byte) 10);
         }
 
         return compound;
