--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/SignStrictJSON.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/SignStrictJSON.java
@@ -1,13 +1,6 @@
 package net.minecraft.util.datafix.fixes;
 
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonDeserializer;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonParseException;
-import java.lang.reflect.Type;
+import com.google.gson.*;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.JsonUtils;
 import net.minecraft.util.StringUtils;
@@ -15,6 +8,8 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentString;
 
+import java.lang.reflect.Type;
+
 public class SignStrictJSON implements IFixableData
 {
     public static final Gson GSON_INSTANCE = (new GsonBuilder()).registerTypeAdapter(ITextComponent.class, new JsonDeserializer<ITextComponent>()
