--- ../src-base/minecraft/net/minecraft/util/datafix/fixes/StringToUUID.java
+++ ../src-work/minecraft/net/minecraft/util/datafix/fixes/StringToUUID.java
@@ -1,9 +1,10 @@
 package net.minecraft.util.datafix.fixes;
 
-import java.util.UUID;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.datafix.IFixableData;
 
+import java.util.UUID;
+
 public class StringToUUID implements IFixableData
 {
     public int getFixVersion()
