--- ../src-base/minecraft/net/minecraftforge/common/IMinecartCollisionHandler.java
+++ ../src-work/minecraft/net/minecraftforge/common/IMinecartCollisionHandler.java
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.common;
 
-import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityMinecart;
+import net.minecraft.util.math.AxisAlignedBB;
 
 /**
  * This class defines a replacement for the default minecart collision code.
@@ -29,8 +29,7 @@
  * If you use this, make it a configuration option.
  * @author CovertJaguar
  */
-public interface IMinecartCollisionHandler
-{
+public interface IMinecartCollisionHandler {
 
     /**
      * This basically replaces the function of the same name in EntityMinecart.
