--- ../src-base/minecraft/net/minecraftforge/common/IMinecartCollisionHandler.java
+++ ../src-work/minecraft/net/minecraftforge/common/IMinecartCollisionHandler.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,9 +19,9 @@
 
 package net.minecraftforge.common;
 
-import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityMinecart;
+import net.minecraft.util.math.AxisAlignedBB;
 
 /**
  * This class defines a replacement for the default minecart collision code.
@@ -57,7 +57,6 @@
      * @return The collision box, cannot be null.
      */
     AxisAlignedBB getMinecartCollisionBox(EntityMinecart cart);
-
     /**
      * This function replaces the function of the same name in EntityMinecart.
      * It defines whether minecarts are solid to the player.
@@ -66,4 +65,3 @@
      */
     AxisAlignedBB getBoundingBox(EntityMinecart cart);
 }
-
