--- ../src-base/minecraft/net/minecraftforge/common/IShearable.java
+++ ../src-work/minecraft/net/minecraftforge/common/IShearable.java
@@ -19,13 +19,12 @@
 
 package net.minecraftforge.common;
 
-import java.util.List;
-
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
 
 import javax.annotation.Nonnull;
+import java.util.List;
 
 /**
  *
@@ -37,8 +36,7 @@
  */
 //TODO Change to World, not IBlockAccess and make Implementor responsible for removing itself from the world.
 //Better mimics vanilla behavior and allows more control for the user.
-public interface IShearable
-{
+public interface IShearable {
     /**
      * Checks if the object is currently shearable
      * Example: Sheep return false when they have no wool
