--- ../src-base/minecraft/net/minecraftforge/common/IShearable.java
+++ ../src-work/minecraft/net/minecraftforge/common/IShearable.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,13 +19,12 @@
 
 package net.minecraftforge.common;
 
-import java.util.List;
-
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
 
 import javax.annotation.Nonnull;
+import java.util.List;
 
 /**
  *
