--- ../src-base/minecraft/net/minecraftforge/common/ISpecialArmor.java
+++ ../src-work/minecraft/net/minecraftforge/common/ISpecialArmor.java
@@ -131,6 +131,10 @@
          */
         public static float applyArmor(EntityLivingBase entity, NonNullList<ItemStack> inventory, DamageSource source, double damage)
         {
+            return applyArmor(entity, inventory, source, damage, true);
+        }
+
+        public static float applyArmor(EntityLivingBase entity, NonNullList<ItemStack> inventory, DamageSource source, double damage, boolean damageArmor) {
             if (DEBUG)
             {
                 System.out.println("Start: " + damage);
@@ -199,17 +203,15 @@
                     {
                         ItemStack stack = inventory.get(prop.Slot);
                         int itemDamage = (int)Math.max(1, absorb);
-                        if (stack.getItem() instanceof ISpecialArmor)
-                        {
-                            ((ISpecialArmor)stack.getItem()).damageArmor(entity, stack, source, itemDamage, prop.Slot);
-                        }
-                        else
-                        {
-                            if (DEBUG)
-                            {
-                                System.out.println("Item: " + stack.toString() + " Absorbed: " + absorb + " Damaged: " + itemDamage);
+                        if (damageArmor) {
+                            if (stack.getItem() instanceof ISpecialArmor) {
+                                ((ISpecialArmor) stack.getItem()).damageArmor(entity, stack, source, itemDamage, prop.Slot);
+                            } else {
+                                if (DEBUG) {
+                                    System.out.println("Item: " + stack.toString() + " Absorbed: " + absorb + " Damaged: " + itemDamage);
+                                }
+                                stack.damageItem(itemDamage, entity);
                             }
-                            stack.damageItem(itemDamage, entity);
                         }
                         if (stack.isEmpty())
                         {
@@ -226,8 +228,8 @@
             if (damage > 0 && (totalArmor > 0 || totalToughness > 0))
             {
                 double armorDamage = Math.max(1.0F, damage / 4.0F);
-                
-                for (int i = 0; i < inventory.size(); i++)
+
+                for (int i = 0; (damageArmor) && (i < inventory.size()); i++)
                 {
                     if (inventory.get(i).getItem() instanceof ItemArmor)
                     {
