--- ../src-base/minecraft/net/minecraftforge/common/ISpecialArmor.java
+++ ../src-work/minecraft/net/minecraftforge/common/ISpecialArmor.java
@@ -131,7 +131,7 @@
          */
         public static float applyArmor(EntityLivingBase entity, NonNullList<ItemStack> inventory, DamageSource source, double damage)
         {
-            if (DEBUG)
+			if (DEBUG)
             {
                 System.out.println("Start: " + damage);
             }
@@ -199,18 +199,18 @@
                     {
                         ItemStack stack = inventory.get(prop.Slot);
                         int itemDamage = (int)Math.max(1, absorb);
-                        if (stack.getItem() instanceof ISpecialArmor)
-                        {
-                            ((ISpecialArmor)stack.getItem()).damageArmor(entity, stack, source, itemDamage, prop.Slot);
+                        if (stack.getItem() instanceof ISpecialArmor) 
+						{
+                        ((ISpecialArmor) stack.getItem()).damageArmor(entity, stack, source, itemDamage, prop.Slot);
+                        } 
+						else 
+						{
+                        	if (DEBUG) 
+							{
+                        		System.out.println("Item: " + stack.toString() + " Absorbed: " + absorb + " Damaged: " + itemDamage);
+                        	}
+                        	stack.damageItem(itemDamage, entity);
                         }
-                        else
-                        {
-                            if (DEBUG)
-                            {
-                                System.out.println("Item: " + stack.toString() + " Absorbed: " + absorb + " Damaged: " + itemDamage);
-                            }
-                            stack.damageItem(itemDamage, entity);
-                        }
                         if (stack.isEmpty())
                         {
                             /*if (entity instanceof EntityPlayer)
@@ -226,7 +226,7 @@
             if (damage > 0 && (totalArmor > 0 || totalToughness > 0))
             {
                 double armorDamage = Math.max(1.0F, damage / 4.0F);
-                
+
                 for (int i = 0; i < inventory.size(); i++)
                 {
                     if (inventory.get(i).getItem() instanceof ItemArmor)
