--- ../src-base/minecraft/net/minecraftforge/common/MinecraftForge.java
+++ ../src-work/minecraft/net/minecraftforge/common/MinecraftForge.java
@@ -1,6 +1,6 @@
 /*
  * Minecraft Forge
- * Copyright (c) 2016-2018.
+ * Copyright (c) 2016.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -19,30 +19,25 @@
 
 package net.minecraftforge.common;
 
-import net.minecraftforge.fml.common.FMLLog;
+import cn.pfcraft.server.Mohist;
+import com.google.common.collect.Lists;
+import net.minecraft.crash.CrashReport;
+import net.minecraft.crash.ICrashReportDetail;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.common.ForgeHooks.SeedEntry;
+import net.minecraftforge.fluids.FluidRegistry;
 import net.minecraftforge.fml.common.ICrashCallable;
 import net.minecraftforge.fml.common.Loader;
 import net.minecraftforge.fml.common.discovery.ASMDataTable;
 import net.minecraftforge.fml.common.discovery.ASMDataTable.ASMData;
 import net.minecraftforge.fml.common.eventhandler.EventBus;
+import net.minecraftforge.oredict.OreDictionary;
 
+import javax.annotation.Nonnull;
 import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 
-import org.apache.logging.log4j.Level;
-
-import com.google.common.collect.Lists;
-
-import net.minecraft.crash.CrashReport;
-import net.minecraft.crash.ICrashReportDetail;
-import net.minecraft.item.ItemStack;
-import net.minecraftforge.common.ForgeHooks.SeedEntry;
-import net.minecraftforge.fluids.FluidRegistry;
-import net.minecraftforge.oredict.OreDictionary;
-
-import javax.annotation.Nonnull;
-
 public class MinecraftForge
 {
     /**
@@ -83,7 +78,7 @@
     */
    public static void initialize()
    {
-       FMLLog.log.info("MinecraftForge v{} Initialized", ForgeVersion.getVersion());
+       Mohist.LOGGER.info("MinecraftForge v{} Initialized", ForgeVersion.getVersion());
 
        OreDictionary.getOreName(0);
 
@@ -124,7 +119,7 @@
            }
            catch (Exception e)
            {
-               FMLLog.log.error("Could not find class for name '{}'.", name, e);
+               Mohist.LOGGER.error("Could not find class for name '{}'.", name, e);
            }
        }
    }
